diff --git a/third_party/jitterentropy/jitterentropy-library/jitterentropy-base-user.h b/third_party/jitterentropy/jitterentropy-library/jitterentropy-base-user.h
--- third_party/jitterentropy/jitterentropy-library/jitterentropy-base-user.h
+++ third_party/jitterentropy/jitterentropy-library/jitterentropy-base-user.h
@@ -87,6 +87,5 @@
 #ifdef __MACH__
 #include <assert.h>
-#include <CoreServices/CoreServices.h>
 #include <mach/mach.h>
 #include <mach/mach_time.h>
 #include <unistd.h>
