--- MODULE.bazel
+++ MODULE.bazel
@@ -1,6 +1,7 @@
 module(
     name = "cpp2sky",
-    version = "0.6.1-dev",
+    version = "0.6.1-20251203-dfc5bd9",
+    bazel_compatibility = [">=7.0.0", "<9.0.0"],
 )
 
 bazel_dep(name = "abseil-cpp", version = "20250512.1", repo_name = "com_google_absl")
@@ -13,9 +14,3 @@
 bazel_dep(name = "spdlog", version = "1.10.0", repo_name = "com_github_gabime_spdlog")
 
 bazel_dep(name = "googletest", version = "1.17.0", dev_dependency = True, repo_name = "com_google_googletest")
-bazel_dep(name = "hedron_compile_commands", dev_dependency = True)
-git_override(
-    module_name = "hedron_compile_commands",
-    commit = "abb61a688167623088f8768cc9264798df6a9d10",
-    remote = "https://github.com/hedronvision/bazel-compile-commands-extractor.git",
-)
