diff --git a/MODULE.bazel b/MODULE.bazel
index 7e8f183..9526e70 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -74,7 +74,7 @@ use_repo(
 pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
 [
   pip.parse(
-    hub_name = "pypi",
+    hub_name = "pybind11_protobuf_py_deps",
     python_version = version,
     requirements_lock = "//pybind11_protobuf/requirements:requirements_lock_" + version.replace('.','_') + ".txt",
   )
@@ -82,4 +82,4 @@ pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
 
 ]
 
-use_repo(pip, "pypi")
+use_repo(pip, "pybind11_protobuf_py_deps")
diff --git a/pybind11_protobuf/tests/BUILD b/pybind11_protobuf/tests/BUILD
index 5166c82..ae20022 100644
--- a/pybind11_protobuf/tests/BUILD
+++ b/pybind11_protobuf/tests/BUILD
@@ -7,7 +7,7 @@ load("@com_github_grpc_grpc//bazel:python_rules.bzl", "py_proto_library")
 load("@pybind11_bazel//:build_defs.bzl", "pybind_extension")
 
 # [internal] load cc_proto_library.bzl
-load("@pypi//:requirements.bzl", "requirement")
+load("@pybind11_protobuf_py_deps//:requirements.bzl", "requirement")
 
 licenses(["notice"])
 
