diff --git BUILD.bazel BUILD.bazel
new file mode 100644
index 000000000..14b84cd84
--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,40 @@
+licenses(["notice"])  # BSD-3-Clause
+
+package(default_visibility = ["//visibility:private"])
+
+cc_library(
+    name = "config",
+    srcs = ["SuiteSparse_config/SuiteSparse_config.c"],
+    hdrs = ["SuiteSparse_config/SuiteSparse_config.h"],
+    strip_include_prefix = "SuiteSparse_config",
+    copts = ["-fvisibility=hidden"],
+    visibility = ["//visibility:public"],
+)
+
+# The header for libamd.
+cc_library(
+    name = "amd_hdrs",
+    hdrs = [
+        "AMD/Include/amd_internal.h",
+        "AMD/Include/amd.h",
+    ],
+    strip_include_prefix = "AMD/Include",
+    deps = [
+        ":config",
+    ],
+)
+
+# The libamd sources are compiled twice, once as 32-bit and once as 64-bit.
+_AMD_SRCS = glob(
+    ["AMD/Source/amd_*.c"],
+    exclude = ["AMD/Source/amd_l*.c"],
+    allow_empty = False,
+)
+
+cc_library(
+    name = "amd",
+    srcs = _AMD_SRCS,
+    copts = ["-fvisibility=hidden", "-DDLONG"],
+    deps = [":amd_hdrs"],
+    visibility = ["//visibility:public"],
+)
