--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -22,6 +22,13 @@ module(
 # Regular dependencies
 # ====================

+# Override version of rules_swift to avoid an error in BCR CI of not
+# finding switfc.exe on Windows with bazel 7.
+single_version_override(
+    module_name = "rules_swift",
+    version = "2.5.0",
+)
+
 # -- Abseil library.
 # Pinning version to be consistent with third_party/abseil-cpp
 bazel_dep(name = "abseil-cpp", version = "20250512.1", repo_name = "com_google_absl")
