diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 00000000..32fbd345
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -0,0 +1,9 @@
+module(
+    name = "universal-robots-client-library",
+    version = "1.4.0",
+    compatibility_level = 1,
+)
+
+# Most recent version from https://registry.bazel.build/modules/googletest
+bazel_dep(name = "googletest", version = "1.15.2")
+bazel_dep(name = "rules_license", version = "0.0.8")
