diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..c8391a7
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,31 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "iostreams",
+    deps = [
+        "@xz//:lzma",
+        "@bzip2",
+        "@zlib",
+        "@zstd",
+        "@boost//:assert",
+        "@boost//:config",
+        "@boost//:core",
+        "@boost//:detail",
+        "@boost//:function",
+        "@boost//:integer",
+        "@boost//:iterator",
+        "@boost//:mpl",
+        "@boost//:numeric.conversion",
+        "@boost//:preprocessor",
+        "@boost//:random",
+        "@boost//:range",
+        "@boost//:regex",
+        "@boost//:smart_ptr",
+        "@boost//:static_assert",
+        "@boost//:throw_exception",
+        "@boost//:type_traits",
+        "@boost//:utility",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..f69beaa
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,36 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.iostreams",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "bzip2", version = "1.0.8")
+bazel_dep(name = "zlib", version = "1.3")
+bazel_dep(name = "zstd", version = "1.5.5.bcr.2")
+bazel_dep(name = "xz", version = "5.4.5")
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..8bf79d5
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,12 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        negative_test_names = [],
+        deps = DEPS,
+    ),
+)
