===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -14,9 +14,9 @@
 
 module(
     name = "rules_fuzzing",
     # Automatically updated during releases by the Publish to BCR app.
-    version = "0.0.0",
+    version = "0.6.0",
 )
 
 bazel_dep(name = "bazel_features", version = "1.20.0")
 bazel_dep(name = "abseil-cpp", version = "20240116.1", repo_name = "com_google_absl")
