diff --git a/MODULE.bazel b/MODULE.bazel
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -23,15 +23,15 @@
 
 # For local development we include LLVM and Hedron-Compile-Commands.
 # For bazelmod usage these have to be provided by the main module - if necessary.
-include("//bazelmod:hedron.MODULE.bazel")
+# include("//bazelmod:hedron.MODULE.bazel")
 
-include("//bazelmod:llvm.MODULE.bazel")
+# include("//bazelmod:llvm.MODULE.bazel")
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
-bazel_dep(name = "rules_cc", version = "0.1.1")
-bazel_dep(name = "platforms", version = "0.0.11")
+bazel_dep(name = "rules_cc", version = "0.0.17")
+bazel_dep(name = "platforms", version = "0.0.10")
 bazel_dep(name = "abseil-cpp", version = "20250127.0", repo_name = "com_google_absl")
 bazel_dep(name = "re2", version = "2024-07-02.bcr.1", repo_name = "com_googlesource_code_re2")
 bazel_dep(name = "googletest", version = "1.16.0", repo_name = "com_google_googletest")
-bazel_dep(name = "rules_python", version = "1.2.0")
-bazel_dep(name = "protobuf", version = "30.0", repo_name = "com_google_protobuf")
+bazel_dep(name = "rules_python", version = "0.33.2")
+bazel_dep(name = "protobuf", version = "27.0", repo_name = "com_google_protobuf")
