===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -13,8 +13,9 @@
 # in the same commit that introduces the incompatible change.
 module(
     name = "rules_dart",
     bazel_compatibility = [">=9.0.0"],
+    version = "0.6.5",
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.9.2")
 bazel_dep(name = "package_metadata", version = "0.0.13")
