diff --git BUILD.bazel BUILD.bazel
index 15284b7..a410135 100644
--- BUILD.bazel
+++ BUILD.bazel
@@ -1,4 +1,5 @@
 load("@buildifier_prebuilt//:rules.bzl", "buildifier", "buildifier_test")
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 load("@rules_gazebo//gazebo:headers.bzl", "gz_configure_header", "gz_export_header")
 load("@rules_license//rules:license.bzl", "license")
 
diff --git MODULE.bazel MODULE.bazel
index 2d81232..c1e330e 100644
--- MODULE.bazel
+++ MODULE.bazel
@@ -1,12 +1,11 @@
-"gz-utils"
-
 module(
     name = "gz-utils",
-    repo_name = "org_gazebosim_gz-utils",
+    version = "4.0.0.bcr.1",
 )
 
-bazel_dep(name = "buildifier_prebuilt", version = "7.3.1")
-bazel_dep(name = "googletest", version = "1.14.0")
+bazel_dep(name = "buildifier_prebuilt", version = "8.2.0.2")
+bazel_dep(name = "googletest", version = "1.17.0.bcr.2")
+bazel_dep(name = "rules_cc", version = "0.1.1")
 bazel_dep(name = "rules_gazebo", version = "0.0.2")
 bazel_dep(name = "rules_license", version = "1.0.0")
-bazel_dep(name = "spdlog", version = "1.12.0")
+bazel_dep(name = "spdlog", version = "1.17.0")
diff --git cli/BUILD.bazel cli/BUILD.bazel
index 3d57df4..29e7e2f 100644
--- cli/BUILD.bazel
+++ cli/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+
 package(
     default_applicable_licenses = ["//:license"],
     default_visibility = ["__subpackages__"],
diff --git examples/BUILD.bazel examples/BUILD.bazel
index 7f4da34..cd8e2d1 100644
--- examples/BUILD.bazel
+++ examples/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary")
+
 package(
     default_applicable_licenses = ["//:license"],
     default_visibility = ["//visibility:private"],
diff --git log/BUILD.bazel log/BUILD.bazel
index 876d7d5..25ad497 100644
--- log/BUILD.bazel
+++ log/BUILD.bazel
@@ -1,3 +1,4 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_gazebo//gazebo:headers.bzl", "gz_export_header")
 
 package(
