===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,8 +1,8 @@
 # LICENSE sha265: c71d239df91726fc519c6eb72d318ec65820627232b2f796219e87dcf35d0ab4
 module(
     name = "xrootfs",
-    version = "0.0.0",
+    version = "0.5.0",
 )
 
 # A consumer builds //cmd/xrootfs and //cmd/imagecfg, which load rules_go,
 # rules_pkg and, through //build, rules_shell.
