--- cmake/modules/Findfastcdr.cmake
+++ cmake/modules/Findfastcdr.cmake
@@ -0,0 +1,45 @@
+# Findfastcdr.cmake
+
+# Look for the header file
+find_path(FASTCDR_INCLUDE_DIR
+    NAMES fastcdr/Cdr.h
+    PATHS
+    /usr/include
+    /usr/local/include
+    $ENV{FASTCDR_HOME}/include
+    ${CMAKE_INSTALL_PREFIX}/include
+    PATH_SUFFIXES fastcdr fastcdr+
+)
+
+# Look for the library
+find_library(FASTCDR_LIBRARY
+    NAMES fastcdr
+    PATHS
+    /usr/lib
+    /usr/lib64
+    /usr/local/lib
+    /usr/local/lib64
+    $ENV{FASTCDR_HOME}
+    $ENV{FASTCDR_HOME}/lib
+    ${CMAKE_INSTALL_PREFIX}/lib
+    ${CMAKE_INSTALL_PREFIX}/lib64
+)
+
+# Handle the QUIETLY and REQUIRED arguments and set FASTCDR_FOUND to TRUE if
+# all listed variables are TRUE
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(fastcdr
+    FOUND_VAR FASTCDR_FOUND
+    REQUIRED_VARS FASTCDR_LIBRARY FASTCDR_INCLUDE_DIR
+)
+
+# Mark these variables as advanced so they don't show up in the GUI by default
+mark_as_advanced(FASTCDR_INCLUDE_DIR FASTCDR_LIBRARY)
+
+if(FASTCDR_FOUND)
+    # If found, set these variables for consumers
+    set(fastcdr_LIBRARY ${FASTCDR_LIBRARY})
+    set(fastcdr_INCLUDE_DIR ${FASTCDR_INCLUDE_DIR})
+    message("-- Found fastcdr dir: ${fastcdr_INCLUDE_DIR}")
+    message("-- Found fastcdr lib: ${fastcdr_LIBRARY}")
+endif()
\ No newline at end of file
