--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,44 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+package(default_visibility = ["//visibility:public"])
+
+licenses(["notice"])
+
+filegroup(
+    name = "fastdds_srcs",
+    srcs = glob(
+        ["**"],
+        exclude = [
+            "bazel-bin/**",
+            "bazel-Fast-DDS/**",
+            "bazel-out/**",
+            "bazel-testlogs/**",
+            "bazel-qnx/**",
+            "*.log"
+        ]),
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "include",
+    srcs = glob(["include/**"], allow_empty=True),
+    visibility = ["//visibility:public"],
+)
+
+cmake(
+    name = "fastdds",
+    cache_entries = {
+        "CMAKE_C_FLAGS": "-fPIC",
+        "COMPILE_TOOLS": "OFF",
+        "BUILD_SHARED_LIBS": "OFF",
+        "NO_TLS": "ON"
+    },
+    lib_source = "//:fastdds_srcs",
+    out_static_libs = ["libfastdds.a"],
+    deps = [
+        "@fastcdr//:fastcdr",
+        "@asio//:asio",
+        "@tinyxml2//:tinyxml2",
+        "@foonathan_memory//:foonathan_memory"
+    ]
+)
