diff --git cmake/modules/FindThirdpartyBoost.cmake cmake/modules/FindThirdpartyBoost.cmake
index 163a91a..9a18669 100644
--- cmake/modules/FindThirdpartyBoost.cmake
+++ cmake/modules/FindThirdpartyBoost.cmake
@@ -16,7 +16,7 @@ find_package(Threads REQUIRED)
 if(WIN32 OR APPLE OR ANDROID OR QNXNTO)
     set(THIRDPARTY_BOOST_LINK_LIBS ${CMAKE_THREAD_LIBS_INIT})
 else() # Posix
-    set(THIRDPARTY_BOOST_LINK_LIBS ${CMAKE_THREAD_LIBS_INIT} rt)
+    set(THIRDPARTY_BOOST_LINK_LIBS ${CMAKE_THREAD_LIBS_INIT} rt stdc++)
 endif()
 
 try_compile(IS_THIRDPARTY_BOOST_OK
