diff --git a/BUILD.bazel b/BUILD.bazel
index ec299a3..dbdd861 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -123,6 +123,7 @@ cc_library(
     }),
     linkopts = select({
         "@platforms//os:android": [],
+        "@platforms//os:windows": [],
         "//conditions:default": ["-lpthread"],
     }),
     linkstatic = select({
diff --git a/MODULE.bazel b/MODULE.bazel
index a934add..775b543 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,6 @@
 module(
     name = "xz",
-    version = "5.4.5.bcr.2",
+    version = "5.4.5.bcr.3",
 )
 
 bazel_dep(name = "platforms", version = "0.0.8")
