diff --git a/MODULE.bazel b/MODULE.bazel
index b613ac12..287f8dc5 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -83,6 +83,7 @@ haskell_toolchains.bindists()
 use_repo(
     haskell_toolchains,
     "all_bindist_toolchains",
+    "rules_haskell_ghc_windows_amd64_cc_toolchain",
     "rules_haskell_python_local",
 )
 
@@ -95,6 +96,7 @@ register_toolchains(
 
 register_toolchains(
     "@all_bindist_toolchains//:all",
+    "@rules_haskell_ghc_windows_amd64_cc_toolchain//:all",
     "@rules_haskell_python_local//:toolchain",
 )
 
