diff --git a/haskell/BUILD.bazel b/haskell/BUILD.bazel
index 046d3d3e..17b92b65 100644
--- a/haskell/BUILD.bazel
+++ b/haskell/BUILD.bazel
@@ -100,8 +100,6 @@ bzl_library(
         "@bazel_skylib//lib:shell",
         "@bazel_skylib//lib:versions",
         "@bazel_skylib//rules:expand_template",
-        "@rules_nixpkgs_core//:nixpkgs",
-        "@rules_nixpkgs_posix//:posix",
         "@rules_sh//sh:posix",
     ],
 )
diff --git a/haskell/platforms/BUILD.bazel b/haskell/platforms/BUILD.bazel
index 28fc88d4..6ccc0224 100644
--- a/haskell/platforms/BUILD.bazel
+++ b/haskell/platforms/BUILD.bazel
@@ -11,26 +11,6 @@ load(":list.bzl", "declare_config_settings")
 
 declare_config_settings()
 
-alias(
-    name = "nixpkgs",
-    actual = "@rules_nixpkgs_core//constraints:support_nix",
-    visibility = ["//visibility:public"],
-)
-
-alias(
-    name = "linux_x86_64_nixpkgs",
-    actual = "@rules_nixpkgs_core//platforms:host",
-    deprecation = "Use @rules_nixpkgs_core//platforms:host instead.",
-    visibility = ["//visibility:public"],
-)
-
-alias(
-    name = "darwin_x86_64_nixpkgs",
-    actual = "@rules_nixpkgs_core//platforms:host",
-    deprecation = "Use @rules_nixpkgs_core//platforms:host instead.",
-    visibility = ["//visibility:public"],
-)
-
 bzl_library(
     name = "platforms",
     srcs = ["list.bzl"],
