diff --git a/extensions/stack_snapshot.bzl b/extensions/stack_snapshot.bzl
index 1e56d3b3..85593577 100644
--- a/extensions/stack_snapshot.bzl
+++ b/extensions/stack_snapshot.bzl
@@ -243,7 +243,7 @@ def _stack_snapshot_impl(mctx):
                     if snapshot_tag.local_snapshot:
                         kwargs["local_snapshot"] = snapshot_tag.local_snapshot
                     if snapshot_tag.name:
-                        kwargs["snapshot_tag"] = snapshot_tag.name
+                        kwargs["snapshot"] = snapshot_tag.name
         else:
             _assert_non_rules_haskell_tags(module)
         if module == root_module:
