diff --git a/BUILD.bazel b/BUILD.bazel
index 9faec8a..653e1a4 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,6 +1,7 @@
-package(
-    default_visibility = ["//visibility:public"],
-)
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_python//python:defs.bzl", "py_test")
 
 # Description:
 #
@@ -18,39 +19,41 @@ exports_files([
 cc_library(
     name = "effcee",
     srcs = glob(
-        ["effcee/*.cc"],
+        include = ["effcee/*.cc"],
         exclude = ["effcee/*_test.cc"],
     ),
     hdrs = glob(["effcee/*.h"]),
-    compatible_with = [
-    ],
     deps = [
-        "@re2//:re2",
+        "@re2",
     ],
+    visibility = ["//visibility:public"],
 )
 
 ## An example binary showing usage
 cc_binary(
     name = "effcee_example",
     srcs = ["examples/main.cc"],
-    deps = [ ":effcee" ],
+    deps = [":effcee"],
 )
 
 # Test effcee_example executable
 py_test(
     name = "effcee_example_test",
     srcs = ["examples/effcee-example-driver.py"],
-    main = "examples/effcee-example-driver.py",
-    data = [ ":effcee_example", "examples/example_data.txt" ],
     args = [
-        "$(location effcee_example)",
-        "examples/example_data.txt",
+        "$(rootpath effcee_example)",
+        "$(rootpath examples/example_data.txt)",
         "'CHECK: Hello'",
         "'CHECK-SAME: world'",
         "'CHECK-NEXT: Bees'",
         "'CHECK-NOT: Sting'",
         "'CHECK: Honey'",
     ],
+    data = [
+        "examples/example_data.txt",
+        ":effcee_example",
+    ],
+    main = "examples/effcee-example-driver.py",
 )
 
 # Unit tests
@@ -60,8 +63,8 @@ cc_test(
     srcs = ["effcee/check_test.cc"],
     deps = [
         ":effcee",
-        "@googletest//:gtest_main",
         "@googletest//:gtest",
+        "@googletest//:gtest_main",
     ],
 )
 
@@ -70,8 +73,8 @@ cc_test(
     srcs = ["effcee/cursor_test.cc"],
     deps = [
         ":effcee",
-        "@googletest//:gtest_main",
         "@googletest//:gtest",
+        "@googletest//:gtest_main",
     ],
 )
 
@@ -80,8 +83,8 @@ cc_test(
     srcs = ["effcee/diagnostic_test.cc"],
     deps = [
         ":effcee",
-        "@googletest//:gtest_main",
         "@googletest//:gtest",
+        "@googletest//:gtest_main",
     ],
 )
 
@@ -90,8 +93,8 @@ cc_test(
     srcs = ["effcee/match_test.cc"],
     deps = [
         ":effcee",
-        "@googletest//:gtest_main",
         "@googletest//:gtest",
+        "@googletest//:gtest_main",
     ],
 )
 
@@ -100,8 +103,8 @@ cc_test(
     srcs = ["effcee/options_test.cc"],
     deps = [
         ":effcee",
-        "@googletest//:gtest_main",
         "@googletest//:gtest",
+        "@googletest//:gtest_main",
     ],
 )
 
@@ -110,7 +113,7 @@ cc_test(
     srcs = ["effcee/result_test.cc"],
     deps = [
         ":effcee",
-        "@googletest//:gtest_main",
         "@googletest//:gtest",
+        "@googletest//:gtest_main",
     ],
 )
diff --git a/MODULE.bazel b/MODULE.bazel
index 22124d9..43ca161 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,17 +1,18 @@
-module(name = "effcee")
+"""effcee"""
 
-bazel_dep(name = "rules_python", version = "0.31.0")
-python = use_extension("@rules_python//python/extensions:python.bzl", "python")
+module(
+    name = "effcee",
+    version = "0.0.0-20250222",
+)
+
+bazel_dep(name = "rules_python", version = "1.2.0")
+bazel_dep(name = "rules_cc", version = "0.1.1")
+bazel_dep(name = "googletest", version = "1.14.0.bcr.1")
+bazel_dep(name = "re2", version = "2024-07-02.bcr.1")
+
+python = use_extension("@rules_python//python/extensions:python.bzl", "python", dev_dependency = True)
 python.toolchain(
     configure_coverage_tool = False,
     ignore_root_user_error = True,
     python_version = "3.11",
 )
-
-bazel_dep(
-    name = "googletest",
-    version = "1.14.0",
-    dev_dependency = True,
-)
-
-bazel_dep(name = "re2", version = "2024-04-01")
