===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -3,9 +3,11 @@
 module(
     name = "rules_d",
 
     # NOTE:
-    # Always leave version unset or set to "" (the default). The default value
+    # Always leave version unset or set to "" (the default,
+    version = "0.2.0",
+). The default value
     # can prevent issues when the module is used via non-registry overrides
     # (e.g. https://github.com/bazel-contrib/rules_go/issues/4380).
     #
     # The publish.yaml GitHub Action sets the version in the registry to the
