diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..afad4df
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,13 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "foreach",
+    deps = [
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.iterator//:iterator",
+        "@boost.mpl//:mpl",
+        "@boost.range//:range",
+        "@boost.type_traits//:type_traits",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..25619c3
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+module(
+    name = "boost.foreach",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.config", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.iterator", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.mpl", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.range", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..e9e7900
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost.foreach//:foreach",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "foreach_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/foreach_test.cc b/bazel_test/foreach_test.cc
new file mode 100644
index 0000000..735edfb
--- /dev/null
+++ b/bazel_test/foreach_test.cc
@@ -0,0 +1,15 @@
+#include <boost/foreach.hpp>
+#include <boost/foreach_fwd.hpp>
+#include <vector>
+
+int main()
+{
+  std::vector<int> vec{ 1, 2 };
+  int sum = 0;
+  BOOST_FOREACH(int n, vec)
+  {
+    sum += n;
+  }
+
+  return sum == 3 ? 0 : 1;
+}
