--- BUILD.bazel
+++ BUILD.bazel
@@ -58,9 +58,13 @@ cc_binary(
         "core/generator_lite/Generator.hh",
         "core/generator_lite/generator_main.cc",
     ],
+    visibility = ["//:__subpackages__"],
     deps = [
         "@com_google_protobuf//:protobuf",
         "@com_google_protobuf//:protoc_lib",
+        "@com_google_protobuf//src/google/protobuf/compiler:code_generator",
+        "@com_google_protobuf//src/google/protobuf/io",
+        "@com_google_protobuf//src/google/protobuf/io:printer",
     ],
 )
 
@@ -83,10 +87,18 @@ py_proto_library(
 
 gz_proto_library(
     name = "gzmsgs_cc_proto",
+    features = [
+        # Layering check fails for clang build due to no module exporting
+        # `port_def.inc` and `port_undef.inc`. Unfortunately, the bazel target
+        # that exports these headers in protobuf is private.
+        "-layering_check",
+    ],
     proto_deps = [":gzmsgs_proto"],
     visibility = ["//visibility:public"],
     deps = [
         "@com_google_protobuf//:protobuf",
+        "@com_google_protobuf//:protobuf_lite",
+        "@com_google_protobuf//src/google/protobuf/io",
     ],
 )
 
@@ -128,7 +140,9 @@ cc_library(
         ":gzmsgs_proto_factory",
         "@com_google_protobuf//:protobuf",
         "@gz-math",
-        "@gz-utils",
+        "@gz-utils//:Environment",
+        "@gz-utils//:ImplPtr",
+        "@gz-utils//:NeverDestroyed",
         "@tinyxml2",
     ],
 )
@@ -160,6 +174,7 @@ cc_test(
     srcs = ["test/integration/headers.cc"],
     deps = [
         ":gzmsgs_cc_proto",
+        "@googletest//:gtest",
         "@googletest//:gtest_main",
     ],
 )
@@ -169,6 +184,7 @@ cc_test(
     srcs = ["test/integration/image_msg.cc"],
     deps = [
         ":gzmsgs_cc_proto",
+        "@googletest//:gtest",
         "@googletest//:gtest_main",
     ],
 )
@@ -246,6 +262,7 @@ cc_binary(
     visibility = ["//visibility:public"],
     deps = [
         ":gz-msgs",
-        "@gz-utils//cli",
+        "@gz-utils//cli:GzFormatter",
+        "@gz-utils//cli:cli11",
     ],
 )
