diff --git a/MODULE.bazel b/MODULE.bazel
index 5dbba8864..9798f568a 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -3,7 +3,7 @@
 
 module(
     name = "protobuf",
-    version = "29.0-rc2",  # Automatically updated on release
+    version = "29.0-rc2.bcr.1",  # Automatically updated on release
     compatibility_level = 1,
     repo_name = "com_google_protobuf",
 )
diff --git a/java/core/BUILD.bazel b/java/core/BUILD.bazel
index 423f0bbb8..63b5f4af5 100644
--- a/java/core/BUILD.bazel
+++ b/java/core/BUILD.bazel
@@ -1,4 +1,5 @@
 load("@bazel_skylib//rules:build_test.bzl", "build_test")
+load("@rules_java//java:java_library.bzl", "java_library")
 load("@rules_pkg//pkg:mappings.bzl", "pkg_files", "strip_prefix")
 load("//:protobuf.bzl", "internal_gen_well_known_protos_java")
 load("//:protobuf_version.bzl", "PROTOBUF_JAVA_VERSION")
@@ -705,3 +706,4 @@ pkg_files(
     strip_prefix = strip_prefix.from_root(""),
     visibility = ["//java:__pkg__"],
 )
+
