diff --git a/bazel/protobuf.bzl b/bazel/protobuf.bzl
index 08d9d56..d131bd6 100644
--- a/bazel/protobuf.bzl
+++ b/bazel/protobuf.bzl
@@ -1,6 +1,8 @@
 load("@bazel_skylib//rules:common_settings.bzl", "BuildSettingInfo")
-load("@bazel_tools//tools/jdk:toolchain_utils.bzl", "find_java_runtime_toolchain", "find_java_toolchain")
-load("@rules_proto//proto:defs.bzl", "ProtoInfo")
+load("@bazel_tools//tools/jdk:toolchain_utils.bzl", "find_java_toolchain")
+load("@com_google_protobuf//bazel/common:proto_info.bzl", "ProtoInfo")
+load("@rules_java//java/common:java_common.bzl", "java_common")
+load("@rules_java//java/common:java_info.bzl", "JavaInfo")

 # Borrowed from https://github.com/grpc/grpc-java/blob/v1.28.0/java_grpc_library.bzl#L59
 # "repository" here is for Bazel builds that span multiple WORKSPACES.
@@ -110,13 +112,13 @@ cc_proto_gen_validate = rule(
             default = [Label("@com_googlesource_code_re2//:re2")],
         ),
         "_protoc": attr.label(
-            cfg = "host",
+            cfg = "exec",
             default = Label("@com_google_protobuf//:protoc"),
             executable = True,
             allow_single_file = True,
         ),
         "_plugin": attr.label(
-            cfg = "host",
+            cfg = "exec",
             default = Label("@com_envoyproxy_protoc_gen_validate//:protoc-gen-validate"),
             allow_files = True,
             executable = True,
@@ -171,13 +173,13 @@ _java_proto_gen_validate_aspect = aspect(
     attr_aspects = ["deps"],
     attrs = {
         "_protoc": attr.label(
-            cfg = "host",
+            cfg = "exec",
             default = Label("@com_google_protobuf//:protoc"),
             executable = True,
             allow_single_file = True,
         ),
         "_plugin": attr.label(
-            cfg = "host",
+            cfg = "exec",
             default = Label("@com_envoyproxy_protoc_gen_validate//:protoc-gen-validate"),
             allow_files = True,
             executable = True,
