diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..1188401
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,57 @@
+load("@bazel_skylib//lib:selects.bzl", "selects")
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "fiber",
+    srcs = select({
+        # BOOST_FIBER_NUMA_SRCS
+        "@boost.rules.tools//:linux_arm": [
+        ],
+        "@boost.rules.tools//:linux_ppc": [
+            "src/numa/linux/pin_thread.cpp",
+            "src/numa/linux/topology.cpp",
+        ],  # MAYBE SHOULD BE BLANK
+        "@boost.rules.tools//:linux_x86_64": [
+            "src/numa/linux/pin_thread.cpp",
+            "src/numa/linux/topology.cpp",
+        ],
+        "@boost.rules.tools//:windows_x86_64": [
+            "src/numa/windows/pin_thread.cpp",
+            "src/numa/windows/topology.cpp",
+        ],
+        "//conditions:default": [],
+    }) + glob([
+        "src/algo/**/*.cpp",
+    ]) + [
+        "examples/asio/detail/yield.hpp",
+    ],
+    hdrs = [
+        # "examples/asio/round_robin.hpp", # TODO Why are these here??
+        # "examples/asio/yield.hpp",
+    ],
+    copts = select({
+        "@boost.rules.tools//:windows_x86_64": [
+            "/D_WIN32_WINNT=0x0601",
+        ],
+        "//conditions:default": [],
+    }),
+    exclude_src = ["src/numa/**/*.cpp"],
+    linkopts = selects.with_or({
+        ("@platforms//os:linux", "@platforms//os:osx", "@platforms//os:ios", "@platforms//os:watchos", "@platforms//os:tvos"): ["-lpthread"],
+        "//conditions:default": [],
+    }),
+    deps = [
+        "@boost//:algorithm",
+        "@boost//:assert",
+        "@boost//:config",
+        "@boost//:context",
+        "@boost//:core",
+        "@boost//:filesystem",
+        "@boost//:format",
+        "@boost//:intrusive",
+        "@boost//:predef",
+        "@boost//:smart_ptr",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..b3ffef9
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,25 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.fiber",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "bazel_skylib", version = "1.5.0")
+
+
+
+
+
+
+
+
+
+
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..003015b
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,14 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:fiber",
+    "@boost//:test",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        negative_test_names = [],
+        deps = DEPS,
+    ),
+)
