diff --git i/test/unit/test_async_client.cpp w/test/unit/test_async_client.cpp
index b7c4b4f..faf0125 100644
--- i/test/unit/test_async_client.cpp
+++ w/test/unit/test_async_client.cpp
@@ -36,7 +36,8 @@ using namespace mqtt;
 // NOTE: This test case requires network access. It uses one of
 //  	 the public available MQTT brokers
 #if defined(TEST_EXTERNAL_SERVER)
-	static const std::string GOOD_SERVER_URI { "tcp://mqtt.eclipse.org:1883" };
+	static const std::string GOOD_SERVER_URI{"tcp://test.mosquitto.org:1883"};
+	static const std::string GOOD_SSL_SERVER_URI{"ssl://test.mosquitto.org:8883"};
 #else
 	static const std::string GOOD_SERVER_URI { "tcp://localhost:1883" };
 	static const std::string GOOD_SSL_SERVER_URI { "ssl://localhost:18885" };
diff --git i/test/unit/test_async_client_v3.cpp w/test/unit/test_async_client_v3.cpp
index 875e8bb..84461b0 100644
--- i/test/unit/test_async_client_v3.cpp
+++ w/test/unit/test_async_client_v3.cpp
@@ -99,7 +99,7 @@ class async_client_v3_test : public CppUnit::TestFixture
 	// NOTE: This test case requires network access. It uses one of
 	//  	 the public available MQTT brokers
 	#if defined(TEST_EXTERNAL_SERVER)
-		const std::string GOOD_SERVER_URI { "tcp://mqtt.eclipse.org:1883" };
+		const std::string GOOD_SERVER_URI { "tcp://test.mosquitto.org:1883" };
 	#else
 		const std::string GOOD_SERVER_URI { "tcp://localhost:1883" };
 		const std::string GOOD_SSL_SERVER_URI { "ssl://localhost:18885" };
diff --git i/test/unit/test_client.cpp w/test/unit/test_client.cpp
index 13cf3e6..e1207f8 100644
--- i/test/unit/test_client.cpp
+++ w/test/unit/test_client.cpp
@@ -40,7 +40,7 @@ using namespace mqtt;
 // NOTE: This test case requires network access. It uses one of
 //  	 the public available MQTT brokers
 #if defined(TEST_EXTERNAL_SERVER)
-	static const std::string GOOD_SERVER_URI { "tcp://mqtt.eclipse.org:1883" };
+	static const std::string GOOD_SERVER_URI { "tcp://test.mosquitto.org:1883" };
 #else
 	static const std::string GOOD_SERVER_URI { "tcp://localhost:1883" };
 #endif
