diff --git a/src/papilo/io/Message.hpp b/src/papilo/io/Message.hpp
index 0693b929..487f97a4 100644
--- a/src/papilo/io/Message.hpp
+++ b/src/papilo/io/Message.hpp
@@ -77,7 +77,7 @@ class Message
 #else
          buf,
 #endif
-         format_str, { fmt::make_format_args( std::forward<Args>( args )... ) } );
+         format_str, { fmt::make_format_args( args... ) } );
       std::size_t size = buf.size();
 
       if( write != nullptr )
