diff --git a/src/papilo/misc/MultiPrecision.hpp b/src/papilo/misc/MultiPrecision.hpp
index cd14de9b..16fea7d0 100644
--- a/src/papilo/misc/MultiPrecision.hpp
+++ b/src/papilo/misc/MultiPrecision.hpp
@@ -26,6 +26,7 @@
 #define _PAPILO_MISC_MULTIPRECISION_HPP_
 
 #include "papilo/Config.hpp"
+#include "papilo/misc/fmt.hpp"
 
 // work around build failure with boost on Fedora 37
 #include <memory>
@@ -158,4 +159,7 @@ using Float1000 =
 } // namespace papilo
 #endif
 
+template <typename Backend, boost::multiprecision::expression_template_option ExpressionTemplates>
+struct fmt::formatter<boost::multiprecision::number<Backend, ExpressionTemplates>> : fmt::ostream_formatter {};
+
 #endif
