diff --git a/src/papilo/core/ProblemUpdate.hpp b/src/papilo/core/ProblemUpdate.hpp
index 615a5421..68085ad6 100644
--- a/src/papilo/core/ProblemUpdate.hpp
+++ b/src/papilo/core/ProblemUpdate.hpp
@@ -2434,7 +2434,7 @@ ProblemUpdate<REAL>::applyTransaction( const Reduction<REAL>* first,
             const int length = rowvec.getLength();
             const int* indices = rowvec.getIndices();
 
-            msg.detailed( "modified columns: ", ColReduction::SUBSTITUTE, col,
+            msg.detailed( "modified columns: ", static_cast<int>(ColReduction::SUBSTITUTE), col,
                           equalityrow );
             for( int j = 0; j != length; ++j )
             {
