diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..e47ce00
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,22 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "icl",
+    deps = [
+        "@boost.assert//:assert",
+        "@boost.concept_check//:concept_check",
+        "@boost.config//:config",
+        "@boost.container//:container",
+        "@boost.core//:core",
+        "@boost.date_time//:date_time",
+        "@boost.detail//:detail",
+        "@boost.iterator//:iterator",
+        "@boost.move//:move",
+        "@boost.mpl//:mpl",
+        "@boost.range//:range",
+        "@boost.rational//:rational",
+        "@boost.static_assert//:static_assert",
+        "@boost.type_traits//:type_traits",
+        "@boost.utility//:utility",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..434c63f
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,23 @@
+module(
+    name = "boost.icl",
+    version = "1.83.0.bzl.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.concept_check", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.container", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.date_time", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.detail", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.iterator", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.move", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.mpl", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.range", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.rational", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.utility", version = "1.83.0.bzl.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
