--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,46 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "embree-x86_64-linux",
+    url = "https://github.com/embree/embree/releases/download/v4.1.0/embree-4.1.0.x86_64.linux.tar.gz",
+build_file_content = """
+cc_import(
+    name = "embree-x86_64-linux",
+    shared_library = "lib/libembree4.so.4",
+)
+""",
+    strip_prefix = "embree-4.1.0.x86_64.linux",
+)
+
+http_archive(
+    name = "embree-x86_64-macosx",
+    url = "https://github.com/embree/embree/releases/download/v4.1.0/embree-4.1.0.x86_64.macosx.zip",
+build_file_content = """
+cc_import(
+    name = "embree-x86_64-macosx",
+    shared_library = "lib/libembree4.4.dylib",
+)
+""",
+    strip_prefix = "embree-4.1.0.x86_64.macosx",
+)
+
+http_archive(
+    name = "embree-x64-windows",
+    url = "https://github.com/embree/embree/releases/download/v4.1.0/embree-4.1.0.x64.windows.zip",
+build_file_content = """
+cc_import(
+    name = "embree-x64-windows",
+    shared_library = "lib/embree4.lib",
+)
+""",
+    strip_prefix = "embree-4.1.0.x64.windows",
+)
+
+cc_library(
+    name = "embree",
+    hdrs = glob(["include/embree4/*.h"]),
+    includes = ["include/"],
+    deps = select({
+        "@bazel_tools//src/conditions:linux_x86_64": [":embree-x86_64-linux"],
+        "@bazel_tools//src/conditions:darwin_x86_64": [":embree-x86_64-macosx"],
+        "@bazel_tools//src/conditions:windows_x64": [":embree-x64-windows"],
+    }),
+    visibility = ["//visibility:public"],
+)
