diff --git a/lib/clean-temp-simple.c b/lib/clean-temp-simple.c
--- a/lib/clean-temp-simple.c
+++ b/lib/clean-temp-simple.c
@@ -28,5 +28,5 @@
 #include <unistd.h>
 
-#include <error.h>
+#include <m4-error.h>
 #include "fatal-signal.h"
 #include "asyncsafe-spin.h"
diff --git a/lib/clean-temp.c b/lib/clean-temp.c
--- a/lib/clean-temp.c
+++ b/lib/clean-temp.c
@@ -37,5 +37,5 @@
 #include "clean-temp-simple.h"
 #include "clean-temp-private.h"
-#include <error.h>
+#include <m4-error.h>
 #include "fatal-signal.h"
 #include "asyncsafe-spin.h"
diff --git a/lib/closein.c b/lib/closein.c
--- a/lib/closein.c
+++ b/lib/closein.c
@@ -29,5 +29,5 @@
 #include "close-stream.h"
 #include "closeout.h"
-#include <error.h>
+#include <m4-error.h>
 #include "exitfail.h"
 #include "freadahead.h"
diff --git a/lib/closeout.c b/lib/closeout.c
--- a/lib/closeout.c
+++ b/lib/closeout.c
@@ -29,5 +29,5 @@
 
 #include "close-stream.h"
-#include <error.h>
+#include <m4-error.h>
 #include "exitfail.h"
 #include "quotearg.h"
diff --git a/lib/error.c b/lib/error.c
--- a/lib/error.c
+++ b/lib/error.c
@@ -30,5 +30,5 @@
 #endif
 
-#include <error.h>
+#include <m4-error.h>
 
 #include <stdarg.h>
diff --git a/lib/execute.c b/lib/execute.c
--- a/lib/execute.c
+++ b/lib/execute.c
@@ -32,5 +32,5 @@
 
 #include "canonicalize.h"
-#include <error.h>
+#include <m4-error.h>
 #include "fatal-signal.h"
 #include "filename.h"
diff --git a/lib/os2-spawn.c b/lib/os2-spawn.c
--- a/lib/os2-spawn.c
+++ b/lib/os2-spawn.c
@@ -36,5 +36,5 @@
 
 #include "cloexec.h"
-#include <error.h>
+#include <m4-error.h>
 #include "gettext.h"
 
diff --git a/lib/spawn-pipe.c b/lib/spawn-pipe.c
--- a/lib/spawn-pipe.c
+++ b/lib/spawn-pipe.c
@@ -34,5 +34,5 @@
 
 #include "canonicalize.h"
-#include <error.h>
+#include <m4-error.h>
 #include "fatal-signal.h"
 #include "filename.h"
diff --git a/lib/wait-process.c b/lib/wait-process.c
--- a/lib/wait-process.c
+++ b/lib/wait-process.c
@@ -30,5 +30,5 @@
 #include <sys/wait.h>
 
-#include <error.h>
+#include <m4-error.h>
 #include "fatal-signal.h"
 #include "xalloc.h"
diff --git a/lib/xalloc-die.c b/lib/xalloc-die.c
--- a/lib/xalloc-die.c
+++ b/lib/xalloc-die.c
@@ -23,5 +23,5 @@
 #include <stdlib.h>
 
-#include <error.h>
+#include <m4-error.h>
 #include "exitfail.h"
 
diff --git a/lib/xprintf.c b/lib/xprintf.c
--- a/lib/xprintf.c
+++ b/lib/xprintf.c
@@ -23,5 +23,5 @@
 #include <errno.h>
 
-#include <error.h>
+#include <m4-error.h>
 #include "exitfail.h"
 #include "gettext.h"
diff --git a/lib/openat-die.c b/lib/openat-die.c
--- a/lib/openat-die.c
+++ b/lib/openat-die.c
@@ -23,5 +23,5 @@
 
 #ifndef GNULIB_LIBPOSIX
-# include <error.h>
+# include <m4-error.h>
 #endif
 
diff --git a/lib/chdir-long.c b/lib/chdir-long.c
--- a/lib/chdir-long.c
+++ b/lib/chdir-long.c
@@ -210,5 +210,5 @@
 
 # include "closeout.h"
-# include <error.h>
+# include <m4-error.h>
 
 int
diff --git a/src/m4.h b/src/m4.h
--- a/src/m4.h
+++ b/src/m4.h
@@ -29,5 +29,5 @@
 #include <c-ctype.h>
 #include <errno.h>
-#include <error.h>
+#include <m4-error.h>
 #include <limits.h>
 #include <locale.h>
