diff --git a/src/output.c b/src/output.c
--- a/src/output.c
+++ b/src/output.c
@@ -381,6 +381,10 @@ static int cmp_diversion_CB (const void *a, const void *b)
 void
 output_init (void)
 {
+#ifdef SET_BINARY
+  SET_BINARY (STDOUT_FILENO);
+#endif
+
   diversion_table = gl_oset_create_empty (GL_AVLTREE_OSET, cmp_diversion_CB,
                                           NULL);
   div0.u.file = stdout;
