diff --git a/src/plt-mach-o.c b/src/plt-mach-o.c
index 3db34ca..575cfd1 100644
--- a/src/plt-mach-o.c
+++ b/src/plt-mach-o.c
@@ -250,29 +250,59 @@ plt_offset *plt_get_offsets(plt_ctx ctx, plt_lib lib, const char *name, size_t *
     return ot;
 }
 
+struct dyld_interpose_tuple {
+    const void* replacement;
+    const void* replacee;
+};
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+void dyld_dynamic_interpose(const struct mach_header* mh, const struct dyld_interpose_tuple array[], size_t count);
+#ifdef __cplusplus
+}
+#endif
+
 void plt_set_offsets(plt_offset *offset, size_t nb_off, plt_fn *newval)
 {
     for (size_t i = 0; i < nb_off; ++i) {
         if (!offset[i].oldval)
             offset[i].oldval = *offset[i].offset;
-#ifdef MMK_ARCH_ARM64
-        const long page_size = sysconf(_SC_PAGESIZE);
-        mmk_assert(page_size > 0);
-        char *page_start = (char *) (((uintptr_t) offset[i].offset) & ~(page_size - 1));
-        mmk_assert(!mprotect (page_start, page_size, PROT_READ | PROT_WRITE));
-        __clear_cache(page_start, page_start + page_size);
-#endif
-        *offset[i].offset = newval;
+        offset[i].newval = newval;
+
+        struct dyld_interpose_tuple interpose = {
+            .replacement = (const void *) newval,
+            .replacee = (const void *) offset[i].oldval,
+        };
+
+        uint32_t count = _dyld_image_count();
+        for (uint32_t img = 0; img < count; ++img) {
+            const struct mach_header *hdr = _dyld_get_image_header(img);
+            dyld_dynamic_interpose(hdr, &interpose, 1);
+        }
     }
 }
 
 void plt_reset_offsets(plt_offset *offset, size_t nb_off)
 {
     for (size_t i = 0; i < nb_off; ++i) {
-        *offset[i].offset = offset[i].oldval;
+        if (!offset[i].newval)
+            continue;
+
+        struct dyld_interpose_tuple interpose = {
+            .replacement = (const void *) offset[i].oldval,
+            .replacee = (const void *) offset[i].newval,
+        };
+
+        uint32_t count = _dyld_image_count();
+        for (uint32_t img = 0; img < count; ++img) {
+            const struct mach_header *hdr = _dyld_get_image_header(img);
+            dyld_dynamic_interpose(hdr, &interpose, 1);
+        }
     }
 }
 
+
 plt_fn *plt_get_real_fn(plt_ctx ctx, const char *name)
 {
     (void) ctx;
diff --git a/src/plt.h b/src/plt.h
index 8fb888c..8c076e5 100644
--- a/src/plt.h
+++ b/src/plt.h
@@ -39,6 +39,7 @@
 typedef struct plt_offset {
     plt_fn **offset;
     plt_fn *oldval;
+    plt_fn *newval;
 } plt_offset;
 
 enum plt_selector {
