diff --git a/test/src/test.c b/test/src/test.c
--- a/test/src/test.c	(revision 91b4c235f5d228bcf3d6d007644ed9eef60dbfef)
+++ b/test/src/test.c	(revision 02f316ca6dc2fd2a321b1e3c151c3e34b4326287)
@@ -181,7 +181,7 @@
 	    test_status_string(test_status_fail),
 	    test_counts[test_status_fail], test_count);

-	return ret;
+	return (ret == test_status_skip) ? test_status_pass : ret;
 }

 test_status_t
