diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..66eede7
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,31 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "filesystem",
+    defines = [
+        "BOOST_FILESYSTEM_NO_CXX20_ATOMIC_REF",
+    ] + select({
+        "@platforms//os:linux": [
+            "BOOST_FILESYSTEM_HAS_POSIX_AT_APIS",
+        ],
+        "//conditions:default": [],
+    }),
+    deps = [
+        "@boost//:assert",
+        "@boost//:atomic",
+        "@boost//:config",
+        "@boost//:container_hash",
+        "@boost//:core",
+        "@boost//:detail",
+        "@boost//:io",
+        "@boost//:iterator",
+        "@boost//:predef",
+        "@boost//:smart_ptr",
+        "@boost//:system",
+        "@boost//:throw_exception",
+        "@boost//:type_traits",
+        "@boost//:winapi",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..ab3be16
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.filesystem",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..977928c
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,39 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:filesystem",
+    "@boost//:foreach",
+    "@boost//:range",
+    "@boost//:test",
+    "@boost//:timer",
+    "@boost//:thread",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        negative_test_names = [
+            "issues/3332/test",
+            "70-71-copy",
+            "99_canonical_with_junction_point",
+            "4329.-basename",
+            "5300-temp-dir-path-130",
+            "6638-global-init-fails-3",
+            "8930",
+            "9054_static_const_codecvt_segfault_pre_main",
+            "9219",
+            "10038",
+            "10205",
+            "10485",
+            "10641",
+            "11166-remove-race",
+            "11228--filtered-recursive_directory_iterator-range",
+            "copy_file-compilation-error-2015-05-04",
+            "fchmodat_AT_SYMLINK_NOFOLLOW_6659",
+            "hello_filesystem",
+            "recurse_dir_iter_5403",
+            "reparse_tag_file_placeholder",
+        ],
+        deps = DEPS,
+    ),
+)
