diff --git a/private/app_file_test.bzl b/private/app_file_test.bzl
index 777d0d2..1f0002f 100644
--- a/private/app_file_test.bzl
+++ b/private/app_file_test.bzl
@@ -8,7 +8,7 @@ def _app_file_test_impl(ctx):
     target_under_test = analysistest.target_under_test(env)
     asserts.equals(
         env,
-        expected = path_join(ctx.label.package, "basic.app"),
+        expected = path_join(ctx.label.workspace_root, ctx.label.package, "basic.app"),
         actual = target_under_test[DefaultInfo].files.to_list()[0].short_path,
     )
 
