--- MODULE.bazel
+++ MODULE.bazel
@@ -1,16 +1,13 @@
 module(
     name = "phtree-cpp",
     version = "1.7.0",
+    compatibility_level = 0,
 )
 
-# Bazel rules repositories.
 bazel_dep(name = "rules_shell", version = "0.6.1")
 bazel_dep(name = "rules_cc", version = "0.2.16")
 bazel_dep(name = "platforms", version = "1.0.0")
-
-# Dependencies.
 bazel_dep(name = "googletest", version = "1.17.0.bcr.2")
 bazel_dep(name = "spdlog", version = "1.17.0")
 bazel_dep(name = "google_benchmark", version = "1.9.5")
-
-bazel_dep(name = "buildifier_prebuilt", version = "8.2.1", dev_dependency = True)
+bazel_dep(name = "buildifier_prebuilt", version = "8.2.1")
