diff --git a/.bcr/presubmit.yml b/.bcr/presubmit.yml
index c004650..fd2cf91 100644
--- a/.bcr/presubmit.yml
+++ b/.bcr/presubmit.yml
@@ -2,7 +2,7 @@ bcr_test_module:
   module_path: examples/basicapp
   matrix:
     platform: ["ubuntu2004", "macos", "windows"]
-    bazel: ["7.2.1", "7.4.0", "8.x", "rolling"]
+    bazel: ["7.4.1", "8.0.0", "rolling"]
   tasks:
     run_test_module:
       name: "Verify build targets with bzlmod"
diff --git a/MODULE.bazel b/MODULE.bazel
index 43208ba..6ce3046 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,6 @@
 module(
     name = "rules_android",
-    version = "0.6.0",
+    version = "0.6.0-alpha1.bcr.1",
     compatibility_level = 1,
 )
 
diff --git a/examples/basicapp/MODULE.bazel b/examples/basicapp/MODULE.bazel
index 9ae567b..d3fd39c 100644
--- a/examples/basicapp/MODULE.bazel
+++ b/examples/basicapp/MODULE.bazel
@@ -7,7 +7,7 @@ bazel_dep(name = "bazel_skylib", version = "1.3.0")
 
 bazel_dep(
     name = "rules_android",
-    version = "0.6.0",
+    version = "0.6.0-alpha1.bcr.1",
 )
 
 local_path_override(
