diff --git MODULE.bazel MODULE.bazel
index c68a9c2..db73a38 100644
--- MODULE.bazel
+++ MODULE.bazel
@@ -5,24 +5,17 @@ module(
     version = "1.5.0",
 )
 
-bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "bazel_skylib", version = "1.9.0")
+bazel_dep(name = "googleapis", version = "0.0.0-20260130-c0fcb356")
+bazel_dep(name = "googleapis-cc", version = "1.0.0")
+bazel_dep(name = "googleapis-go", version = "1.0.0")
+bazel_dep(name = "googleapis-python", version = "1.0.0")
+bazel_dep(name = "googleapis-grpc-cc", version = "1.0.0")
+bazel_dep(name = "grpc", version = "1.76.0")
+bazel_dep(name = "protobuf", version = "33.5")
 bazel_dep(name = "rules_cc", version = "0.2.16")
-bazel_dep(name = "googleapis", version = "0.0.0-20240819-fe8ba054a")
-bazel_dep(name = "grpc", version = "1.68.0")
-bazel_dep(name = "protobuf", version = "29.1")
+bazel_dep(name = "rules_go", version = "0.60.0")
 bazel_dep(name = "rules_license", version = "1.0.0")
-bazel_dep(name = "rules_proto", version = "7.0.2")
-bazel_dep(name = "rules_go", version = "0.59.0")
-bazel_dep(name = "gazelle", version = "0.45.0")
-
-switched_rules = use_extension("@googleapis//:extensions.bzl", "switched_rules")
-switched_rules.use_languages(
-    cc = True,
-    go = True,
-    grpc = True,
-    python = True,
-)
-use_repo(switched_rules, googleapis_imports = "com_google_googleapis_imports")
 
 # Including this allows us to autoformat all Bazel files by running
 # ```
diff --git proto/p4/config/v1/BUILD.bazel proto/p4/config/v1/BUILD.bazel
index 787e5de..8f09324 100644
--- proto/p4/config/v1/BUILD.bazel
+++ proto/p4/config/v1/BUILD.bazel
@@ -2,7 +2,7 @@ load("@bazel_skylib//rules:build_test.bzl", "build_test")
 load("@grpc//bazel:python_rules.bzl", "py_proto_library")
 load("@protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
 load("@rules_go//proto:def.bzl", "go_proto_library")
-load("@rules_proto//proto:defs.bzl", "proto_library")
+load("@protobuf//bazel:proto_library.bzl", "proto_library")
 
 package(
     default_visibility = ["//visibility:public"],
diff --git proto/p4/v1/BUILD.bazel proto/p4/v1/BUILD.bazel
index a6608ab..57931bb 100644
--- proto/p4/v1/BUILD.bazel
+++ proto/p4/v1/BUILD.bazel
@@ -3,7 +3,7 @@ load("@grpc//bazel:cc_grpc_library.bzl", "cc_grpc_library")
 load("@grpc//bazel:python_rules.bzl", "py_grpc_library", "py_proto_library")
 load("@protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
 load("@rules_go//proto:def.bzl", "go_proto_library")
-load("@rules_proto//proto:defs.bzl", "proto_library")
+load("@protobuf//bazel:proto_library.bzl", "proto_library")
 
 package(
     default_visibility = ["//visibility:public"],
