diff --git a/bazel/grpc_build_system.bzl b/bazel/grpc_build_system.bzl
index ddfb8e36e7..086ebe34b6 100644
--- a/bazel/grpc_build_system.bzl
+++ b/bazel/grpc_build_system.bzl
@@ -26,7 +26,7 @@
 load("//bazel:cc_grpc_library.bzl", "cc_grpc_library")
 load("//bazel:copts.bzl", "GRPC_DEFAULT_COPTS")
 load("@upb//bazel:upb_proto_library.bzl", "upb_proto_library")
-load("@build_bazel_rules_apple//apple:ios.bzl", "ios_unit_test")
+# load("@build_bazel_rules_apple//apple:ios.bzl", "ios_unit_test")
 
 # The set of pollers to test against if a test exercises polling
 POLLERS = ["epollex", "epoll1", "poll"]
@@ -211,13 +211,13 @@ def ios_cc_test(
             testonly = 1,
         )
         ios_test_deps = [ios_test_adapter, ":" + test_lib_ios]
-        ios_unit_test(
-            name = name + "_on_ios",
-            size = kwargs.get("size"),
-            tags = ios_tags,
-            minimum_os_version = "9.0",
-            deps = ios_test_deps,
-        )
+        # ios_unit_test(
+        #     name = name + "_on_ios",
+        #     size = kwargs.get("size"),
+        #     tags = ios_tags,
+        #     minimum_os_version = "9.0",
+        #     deps = ios_test_deps,
+        # )
 
 def grpc_cc_test(name, srcs = [], deps = [], external_deps = [], args = [], data = [], uses_polling = True, language = "C++", size = "medium", timeout = None, tags = [], exec_compatible_with = [], exec_properties = {}, shard_count = None, flaky = None, copts = []):
     copts = copts + if_mac(["-DGRPC_CFSTREAM"])
