--- config.h.in
+++ config.h.in
@@ -176,7 +176,7 @@
 #undef HAVE_VSNPRINTF
 
 /* Define if you have the wcrtomb function.  */
-#undef HAVE_WCRTOMB 
+#undef HAVE_WCRTOMB
 
 /* Define if you have the wcscoll function.  */
 #undef HAVE_WCSCOLL
@@ -292,13 +292,13 @@
 #undef HAVE_MBSTATE_T
 
 /* Define if you have wchar_t in <wctype.h>. */
-#undef HAVE_WCHAR_T 
+#undef HAVE_WCHAR_T
 
 /* Define if you have wctype_t in <wctype.h>. */
 #undef HAVE_WCTYPE_T
 
-/* Define if you have wint_t in <wctype.h>. */  
-#undef HAVE_WINT_T  
+/* Define if you have wint_t in <wctype.h>. */
+#undef HAVE_WINT_T
 
 /* Define if you have <langinfo.h> and nl_langinfo(CODESET). */
 #undef HAVE_LANGINFO_CODESET
