===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,8 +1,10 @@
 module(
     name = "windows_support",
 
-    # NOTE: Always leave version unset or set to "" (the default) https://bazelbuild.slack.com/archives/CA31HN1T3/p1750406404452179
+    # NOTE: Always leave version unset or set to "" (the default,
+    version = "0.1.6",
+) https://bazelbuild.slack.com/archives/CA31HN1T3/p1750406404452179
     version = "",
     compatibility_level = 0,
 )
 
