diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..e1fcda8
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,10 @@
+module(
+    name = "curl",
+    version = "8.8.0.bcr.1",
+    compatibility_level = 0,
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "mbedtls", version = "3.6.0")
+bazel_dep(name = "platforms", version = "0.0.10")
+bazel_dep(name = "boringssl", version = "0.0.0-20230215-5c22014")
