--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,3 +1,2 @@
-load("@bazel_skylib//lib:selects.bzl", "selects")
 load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
 load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
@@ -21,10 +20,10 @@
 # If we're building azure_core_cpp with Curl as a transport, we must ensure curl
 # was built with OpenSSL and not another SSL lib, as azure_core_cpp only
 # supports OpenSSL.
-selects.config_setting_group(
+config_setting(
     name = "build_transport_curl_openssl",
-    match_all = [
-        ":build_transport_curl_setting",
-        "@curl//:use_openssl",
-    ],
+    flag_values = {
+        ":build_transport_curl": "true",
+        "@curl//:ssl_lib": "openssl",
+    },
 )
