diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..4cbb54e
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,5 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "pfr",
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..2c01027
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,8 @@
+module(
+    name = "boost.pfr",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..e5ec4a7
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost.pfr//:pfr",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "pfr_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/pfr_test.cc b/bazel_test/pfr_test.cc
new file mode 100644
index 0000000..d238945
--- /dev/null
+++ b/bazel_test/pfr_test.cc
@@ -0,0 +1,11 @@
+#include <boost/pfr.hpp>
+
+struct Foo {
+  int bar = 1;
+  int baz = 0;
+};
+
+int main() {
+  Foo f{};
+  return boost::pfr::get<1>(f);
+}
