diff --git a/source/icudefs.mk.in b/source/icudefs.mk.in
index c18113c892a..4a517307e5d 100644
--- a/source/icudefs.mk.in
+++ b/source/icudefs.mk.in
@@ -51,13 +51,13 @@ SO_TARGET_VERSION_MAJOR = @LIB_VERSION_MAJOR@
 # The ICU data external name is usually icudata; the entry point name is
 # the version-dependent name (for no particular reason except it was easier
 # to change the build this way). When building in common mode, the data
-# name is the versioned platform-dependent one. 
+# name is the versioned platform-dependent one.
 
 ICUDATA_DIR = @pkgicudatadir@/$(PACKAGE)$(ICULIBSUFFIX)/$(VERSION)
 
 ICUDATA_BASENAME_VERSION = $(ICUPREFIX)dt@LIB_VERSION_MAJOR@
-# the entry point is almost like the basename, but has the lib suffix.  
-ICUDATA_ENTRY_POINT = $(ICUPREFIX)dt@ICULIBSUFFIXCNAME@@LIB_VERSION_MAJOR@ 
+# the entry point is almost like the basename, but has the lib suffix.
+ICUDATA_ENTRY_POINT = $(ICUPREFIX)dt@ICULIBSUFFIXCNAME@@LIB_VERSION_MAJOR@
 ICUDATA_CHAR = @ICUDATA_CHAR@
 ICUDATA_PLATFORM_NAME = $(ICUDATA_BASENAME_VERSION)$(ICUDATA_CHAR)
 PKGDATA_LIBSTATICNAME = -L $(STATIC_PREFIX)$(ICUPREFIX)$(DATA_STUBNAME)$(ICULIBSUFFIX)
@@ -118,7 +118,7 @@ EXEEXT = @EXEEXT@
 CC = @CC@
 CXX = @CXX@
 AR = @AR@
-ARFLAGS = @ARFLAGS@ r
+ARFLAGS = @ARFLAGS@
 RANLIB = @RANLIB@
 COMPILE_LINK_ENVVAR = @COMPILE_LINK_ENVVAR@
 UCLN_NO_AUTO_CLEANUP = @UCLN_NO_AUTO_CLEANUP@
@@ -211,7 +211,7 @@ LIBICU = $(LIBPREFIX)$(ICUPREFIX)
 ifneq ($(ENABLE_SHARED),YES)
 STATIC_PREFIX_WHEN_USED = s
 else
-STATIC_PREFIX_WHEN_USED = 
+STATIC_PREFIX_WHEN_USED =
 endif
 
 # Static library prefix and file extension
