diff --git a/MODULE.bazel b/MODULE.bazel
index 0a6b8c3..90ec72c 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -14,6 +14,6 @@
 # limitations under the License.
 #
 
-module(name = "protobuf-matchers")
+module(name = "protobuf-matchers", version = "0.0.0-20240603-9f688b0")
 bazel_dep(name = "protobuf", version = "23.1", repo_name = "com_google_protobuf")
 bazel_dep(name = "googletest", version = "1.11.0", repo_name = "com_google_googletest")
