diff --git a/powershell/private/entrypoint.bat b/powershell/private/entrypoint.bat
old mode 100644
new mode 100755
index 3c23e8c..7d73b2a
--- a/powershell/private/entrypoint.bat
+++ b/powershell/private/entrypoint.bat
@@ -63,6 +63,13 @@ exit /b 0
 call :rlocation "{PWSH_INTERPRETER}" PWSH_INTERPRETER
 call :rlocation "{MAIN}" MAIN
 
+# Powershell tries to cache files in the user's `HOME` directory. When running
+# tests, try to contain this cache to an isolated location.
+if defined TEST_TMPDIR (
+    set "HOME=%TEST_TMPDIR%\powershell"
+    set "USERPROFILE=%TEST_TMPDIR%\powershell"
+)
+
 %PWSH_INTERPRETER% ^
     %MAIN% ^
     %*
diff --git a/tools/version_test/version_test.ps1 b/tools/version_test/version_test.ps1
index 8c0a7dd..5b70eb3 100644
--- a/tools/version_test/version_test.ps1
+++ b/tools/version_test/version_test.ps1
@@ -30,6 +30,7 @@ function Get-Runfile {
     if ($RlocationPath -like "rules_powershell*") {
         $altPath = $RlocationPath -replace "^rules_powershell", "_main"
         $runfileAlt = Join-Path -Path $parent3 -ChildPath $altPath
+        Write-Output "$runfileAlt"
         if (Test-Path $runfileAlt) {
             return $runfileAlt
         }
