diff --git a/powershell/private/entrypoint.bat b/powershell/private/entrypoint.bat
old mode 100644
new mode 100755
index 3c23e8c..792f95d
--- a/powershell/private/entrypoint.bat
+++ b/powershell/private/entrypoint.bat
@@ -63,6 +63,13 @@ exit /b 0
 call :rlocation "{PWSH_INTERPRETER}" PWSH_INTERPRETER
 call :rlocation "{MAIN}" MAIN
 
+@REM Powershell tries to cache files in the user's `HOME` directory. When running
+@REM tests, try to contain this cache to an isolated location.
+if defined TEST_TMPDIR (
+    set "HOME=%TEST_TMPDIR%\powershell"
+    set "USERPROFILE=%TEST_TMPDIR%\powershell"
+)
+
 %PWSH_INTERPRETER% ^
     %MAIN% ^
     %*
diff --git a/powershell/private/entrypoint.sh b/powershell/private/entrypoint.sh
old mode 100644
new mode 100755
index 9367613..8aeee70
--- a/powershell/private/entrypoint.sh
+++ b/powershell/private/entrypoint.sh
@@ -19,6 +19,12 @@ fi
 
 runfiles_export_envvars
 
+# Powershell tries to cache files in the user's `HOME` directory. When running
+# tests, try to contain this cache to an isolated location.
+if [[ -n "${TEST_TMPDIR}" ]]; then
+    export HOME="${TEST_TMPDIR}/powershell"
+fi
+
 exec \
     "$(rlocation "{PWSH_INTERPRETER}")" \
     "$(rlocation "{MAIN}")" \
diff --git a/tools/version_test/version_test.ps1 b/tools/version_test/version_test.ps1
index 8c0a7dd..bcbdd4d 100644
--- a/tools/version_test/version_test.ps1
+++ b/tools/version_test/version_test.ps1
@@ -16,10 +16,10 @@ function Get-Runfile {
     # Move up three directories
     $parent1 = Split-Path -Parent $scriptDir
     $parent2 = Split-Path -Parent $parent1
-    $parent3 = Split-Path -Parent $parent2
+    $runfiles_root = Split-Path -Parent $parent2
 
     # Combine with the original path
-    $runfile = Join-Path -Path $parent3 -ChildPath $RlocationPath
+    $runfile = Join-Path -Path $runfiles_root -ChildPath $RlocationPath
 
     # If it exists, return it
     if (Test-Path $runfile) {
@@ -29,14 +29,15 @@ function Get-Runfile {
     # If not, try replacing 'rules_powershell' with '_main' at the start
     if ($RlocationPath -like "rules_powershell*") {
         $altPath = $RlocationPath -replace "^rules_powershell", "_main"
-        $runfileAlt = Join-Path -Path $parent3 -ChildPath $altPath
+        $runfileAlt = Join-Path -Path $runfiles_root -ChildPath $altPath
+        Write-Output "second: $runfileAlt"
         if (Test-Path $runfileAlt) {
             return $runfileAlt
         }
     }
 
     # If neither exists, throw an error
-    throw "Runfile does not exist: ($RlocationPath)"
+    throw "Runfile does not exist: ($RlocationPath) from ($runfiles_root)"
 }
 
 function Main {
