diff --git a/powershell/private/entrypoint.bat b/powershell/private/entrypoint.bat
index 3c23e8c..e43000c 100644
--- a/powershell/private/entrypoint.bat
+++ b/powershell/private/entrypoint.bat
@@ -63,6 +63,10 @@ exit /b 0
 call :rlocation "{PWSH_INTERPRETER}" PWSH_INTERPRETER
 call :rlocation "{MAIN}" MAIN
 
+if defined TEST_TMPDIR (
+    set "XDG_CACHE_HOME=%TEST_TMPDIR%\powershell"
+)
+
 %PWSH_INTERPRETER% ^
     %MAIN% ^
     %*
diff --git a/powershell/private/entrypoint.sh b/powershell/private/entrypoint.sh
index 9367613..d814e5c 100644
--- a/powershell/private/entrypoint.sh
+++ b/powershell/private/entrypoint.sh
@@ -19,6 +19,12 @@ fi
 
 runfiles_export_envvars
 
+set -x
+
+if [[ -n "${TEST_TMPDIR}" ]]; then
+    export XDG_CACHE_HOME="${TEST_TMPDIR}/powershell"
+fi
+
 exec \
     "$(rlocation "{PWSH_INTERPRETER}")" \
     "$(rlocation "{MAIN}")" \
