diff -ruN a/libveriuser/config.h.in b/libveriuser/config.h.in
--- a/libveriuser/config.h.in	2026-03-01 23:38:29.000000000 +0000
+++ b/libveriuser/config.h.in	2026-04-17 19:26:33.258538087 +0000
@@ -1,5 +1,5 @@
-#ifndef IVL_config_H
-#define IVL_config_H
+#ifndef IVL_libveriuser_config_H
+#define IVL_libveriuser_config_H
 /*
  * Copyright (c) 2003-2014 Stephen Williams (steve@icarus.com)
  *
@@ -37,4 +37,4 @@
 # endif
 #endif
 
-#endif /* IVL_config_H */
+#endif /* IVL_libveriuser_config_H */
diff -ruN a/libveriuser/getsimtime.c b/libveriuser/getsimtime.c
--- a/libveriuser/getsimtime.c	2026-03-01 23:38:29.000000000 +0000
+++ b/libveriuser/getsimtime.c	2026-04-17 19:26:33.259538085 +0000
@@ -21,7 +21,7 @@
 #include  <vpi_user.h>
 #include  <stdlib.h>
 #include  <math.h>
-#include  "config.h"
+#include  "lvr_config.h"
 #include  "priv.h"
 #include  <assert.h>
 
diff -ruN a/libveriuser/math.c b/libveriuser/math.c
--- a/libveriuser/math.c	2026-03-01 23:38:29.000000000 +0000
+++ b/libveriuser/math.c	2026-04-17 19:26:33.259538085 +0000
@@ -20,7 +20,7 @@
 # include <string.h>
 # include <stdlib.h>
 # include <assert.h>
-# include "config.h"
+# include "lvr_config.h"
 # include "vpi_user.h"
 # include "veriuser.h"
 
diff -ruN a/libveriuser/veriusertfs.c b/libveriuser/veriusertfs.c
--- a/libveriuser/veriusertfs.c	2026-03-01 23:38:29.000000000 +0000
+++ b/libveriuser/veriusertfs.c	2026-04-17 19:26:33.260538083 +0000
@@ -27,7 +27,7 @@
 # include <stdlib.h>
 # include <assert.h>
 # include <math.h>
-# include "config.h"
+# include "lvr_config.h"
 # include "priv.h"
 # include "vpi_user.h"
 # include "veriuser.h"
diff -ruN a/vhdlpp/parse_wrap.h b/vhdlpp/parse_wrap.h
--- a/vhdlpp/parse_wrap.h	2026-03-01 23:38:29.000000000 +0000
+++ b/vhdlpp/parse_wrap.h	2026-04-17 19:26:33.263538078 +0000
@@ -36,6 +36,6 @@
 
 class VType;
 
-# include  "parse.h"
+# include  "vhdlpp_parse.h"
 
 #endif /* IVL_parse_wrap_H */
diff -ruN a/vvp/config.h.in b/vvp/config.h.in
--- a/vvp/config.h.in	2026-04-17 19:08:55.711062174 +0000
+++ b/vvp/config.h.in	2026-04-17 19:26:33.262538079 +0000
@@ -1,5 +1,5 @@
-#ifndef IVL_config_H
-#define IVL_config_H
+#ifndef IVL_vvp_config_H
+#define IVL_vvp_config_H
 /*
  * Copyright (c) 2001-2025 Stephen Williams (steve@icarus.com)
  *
@@ -213,4 +213,4 @@
 }
 #endif
 
-#endif /* IVL_config_H */
+#endif /* IVL_vvp_config_H */
