diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..97be996
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,28 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "python",
+    deps = [
+        "@boost.align//:align",
+        "@boost.bind//:bind",
+        "@boost.config//:config",
+        "@boost.conversion//:conversion",
+        "@boost.core//:core",
+        "@boost.detail//:detail",
+        "@boost.foreach//:foreach",
+        "@boost.function//:function",
+        "@boost.graph//:graph",
+        "@boost.integer//:integer",
+        "@boost.iterator//:iterator",
+        "@boost.lexical_cast//:lexical_cast",
+        "@boost.mpl//:mpl",
+        "@boost.numeric.conversion//:conversion",
+        "@boost.preprocessor//:preprocessor",
+        "@boost.property_map//:property_map",
+        "@boost.smart_ptr//:smart_ptr",
+        "@boost.static_assert//:static_assert",
+        "@boost.tuple//:tuple",
+        "@boost.type_traits//:type_traits",
+        "@boost.utility//:utility",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..aa2197c
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,29 @@
+module(
+    name = "boost.python",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.align", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.bind", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.conversion", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.detail", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.foreach", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.function", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.graph", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.integer", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.iterator", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.lexical_cast", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.mpl", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.numeric.conversion", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.preprocessor", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.property_map", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.smart_ptr", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.tuple", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.utility", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
