diff --git dart/common/detail/SharedLibraryManager.cpp dart/common/detail/SharedLibraryManager.cpp
index c10e7dfb57..8991a61b92 100644
--- dart/common/detail/SharedLibraryManager.cpp
+++ dart/common/detail/SharedLibraryManager.cpp
@@ -32,6 +32,7 @@
 
 #include "dart/common/detail/SharedLibraryManager.hpp"
 
+#include <cassert>
 #include <fstream>
 
 #include "dart/common/Console.hpp"
diff --git dart/optimizer/GenericMultiObjectiveProblem.cpp dart/optimizer/GenericMultiObjectiveProblem.cpp
index 187200bc8a..3c999b6989 100644
--- dart/optimizer/GenericMultiObjectiveProblem.cpp
+++ dart/optimizer/GenericMultiObjectiveProblem.cpp
@@ -33,6 +33,7 @@
 #include "dart/optimizer/GenericMultiObjectiveProblem.hpp"
 
 #include <algorithm>
+#include <cassert>
 #include <limits>
 #include <numeric>
 
diff --git dart/optimizer/GradientDescentSolver.cpp dart/optimizer/GradientDescentSolver.cpp
index 8b1f4b13fd..b58a6fb879 100644
--- dart/optimizer/GradientDescentSolver.cpp
+++ dart/optimizer/GradientDescentSolver.cpp
@@ -32,6 +32,7 @@
 
 #include "dart/optimizer/GradientDescentSolver.hpp"
 
+#include <cassert>
 #include <iostream>
 
 #include "dart/common/Console.hpp"
diff --git dart/optimizer/MultiObjectiveProblem.cpp dart/optimizer/MultiObjectiveProblem.cpp
index 70cfbdbfa7..6bad6b272c 100644
--- dart/optimizer/MultiObjectiveProblem.cpp
+++ dart/optimizer/MultiObjectiveProblem.cpp
@@ -33,6 +33,7 @@
 #include "dart/optimizer/MultiObjectiveProblem.hpp"
 
 #include <algorithm>
+#include <cassert>
 #include <limits>
 #include <numeric>
 
diff --git dart/optimizer/Population.cpp dart/optimizer/Population.cpp
index ccffcad27e..6b9909f114 100644
--- dart/optimizer/Population.cpp
+++ dart/optimizer/Population.cpp
@@ -32,6 +32,8 @@
 
 #include "dart/optimizer/Population.hpp"
 
+#include <cassert>
+
 #include "dart/common/Console.hpp"
 #include "dart/math/Random.hpp"
 
diff --git dart/optimizer/Problem.cpp dart/optimizer/Problem.cpp
index 94d9794dd0..8d6a27a5c2 100644
--- dart/optimizer/Problem.cpp
+++ dart/optimizer/Problem.cpp
@@ -33,6 +33,7 @@
 #include "dart/optimizer/Problem.hpp"
 
 #include <algorithm>
+#include <cassert>
 #include <limits>
 
 #include "dart/common/Console.hpp"
