diff --git a/cpp/foxglove-websocket/include/foxglove/websocket/websocket_server.hpp b/cpp/foxglove-websocket/include/foxglove/websocket/websocket_server.hpp
index 95a5a67..fb8aac8 100644
--- a/cpp/foxglove-websocket/include/foxglove/websocket/websocket_server.hpp
+++ b/cpp/foxglove-websocket/include/foxglove/websocket/websocket_server.hpp
@@ -129,6 +129,9 @@ public:
   Server& operator=(Server&&) = delete;
 
   void start(const std::string& host, uint16_t port) override;
+  size_t run_for(std::chrono::nanoseconds duration) {
+    return _server.get_io_service().run_for(duration);
+  }
   void stop() override;
 
   std::vector<ChannelId> addChannels(const std::vector<ChannelWithoutId>& channels) override;
