diff --git a/MODULE.bazel b/MODULE.bazel
index cb75dd1..071898b 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -34,8 +34,6 @@ bazel_dep(
     version = "2.11.1.bzl.2"
 )
 
-#### DEV ONLY DEPENDENCIES BELOW HERE ####
-
 SUPPORTED_PYTHON_VERSIONS = [
   "3.12",
   "3.11",
@@ -46,7 +46,7 @@ SUPPORTED_PYTHON_VERSIONS = [
 
 DEFAULT_PYTHON = "3.11"
 
-python = use_extension("@rules_python//python/extensions:python.bzl", "python", dev_dependency=True)
+python = use_extension("@rules_python//python/extensions:python.bzl", "python")
 [
   python.toolchain(
     python_version = version,
@@ -60,7 +60,7 @@ use_repo(
     python = "python_versions",
 )
 
-pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip", dev_dependency=True)
+pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
 [
   pip.parse(
     hub_name = "pypi",
