--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -37,16 +37,19 @@
 config_setting(
     name = "tls_boringssl",
     flag_values = {":tls": "boringssl"},
+    visibility = ["//visibility:public"],
 )
 
 config_setting(
     name = "tls_openssl",
     flag_values = {":tls": "openssl"},
+    visibility = ["//visibility:public"],
 )
 
 config_setting(
     name = "tls_no",
     flag_values = {":tls": "no"},
+    visibility = ["//visibility:public"],
 )
 
 # OpenSSL on Windows pulls in Win32 system libraries (the cert store via
