diff --git gpu/surface/src/cuda/convex_hull.cu gpu/surface/src/cuda/convex_hull.cu
index 98e89cd70..104e3d84c 100644
--- gpu/surface/src/cuda/convex_hull.cu
+++ gpu/surface/src/cuda/convex_hull.cu
@@ -138,9 +138,9 @@ namespace pcl
 	    thrust::counting_iterator<int> cbeg(0);
 		thrust::counting_iterator<int> cend = cbeg + thrust::distance(beg, end);
 
-	    thrust::tuple<float, int> t = transform_reduce(
-		  make_zip_iterator(thrust::make_tuple(beg, cbeg)),
-		  make_zip_iterator(thrust::make_tuple(end, cend)),
+	    thrust::tuple<float, int> t = thrust::transform_reduce(
+		  thrust::make_zip_iterator(thrust::make_tuple(beg, cbeg)),
+		  thrust::make_zip_iterator(thrust::make_tuple(end, cend)),
 		  unop, init, binary);
 
 		return thrust::get<1>(t);
