diff --git filters/src/convolution.cpp filters/src/convolution.cpp
index fbdc0b44a..ad1d43181 100644
--- filters/src/convolution.cpp
+++ filters/src/convolution.cpp
@@ -36,6 +36,18 @@
  */
 
 #include <pcl/filters/convolution.h>
+#include <pcl/filters/impl/convolution.hpp>
+#include <pcl/point_types.h>
+
+// Explicitly instantiate the template classes for the types used
+namespace pcl
+{
+  namespace filters
+  {
+    template class Convolution<pcl::PointXYZRGB, pcl::PointXYZRGB>;
+    template class Convolution<pcl::RGB, pcl::RGB>;
+  }
+}
 
 namespace pcl {
 namespace filters {
