diff --git a/BUILD.bazel b/BUILD.bazel
index e5235a2a2..999bd9fa9 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -496,7 +496,7 @@ xnnpack_cc_library(
         ":microparams",
         ":unaligned",
         ":xnnpack_h",
-        "@FP16",
+        "@fp16",
     ],
 )
 
@@ -601,7 +601,7 @@ xnnpack_cc_library(
         ":common",
         ":math",
         ":microparams",
-        "@FP16",
+        "@fp16",
     ],
 )
 
@@ -652,8 +652,8 @@ xnnpack_cc_library(
         ":prefetch",
         ":tables",
         ":unaligned",
-        "@FP16",
-        "@FXdiv",
+        "@fp16",
+        "@fxdiv//:FXdiv",
     ],
 )
 
@@ -676,8 +676,8 @@ xnnpack_cc_library(
         ":prefetch",
         ":tables",
         ":unaligned",
-        "@FP16",
-        "@FXdiv",
+        "@fp16",
+        "@fxdiv//:FXdiv",
     ],
 )
 
@@ -703,8 +703,8 @@ xnnpack_cc_library(
         ":prefetch",
         ":tables",
         ":unaligned",
-        "@FP16",
-        "@FXdiv",
+        "@fp16",
+        "@fxdiv//:FXdiv",
     ],
 )
 
@@ -1306,7 +1306,7 @@ xnnpack_cc_library(
         ":prefetch",
         ":tables",
         ":unaligned",
-        "@FP16",
+        "@fp16",
     ],
 )
 
@@ -2801,7 +2801,7 @@ xnnpack_cc_library(
         ":prefetch",
         ":tables",
         ":unaligned",
-        "@FP16",
+        "@fp16",
     ],
 )
 
@@ -3226,8 +3226,8 @@ xnnpack_cc_library(
         ":math",
         ":operator_h",
         ":xnnpack_h",
-        "@FP16",
-        "@FXdiv",
+        "@fp16",
+        "@fxdiv//:FXdiv",
     ],
 )
 
@@ -3245,8 +3245,8 @@ xnnpack_cc_library(
         ":common",
         ":math",
         ":operator_h",
-        "@FP16",
-        "@FXdiv",
+        "@fp16",
+        "@fxdiv//:FXdiv",
     ],
 )
 
@@ -3261,7 +3261,7 @@ xnnpack_cc_library(
         ":math",
         ":operator_h",
         ":unaligned",
-        "@FP16",
+        "@fp16",
     ],
 )
 
@@ -3280,7 +3280,7 @@ xnnpack_cc_library(
         ":math",
         ":operator_h",
         ":unaligned",
-        "@FP16",
+        "@fp16",
     ],
 )
 
@@ -3510,7 +3510,7 @@ xnnpack_cc_library(
         ":params",
         ":post_operation",
         ":xnnpack_h",
-        "@FP16",
+        "@fp16",
         "@pthreadpool",
     ] + select({
         ":jit_enabled": [":jit"],
@@ -3557,7 +3557,7 @@ xnnpack_cc_library(
         ":params",
         ":post_operation_test_mode",
         ":xnnpack_h",
-        "@FP16",
+        "@fp16",
         "@pthreadpool",
     ],
 )
@@ -3672,7 +3672,7 @@ xnnpack_cc_library(
         ":params",
         ":requantization",
         ":xnnpack_h",
-        "@FP16",
+        "@fp16",
     ] + xnnpack_slinky_deps(),
 )
 
@@ -3710,7 +3710,7 @@ xnnpack_cc_library(
         ":params",
         ":requantization",
         ":xnnpack_h",
-        "@FP16",
+        "@fp16",
     ] + xnnpack_slinky_deps(),
 )
 
@@ -3927,7 +3927,7 @@ xnnpack_cc_library(
         ":XNNPACK",
         ":aligned_allocator",
         ":common",
-        "@FP16",
+        "@fp16",
     ],
 )
 
diff --git a/bench/BUILD.bazel b/bench/BUILD.bazel
index 349b2f311..1f745319a 100644
--- a/bench/BUILD.bazel
+++ b/bench/BUILD.bazel
@@ -31,7 +31,7 @@ MICROKERNEL_BENCHMARK_DEPS = [
     "//:microparams",
     "//:microparams_init",
     "//:xnnpack_h",
-    "@FP16",
+    "@fp16",
 ]
 
 OPERATOR_BENCHMARK_DEPS = [
@@ -40,7 +40,7 @@ OPERATOR_BENCHMARK_DEPS = [
     "//:aligned_allocator",
     "//:cache",
     "//:common",
-    "@FP16",
+    "@fp16",
 ]
 
 xnnpack_cc_library(
diff --git a/eval/BUILD.bazel b/eval/BUILD.bazel
index bcb63dc44..eea18fef2 100644
--- a/eval/BUILD.bazel
+++ b/eval/BUILD.bazel
@@ -18,7 +18,7 @@ ACCURACY_EVAL_DEPS = [
     "//:math",
     "//:math_stubs",
     "//bench:bench_utils",
-    "@FP16",
+    "@fp16",
     "@pthreadpool",
 ]
 
@@ -227,7 +227,7 @@ xnnpack_cc_library(
     srcs = ["math-evaluation-tester.cc"],
     hdrs = ["math-evaluation-tester.h"],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:aligned_allocator",
         "//:common",
         "//:math",
diff --git a/models/BUILD b/models/BUILD
index 9a0f34d4a..73a672345 100644
--- a/models/BUILD
+++ b/models/BUILD
@@ -16,7 +16,7 @@ xnnpack_cc_library(
     copts = xnnpack_std_cxxopts(),
     visibility = xnnpack_visibility(),
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK",
         "//:aligned_allocator",
         "//:cache",
@@ -199,7 +199,7 @@ xnnpack_cc_library(
     copts = xnnpack_std_cxxopts(),
     visibility = xnnpack_visibility(),
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK",
         "//:aligned_allocator",
         "//:cache",
@@ -269,7 +269,7 @@ xnnpack_cc_library(
     copts = xnnpack_std_cxxopts(),
     visibility = xnnpack_visibility(),
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK",
         "//:aligned_allocator",
         "//:cache",
@@ -340,7 +340,7 @@ xnnpack_cc_library(
     copts = xnnpack_std_cxxopts(),
     visibility = xnnpack_visibility(),
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK",
         "//:aligned_allocator",
         "//:cache",
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
index 87d777657..8c176b03e 100644
--- a/test/BUILD.bazel
+++ b/test/BUILD.bazel
@@ -27,7 +27,7 @@ MICROKERNEL_TEST_DEPS = [
     "//:requantization",
     "//:test_microkernels",
     "//:xnnpack_h",
-    "@FP16",
+    "@fp16",
 ]
 
 OPERATOR_TEST_DEPS = [
@@ -40,7 +40,7 @@ OPERATOR_TEST_DEPS = [
     "//:jit_test_mode",
     "//:normalization_test_mode",
     "@pthreadpool",
-    "@FP16",
+    "@fp16",
 ]
 
 ######################### Unit tests for micro-kernels #########################
@@ -3124,7 +3124,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3139,7 +3139,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_binary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3167,7 +3167,7 @@ xnnpack_unit_test(
         "argmax-pooling-2d.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:common",
@@ -3183,7 +3183,7 @@ xnnpack_unit_test(
         "average-pooling-2d.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:common",
@@ -3217,7 +3217,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3231,7 +3231,7 @@ xnnpack_unit_test(
         "batch-matrix-multiply.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:common",
@@ -3248,7 +3248,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3263,7 +3263,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3277,7 +3277,7 @@ xnnpack_unit_test(
         "concatenate2.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3291,7 +3291,7 @@ xnnpack_unit_test(
         "concatenate3.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3305,7 +3305,7 @@ xnnpack_unit_test(
         "concatenate4.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3319,7 +3319,7 @@ xnnpack_unit_test(
         "concatenate5.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3334,7 +3334,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3349,7 +3349,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":convolution_test_helpers",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:operator_utils",
@@ -3366,7 +3366,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3382,7 +3382,7 @@ xnnpack_unit_test(
     ],
     shard_count = 5,
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:operator_utils",
         "//:operators_test_mode",
@@ -3397,7 +3397,7 @@ xnnpack_unit_test(
         "depth-to-space-2d.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3412,7 +3412,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":convolution_test_helpers",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:operator_utils",
@@ -3429,7 +3429,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_binary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3444,7 +3444,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3458,7 +3458,7 @@ xnnpack_unit_test(
         "even-split2.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3472,7 +3472,7 @@ xnnpack_unit_test(
         "even-split3.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3486,7 +3486,7 @@ xnnpack_unit_test(
         "even-split4.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3501,7 +3501,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3515,7 +3515,7 @@ xnnpack_unit_test(
         "fully-connected.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:common",
@@ -3533,7 +3533,7 @@ xnnpack_unit_test(
         "global-average-pooling-1d.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:node_type",
@@ -3549,7 +3549,7 @@ xnnpack_unit_test(
         "global-average-pooling-2d.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:node_type",
@@ -3565,7 +3565,7 @@ xnnpack_unit_test(
         "global-sum-pooling-1d.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:node_type",
@@ -3580,7 +3580,7 @@ xnnpack_unit_test(
         "global-sum-pooling-2d.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:node_type",
@@ -3596,7 +3596,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3611,7 +3611,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3625,7 +3625,7 @@ xnnpack_unit_test(
         "max-pooling-2d.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operator_utils",
@@ -3642,7 +3642,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_binary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3657,7 +3657,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_binary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3672,7 +3672,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_binary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3687,7 +3687,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3701,7 +3701,7 @@ xnnpack_unit_test(
         "prelu.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3715,7 +3715,7 @@ xnnpack_unit_test(
         "rope.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3730,7 +3730,7 @@ xnnpack_unit_test(
         "scaled-dot-product-attention.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:common",
@@ -3747,7 +3747,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3762,7 +3762,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3777,7 +3777,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3792,7 +3792,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3807,7 +3807,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3822,7 +3822,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_binary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3837,7 +3837,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3851,7 +3851,7 @@ xnnpack_unit_test(
         "static-mean.cc",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:aligned_allocator",
         "//:common",
@@ -3868,7 +3868,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3895,7 +3895,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3910,7 +3910,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3925,7 +3925,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3940,7 +3940,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3955,7 +3955,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_binary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3970,7 +3970,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -3985,7 +3985,7 @@ xnnpack_unit_test(
     ],
     deps = [
         ":subgraph_unary_tester",
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:node_type",
         "//:operators_test_mode",
@@ -4086,7 +4086,7 @@ xnnpack_unit_test(
         "subgraph-tester.h",
     ],
     deps = [
-        "@FP16",
+        "@fp16",
         "//:XNNPACK_test_mode",
         "//:allocator",
         "//:node_type",
