diff --git a/BUILD.bazel b/BUILD.bazel
index 6d7ac3d..b013e4f 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -14,9 +14,9 @@ cc_library(
         "include/json/allocator.h",
         "include/json/assertions.h",
         "include/json/config.h",
-        "include/json/json_features.h",
         "include/json/forwards.h",
         "include/json/json.h",
+        "include/json/json_features.h",
         "include/json/reader.h",
         "include/json/value.h",
         "include/json/version.h",
