diff --git a/MODULE.bazel b/MODULE.bazel
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,5 +1,6 @@
 module(
     name = "riegeli",
+    version = "0.0.0-20260924-2de8c84",
     repo_name = "com_google_riegeli",
 )
 
@@ -23,7 +24,7 @@
 )
 bazel_dep(
     name = "brotli",
-    version = "1.2.0.bcr.1",
+    version = "1.2.0.bcr.3",
     repo_name = "org_brotli",
 )
 bazel_dep(
@@ -44,7 +45,7 @@
 )
 bazel_dep(
     name = "protobuf",
-    version = "34.1",
+    version = "35.0",
     repo_name = "com_google_protobuf",
 )
 bazel_dep(
