diff --git a/BUILD.bazel b/BUILD.bazel
index 97c9f3a..4dc23aa 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -114,7 +114,7 @@ cc_test(
     deps = [
         ":snappy",
         ":snappy-test",
-        "//third_party/benchmark:benchmark_main",
+        "@com_google_benchmark//:benchmark_main",
     ],
 )
 
@@ -127,7 +127,7 @@ cc_test(
     deps = [
         ":snappy",
         ":snappy-test",
-        "//third_party/googletest:gtest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
