===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,7 +1,7 @@
 """AppImage rules for Bazel."""
 
-module(name = "rules_appimage", version = "0.0.0")
+module(name = "rules_appimage", version = "1.10.0")
 
 bazel_dep(name = "bazel_skylib", version = "1.6.1")
 bazel_dep(name = "platforms", version = "0.0.8")
 bazel_dep(name = "rules_cc", version = "0.0.9")
