diff -uwr histlib.h
--- histlib.h	2024-08-12 21:10:25.000000000 +0200
+++ histlib.h	2026-02-05 00:12:51.418009274 +0100
@@ -22,6 +22,8 @@
 #if !defined (_HISTLIB_H_)
 #define _HISTLIB_H_
 
+#include "history.h"  /* for histdata_t */
+
 #if defined (HAVE_STRING_H)
 #  include <string.h>
 #else
diff -uwr posixtime.h
--- posixtime.h	2025-06-25 21:48:49.000000000 +0200
+++ posixtime.h	2026-02-05 00:01:08.600116824 +0100
@@ -21,6 +21,8 @@
 #ifndef _POSIXTIME_H_
 #define _POSIXTIME_H_
 
+#include "config.h"
+
 /* include this after config.h */
 /* Some systems require this, mostly for the definition of `struct timezone'.
    For example, Dynix/ptx has that definition in <time.h> rather than
diff -uwr rlprivate.h
--- rlprivate.h	2025-03-08 00:51:02.000000000 +0100
+++ rlprivate.h	2026-02-05 00:14:39.463576066 +0100
@@ -23,6 +23,7 @@
 #if !defined (_RL_PRIVATE_H_)
 #define _RL_PRIVATE_H_
 
+#include "readline.h"	/* for UNDO_LIST */
 #include "rlconf.h"	/* for VISIBLE_STATS */
 #include "rlstdc.h"
 #include "posixjmp.h"	/* defines procenv_t */
diff -uwr rltypedefs.h
--- rltypedefs.h	2023-07-31 21:02:30.000000000 +0200
+++ rltypedefs.h	2026-02-04 23:56:30.872133646 +0100
@@ -22,6 +22,8 @@
 #ifndef _RL_TYPEDEFS_H_
 #define _RL_TYPEDEFS_H_
 
+#include <stdio.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
diff -uwr xmalloc.h
--- xmalloc.h	2023-03-28 16:13:06.000000000 +0200
+++ xmalloc.h	2026-02-04 23:55:13.131207438 +0100
@@ -22,6 +22,8 @@
 #if !defined (_XMALLOC_H_)
 #define _XMALLOC_H_
 
+#include <stddef.h>
+
 #if defined (READLINE_LIBRARY)
 #  include "rlstdc.h"
 #else
