--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,7 +1,10 @@
-module(name = "com_github_mvukov_rules_ros2")
+module(
+    name = "com_github_mvukov_rules_ros2",
+    version = "0.0.0-20260118-78d85ff",
+)
 
 bazel_dep(name = "asio", version = "1.32.0")
 bazel_dep(name = "bazel_skylib", version = "1.9.0")
 bazel_dep(name = "boringssl", version = "0.20251124.0")
 bazel_dep(name = "curl", version = "8.11.0.bcr.4")
 bazel_dep(name = "eigen", version = "3.4.1.bcr.1")
