diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..34ed36a
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,19 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "units",
+    deps = [
+        "@boost.assert//:assert",
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.integer//:integer",
+        "@boost.io//:io",
+        "@boost.lambda//:lambda",
+        "@boost.math//:math",
+        "@boost.mpl//:mpl",
+        "@boost.preprocessor//:preprocessor",
+        "@boost.static_assert//:static_assert",
+        "@boost.type_traits//:type_traits",
+        "@boost.typeof//:typeof",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..05246bc
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,20 @@
+module(
+    name = "boost.units",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.assert", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.integer", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.io", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.lambda", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.math", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.mpl", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.preprocessor", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.typeof", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
