--- MODULE.bazel
+++ MODULE.bazel
@@ -1,7 +1,8 @@
 ## MODULE.bazel
 module(
     name = "gz-msgs",
-    repo_name = "org_gazebosim_gz-msgs",
+    version = "12.0.0-pre1",
+    compatibility_level = 11,
 )
 
 bazel_dep(name = "googletest", version = "1.14.0")
@@ -10,20 +11,6 @@
 bazel_dep(name = "rules_proto", version = "7.1.0")
 bazel_dep(name = "rules_python", version = "0.36.0")
 bazel_dep(name = "tinyxml2", version = "10.0.0")
-
-# Gazebo Dependencies
 bazel_dep(name = "rules_gazebo", version = "0.0.2")
-bazel_dep(name = "gz-math")
-bazel_dep(name = "gz-utils")
-
-archive_override(
-    module_name = "gz-math",
-    strip_prefix = "gz-math-main",
-    urls = ["https://github.com/gazebosim/gz-math/archive/refs/heads/main.tar.gz"],
-)
-
-archive_override(
-    module_name = "gz-utils",
-    strip_prefix = "gz-utils-main",
-    urls = ["https://github.com/gazebosim/gz-utils/archive/refs/heads/main.tar.gz"],
-)
+bazel_dep(name = "gz-utils", version = "4.0.0-pre1")
+bazel_dep(name = "gz-math", version = "9.0.0-pre1")
