--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,51 @@
+package(default_visibility = ["//visibility:public"])
+
+licenses(["notice"])
+
+lmdb_hdrs = glob(["libraries/liblmdb/*.h"])
+
+cc_library(
+    name = "lmdb",
+    srcs = [
+        "libraries/liblmdb/mdb.c",
+        "libraries/liblmdb/midl.c",
+    ],
+    hdrs = lmdb_hdrs,
+    linkopts = ["-pthread"],
+)
+
+cc_binary(
+    name = "mdb_copy",
+    srcs = ["libraries/liblmdb/mdb_copy.c"],
+    deps = [":lmdb"],
+    target_compatible_with = [
+        "@platforms//os:linux",
+    ],
+)
+
+cc_binary(
+    name = "mdb_dump",
+    srcs = ["libraries/liblmdb/mdb_dump.c"],
+    deps = [":lmdb"],
+    target_compatible_with = [
+        "@platforms//os:linux",
+    ],
+)
+
+cc_binary(
+    name = "mdb_load",
+    srcs = ["libraries/liblmdb/mdb_load.c"],
+    deps = [":lmdb"],
+    target_compatible_with = [
+        "@platforms//os:linux",
+    ],
+)
+
+cc_binary(
+    name = "mdb_stat",
+    srcs = ["libraries/liblmdb/mdb_stat.c"],
+    deps = [":lmdb"],
+    target_compatible_with = [
+        "@platforms//os:linux",
+    ],
+)
