From 5aa354a1f1ae6d56ea6d2a1c957ded7dc9885388 Mon Sep 17 00:00:00 2001
From: Mizux Seiha <mizux.dev@gmail.com>
Date: Thu, 9 Jul 2026 14:30:46 +0200
Subject: [PATCH 1/4] fix_src_soplex_fmt_hpp

---
 src/soplex/fmt.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/soplex/fmt.hpp b/src/soplex/fmt.hpp
index 54c7dca8..c93602a4 100644
--- a/src/soplex/fmt.hpp
+++ b/src/soplex/fmt.hpp
@@ -36,8 +36,8 @@
 
 /* to provide backwards compatibility use fmt of PaPILO <= 2.3 due to breaking changes in fmt 7 */
 #if !defined(SOPLEX_WITH_PAPILO) || PAPILO_VERSION_MAJOR > 2 || (PAPILO_VERSION_MAJOR == 2 && PAPILO_VERSION_MINOR > 3)
-#include "soplex/external/fmt/format.h"
-#include "soplex/external/fmt/ostream.h"
+#include "fmt/format.h"
+#include "fmt/ostream.h"
 #else
 #include "papilo/external/fmt/format.h"
 #include "papilo/external/fmt/ostream.h"
