From 773169d291f799f293007687c26cfeba72a60e19 Mon Sep 17 00:00:00 2001
From: Mizux Seiha <mizux.dev@gmail.com>
Date: Thu, 9 Jul 2026 14:31:04 +0200
Subject: [PATCH 2/4] fix_src_soplex_spxfileio_h

---
 src/soplex/spxfileio.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/soplex/spxfileio.h b/src/soplex/spxfileio.h
index 89edb9db..90bc7bee 100644
--- a/src/soplex/spxfileio.h
+++ b/src/soplex/spxfileio.h
@@ -41,7 +41,7 @@
  *-----------------------------------------------------------------------------
  */
 #ifdef SOPLEX_WITH_ZLIB
-#include "soplex/external/zstr/zstr.hpp"
+#include "zstr.hpp"
 #endif // WITH_GSZSTREAM
 
 namespace soplex
