diff --git a/bazel/BUILD b/bazel/BUILD
index eb9f3971..722232b0 100644
--- a/bazel/BUILD
+++ b/bazel/BUILD
@@ -1,4 +1,3 @@
-load("@envoy_toolshed//:macros.bzl", "json_data")
 load("@io_bazel_rules_go//proto:compiler.bzl", "go_proto_compiler")
 load(
     ":external_proto_deps.bzl",
@@ -29,17 +28,3 @@ go_proto_compiler(
     valid_archive = False,
     visibility = ["//visibility:public"],
 )
-
-json_data(
-    name = "repository_locations",
-    data = load_repository_locations_spec(REPOSITORY_LOCATIONS_SPEC),
-)
-
-json_data(
-    name = "external_proto_deps",
-    data = dict(
-        cc = EXTERNAL_PROTO_CC_BAZEL_DEP_MAP,
-        go = EXTERNAL_PROTO_GO_BAZEL_DEP_MAP,
-        imports = EXTERNAL_PROTO_IMPORT_BAZEL_DEP_MAP,
-    ),
-)
diff --git a/bazel/repositories.bzl b/bazel/repositories.bzl
index 95951c21..eb8abead 100644
--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -387,3 +387,5 @@ go_grpc_library(
     ],
 )
 """
+
+non_module_deps = module_extension(implementation = lambda ctx: api_dependencies(bzlmod = True))
