diff --git a/common/BUILD b/common/BUILD
index 095a7ca..2f3b06e 100644
--- a/common/BUILD
+++ b/common/BUILD
@@ -33,7 +33,7 @@ cc_library(
     ],
     copts = DEFAULT_COPTS,
     deps = [
-        "@com_github_fmtlib_fmt//:fmtlib",
+        "@com_github_fmtlib_fmt//:fmt",
         "@com_google_absl//absl/strings",
     ],
 )
diff --git a/demo/BUILD b/demo/BUILD
index 899d3fa..3abbd57 100644
--- a/demo/BUILD
+++ b/demo/BUILD
@@ -1,15 +1,6 @@
 load(
     "@com_google_absl//absl:copts/GENERATED_copts.bzl",
-    "ABSL_GCC_EXCEPTIONS_FLAGS",
     "ABSL_GCC_FLAGS",
-    "ABSL_GCC_TEST_FLAGS",
-    "ABSL_LLVM_EXCEPTIONS_FLAGS",
-    "ABSL_LLVM_FLAGS",
-    "ABSL_LLVM_TEST_FLAGS",
-    "ABSL_MSVC_EXCEPTIONS_FLAGS",
-    "ABSL_MSVC_FLAGS",
-    "ABSL_MSVC_LINKOPTS",
-    "ABSL_MSVC_TEST_FLAGS",
 )
 
 package(default_visibility = ["//visibility:public"])
diff --git a/demo/MODULE.bazel b/demo/MODULE.bazel
new file mode 100644
index 0000000..157593e
--- /dev/null
+++ b/demo/MODULE.bazel
@@ -0,0 +1,21 @@
+module(
+    name = "com_alibaba_hessian2_codec_demo",
+    version = "0.0.0-20231212-d5d1202",
+)
+
+bazel_dep(
+    name = "hessian2-codec",
+    version = "0.0.0-20231212-d5d1202",
+    repo_name = "com_alibaba_hessian2_codec",
+)
+bazel_dep(
+    name = "abseil-cpp",
+    version = "20220623.1",
+    repo_name = "com_google_absl",
+)
+# -- bazel_dep definitions -- #
+
+local_path_override(
+    module_name = "hessian2-codec",
+    path = "..",
+)
\ No newline at end of file
diff --git a/example/BUILD b/example/BUILD
index 0dbd652..a9f7bdd 100644
--- a/example/BUILD
+++ b/example/BUILD
@@ -1,4 +1,4 @@
-load("//:copts.bzl", "DEFAULT_COPTS", "TEST_COPTS")
+load("//:copts.bzl", "DEFAULT_COPTS")
 
 package(default_visibility = ["//visibility:public"])
 
