--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,28 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "avcodec",
+    srcs = glob([
+        "usr/local/lib/libavcodec.so*",
+        "usr/local/lib/libavformat.so*",
+        "usr/local/lib/libswresample.so*",
+    ]),
+    hdrs = glob(["usr/local/include/**/*.h"]),
+    includes = ["usr/local/include"],
+)
+
+cc_library(
+    name = "avutil",
+    srcs = glob(["usr/local/lib/libavutil.so*"]),
+    hdrs = glob(["usr/local/include/**/*.h"]),
+    includes = ["usr/local/include"],
+)
+
+cc_library(
+    name = "swscale",
+    srcs = glob(["usr/local/lib/libswscale.so*"]),
+    hdrs = glob(["usr/local/include/**/*.h"]),
+    includes = ["usr/local/include"],
+)
