diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..af0a960
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,26 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "process",
+    deps = [
+        "@boost//:algorithm",
+        "@boost//:asio",
+        "@boost//:config",
+        "@boost//:core",
+        "@boost//:filesystem",
+        "@boost//:fusion",
+        "@boost//:io",
+        "@boost//:iterator",
+        "@boost//:move",
+        "@boost//:optional",
+        "@boost//:system",
+        "@boost//:throw_exception",
+        "@boost//:tokenizer",
+        "@boost//:type_index",
+        "@boost//:type_traits",
+        "@boost//:utility",
+        "@boost//:winapi",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..536d754
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.process",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..ac6938d
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:process",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "process_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/process_test.cc b/bazel_test/process_test.cc
new file mode 100644
index 0000000..9cfe22b
--- /dev/null
+++ b/bazel_test/process_test.cc
@@ -0,0 +1,28 @@
+// work-around issue in Boost 1.77
+// https://github.com/boostorg/process/pull/215
+#include <algorithm>
+
+#include <boost/process.hpp>
+
+int main() 
+{
+    using namespace boost::process;
+
+    ipstream pipe_stream;
+    child c("echo hi", std_out > pipe_stream);
+
+    std::string line;
+    std::string all;
+
+    while (pipe_stream && std::getline(pipe_stream, line) && !line.empty())
+        all += line + '\n';
+
+    c.wait();
+
+    auto iter = all.find("hi");
+    if (iter != std::string::npos) {
+        return 0;
+    } else {
+        return -1;
+    }
+}
