===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,7 +1,7 @@
 module(
     name = "package_metadata",
-    version = "HEAD",  # Automatically updated by release pipeline.
+    version = "0.0.2",  # Automatically updated by release pipeline.
 )
 
 # This is a fundamental module that's depended on by virtually every bazel
 # module and **MUST NOT** have any dependencies.
