diff --git MODULE.bazel MODULE.bazel
--- MODULE.bazel
+++ MODULE.bazel
@@ -3,6 +3,7 @@
 module(
     name = "opencc",
     version = "1.4.1",
+    compatibility_level = 1,
 )
 
 bazel_dep(name = "aspect_bazel_lib", version = "2.22.5")
@@ -19,7 +20,6 @@
 bazel_dep(name = "rules_nodejs", version = "6.7.4")
 bazel_dep(name = "rules_python", version = "2.0.3")
 bazel_dep(name = "rules_shell", version = "0.8.0")
-bazel_dep(name = "rules_zig", version = "0.14.0")
 bazel_dep(name = "tclap", version = "1.2.5.bcr.1")
 bazel_dep(name = "toolchains_buildbuddy", version = "0.0.4", dev_dependency = True)
 
@@ -50,17 +50,11 @@
 
 buildbuddy = use_extension("@toolchains_buildbuddy//:extensions.bzl", "buildbuddy", dev_dependency = True)
 
-zig = use_extension("@rules_zig//zig:extensions.bzl", "zig")
-zig.toolchain(zig_version = "0.15.2")
-use_repo(zig, "zig_toolchains")
-
 register_toolchains(
     "@toolchains_buildbuddy//toolchains/cc:ubuntu_gcc_arm64",
     "@toolchains_buildbuddy//toolchains/cc:ubuntu_gcc_x86_64",
     dev_dependency = True,
 )
-register_toolchains("@rules_zig//zig/target:all")
-register_toolchains("@zig_toolchains//:all")
 
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
 python.toolchain(
