--- db/db_impl/db_impl.cc
+++ db/db_impl/db_impl.cc
@@ -5952,6 +5952,7 @@ Status DBImpl::IngestExternalFiles(

     num_running_ingest_file_ += static_cast<int>(num_cfs);
     TEST_SYNC_POINT("DBImpl::IngestExternalFile:AfterIncIngestFileCounter");
+    TEST_SYNC_POINT("DBImpl::IngestExternalFile:AfterIncIngestFileCounter:2");

     bool at_least_one_cf_need_flush = false;
     std::vector<bool> need_flush(num_cfs, false);

--- db/external_sst_file_basic_test.cc
+++ db/external_sst_file_basic_test.cc
@@ -265,8 +265,8 @@ TEST_F(ExternalSSTFileBasicTest, Basic) {
   SyncPoint::GetInstance()->LoadDependency({
       {"DBImpl::IngestExternalFile:AfterIncIngestFileCounter",
        "ExternalSSTFileBasicTest.LiveWriteStart"},
-      {"ExternalSSTFileBasicTest.LiveWriteStart",
-       "DBImpl::IngestExternalFiles:InstallSVForFirstCF:0"},
+      {"WriteThread::JoinBatchGroup:Wait",
+       "DBImpl::IngestExternalFile:AfterIncIngestFileCounter:2"},
   });
   SyncPoint::GetInstance()->EnableProcessing();
   PerfContext* write_thread_perf_context;
