diff --git a/folly/memory/UninitializedMemoryHacks.h b/folly/memory/UninitializedMemoryHacks.h
index 6644ca13f..a12077b71 100644
--- a/folly/memory/UninitializedMemoryHacks.h
+++ b/folly/memory/UninitializedMemoryHacks.h
@@ -21,6 +21,17 @@
 #include <type_traits>
 #include <vector>
 
+// Add missing includes for newer libc++ versions.
+#if defined(_LIBCPP_VECTOR)
+#ifndef _LIBCPP_COMPRESSED_PAIR
+#  if __has_include(<__memory/compressed_pair.h>)
+#    include <__memory/compressed_pair.h>
+#  elif __has_include(<__utility/compressed_pair.h>)
+#    include <__utility/compressed_pair.h>
+#  endif
+#endif
+#endif
+
 // On MSVC an incorrect <version> header get's picked up
 #if !defined(_MSC_VER) && __has_include(<version>)
 #include <version>
