diff --git a/folly/fibers/async/test/AsyncTest.cpp b/folly/fibers/async/test/AsyncTest.cpp
index 3683de427..dd195fdf8 100644
--- a/folly/fibers/async/test/AsyncTest.cpp
+++ b/folly/fibers/async/test/AsyncTest.cpp
@@ -92,7 +92,9 @@ TEST(AsyncTest, asyncAwait) {
 
 TEST(AsyncTest, asyncBaton) {
   folly::EventBase evb;
-  auto& fm = getFiberManager(evb);
+  folly::fibers::FiberManager::Options opts;
+  opts.stackSize = 64 * 1024;
+  auto& fm = getFiberManager(evb, opts);
   std::chrono::steady_clock::time_point start;
 
   EXPECT_NO_THROW(
@@ -135,7 +137,9 @@ TEST(AsyncTest, asyncPromise) {
 
 TEST(AsyncTest, asyncFuture) {
   folly::EventBase evb;
-  auto& fm = getFiberManager(evb);
+  folly::fibers::FiberManager::Options opts;
+  opts.stackSize = 64 * 1024;
+  auto& fm = getFiberManager(evb, opts);
 
   // Return format: Info about where continuation runs (thread_id,
   // in_fiber_loop, on_active_fiber)
