diff --git a/folly/io/async/test/TerminateCancellationTokenTest.cpp b/folly/io/async/test/TerminateCancellationTokenTest.cpp
index d3af92225..f32e1fa4f 100644
--- a/folly/io/async/test/TerminateCancellationTokenTest.cpp
+++ b/folly/io/async/test/TerminateCancellationTokenTest.cpp
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+#include <signal.h>
+
 #include <folly/io/async/TerminateCancellationToken.h>
 
 #include <chrono>
