diff --git a/folly/debugging/symbolizer/Dwarf.cpp b/folly/debugging/symbolizer/Dwarf.cpp
index d92a091c6..a4301121f 100644
--- a/folly/debugging/symbolizer/Dwarf.cpp
+++ b/folly/debugging/symbolizer/Dwarf.cpp
@@ -27,7 +27,7 @@
 
 #if FOLLY_HAVE_DWARF && FOLLY_HAVE_ELF
 
-#include <dwarf.h> // @manual=fbsource//third-party/libdwarf:dwarf
+#include <libdwarf/dwarf.h> // @manual=fbsource//third-party/libdwarf:dwarf
 
 namespace folly {
 namespace symbolizer {
