diff --git a/folly/io/async/test/IoTestTempFileUtil.cpp b/folly/io/async/test/IoTestTempFileUtil.cpp
index 0e70b7e69..0ca931837 100644
--- a/folly/io/async/test/IoTestTempFileUtil.cpp
+++ b/folly/io/async/test/IoTestTempFileUtil.cpp
@@ -49,7 +49,11 @@ TemporaryFile TempFileUtil::getTempFile(size_t size) {
     size -= n;
   }
 
+#ifdef __APPLE__
+  CHECK_EQ(::fsync(fd), 0);
+#else
   CHECK_EQ(::fdatasync(fd), 0);
+#endif
 
   // the file was opened with O_EXCL so we need to close to be able
   // to open it again
