--- /dev/null	2025-06-09 16:42:57.588000170 +0000
+++ include/fcl/export.h.in	2025-06-13 17:46:38.605756445 +0000
@@ -0,0 +1,42 @@
+
+#ifndef FCL_EXPORT_H
+#define FCL_EXPORT_H
+
+#ifdef FCL_STATIC_DEFINE
+#  define FCL_EXPORT
+#  define FCL_NO_EXPORT
+#else
+#  ifndef FCL_EXPORT
+#    ifdef fcl_EXPORTS
+        /* We are building this library */
+#      define FCL_EXPORT @FCL_EXPORT_VISIBILITY@
+#    else
+        /* We are using this library */
+#      define FCL_EXPORT @FCL_IMPORT_VISIBILITY@
+#    endif
+#  endif
+
+#  ifndef FCL_NO_EXPORT
+#    define FCL_NO_EXPORT @FCL_HIDDEN_VISIBILITY@
+#  endif
+#endif
+
+#ifndef FCL_DEPRECATED
+#  define FCL_DEPRECATED @FCL_DEPRECATED@
+#endif
+
+#ifndef FCL_DEPRECATED_EXPORT
+#  define FCL_DEPRECATED_EXPORT FCL_EXPORT FCL_DEPRECATED
+#endif
+
+#ifndef FCL_DEPRECATED_NO_EXPORT
+#  define FCL_DEPRECATED_NO_EXPORT FCL_NO_EXPORT FCL_DEPRECATED
+#endif
+
+#if 0 /* DEFINE_NO_DEPRECATED */
+#  ifndef FCL_NO_DEPRECATED
+#    define FCL_NO_DEPRECATED
+#  endif
+#endif
+
+#endif /* FCL_EXPORT_H */
