diff --git include/boost/thread/future.hpp include/boost/thread/future.hpp
index 00f504b9..d15d7ffa 100644
--- include/boost/thread/future.hpp
+++ include/boost/thread/future.hpp
@@ -4668,7 +4668,7 @@ namespace detail
       }
       run_it& operator=(BOOST_THREAD_RV_REF(run_it) x) BOOST_NOEXCEPT {
         if (this != &x) {
-          that_=x.that;
+          that_=x.that_;
           x.that_.reset();
         }
         return *this;
