diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..709a896
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,12 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "tti",
+    deps = [
+        "@boost.config//:config",
+        "@boost.function_types//:function_types",
+        "@boost.mpl//:mpl",
+        "@boost.preprocessor//:preprocessor",
+        "@boost.type_traits//:type_traits",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..7a19ff5
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,13 @@
+module(
+    name = "boost.tti",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.config", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.function_types", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.mpl", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.preprocessor", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..278129e
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost.tti//:tti",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "tti_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/tti_test.cc b/bazel_test/tti_test.cc
new file mode 100644
index 0000000..9ce166c
--- /dev/null
+++ b/bazel_test/tti_test.cc
@@ -0,0 +1,6 @@
+#include <boost/tti/tti.hpp>
+
+int
+main() {
+  return 0;
+}
