--- examples/test.py
+++ examples/test.py
@@ -144,7 +144,7 @@ def main(legacy_workspace: bool) -> int:
     We ony look for the return code of a command. If a command fails as expected we do not analyze if it fails for the
     correct reason. These kind of detailed testing is done in the aspect integration tests.
     """
-    bazel_binary = get_bazel_binary()
+    bazel_binary = "bazel"

     results = [
         execute_example(example=ex, bazel_bin=bazel_binary, legacy_workspace=legacy_workspace) for ex in EXAMPLES
