diff --git a/MODULE.bazel b/MODULE.bazel
index 8b71618..28a31d6 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -13,23 +13,16 @@
 # limitations under the License.
 
 module(
-    name = "proxy_wasm_rust_sdk",
-    version = "0.2.5-dev",
+    name = "proxy-wasm-rust-sdk",
+    repo_name = "proxy_wasm_rust_sdk",
+    version = "0.2.4-251205-5283e57",
 )
 
 # Regular dependencies (sorted alphabetically).
 bazel_dep(name = "bazel_features", version = "1.38.0")
 bazel_dep(name = "bazel_skylib", version = "1.8.2")
 bazel_dep(name = "rules_cc", version = "0.2.14")
-bazel_dep(name = "rules_rust", version = "0.67.0")
-single_version_override(
-    module_name = "rules_rust",
-    patch_strip = 1,
-    patches = [
-        "//bazel:rules_rust.patch",
-    ],
-    version = "0.67.0",
-)
+bazel_dep(name = "rules_rust", version = "0.68.1.envoy")
 
 # Configure Rust toolchain.
 rust = use_extension("@rules_rust//rust:extensions.bzl", "rust")
@@ -47,7 +40,7 @@ register_toolchains(
 )
 
 # Cargo dependencies.
-crates_deps = use_extension("//bazel:extensions.bzl", "crates_deps")
+crates_deps = use_extension(":extensions.bzl", "crates_deps")
 use_repo(
     crates_deps,
     "crates_vendor",
