--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,74 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+package(default_visibility = ["//visibility:public"])
+
+filegroup(
+    name = "srcs",
+    srcs = glob(
+        [
+            "CMakeLists.txt",
+            "CppUnitLite/**/*",
+            "cmake/**/*",
+            "cython/**/*",
+            "docker/**/*",
+            "examples/**/*",
+            "gtsam/**/*",
+            "gtsam.h",
+            "gtsam_extra.cmake.in",
+            "gtsam_unstable/**/*",
+            "makestats.sh",
+            "matlab/**/*",
+            "package.xml",
+            "tests/**/*",
+            "timing/**/*",
+            "wrap/**/*",
+        ],
+    ),
+)
+
+cmake(
+    name = "gtsam",
+    deps = [
+        "@boost.bind",
+        "@boost.array",
+        "@boost.filesystem",
+        "@boost.range",
+        "@boost.algorithm",
+        "@boost.circular_buffer",
+        "@boost.thread",
+        "@boost.function",
+        "@boost.container",
+        "@boost.foreach",
+        "@boost.lexical_cast",
+        "@boost.math",
+        "@boost.random",
+        "@boost.format",
+        "@boost.mpl",
+        "@boost.property_tree",
+        "@boost.program_options",
+        "@boost.signals2",
+        "@boost.tuple",
+        "@eigen",
+    ],
+    lib_source = ":srcs",
+    env = {
+        "CMAKE_BUILD_TYPE": "Release",
+    },
+    cache_entries = {
+        "GTSAM_USE_SYSTEM_EIGEN": "ON",
+        "CMAKE_BUILD_TYPE": "Release",
+        "GTSAM_BUILD_TESTS": "OFF",
+    },
+    out_shared_libs = [
+        "libgtsam.so.4.0.3",
+        "libgtsam.so.4",
+        "libgtsam.so",
+        "libgtsam_unstable.so",
+        "libgtsam_unstable.so.4.0.3",
+        "libgtsam_unstable.so.4",
+        "libmetis-gtsam.so",
+    ],
+    out_lib_dir = "lib",
+    out_include_dir = "include",
+)
+
