--- gtsam/3rdparty/Eigen/Eigen/src/Core/util/ForwardDeclarations.h
+++ gtsam/3rdparty/Eigen/Eigen/src/Core/util/ForwardDeclarations.h
@@ -48,10 +48,7 @@ template<typename Derived> struct EigenB
 template<typename Derived> class DenseBase;
 template<typename Derived> class PlainObjectBase;

-
-template<typename Derived,
-         int Level = internal::accessors_level<Derived>::value >
-class DenseCoeffsBase;
+template<typename Derived, int Level> class DenseCoeffsBase;

 template<typename _Scalar, int _Rows, int _Cols,
          int _Options = AutoAlign |
