--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,22 @@
+cc_library(
+    name = "gsl",
+    hdrs = [
+        "gsl/algorithm",
+        "gsl/assert",
+        "gsl/byte",
+        "gsl/gsl",
+        "gsl/gsl_algorithm",
+        "gsl/gsl_assert",
+        "gsl/gsl_byte",
+        "gsl/gsl_narrow",
+        "gsl/gsl_util",
+        "gsl/narrow",
+        "gsl/pointers",
+        "gsl/span",
+        "gsl/span_ext",
+        "gsl/string_span",
+        "gsl/util",
+    ],
+    includes = ["."],
+    visibility = ["//visibility:public"],
+)
