From cdb2658c7e4521cf7e287cd8a512cb2897230a76 Mon Sep 17 00:00:00 2001
From: Corentin Le Molgat <corentinl@google.com>
Date: Wed, 1 Jul 2026 10:23:23 +0200
Subject: [PATCH 3/5] add_scip_config.h

---
 scip/config.h | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)
 create mode 100644 scip/config.h

diff --git a/scip/config.h b/scip/config.h
new file mode 100644
index 0000000000..f0211b0748
--- /dev/null
+++ b/scip/config.h
@@ -0,0 +1,37 @@
+#ifndef __CONFIG_H__
+#define __CONFIG_H__
+
+#define SCIP_BUILD_TYPE "Release"
+#define SCIP_VERSION_MAJOR 10
+#define SCIP_VERSION_MINOR 0
+#define SCIP_VERSION_PATCH 2
+#define SCIP_VERSION_API 156
+/* #undef BMS_NOBLOCKMEM */
+/* #undef SCIP_NOBUFFERMEM */
+/* #undef WITH_DEBUG_SOLUTION */
+/* #undef SCIP_NO_SIGACTION */
+/* #undef SCIP_NO_STRTOK_R */
+/* #undef TPI_NONE */
+#define TPI_TNY
+/* #undef TPI_OMP */
+#define SCIP_THREADSAFE
+#define WITH_SCIPDEF
+/* #undef SCIP_WITH_LAPACK */
+/* #undef SCIP_WITH_PAPILO */
+#define SCIP_WITH_ZLIB
+/* #undef SCIP_WITH_READLINE */
+/* #undef SCIP_WITH_GMP */
+/* #undef SCIP_WITH_MPFR */
+/* #undef SCIP_WITH_QSOPTEX */
+/* #undef SCIP_WITH_LPSCHECK */
+/* #undef SCIP_WITH_ZIMPL */
+/* #undef SCIP_WITH_AMPL */
+#define SCIP_ROUNDING_FE
+/* #undef SCIP_ROUNDING_FP */
+/* #undef SCIP_ROUNDING_MS */
+/* #undef SCIP_WITH_EXACTSOLVE */
+/* #undef SCIP_WITH_EGLIB */
+/* #undef SCIP_WITH_BOOST */
+/* #undef SCIP_CHECK_STAGE */
+
+#endif
