diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 00000000..f9d52aab
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,41 @@
+# Description:
+# Soplex is a LP solver, a common tool in Operations Research.
+#
+# Copyright 2011 Google Inc.  All rights reserved.
+
+cc_library(
+    name = "libsoplex",
+    srcs = glob(
+        [
+            "src/soplex/*.cpp",
+            "src/soplex/external/fmt/*.cc",
+        ],
+        exclude = [
+            "src/git_hash.cpp",
+        ],
+    ),
+    copts = [
+        "-w",
+        "-fexceptions",
+        "-DGITHASH=\"NoGitInfo\"",
+    ],
+    features = ["-use_header_modules"],  # Incompatible with -fexceptions.
+    includes = [
+        "src",
+        "src/soplex/external",
+    ],
+    textual_hdrs = glob(["src/soplex/*.h"]) + glob(
+        [
+            "src/soplex/*.hpp",
+            "src/soplex/external/fmt/*.h"
+        ],
+    ) + [
+        "src/soplex.h",
+        "src/soplex.hpp",
+        "src/soplex/git_hash.cpp",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [
+#        "//third_party/boost/do_not_include_from_google3_only_third_party/boost",
+    ],
+)
