--- e2e/bzlmod/.bazelrc
+++ e2e/bzlmod/.bazelrc
@@ -1,4 +1,7 @@
-# NB: we don't import common bazelrc shared with e2e workspaces
+# NB: we don't import common bazelrc shared with e2e workspaces so that we can test flag settings
 
+build --enable_runfiles
 common --enable_bzlmod
-common --@aspect_rules_ts//ts:skipLibCheck=honor_tsconfig
+build --@aspect_rules_ts//ts:skipLibCheck=honor_tsconfig
+query --@aspect_rules_ts//ts:skipLibCheck=honor_tsconfig
+fetch --@aspect_rules_ts//ts:skipLibCheck=honor_tsconfig
