diff --git a/folly/debugging/symbolizer/DwarfUtil.cpp b/folly/debugging/symbolizer/DwarfUtil.cpp
index 8dba0e0c1..7c434bb03 100644
--- a/folly/debugging/symbolizer/DwarfUtil.cpp
+++ b/folly/debugging/symbolizer/DwarfUtil.cpp
@@ -27,7 +27,7 @@
 
 #if FOLLY_HAVE_DWARF && FOLLY_HAVE_ELF
 
-#include <dwarf.h> // @manual=fbsource//third-party/libdwarf:dwarf
+#include <libdwarf/dwarf.h> // @manual=fbsource//third-party/libdwarf:dwarf
 
 // We need a single dwarf5 tag, but may not be building against
 // a new enough libdwarf, so just define it ourselves.
