--- MODULE.bazel
+++ MODULE.bazel
@@ -0,0 +1,16 @@
+module(
+    name = "roboticslibrary.rl",
+    version = "0.0.0-20250723-5bbd773",
+    compatibility_level = 0,
+)
+
+bazel_dep(name = "rules_cc", version = "0.1.4")
+bazel_dep(name = "rules_license", version = "1.0.0")
+bazel_dep(name = "eigen", version = "3.4.0.bcr.3", repo_name = "com_gitlab_libeigen_eigen")
+bazel_dep(name = "boost.optional", version = "1.83.0")
+bazel_dep(name = "boost.algorithm", version = "1.83.0")
+bazel_dep(name = "boost.dynamic_bitset", version = "1.83.0")
+bazel_dep(name = "boost.iostreams", version = "1.83.0")
+bazel_dep(name = "boost.lexical_cast", version = "1.83.0")
+bazel_dep(name = "boost.graph", version = "1.83.0")
+bazel_dep(name = "abseil-cpp", version = "20250127.1")
