diff --git a/lib/curl_config.h b/lib/curl_config.h
new file mode 100755
--- /dev/null
+++ b/lib/curl_config.h
@@ -0,0 +1,34 @@
+// added for bzlmod
+// fallback incase other bazel dependencies define HAVE_CONFIG_H
+
+#ifdef _WIN32_WCE
+#	include "config-win32ce.h"
+#else
+#	ifdef _WIN32
+#		include "config-win32.h"
+#	endif
+#endif
+
+#ifdef macintosh
+#	include "config-mac.h"
+#endif
+
+#ifdef __riscos__
+#	include "config-riscos.h"
+#endif
+
+#ifdef __AMIGA__
+#	include "config-amigaos.h"
+#endif
+
+#ifdef __OS400__
+#	include "config-os400.h"
+#endif
+
+#ifdef __PLAN9__
+#	include "config-plan9.h"
+#endif
+
+#ifdef MSDOS
+#	include "config-dos.h"
+#endif
