diff --git a/proxygen/httpserver/tests/HTTPServerTest.cpp b/proxygen/httpserver/tests/HTTPServerTest.cpp
index 65fc9c7db..b68b1a3f7 100644
--- a/proxygen/httpserver/tests/HTTPServerTest.cpp
+++ b/proxygen/httpserver/tests/HTTPServerTest.cpp
@@ -37,7 +37,7 @@ using namespace CurlService;
 
 namespace {
 
-const std::string kTestDir = getContainingDirectory(XLOG_FILENAME).str();
+const std::string kTestDir = getContainingDirectory(BAZEL_TEST_RPATH).str();
 
 }
 
diff --git a/proxygen/lib/http/session/test/HTTPSessionAcceptorTest.cpp b/proxygen/lib/http/session/test/HTTPSessionAcceptorTest.cpp
index 4d54db6b0..d225bb261 100644
--- a/proxygen/lib/http/session/test/HTTPSessionAcceptorTest.cpp
+++ b/proxygen/lib/http/session/test/HTTPSessionAcceptorTest.cpp
@@ -27,7 +27,7 @@ using folly::test::MockAsyncSocket;
 
 namespace {
 
-const std::string kTestDir = getContainingDirectory(XLOG_FILENAME).str();
+const std::string kTestDir = getContainingDirectory(BAZEL_TEST_RPATH).str();
 
 }
 
