--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,75 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
+
+licenses(["reciprocal"])
+
+exports_files(["LICENSE"])
+
+expand_template(
+    name = "config_hpp",
+    template = "include/open62541pp/config.hpp.in",
+    substitutions = {
+        "@PROJECT_NAME@": "open62541pp",
+        "@VERSION@": "0.19.0",
+        "@PROJECT_VERSION_MAJOR@": "0",
+        "@PROJECT_VERSION_MINOR@": "19",
+        "@PROJECT_VERSION_PATCH@": "0",
+
+        "#cmakedefine UAPP_ENABLE_NODESETLOADER": "/* #undef UAPP_ENABLE_NODESETLOADER */",
+    },
+    out = "include_generated/open62541pp/config.hpp",
+)
+
+cc_library(
+    name = "open62541pp_config_header",
+    hdrs = [":config_hpp"],
+    includes = ["include_generated"],
+    visibility = ["//:__pkg__"],
+)
+
+cc_library(
+    name = "open62541pp",
+    hdrs = glob([
+        "include/open62541pp/**/*.hpp",
+        "include/open62541pp/detail/open62541/*.h",
+    ]),
+    srcs = [
+        "src/client.cpp",
+        "src/datatype.cpp",
+        "src/event.cpp",
+        "src/monitoreditem.cpp",
+        "src/node.cpp",
+        "src/plugin/accesscontrol.cpp",
+        "src/plugin/accesscontrol_default.cpp",
+        "src/plugin/create_certificate.cpp",
+        "src/plugin/log.cpp",
+        "src/plugin/nodesetloader.cpp",
+        "src/plugin/nodestore.cpp",
+        "src/server.cpp",
+        "src/services_attribute.cpp",
+        "src/services_method.cpp",
+        "src/services_monitoreditem.cpp",
+        "src/services_nodemanagement.cpp",
+        "src/services_subscription.cpp",
+        "src/services_view.cpp",
+        "src/session.cpp",
+        "src/string_utils.cpp",
+        "src/subscription.cpp",
+        "src/types.cpp",
+        "src/ua_types.cpp",
+    ],
+    deps = [
+        ":open62541pp_config_header",
+        "@open62541//:open62541",
+    ],
+    includes = ["include"],
+    defines = [
+        "UA_ENABLE_JSON_ENCODING=1",
+        "UA_ENABLE_PARSING=1",
+        "UA_ENABLE_TYPEDESCRIPTION=1",
+        "UA_ENABLE_DA=1",
+        "UA_ENABLE_SUBSCRIPTIONS=1",
+        "UA_ENABLE_ENCRYPTION_OPENSSL=1",
+    ],
+    visibility = ["//visibility:public"],
+)
