diff --git a/example/BUILD b/example/BUILD
new file mode 100644
index 0000000..7a08c2c
--- /dev/null
+++ b/example/BUILD
@@ -0,0 +1,17 @@
+cc_binary(
+    name = "rsa_create",
+    srcs = ["rsa-create.cpp"],
+    deps = [
+        "//include/jwt-cpp:jwt",
+        "//include/jwt-cpp/traits/nlohmann-json:specialization",
+    ],
+)
+
+cc_binary(
+    name = "rsa_verify",
+    srcs = ["rsa-verify.cpp"],
+    deps = [
+        "//include/jwt-cpp:jwt",
+        "//include/jwt-cpp/traits/nlohmann-json:specialization",
+    ],
+)
