diff --git a/include/jwt-cpp/BUILD b/include/jwt-cpp/BUILD
new file mode 100644
index 0000000..4c52f78
--- /dev/null
+++ b/include/jwt-cpp/BUILD
@@ -0,0 +1,27 @@
+cc_library(
+    name = "base64",
+    hdrs = ["base.h"],
+    include_prefix = "jwt-cpp",
+    copts = ["-fexceptions"],
+    features = ["-use_header_modules"],
+)
+
+cc_library(
+    name = "jwt",
+    hdrs = [
+        "jwt.h",
+    ],
+    copts = [
+        "-fexceptions",
+    ],
+    defines = [
+        "JWT_DISABLE_PICOJSON",
+    ],
+    features = ["-use_header_modules"],
+    include_prefix = "jwt-cpp",
+    deps = [
+        ":base64",
+        "@boringssl//:crypto",
+    ],
+    visibility = ["//visibility:public"],
+)
