diff --git a/include/jwt-cpp/traits/nlohmann-json/BUILD b/include/jwt-cpp/traits/nlohmann-json/BUILD
new file mode 100644
index 0000000..8779ccb
--- /dev/null
+++ b/include/jwt-cpp/traits/nlohmann-json/BUILD
@@ -0,0 +1,23 @@
+cc_library(
+    name = "traits",
+    hdrs = ["traits.h"],
+    copts = ["-fexceptions"],
+    features = ["-use_header_modules"],
+    include_prefix = "jwt-cpp/traits/nlohmann-json",
+    deps = [
+        "//include/jwt-cpp:jwt",
+        "@nlohmann_json//:json",
+    ],
+)
+
+cc_library(
+    name = "specialization",
+    hdrs = ["defaults.h"],
+    copts = ["-fexceptions"],
+    features = ["-use_header_modules"],
+    include_prefix = "jwt-cpp/traits/nlohmann-json",
+    deps = [
+        ":traits",
+    ],
+    visibility = ["//visibility:public"],
+)
