diff --git a/include/jwt-cpp/jwt.h b/include/jwt-cpp/jwt.h
index b2b998a..867cadb 100644
--- a/include/jwt-cpp/jwt.h
+++ b/include/jwt-cpp/jwt.h
@@ -9,17 +9,17 @@
 #endif
 
 #ifndef JWT_DISABLE_BASE64
-#include "base.h"
+#include "jwt-cpp/base.h"
 #endif
 
-#include <openssl/ec.h>
-#include <openssl/ecdsa.h>
-#include <openssl/err.h>
-#include <openssl/evp.h>
-#include <openssl/hmac.h>
-#include <openssl/pem.h>
-#include <openssl/rsa.h>
-#include <openssl/ssl.h>
+#include "openssl/ec.h"
+#include "openssl/ecdsa.h"
+#include "openssl/err.h"
+#include "openssl/evp.h"
+#include "openssl/hmac.h"
+#include "openssl/pem.h"
+#include "openssl/rsa.h"
+#include "openssl/is_boringssl.h"
 
 #include <algorithm>
 #include <chrono>
