diff --git a/include/jwt-cpp/traits/nlohmann-json/defaults.h b/include/jwt-cpp/traits/nlohmann-json/defaults.h
index c324075..7ed0a85 100644
--- a/include/jwt-cpp/traits/nlohmann-json/defaults.h
+++ b/include/jwt-cpp/traits/nlohmann-json/defaults.h
@@ -5,7 +5,7 @@
 #define JWT_DISABLE_PICOJSON
 #endif
 
-#include "traits.h"
+#include "jwt-cpp/traits/nlohmann-json/traits.h"
 
 namespace jwt {
 	/**
