--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,25 @@
+load("@rules_license//rules:license.bzl", "license")
+
+package(
+    default_applicable_licenses = [":license"],
+)
+
+license(
+    name = "license",
+    package_name = "remotery",
+)
+
+licenses(["notice"])
+
+exports_files(["LICENSE"])
+
+cc_library(
+    name = "remotery",
+    srcs = ["lib/Remotery.c"],
+    hdrs = ["lib/Remotery.h"],
+    includes = ["lib"],
+    copts = [
+        "-w",
+    ],
+    visibility = ["//visibility:public"],
+)
