From 51c98cb671084be03431d248dd54eaa2de066c09 Mon Sep 17 00:00:00 2001
From: Ivo List <ilist@google.com>
Date: Fri, 29 Dec 2023 14:47:06 +0000
Subject: [PATCH 4/7] Relative labels

---
 protobuf.bzl      | 4 ++--
 protobuf_deps.bzl | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/protobuf.bzl b/protobuf.bzl
index d96eeb4c9..e7b7a14eb 100644
--- a/protobuf.bzl
+++ b/protobuf.bzl
@@ -493,8 +493,8 @@ def internal_ruby_proto_library(
         srcs = [],
         deps = [],
         includes = ["."],
-        default_runtime = "@com_google_protobuf//ruby:protobuf",
-        protoc = "@com_google_protobuf//:protoc",
+        default_runtime = Label("//ruby:protobuf"),
+        protoc = Label("//:protoc"),
         testonly = None,
         visibility = ["//visibility:public"],
         **kwargs):
diff --git a/protobuf_deps.bzl b/protobuf_deps.bzl
index 37da217ee..7d825b4c7 100644
--- a/protobuf_deps.bzl
+++ b/protobuf_deps.bzl
@@ -153,5 +153,5 @@ def protobuf_deps():
             repo = "https://github.com/protocolbuffers/upb",
             commit = "61a97efa24a5ce01fb8cc73c9d1e6e7060f8ea98",
             sha256 = "0c57aac04d62eeabe097513593c800a99aa3f5d8ac1e7871c7afadbe1d39851a",
-            patches = ["@com_google_protobuf//build_defs:upb.patch"],
+            patches = ["//build_defs:upb.patch"],
         )
-- 
2.42.1

