--- a/python/BUILD.bazel
+++ b/python/BUILD.bazel
@@ -77,7 +77,7 @@ cc_binary(
     ],
     deps = select({
         "//conditions:default": [],
-        ":use_fast_cpp_protos": ["//external:python_headers"],
+        ":use_fast_cpp_protos": ["//third_party:python_headers"],
     }),
 )
 
@@ -116,7 +116,7 @@ cc_binary(
         "//src/google/protobuf:descriptor_legacy",
     ] + select({
         "//conditions:default": [],
-        ":use_fast_cpp_protos": ["//external:python_headers"],
+        ":use_fast_cpp_protos": ["//third_party:python_headers"],
     }),
 )
 
@@ -379,7 +379,7 @@ cc_library(
     hdrs = ["google/protobuf/proto_api.h"],
     visibility = ["//visibility:public"],
     deps = [
-        "//external:python_headers",
+        "//third_party:python_headers",
     ],
 )
 
diff --git a/third_party/BUILD.bazel b/third_party/BUILD.bazel
index d12b3e37f..2abc1e7ec 100644
--- a/third_party/BUILD.bazel
+++ b/third_party/BUILD.bazel
@@ -2,3 +2,9 @@ exports_files([
     "BUILD.bazel",
     "zlib.BUILD",
 ])
+
+alias(
+    name = "python_headers",
+    actual = "@rules_python//python/cc:current_py_cc_headers",
+    visibility = ["//visibility:public"],
+)
