diff --git a/src/rl/math/Matrix.h b/src/rl/math/Matrix.h
--- src/rl/math/Matrix.h
+++ src/rl/math/Matrix.h
@@ -46,7 +46,8 @@ namespace rl
 {
 	namespace math
 	{
-		typedef ::Eigen::Matrix<Real, ::Eigen::Dynamic, ::Eigen::Dynamic> Matrix;
+		// Updated template parameters to contain max sizes for hard real time safe construction.
+		typedef ::Eigen::Matrix<Real, ::Eigen::Dynamic, ::Eigen::Dynamic, ::Eigen::DontAlign, 16, 16> Matrix;
 		
 		typedef ::Eigen::Matrix<Real, 2, 2> Matrix22;
 		
diff --git a/src/rl/math/Vector.h b/src/rl/math/Vector.h
--- src/rl/math/Vector.h
+++ src/rl/math/Vector.h
@@ -46,7 +46,8 @@ namespace rl
 {
 	namespace math
 	{
-		typedef ::Eigen::Matrix<Real, ::Eigen::Dynamic, 1> Vector;
+		// Updated template parameters to contain max sizes for hard real time safe construction.
+		typedef ::Eigen::Matrix<Real, ::Eigen::Dynamic, 1, ::Eigen::DontAlign, 16, 1> Vector;
 		
 		typedef ::Eigen::Matrix<Real, 2, 1> Vector2;
 		
