--- a/BUILD.bazel	2026-03-18 15:10:47.292491743 +0000
+++ b/BUILD.bazel	2026-03-19 11:09:02.801837516 +0000
@@ -1,9 +1,20 @@
 load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
+load("@rules_license//rules:license.bzl", "license")
+
+package(
+    default_applicable_licenses = [":license"],
+)

 licenses(["notice"])

 exports_files(["LICENSE"])

+license(
+    name = "license",
+    license_kinds = ["@rules_license//licenses/spdx:BSD-2-Clause"],
+    license_text = "LICENSE",
+)
+
 ############################## pthreadpool library #############################

 INTERNAL_HDRS = [
@@ -246,34 +236,38 @@

 config_setting(
     name = "android_x86",
-    values = {
-        "crosstool_top": "//external:android/crosstool",
-        "cpu": "x86",
-    },
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:x86_32",
+    ],
+    visibility = ["//visibility:public"],
 )

 config_setting(
     name = "android_x86_64",
-    values = {
-        "crosstool_top": "//external:android/crosstool",
-        "cpu": "x86_64",
-    },
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:x86_64",
+    ],
+    visibility = ["//visibility:public"],
 )

 config_setting(
     name = "android_armv7",
-    values = {
-        "crosstool_top": "//external:android/crosstool",
-        "cpu": "armeabi-v7a",
-    },
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:armv7",
+    ],
+    visibility = ["//visibility:public"],
 )

 config_setting(
     name = "android_arm64",
-    values = {
-        "crosstool_top": "//external:android/crosstool",
-        "cpu": "arm64-v8a",
-    },
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:arm64",
+    ],
+    visibility = ["//visibility:public"],
 )

 # Note: we need to individually match x86 and x86-64 macOS rather than use
@@ -281,114 +275,118 @@
 # "macos" even when building on Linux!
 config_setting(
     name = "macos_x86",
-    values = {
-        "apple_platform_type": "macos",
-        "cpu": "darwin",
-    },
+    constraint_values = [
+        "@platforms//os:macos",
+        "@platforms//cpu:x86_32",
+    ],
 )

 config_setting(
     name = "macos_x86_64",
-    values = {
-        "apple_platform_type": "macos",
-        "cpu": "darwin_x86_64",
-    },
+    constraint_values = [
+        "@platforms//os:macos",
+        "@platforms//cpu:x86_64",
+    ],
 )

 config_setting(
     name = "macos_arm64",
-    values = {
-        "apple_platform_type": "macos",
-        "cpu": "darwin_arm64",
-    },
+    constraint_values = [
+        "@platforms//cpu:arm64",
+        "@platforms//os:macos",
+    ],
 )

 config_setting(
     name = "ios",
-    values = {
-        "apple_platform_type": "ios",
-    },
+    constraint_values = [
+        "@platforms//os:ios",
+    ],
 )

 config_setting(
     name = "ios_x86",
-    values = {
-        "apple_platform_type": "ios",
-        "cpu": "ios_i386",
-    },
+    constraint_values = [
+        "@platforms//cpu:x86_32",
+        "@platforms//os:ios",
+    ],
 )

 config_setting(
     name = "ios_x86_64",
-    values = {
-        "apple_platform_type": "ios",
-        "cpu": "ios_x86_64",
-    },
+    constraint_values = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:ios",
+    ],
 )

 config_setting(
     name = "watchos",
-    values = {
-        "apple_platform_type": "watchos",
-    },
+    constraint_values = [
+        "@platforms//os:watchos",
+    ],
 )

 config_setting(
     name = "watchos_x86",
-    values = {
-        "apple_platform_type": "watchos",
-        "cpu": "watchos_i386",
-    },
+    constraint_values = [
+        "@platforms//os:watchos",
+        "@platforms//cpu:x86_32",
+    ],
 )

 config_setting(
     name = "watchos_x86_64",
-    values = {
-        "apple_platform_type": "watchos",
-        "cpu": "watchos_x86_64",
-    },
+    constraint_values = [
+        "@platforms//os:watchos",
+        "@platforms//cpu:x86_64",
+    ],
 )

 config_setting(
     name = "tvos",
-    values = {
-        "apple_platform_type": "tvos",
-    },
+    constraint_values = [
+        "@platforms//os:tvos",
+    ],
 )

 config_setting(
     name = "tvos_x86_64",
-    values = {
-        "apple_platform_type": "tvos",
-        "cpu": "tvos_x86_64",
-    },
+    constraint_values = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:tvos",
+    ],
 )

 config_setting(
     name = "windows_x86_64",
-    values = {
-        "cpu": "x64_windows",
-    },
+    constraint_values = [
+        "@platforms//os:windows",
+        "@platforms//cpu:x86_64",
+    ],
 )

 config_setting(
     name = "windows_arm64",
-    values = {
-        "cpu": "arm64_windows",
-    },
+    constraint_values = [
+        "@platforms//os:windows",
+        "@platforms//cpu:arm64",
+    ],
 )

 config_setting(
     name = "emscripten",
-    values = {
-        "crosstool_top": "//toolchain:emscripten",
-    },
+    constraint_values = [
+        "@platforms//os:emscripten",
+    ],
 )

 config_setting(
     name = "emscripten_with_threads",
+    constraint_values = [
+        "@platforms//os:emscripten",
+    ],
     values = {
-        "crosstool_top": "//toolchain:emscripten",
-        "copt": "-pthread",
+        "features": "use_pthreads",
     },
 )
