--- a/BUILD.bazel	2026-03-18 15:13:27.590192282 +0000
+++ b/BUILD.bazel	2026-03-18 15:17:27.676747537 +0000
@@ -1,9 +1,20 @@
 load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
+load("@rules_license//rules:license.bzl", "license")
+
+package(
+    default_applicable_licenses = [":license"],
+)
 
 licenses(["notice"])
 
 exports_files(["LICENSE"])
 
+license(
+    name = "license",
+    license_kinds = ["@rules_license//licenses/spdx:BSD-2-Clause"],
+    license_text = "LICENSE",
+)
+
 ############################## pthreadpool library #############################
 
 INTERNAL_HDRS = [
@@ -24,13 +35,30 @@
 
 PTHREADS_IMPL_SRCS = PORTABLE_SRCS + ["src/pthreads.c"]
 
+GCD_IMPL_SRCS = PORTABLE_SRCS + ["src/gcd.c"]
+
+WINDOWS_IMPL_SRCS = PORTABLE_SRCS + ["src/windows.c"]
+
 SHIM_IMPL_SRCS = ["src/shim.c"]
 
 cc_library(
     name = "pthreadpool",
     srcs = select({
-        ":emscripten": SHIM_IMPL_SRCS,
-        "//conditions:default": PTHREADS_IMPL_SRCS,
+        ":pthreadpool_sync_primitive_explicit_condvar": INTERNAL_HDRS + PTHREADS_IMPL_SRCS,
+        ":pthreadpool_sync_primitive_explicit_futex": INTERNAL_HDRS + PTHREADS_IMPL_SRCS,
+        ":pthreadpool_sync_primitive_explicit_gcd": INTERNAL_HDRS + GCD_IMPL_SRCS,
+        ":pthreadpool_sync_primitive_explicit_event": INTERNAL_HDRS + WINDOWS_IMPL_SRCS,
+        ":emscripten_with_threads": INTERNAL_HDRS + PTHREADS_IMPL_SRCS,
+        ":emscripten": INTERNAL_HDRS + SHIM_IMPL_SRCS,
+        ":macos_x86": INTERNAL_HDRS + GCD_IMPL_SRCS,
+        ":macos_x86_64": INTERNAL_HDRS + GCD_IMPL_SRCS,
+        ":macos_arm64": INTERNAL_HDRS + GCD_IMPL_SRCS,
+        ":ios": INTERNAL_HDRS + GCD_IMPL_SRCS,
+        ":watchos": INTERNAL_HDRS + GCD_IMPL_SRCS,
+        ":tvos": INTERNAL_HDRS + GCD_IMPL_SRCS,
+        ":windows_x86_64": INTERNAL_HDRS + WINDOWS_IMPL_SRCS,
+        ":windows_arm64": INTERNAL_HDRS + WINDOWS_IMPL_SRCS,
+        "//conditions:default": INTERNAL_HDRS + PTHREADS_IMPL_SRCS,
     }) + select({
         ":linux_x86_64": ARCH_SPECIFIC_SRCS,
         ":android_x86": ARCH_SPECIFIC_SRCS,
@@ -47,50 +75,68 @@
         ":tvos_x86_64": ARCH_SPECIFIC_SRCS,
         "//conditions:default": [],
     }),
-    hdrs = [
-        "include/pthreadpool.h",
-    ],
     copts = [
-        "-std=c11",
+        "-std=gnu11",
     ] + select({
         ":optimized_build": ["-O2"],
         "//conditions:default": [],
-    }),
-    defines = select({
-        ":linux_arm": ["PTHREADPOOL_USE_CPUINFO=1"],
-        ":linux_armeabi": ["PTHREADPOOL_USE_CPUINFO=1"],
-        ":linux_armhf": ["PTHREADPOOL_USE_CPUINFO=1"],
-        ":linux_armv7a": ["PTHREADPOOL_USE_CPUINFO=1"],
-        ":linux_aarch64": ["PTHREADPOOL_USE_CPUINFO=1"],
-        ":android_armv7": ["PTHREADPOOL_USE_CPUINFO=1"],
-        ":android_arm64": ["PTHREADPOOL_USE_CPUINFO=1"],
-        "//conditions:default": ["PTHREADPOOL_USE_CPUINFO=0"],
+    }) + select({
+        ":linux_arm": ["-DPTHREADPOOL_USE_CPUINFO=1"],
+        ":linux_armeabi": ["-DPTHREADPOOL_USE_CPUINFO=1"],
+        ":linux_armhf": ["-DPTHREADPOOL_USE_CPUINFO=1"],
+        ":linux_armv7a": ["-DPTHREADPOOL_USE_CPUINFO=1"],
+        ":linux_aarch64": ["-DPTHREADPOOL_USE_CPUINFO=1"],
+        ":android_armv7": ["-DPTHREADPOOL_USE_CPUINFO=1"],
+        ":android_arm64": ["-DPTHREADPOOL_USE_CPUINFO=1"],
+        "//conditions:default": ["-DPTHREADPOOL_USE_CPUINFO=0"],
     }) + select({
         ":pthreadpool_sync_primitive_explicit_condvar": [
-            "PTHREADPOOL_USE_CONDVAR=1",
-            "PTHREADPOOL_USE_FUTEX=0",
+            "-DPTHREADPOOL_USE_CONDVAR=1",
+            "-DPTHREADPOOL_USE_FUTEX=0",
+            "-DPTHREADPOOL_USE_GCD=0",
+            "-DPTHREADPOOL_USE_EVENT=0",
         ],
         ":pthreadpool_sync_primitive_explicit_futex": [
-            "PTHREADPOOL_USE_CONDVAR=0",
-            "PTHREADPOOL_USE_FUTEX=1",
+            "-DPTHREADPOOL_USE_CONDVAR=0",
+            "-DPTHREADPOOL_USE_FUTEX=1",
+            "-DPTHREADPOOL_USE_GCD=0",
+            "-DPTHREADPOOL_USE_EVENT=0",
+        ],
+        ":pthreadpool_sync_primitive_explicit_gcd": [
+            "-DPTHREADPOOL_USE_CONDVAR=0",
+            "-DPTHREADPOOL_USE_FUTEX=0",
+            "-DPTHREADPOOL_USE_GCD=1",
+            "-DPTHREADPOOL_USE_EVENT=0",
+        ],
+        ":pthreadpool_sync_primitive_explicit_event": [
+            "-DPTHREADPOOL_USE_CONDVAR=0",
+            "-DPTHREADPOOL_USE_FUTEX=0",
+            "-DPTHREADPOOL_USE_GCD=0",
+            "-DPTHREADPOOL_USE_EVENT=1",
         ],
         "//conditions:default": [],
     }) + select({
-        ":linux_x86_64": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":android_x86": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":android_x86_64": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":windows_x86_64": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":windows_arm64": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":macos_x86": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":macos_x86_64": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":macos_arm64": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":ios_x86": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":ios_x86_64": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":watchos_x86": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":watchos_x86_64": ["PTHREADPOOL_USE_FASTPATH=1"],
-        ":tvos_x86_64": ["PTHREADPOOL_USE_FASTPATH=1"],
-        "//conditions:default": ["PTHREADPOOL_USE_FASTPATH=0"],
-    }) + ["PTHREADPOOL_NO_DEPRECATED_API"],
+        ":linux_x86_64": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":android_x86": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":android_x86_64": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":windows_x86_64": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":windows_arm64": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":macos_x86": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":macos_x86_64": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":macos_arm64": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":ios_x86": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":ios_x86_64": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":watchos_x86": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":watchos_x86_64": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        ":tvos_x86_64": ["-DPTHREADPOOL_USE_FASTPATH=1"],
+        "//conditions:default": ["-DPTHREADPOOL_USE_FASTPATH=0"],
+    }),
+    hdrs = [
+        "include/pthreadpool.h",
+    ],
+    defines = [
+        "PTHREADPOOL_NO_DEPRECATED_API",
+    ],
     includes = [
         "include",
     ],
@@ -102,13 +148,8 @@
         "//conditions:default": [],
     }),
     strip_include_prefix = "include",
-    # These headers are listed here to prevent them from being parsed. G3 toolchain runs the
-    # compiler in C++ mode on this step and can not complete correctly.
-    # These are NOT public headers and should not be included from other targets.
-    textual_hdrs = INTERNAL_HDRS,
-    visibility = ["//visibility:public"],
     deps = [
-        "@FXdiv",
+        "@FXdiv//:FXdiv",
     ] + select({
         ":linux_arm": ["@cpuinfo"],
         ":linux_armeabi": ["@cpuinfo"],
@@ -119,6 +160,7 @@
         ":android_arm64": ["@cpuinfo"],
         "//conditions:default": [],
     }),
+    visibility = ["//visibility:public"],
 )
 
 ################################## Unit tests ##################################
@@ -207,6 +249,18 @@
     define_values = {"pthreadpool_sync_primitive": "futex"},
 )
 
+# Synchronize workers using Grand Central Dispatch.
+config_setting(
+    name = "pthreadpool_sync_primitive_explicit_gcd",
+    define_values = {"pthreadpool_sync_primitive": "gcd"},
+)
+
+# Synchronize workers using WinAPI event.
+config_setting(
+    name = "pthreadpool_sync_primitive_explicit_event",
+    define_values = {"pthreadpool_sync_primitive": "event"},
+)
+
 config_setting(
     name = "optimized_build",
     values = {
@@ -216,64 +270,80 @@
 
 config_setting(
     name = "linux_x86_64",
-    values = {"cpu": "k8"},
+    constraint_values = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:linux",
+    ],
 )
 
 config_setting(
     name = "linux_arm",
-    values = {"cpu": "arm"},
+    constraint_values = [
+        "@platforms//cpu:arm",
+        "@platforms//os:linux",
+    ],
 )
 
-config_setting(
+alias(
     name = "linux_armeabi",
-    values = {"cpu": "armeabi"},
+    actual = ":linux_arm",
 )
 
-config_setting(
+alias(
     name = "linux_armhf",
-    values = {"cpu": "armhf"},
+    actual = ":linux_arm",
 )
 
 config_setting(
     name = "linux_armv7a",
-    values = {"cpu": "armv7a"},
+    constraint_values = [
+        "@platforms//cpu:armv7",
+        "@platforms//os:linux",
+    ],
 )
 
 config_setting(
     name = "linux_aarch64",
-    values = {"cpu": "aarch64"},
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:aarch64",
+    ],
 )
 
 config_setting(
     name = "android_x86",
-    values = {
-        "crosstool_top": "//external:android/crosstool",
-        "cpu": "x86",
-    },
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:x86_32",
+    ],
+    visibility = ["//visibility:public"],
 )
 
 config_setting(
     name = "android_x86_64",
-    values = {
-        "crosstool_top": "//external:android/crosstool",
-        "cpu": "x86_64",
-    },
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:x86_64",
+    ],
+    visibility = ["//visibility:public"],
 )
 
 config_setting(
     name = "android_armv7",
-    values = {
-        "crosstool_top": "//external:android/crosstool",
-        "cpu": "armeabi-v7a",
-    },
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:armv7",
+    ],
+    visibility = ["//visibility:public"],
 )
 
 config_setting(
     name = "android_arm64",
-    values = {
-        "crosstool_top": "//external:android/crosstool",
-        "cpu": "arm64-v8a",
-    },
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:arm64",
+    ],
+    visibility = ["//visibility:public"],
 )
 
 # Note: we need to individually match x86 and x86-64 macOS rather than use
@@ -281,114 +351,118 @@
 # "macos" even when building on Linux!
 config_setting(
     name = "macos_x86",
-    values = {
-        "apple_platform_type": "macos",
-        "cpu": "darwin",
-    },
+    constraint_values = [
+        "@platforms//os:macos",
+        "@platforms//cpu:x86_32",
+    ],
 )
 
 config_setting(
     name = "macos_x86_64",
-    values = {
-        "apple_platform_type": "macos",
-        "cpu": "darwin_x86_64",
-    },
+    constraint_values = [
+        "@platforms//os:macos",
+        "@platforms//cpu:x86_64",
+    ],
 )
 
 config_setting(
     name = "macos_arm64",
-    values = {
-        "apple_platform_type": "macos",
-        "cpu": "darwin_arm64",
-    },
+    constraint_values = [
+        "@platforms//cpu:arm64",
+        "@platforms//os:macos",
+    ],
 )
 
 config_setting(
     name = "ios",
-    values = {
-        "apple_platform_type": "ios",
-    },
+    constraint_values = [
+        "@platforms//os:ios",
+    ],
 )
 
 config_setting(
     name = "ios_x86",
-    values = {
-        "apple_platform_type": "ios",
-        "cpu": "ios_i386",
-    },
+    constraint_values = [
+        "@platforms//cpu:x86_32",
+        "@platforms//os:ios",
+    ],
 )
 
 config_setting(
     name = "ios_x86_64",
-    values = {
-        "apple_platform_type": "ios",
-        "cpu": "ios_x86_64",
-    },
+    constraint_values = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:ios",
+    ],
 )
 
 config_setting(
     name = "watchos",
-    values = {
-        "apple_platform_type": "watchos",
-    },
+    constraint_values = [
+        "@platforms//os:watchos",
+    ],
 )
 
 config_setting(
     name = "watchos_x86",
-    values = {
-        "apple_platform_type": "watchos",
-        "cpu": "watchos_i386",
-    },
+    constraint_values = [
+        "@platforms//os:watchos",
+        "@platforms//cpu:x86_32",
+    ],
 )
 
 config_setting(
     name = "watchos_x86_64",
-    values = {
-        "apple_platform_type": "watchos",
-        "cpu": "watchos_x86_64",
-    },
+    constraint_values = [
+        "@platforms//os:watchos",
+        "@platforms//cpu:x86_64",
+    ],
 )
 
 config_setting(
     name = "tvos",
-    values = {
-        "apple_platform_type": "tvos",
-    },
+    constraint_values = [
+        "@platforms//os:tvos",
+    ],
 )
 
 config_setting(
     name = "tvos_x86_64",
-    values = {
-        "apple_platform_type": "tvos",
-        "cpu": "tvos_x86_64",
-    },
+    constraint_values = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:tvos",
+    ],
 )
 
 config_setting(
     name = "windows_x86_64",
-    values = {
-        "cpu": "x64_windows",
-    },
+    constraint_values = [
+        "@platforms//os:windows",
+        "@platforms//cpu:x86_64",
+    ],
 )
 
 config_setting(
     name = "windows_arm64",
-    values = {
-        "cpu": "arm64_windows",
-    },
+    constraint_values = [
+        "@platforms//os:windows",
+        "@platforms//cpu:arm64",
+    ],
 )
 
 config_setting(
     name = "emscripten",
-    values = {
-        "crosstool_top": "//toolchain:emscripten",
-    },
+    constraint_values = [
+        "@platforms//os:emscripten",
+    ],
 )
 
 config_setting(
     name = "emscripten_with_threads",
+    constraint_values = [
+        "@platforms//os:emscripten",
+    ],
     values = {
-        "crosstool_top": "//toolchain:emscripten",
-        "copt": "-pthread",
+        "features": "use_pthreads",
     },
 )
