diff --git google/cloud/asset/v1p7beta1/BUILD.bazel google/cloud/asset/v1p7beta1/BUILD.bazel
--- google/cloud/asset/v1p7beta1/BUILD.bazel
+++ google/cloud/asset/v1p7beta1/BUILD.bazel
@@ -13,4 +13,6 @@
 # Common
 ##############################################################################
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load(
diff --git google/cloud/bigquery/connection/v1beta1/BUILD.bazel google/cloud/bigquery/connection/v1beta1/BUILD.bazel
--- google/cloud/bigquery/connection/v1beta1/BUILD.bazel
+++ google/cloud/bigquery/connection/v1beta1/BUILD.bazel
@@ -7,4 +7,6 @@
 # Common
 ##############################################################################
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load("@com_google_googleapis_imports//:imports.bzl", "proto_library_with_info")
diff --git google/cloud/bigquery/storage/v1/BUILD.bazel google/cloud/bigquery/storage/v1/BUILD.bazel
--- google/cloud/bigquery/storage/v1/BUILD.bazel
+++ google/cloud/bigquery/storage/v1/BUILD.bazel
@@ -2,4 +2,6 @@
 # Common
 ##############################################################################
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load(
diff --git google/cloud/bigquery/storage/v1beta2/BUILD.bazel google/cloud/bigquery/storage/v1beta2/BUILD.bazel
--- google/cloud/bigquery/storage/v1beta2/BUILD.bazel
+++ google/cloud/bigquery/storage/v1beta2/BUILD.bazel
@@ -5,4 +5,6 @@
 # Common
 ##############################################################################
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load("@com_google_googleapis_imports//:imports.bzl", "proto_library_with_info")
diff --git google/cloud/oslogin/common/BUILD.bazel google/cloud/oslogin/common/BUILD.bazel
--- google/cloud/oslogin/common/BUILD.bazel
+++ google/cloud/oslogin/common/BUILD.bazel
@@ -7,4 +7,6 @@
 # Common
 ##############################################################################
+load("@rules_proto//proto:defs.bzl", "proto_library")
+
 load("@com_google_googleapis_imports//:imports.bzl", "proto_library_with_info")
 
diff --git google/cloud/recommendationengine/v1beta1/BUILD.bazel google/cloud/recommendationengine/v1beta1/BUILD.bazel
--- google/cloud/recommendationengine/v1beta1/BUILD.bazel
+++ google/cloud/recommendationengine/v1beta1/BUILD.bazel
@@ -4,4 +4,6 @@
 # Common
 ##############################################################################
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load(
diff --git google/devtools/build/v1/BUILD.bazel google/devtools/build/v1/BUILD.bazel
--- google/devtools/build/v1/BUILD.bazel
+++ google/devtools/build/v1/BUILD.bazel
@@ -13,4 +13,6 @@
 # Common
 ##############################################################################
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load(
diff --git google/devtools/remoteworkers/v1test2/BUILD.bazel google/devtools/remoteworkers/v1test2/BUILD.bazel
--- google/devtools/remoteworkers/v1test2/BUILD.bazel
+++ google/devtools/remoteworkers/v1test2/BUILD.bazel
@@ -9,4 +9,6 @@
 #    * extra_protoc_file_parameters
 # The complete list of preserved parameters can be found in the source code.
+
+load("@rules_python//python:defs.bzl", "py_test")
 
 load(
diff --git google/devtools/source/v1/BUILD.bazel google/devtools/source/v1/BUILD.bazel
--- google/devtools/source/v1/BUILD.bazel
+++ google/devtools/source/v1/BUILD.bazel
@@ -4,4 +4,6 @@
 # Common
 ##############################################################################
+load("@rules_proto//proto:defs.bzl", "proto_library")
+
 load(
     "@com_google_googleapis_imports//:imports.bzl",
diff --git google/home/graph/v1/BUILD.bazel google/home/graph/v1/BUILD.bazel
--- google/home/graph/v1/BUILD.bazel
+++ google/home/graph/v1/BUILD.bazel
@@ -4,4 +4,6 @@
 # Common
 ##############################################################################
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load(
diff --git google/maps/BUILD.bazel google/maps/BUILD.bazel
--- google/maps/BUILD.bazel
+++ google/maps/BUILD.bazel
@@ -1,2 +1,4 @@
+load("@rules_shell//shell:sh_binary.bzl", "sh_binary")
+
 package(default_visibility = ["//google/maps:__subpackages__"])
 
diff --git google/maps/fleetengine/delivery/v1/BUILD.bazel google/maps/fleetengine/delivery/v1/BUILD.bazel
--- google/maps/fleetengine/delivery/v1/BUILD.bazel
+++ google/maps/fleetengine/delivery/v1/BUILD.bazel
@@ -1,2 +1,4 @@
+load("@rules_python//python:defs.bzl", "py_test")
+
 load(
     "@com_google_googleapis_imports//:imports.bzl",
diff --git google/maps/fleetengine/v1/BUILD.bazel google/maps/fleetengine/v1/BUILD.bazel
--- google/maps/fleetengine/v1/BUILD.bazel
+++ google/maps/fleetengine/v1/BUILD.bazel
@@ -1,2 +1,4 @@
+load("@rules_python//python:defs.bzl", "py_test")
+
 load(
     "@com_google_googleapis_imports//:imports.bzl",
diff --git google/shopping/merchant/notifications/v1beta/BUILD.bazel google/shopping/merchant/notifications/v1beta/BUILD.bazel
--- google/shopping/merchant/notifications/v1beta/BUILD.bazel
+++ google/shopping/merchant/notifications/v1beta/BUILD.bazel
@@ -19,4 +19,6 @@
 ##############################################################################
 # buildifier: disable=same-origin-load
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@com_google_googleapis_imports//:imports.bzl", "proto_library_with_info")
 load("@rules_proto//proto:defs.bzl", "proto_library")
diff --git google/shopping/merchant/reports/v1alpha/BUILD.bazel google/shopping/merchant/reports/v1alpha/BUILD.bazel
--- google/shopping/merchant/reports/v1alpha/BUILD.bazel
+++ google/shopping/merchant/reports/v1alpha/BUILD.bazel
@@ -19,4 +19,6 @@
 ##############################################################################
 # buildifier: disable=same-origin-load
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@com_google_googleapis_imports//:imports.bzl", "proto_library_with_info")
 load("@rules_proto//proto:defs.bzl", "proto_library")
diff --git google/shopping/merchant/reports/v1beta/BUILD.bazel google/shopping/merchant/reports/v1beta/BUILD.bazel
--- google/shopping/merchant/reports/v1beta/BUILD.bazel
+++ google/shopping/merchant/reports/v1beta/BUILD.bazel
@@ -19,4 +19,6 @@
 ##############################################################################
 # buildifier: disable=same-origin-load
+load("@rules_python//python:defs.bzl", "py_test")
+
 load("@com_google_googleapis_imports//:imports.bzl", "proto_library_with_info")
 load("@rules_proto//proto:defs.bzl", "proto_library")
diff --git google/storage/v2/BUILD.bazel google/storage/v2/BUILD.bazel
--- google/storage/v2/BUILD.bazel
+++ google/storage/v2/BUILD.bazel
@@ -9,4 +9,6 @@
 #    * extra_protoc_file_parameters
 # The complete list of preserved parameters can be found in the source code.
+
+load("@rules_python//python:defs.bzl", "py_test")
 
 load(
