diff --git google/cloud/asset/v1p7beta1/BUILD.bazel google/cloud/asset/v1p7beta1/BUILD.bazel
--- google/cloud/asset/v1p7beta1/BUILD.bazel
+++ google/cloud/asset/v1p7beta1/BUILD.bazel
@@ -33,6 +33,7 @@
     "proto_library_with_info",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "ruby_cloud_gapic_library",
     "ruby_gapic_assembly_pkg",
     "ruby_grpc_library",
diff --git google/cloud/bigquery/connection/v1beta1/BUILD.bazel google/cloud/bigquery/connection/v1beta1/BUILD.bazel
--- google/cloud/bigquery/connection/v1beta1/BUILD.bazel
+++ google/cloud/bigquery/connection/v1beta1/BUILD.bazel
@@ -152,6 +152,7 @@
     "@com_google_googleapis_imports//:imports.bzl",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
 )

 py_gapic_library(
diff --git google/cloud/bigquery/storage/v1/BUILD.bazel google/cloud/bigquery/storage/v1/BUILD.bazel
--- google/cloud/bigquery/storage/v1/BUILD.bazel
+++ google/cloud/bigquery/storage/v1/BUILD.bazel
@@ -26,6 +26,7 @@
     "proto_library_with_info",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "ruby_cloud_gapic_library",
     "ruby_gapic_assembly_pkg",
     "ruby_grpc_library",
diff --git google/cloud/bigquery/storage/v1beta2/BUILD.bazel google/cloud/bigquery/storage/v1beta2/BUILD.bazel
--- google/cloud/bigquery/storage/v1beta2/BUILD.bazel
+++ google/cloud/bigquery/storage/v1beta2/BUILD.bazel
@@ -153,6 +153,7 @@
     "@com_google_googleapis_imports//:imports.bzl",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
 )

 py_gapic_library(
diff --git google/cloud/oslogin/common/BUILD.bazel google/cloud/oslogin/common/BUILD.bazel
--- google/cloud/oslogin/common/BUILD.bazel
+++ google/cloud/oslogin/common/BUILD.bazel
@@ -7,6 +7,7 @@
 # Common
 ##############################################################################
 load("@com_google_googleapis_imports//:imports.bzl", "proto_library_with_info")
+load("@rules_proto//proto:defs.bzl", "proto_library")

 proto_library(
     name = "common_proto",
diff --git google/cloud/recommendationengine/v1beta1/BUILD.bazel google/cloud/recommendationengine/v1beta1/BUILD.bazel
--- google/cloud/recommendationengine/v1beta1/BUILD.bazel
+++ google/cloud/recommendationengine/v1beta1/BUILD.bazel
@@ -26,6 +26,7 @@
     "proto_library_with_info",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "ruby_cloud_gapic_library",
     "ruby_gapic_assembly_pkg",
     "ruby_grpc_library",
diff --git google/devtools/build/v1/BUILD.bazel google/devtools/build/v1/BUILD.bazel
--- google/devtools/build/v1/BUILD.bazel
+++ google/devtools/build/v1/BUILD.bazel
@@ -37,6 +37,7 @@
     "proto_library_with_info",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "ruby_cloud_gapic_library",
     "ruby_gapic_assembly_pkg",
     "ruby_grpc_library",
diff --git google/devtools/containeranalysis/v1beta1/build/BUILD.bazel google/devtools/containeranalysis/v1beta1/build/BUILD.bazel
--- google/devtools/containeranalysis/v1beta1/build/BUILD.bazel
+++ google/devtools/containeranalysis/v1beta1/build/BUILD.bazel
@@ -29,6 +29,7 @@
 load(
     "@com_google_googleapis_imports//:imports.bzl",
     "java_grpc_library",
+    "java_proto_library",
 )

 java_proto_library(
diff --git google/devtools/containeranalysis/v1beta1/deployment/BUILD.bazel google/devtools/containeranalysis/v1beta1/deployment/BUILD.bazel
--- google/devtools/containeranalysis/v1beta1/deployment/BUILD.bazel
+++ google/devtools/containeranalysis/v1beta1/deployment/BUILD.bazel
@@ -29,6 +29,7 @@
 load(
     "@com_google_googleapis_imports//:imports.bzl",
     "java_grpc_library",
+    "java_proto_library",
 )

 java_proto_library(
diff --git google/devtools/containeranalysis/v1beta1/discovery/BUILD.bazel google/devtools/containeranalysis/v1beta1/discovery/BUILD.bazel
--- google/devtools/containeranalysis/v1beta1/discovery/BUILD.bazel
+++ google/devtools/containeranalysis/v1beta1/discovery/BUILD.bazel
@@ -31,6 +31,7 @@
 load(
     "@com_google_googleapis_imports//:imports.bzl",
     "java_grpc_library",
+    "java_proto_library",
 )

 java_proto_library(
diff --git google/devtools/containeranalysis/v1beta1/image/BUILD.bazel google/devtools/containeranalysis/v1beta1/image/BUILD.bazel
--- google/devtools/containeranalysis/v1beta1/image/BUILD.bazel
+++ google/devtools/containeranalysis/v1beta1/image/BUILD.bazel
@@ -27,6 +27,7 @@
 load(
     "@com_google_googleapis_imports//:imports.bzl",
     "java_grpc_library",
+    "java_proto_library",
 )

 java_proto_library(
diff --git google/devtools/remoteworkers/v1test2/BUILD.bazel google/devtools/remoteworkers/v1test2/BUILD.bazel
--- google/devtools/remoteworkers/v1test2/BUILD.bazel
+++ google/devtools/remoteworkers/v1test2/BUILD.bazel
@@ -31,6 +31,7 @@
     "proto_library_with_info",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "ruby_cloud_gapic_library",
     "ruby_gapic_assembly_pkg",
     "ruby_grpc_library",
diff --git google/devtools/source/v1/BUILD.bazel google/devtools/source/v1/BUILD.bazel
--- google/devtools/source/v1/BUILD.bazel
+++ google/devtools/source/v1/BUILD.bazel
@@ -21,6 +21,8 @@
     "ruby_grpc_library",
     "ruby_proto_library",
 )
+
+load("@rules_proto//proto:defs.bzl", "proto_library")

 # This is an API workspace, having public visibility by default makes perfect sense.
 package(default_visibility = ["//visibility:public"])
diff --git google/home/graph/v1/BUILD.bazel google/home/graph/v1/BUILD.bazel
--- google/home/graph/v1/BUILD.bazel
+++ google/home/graph/v1/BUILD.bazel
@@ -26,6 +26,7 @@
     "proto_library_with_info",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "ruby_cloud_gapic_library",
     "ruby_gapic_assembly_pkg",
     "ruby_grpc_library",
diff --git google/maps/BUILD.bazel google/maps/BUILD.bazel
--- google/maps/BUILD.bazel
+++ google/maps/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_shell//shell:sh_binary.bzl", "sh_binary")
+
 package(default_visibility = ["//google/maps:__subpackages__"])

 sh_binary(
diff --git google/maps/fleetengine/delivery/v1/BUILD.bazel google/maps/fleetengine/delivery/v1/BUILD.bazel
--- google/maps/fleetengine/delivery/v1/BUILD.bazel
+++ google/maps/fleetengine/delivery/v1/BUILD.bazel
@@ -20,6 +20,7 @@
     "proto_library_with_info",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "py_import",
     "ruby_cloud_gapic_library",
     "ruby_gapic_assembly_pkg",
diff --git google/maps/fleetengine/v1/BUILD.bazel google/maps/fleetengine/v1/BUILD.bazel
--- google/maps/fleetengine/v1/BUILD.bazel
+++ google/maps/fleetengine/v1/BUILD.bazel
@@ -20,6 +20,7 @@
     "proto_library_with_info",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "py_import",
     "ruby_cloud_gapic_library",
     "ruby_gapic_assembly_pkg",
diff --git google/shopping/merchant/notifications/v1beta/BUILD.bazel google/shopping/merchant/notifications/v1beta/BUILD.bazel
--- google/shopping/merchant/notifications/v1beta/BUILD.bazel
+++ google/shopping/merchant/notifications/v1beta/BUILD.bazel
@@ -164,6 +164,7 @@
     "@com_google_googleapis_imports//:imports.bzl",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "py_import",
 )

diff --git google/shopping/merchant/reports/v1alpha/BUILD.bazel google/shopping/merchant/reports/v1alpha/BUILD.bazel
--- google/shopping/merchant/reports/v1alpha/BUILD.bazel
+++ google/shopping/merchant/reports/v1alpha/BUILD.bazel
@@ -165,6 +165,7 @@
     "@com_google_googleapis_imports//:imports.bzl",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "py_import",
 )

diff --git google/shopping/merchant/reports/v1beta/BUILD.bazel google/shopping/merchant/reports/v1beta/BUILD.bazel
--- google/shopping/merchant/reports/v1beta/BUILD.bazel
+++ google/shopping/merchant/reports/v1beta/BUILD.bazel
@@ -165,6 +165,7 @@
     "@com_google_googleapis_imports//:imports.bzl",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "py_import",
 )

diff --git google/storage/v2/BUILD.bazel google/storage/v2/BUILD.bazel
--- google/storage/v2/BUILD.bazel
+++ google/storage/v2/BUILD.bazel
@@ -33,6 +33,7 @@
     "proto_library_with_info",
     "py_gapic_assembly_pkg",
     "py_gapic_library",
+    "py_test",
     "ruby_cloud_gapic_library",
     "ruby_gapic_assembly_pkg",
     "ruby_grpc_library",
