diff --git a/lib/xstrndup.c b/lib/xstrndup.c
--- a/lib/xstrndup.c
+++ b/lib/xstrndup.c
@@ -21,8 +21,13 @@
 #include "xstrndup.h"
 
 #include <string.h>
+#include <stdlib.h>
 #include "xalloc.h"
 
+#ifdef _MSC_VER
+extern char *strndup (const char *, size_t);
+#endif
+
 /* Return a newly allocated copy of at most N bytes of STRING.
    In other words, return a copy of the initial segment of length N of
    STRING.  */
