diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 00000000..d4140ace
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,30 @@
+licenses(["notice"])  # MIT
+
+cc_library(
+    name = "rapidjson",
+    srcs = glob(["include/rapidjson/internal/*.h"]),
+    hdrs = glob(
+        [
+            "include/rapidjson/**/*.h",
+        ],
+        exclude = [
+            "include/rapidjson/internal/*.h",
+        ],
+    ),
+    includes = ["include"],
+    visibility = ["//visibility:public"],
+)
+
+cc_test(
+    name = "unittests",
+    srcs = glob([
+        "test/unittest/**/*.cpp",
+        "test/unittest/**/*.h",
+    ]),
+    data = glob(["bin/**/*"]),
+    local_defines = ["RAPIDJSON_HAS_STDSTRING"],
+    deps = [
+        ":rapidjson",
+        "@googletest//:gtest",
+    ],
+)
