--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,23 @@
+cc_library(
+    name = "flip",
+    hdrs = ["cpp/FLIP.h"],
+    includes = ["cpp"],
+    visibility = ["//visibility:public"],
+)
+
+cc_binary(
+    name = "flip-cli",
+    srcs = [
+        "cpp/tool/FLIP-tool.cpp",
+        "cpp/tool/FLIPToolHelpers.h",
+        "cpp/tool/commandline.h",
+        "cpp/tool/filename.h",
+        "cpp/tool/imagehelpers.h",
+        "cpp/tool/pooling.h",
+        "cpp/tool/stb_image.h",
+        "cpp/tool/stb_image_write.h",
+        "cpp/tool/tinyexr.h",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [":flip"],
+)
