--- MODULE.bazel
+++ MODULE.bazel
@@ -0,0 +1,57 @@
+module(
+    name = "cpp2sky",
+    version = "0.5.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(
+    name = "abseil-cpp",
+    version = "20240116.2",
+    repo_name = "com_google_absl",
+)
+bazel_dep(
+    name = "cpp-httplib",
+    version = "0.7.15",
+    repo_name = "com_github_httplib",
+)
+bazel_dep(
+    name = "fmt",
+    version = "8.1.1",
+    repo_name = "com_github_fmtlib_fmt",
+)
+
+bazel_dep(
+    name = "googletest",
+    version = "1.14.0.bcr.1",
+    dev_dependency = True,
+    repo_name = "com_google_googletest",
+)
+
+bazel_dep(
+    name = "grpc",
+    version = "1.56.3.bcr.1",
+    repo_name = "com_github_grpc_grpc",
+)
+bazel_dep(
+    name = "protobuf",
+    version = "3.19.2",
+    repo_name = "com_google_protobuf",
+)
+bazel_dep(
+    name = "rules_cc",
+    version = "0.0.9",
+)
+bazel_dep(
+    name = "rules_proto",
+    version = "4.0.0",
+)
+bazel_dep(
+    name = "skywalking-data-collect-protocol",
+    version = "9.7.0",
+    repo_name = "skywalking_data_collect_protocol",
+)
+bazel_dep(
+    name = "spdlog",
+    version = "1.10.0",
+    repo_name = "com_github_gabime_spdlog",
+)
