diff --git a/MODULE.bazel b/MODULE.bazel
index 2d43e46df..b50917142 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,8 +1,6 @@
-# TODO: migrate all dependencies from WORKSPACE to MODULE.bazel
-# https://github.com/protocolbuffers/protobuf/issues/14313
 module(
     name = "protobuf",
-    version = "27.1", # Automatically updated on release
+    version = "27.1.bcr.1",
     compatibility_level = 1,
     repo_name = "com_google_protobuf",
 )
