diff --git a/MODULE.bazel b/MODULE.bazel
index 7d5045b..b7e054b 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -5,8 +5,7 @@ module(
 )
 
 bazel_dep(name = "rules_cc", version = "0.1.1")
-
-bazel_dep(name = "rules_python", version = "1.4.1", dev_dependency = True)
+bazel_dep(name = "rules_python", version = "1.4.1")
 
 # Setup Python toolchain (required by rules_python)
 python = use_extension("@rules_python//python:extensions.bzl", "python")
