diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..70f393e
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,5 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "leaf",
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..544d3b4
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,8 @@
+module(
+    name = "boost.leaf",
+    version = "1.83.0.bzl.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
