--- MODULE.bazel
+++ MODULE.bazel
@@ -0,0 +1,18 @@
+module(
+    name = "emboss",
+    version = "2024.0304.184309",
+    compatibility_level = 1,
+    repo_name = "com_google_emboss",
+)
+
+bazel_dep(
+    name = "abseil-cpp",
+    repo_name = "com_google_absl",
+    version = "20230125.1",
+)
+bazel_dep(
+    name = "googletest",
+    version = "1.14.0.bcr.1",
+    repo_name = "com_google_googletest",
+)
+# -- bazel_dep definitions -- #
