===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,7 +1,7 @@
 module(
     name = "with_cfg.bzl",
-    version = "0.0.0",
+    version = "0.4.1",
     compatibility_level = 1,
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.4.1")
--- a/examples/.bazelrc
+++ b/examples/.bazelrc
@@ -16,9 +16,6 @@ common --incompatible_sandbox_hermetic_tmp
 common --java_language_version=17
 common --java_runtime_version=remotejdk_17

-# Use Bazel's embedded JDK for r_j_e's coursier.
-common --repo_env=JAVA_HOME=../../install/embedded_tools/jdk
-
 # Improve build caching by redacting environment variables.
 common --incompatible_strict_action_env
 # On Windows, PATH doubles as the dynamic library search path. We need its
