diff --git a/BUILD.bazel b/BUILD.bazel
index b98441a..316e306 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -3,6 +3,7 @@ load("@bazel_skylib//rules:copy_file.bzl", "copy_file")
 # Note: if you are still using WORKSPACE, you will need to patch this file to use the following instead
 # load("@python_3_11//:defs.bzl", "py_binary")
 load("@python_versions//3.11:defs.bzl", "py_binary")
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_proto//proto:defs.bzl", "proto_library")
 load("@rules_proto_grpc//:defs.bzl", "proto_plugin")
 load("@rules_python//python:proto.bzl", "py_proto_library")
diff --git a/MODULE.bazel b/MODULE.bazel
index 8aaa68e..3ca1685 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,6 @@
 module(
     name = "nanopb",
-    version = "0.4.9",
+    version = "0.4.9.1.bcr.3",
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
