From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shantanu Gontia <gontia.shantanu@gmail.com>
Date: Fri, 29 May 2026 08:13:40 +0530
Subject: fix printing warning


diff --git src/circllhist_print.c src/circllhist_print.c
index bce97d2..46a614c 100644
--- src/circllhist_print.c
+++ src/circllhist_print.c
@@ -106,7 +106,7 @@ void print(const histogram_t *hist) {
     for(i=0; i<invquantiles.cnt; i++) {
       printf("\"invq(%f)\":%g,", invquantiles.elements[i], qvals[i]);
     }
-    printf("\"count\":%zu}\n", hist_sample_count(hist));
+    printf("\"count\":%" PRIu64 "}\n", hist_sample_count(hist));
   }
   else print_hist(hist);
 }
-- 
2.54.0
