From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shantanu Gontia <gontia.shantanu@gmail.com>
Date: Fri, 29 May 2026 08:14:07 +0530
Subject: add module.bazel


diff --git MODULE.bazel MODULE.bazel
new file mode 100644
index 0000000..da9c01b
--- /dev/null
+++ MODULE.bazel
@@ -0,0 +1,16 @@
+module(
+    name = "libcircllhist",
+    version = "0.3.2.bcr.1",
+    compatibility_level = 1,
+    bazel_compatibility = [">=7.2.1"],
+)
+
+bazel_dep(
+    name = "platforms",
+    version = "1.1.0",
+)
+
+bazel_dep(
+    name = "rules_cc",
+    version = "0.2.17",
+)
-- 
2.54.0
