From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shantanu Gontia <gontia.shantanu@gmail.com>
Date: Fri, 29 May 2026 08:15:13 +0530
Subject: add build file


diff --git BUILD.bazel BUILD.bazel
new file mode 100644
index 0000000..765743f
--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,40 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_binary")
+
+licenses(["notice"])  # Apache 2
+
+config_setting(
+    name = "windows_x86_64",
+    constraint_values = [
+        "@platforms//os:windows",
+        "@platforms//cpu:x86_64",
+    ],
+)
+
+cc_library(
+    name = "circlhist",
+    srcs = ["src/circllhist.c"],
+    hdrs = [
+        "src/circllhist.h",
+    ],
+    copts = select({
+        "//:windows_x86_64": ["-DWIN32"],
+        "//conditions:default": [],
+    }),
+    strip_include_prefix = "src",
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "libcirclhist",
+    actual = "//:circlhist",
+    visibility = ["//visibility:public"],
+)
+
+cc_binary(
+    name = "circlhist_print",
+    srcs = ["src/circllhist_print.c"],
+    deps = [
+        ":circlhist",
+    ],
+    visibility = ["//visibility:public"],
+)
-- 
2.54.0
