diff --git a/src/scip/scip_export.h b/src/scip/scip_export.h
new file mode 100644
index 0000000000..871fde8e55
--- /dev/null
+++ b/src/scip/scip_export.h
@@ -0,0 +1,24 @@
+
+#ifndef SCIP_EXPORT_H
+#define SCIP_EXPORT_H
+
+#ifdef SCIP_STATIC_DEFINE
+#  define SCIP_EXPORT
+#  define SCIP_NO_EXPORT
+#else
+#  ifndef SCIP_EXPORT
+#    ifdef libscip_EXPORTS
+/* We are building this library */
+#      define SCIP_EXPORT
+#    else
+/* We are using this library */
+#      define SCIP_EXPORT
+#    endif
+#  endif
+
+#  ifndef SCIP_NO_EXPORT
+#    define SCIP_NO_EXPORT
+#  endif
+#endif
+
+#endif /* SCIP_EXPORT_H */
