--- a/build_defs.bzl
+++ b/build_defs.bzl
@@ -11,14 +11,14 @@
         "-fvisibility=hidden",
         "-fno-exceptions",
         "-fno-rtti",
+        "-std=c++17",
         "-Wall",
         "-Wextra",
         "-Wnon-virtual-dtor",
         "-Wno-missing-field-initializers",
-        "-Werror",
         "-Wno-long-long",
         "-Wshadow",
         "-Wundef",
         "-Wconversion",
         "-Wno-sign-conversion",
     ],
