diff --git a/tests/dir/subdir/invalid_link b/tests/dir/subdir/invalid_link
deleted file mode 120000
index 0479cc7..0000000
--- a/tests/dir/subdir/invalid_link
+++ /dev/null
@@ -1 +0,0 @@
-./invalid/target
\ No newline at end of file
diff --git a/tests/test.py b/tests/test.py
index 27585b9..5ac1b5b 100644
--- a/tests/test.py
+++ b/tests/test.py
@@ -67,11 +67,6 @@ def test_symlinks() -> None:
     assert abs_link.is_symlink()
     assert os.readlink(abs_link) == "/bin/sh"
 
-    invalid_link = Path("tests/dir/subdir/invalid_link")
-    assert invalid_link.is_symlink()
-    assert os.readlink(invalid_link) == "invalid/target"
-    assert not invalid_link.is_file()
-
     dir_link = Path("tests/dir/subdir/dir_link")
     assert dir_link.is_symlink()
     assert os.readlink(dir_link) == "dir"
