From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shantanu Gontia <gontia.shantanu@gmail.com>
Date: Thu, 28 May 2026 22:06:59 +0530
Subject: Update dependencies and fix boringssl name


diff --git c++/MODULE.bazel c++/MODULE.bazel
index 3f47e58..89a25ad 100644
--- c++/MODULE.bazel
+++ c++/MODULE.bazel
@@ -1,11 +1,11 @@
 module(
     name = "capnp-cpp",
-    version = "1.1.0",
+    version = "1.4.0",
 )
 
-bazel_dep(name = "platforms", version = "0.0.11")
+bazel_dep(name = "platforms", version = "1.0.0")
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
-bazel_dep(name = "rules_cc", version = "0.1.1")
+bazel_dep(name = "rules_cc", version = "0.2.17")
 bazel_dep(name = "zlib", version = "1.3.1.bcr.5")
-bazel_dep(name = "boringssl", version = "0.20241024.0")
-bazel_dep(name = "brotli", version = "1.1.0")
+bazel_dep(name = "boringssl", version = "0.20241024.0", repo_name = "ssl")
+bazel_dep(name = "brotli", version = "1.2.0.bcr.1")
-- 
2.54.0
