From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shantanu Gontia <gontia.shantanu@gmail.com>
Date: Thu, 28 May 2026 23:24:39 +0530
Subject: add bazel version compatibility


diff --git c++/MODULE.bazel c++/MODULE.bazel
index 89a25ad..562252c 100644
--- c++/MODULE.bazel
+++ c++/MODULE.bazel
@@ -1,6 +1,7 @@
 module(
     name = "capnp-cpp",
     version = "1.4.0",
+    bazel_compatibility = [">=7.2.1"]
 )
 
 bazel_dep(name = "platforms", version = "1.0.0")
-- 
2.54.0
