===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -13,9 +13,9 @@
 # https://bazel.build/versions/6.0.0/build/bzlmod#version-resolution
 # Thus the highest version in their module graph is resolved.
 
 # These dependencies must be declared before the other rules dependencies.
-bazel_dep(name = "rules_apple", version = "3.16.0", repo_name = "build_bazel_rules_apple")
+bazel_dep(name = "rules_apple", version = "34.0-rc1.1", repo_name = "build_bazel_rules_apple")
 bazel_dep(name = "apple_support", version = "1.15.1", repo_name = "build_bazel_apple_support")
 
 # Keep apple_support on 1.15.1 for now to avoid this issue:
 # https://github.com/bazelbuild/apple_support/issues/316

--- a/MODULE.bazel
    +++ a/MODULE.bazel
    @@ -3,7 +3,7 @@
     module(
         name = "protobuf",
    -    version = "34.0-rc1",  # Automatically updated on release
    +    version = "34.0-rc1.1",  # Automatically updated on release
         compatibility_level = 1,
         repo_name = "com_google_protobuf",
     )
