diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..a77037b
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,17 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "conversion",
+    alias_repo_name = "numeric.conversion",
+    deps = [
+        "@boost//:config",
+        "@boost//:conversion",
+        "@boost//:core",
+        "@boost//:mpl",
+        "@boost//:preprocessor",
+        "@boost//:throw_exception",
+        "@boost//:type_traits",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..01b45a6
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.numeric.conversion",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..90ba3f6
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:numeric.conversion",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "numeric_conversion_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/numeric_conversion_test.cc b/bazel_test/numeric_conversion_test.cc
new file mode 100644
index 0000000..48f93ed
--- /dev/null
+++ b/bazel_test/numeric_conversion_test.cc
@@ -0,0 +1,13 @@
+#include <boost/numeric/conversion/cast.hpp>
+
+int main()
+{
+  short a = 42;
+  int b = 42;
+
+  if (boost::numeric_cast<int>(a) != b) {
+    return 1;
+  }
+
+  return 0;
+}
