--- Source/FreeImage/PluginTIFF.cpp
+++ Source/FreeImage/PluginTIFF.cpp
@@ -197,6 +197,8 @@ TIFFFdOpen(thandle_t handle, const char *name, const char *mode) {
 	return tif;
 }
 
+#ifndef USE_INTERNAL_TIFF_UNIX_ROUTINES
+
 // ----------------------------------------------------------
 //   TIFF library FreeImage-specific routines.
 // ----------------------------------------------------------
@@ -267,6 +269,7 @@ msdosErrorHandler(const char* module, const char* fmt, va_list ap) {
 
 TIFFErrorHandler _TIFFerrorHandler = msdosErrorHandler;
 #endif // ndef _WIN32
+#endif // ndef USE_INTERNAL_TIFF_UNIX_ROUTINES
 // ----------------------------------------------------------
 
 #define CVT(x)      (((x) * 255L) / ((1L<<16)-1))
