diff --git a/lanelet2_core/test/test_linestring.cpp b/lanelet2_core/test/test_linestring.cpp
index 7ef3d7c..42928b3 100644
--- a/lanelet2_core/test/test_linestring.cpp
+++ b/lanelet2_core/test/test_linestring.cpp
@@ -5,7 +5,7 @@
 
 #include "lanelet2_core/geometry/LineString.h"
 #include "lanelet2_core/primitives/LineString.h"
-#include "primitives/Traits.h"
+#include "lanelet2_core/primitives/Traits.h"
 using namespace lanelet;
 
 class LineStringPoints : public ::testing::Test {
diff --git a/lanelet2_matching/test/lanelet2_matching.cpp b/lanelet2_matching/test/lanelet2_matching.cpp
index c47194e..5b84b54 100644
--- a/lanelet2_matching/test/lanelet2_matching.cpp
+++ b/lanelet2_matching/test/lanelet2_matching.cpp
@@ -33,7 +33,7 @@
 #include <lanelet2_traffic_rules/TrafficRules.h>
 #include <lanelet2_traffic_rules/TrafficRulesFactory.h>
 
-#include "LaneletMatching.h"
+#include "lanelet2_matching/LaneletMatching.h"
 #include "gtest/gtest.h"
 
 using namespace lanelet;
diff --git a/lanelet2_routing/test/test_routing_map.h b/lanelet2_routing/test/test_routing_map.h
index 544b22e..cbebc2f 100644
--- a/lanelet2_routing/test/test_routing_map.h
+++ b/lanelet2_routing/test/test_routing_map.h
@@ -7,7 +7,7 @@
 #include <memory>
 #include <utility>
 
-#include "RoutingCost.h"
+#include "lanelet2_routing/RoutingCost.h"
 #include "lanelet2_routing/Forward.h"
 #include "lanelet2_routing/RoutingGraph.h"
 
