diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..726bdeb
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,18 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "lexical_cast",
+    deps = [
+        "@boost.array//:array",
+        "@boost.assert//:assert",
+        "@boost.config//:config",
+        "@boost.container//:container",
+        "@boost.core//:core",
+        "@boost.integer//:integer",
+        "@boost.numeric.conversion//:conversion",
+        "@boost.range//:range",
+        "@boost.static_assert//:static_assert",
+        "@boost.throw_exception//:throw_exception",
+        "@boost.type_traits//:type_traits",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..26411c5
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,19 @@
+module(
+    name = "boost.lexical_cast",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.array", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.assert", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.container", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.integer", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.numeric.conversion", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.range", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.throw_exception", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
