===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -25,9 +25,9 @@
     compatibility_level = 1,
 )
 
 # --registry=https://bcr.bazel.build
-bazel_dep(name = "protobuf", version = "27.3")
+bazel_dep(name = "protobuf", version = "20241225.0-dev")
 bazel_dep(name = "rules_proto", version = "4.0.0")
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
 bazel_dep(name = "apple_support", version = "1.17.1")
 bazel_dep(name = "rules_cc", version = "0.0.13")
