diff --git a/folly/debugging/symbolizer/DwarfUtil.h b/folly/debugging/symbolizer/DwarfUtil.h
index b1ff6819e..c8cdcf741 100644
--- a/folly/debugging/symbolizer/DwarfUtil.h
+++ b/folly/debugging/symbolizer/DwarfUtil.h
@@ -26,7 +26,7 @@
 #include <folly/experimental/symbolizer/ElfCache.h>
 
 #if FOLLY_HAVE_DWARF && FOLLY_HAVE_ELF
-#include <dwarf.h> // @manual=fbsource//third-party/libdwarf:dwarf
+#include <libdwarf/dwarf.h> // @manual=fbsource//third-party/libdwarf:dwarf
 
 namespace folly {
 namespace symbolizer {
