diff --git a/folly/Conv.h b/folly/Conv.h
index abd396c7d..93b06dd79 100644
--- a/folly/Conv.h
+++ b/folly/Conv.h
@@ -127,7 +127,7 @@
 #include <charconv>
 #endif
 
-#include <double-conversion/double-conversion.h> // V8 JavaScript implementation
+#include "double-conversion/double-conversion.h" // V8 JavaScript implementation
 
 #include <folly/CPortability.h>
 
diff --git a/folly/Format.cpp b/folly/Format.cpp
index 6ab655ba5..203d5fd06 100644
--- a/folly/Format.cpp
+++ b/folly/Format.cpp
@@ -23,7 +23,7 @@
 #include <folly/Portability.h>
 #include <folly/container/Array.h>
 
-#include <double-conversion/double-conversion.h>
+#include "double-conversion/double-conversion.h"
 
 namespace folly {
 namespace detail {
diff --git a/folly/test/StringToFloatBenchmark.cpp b/folly/test/StringToFloatBenchmark.cpp
index b2b4055fd..b0790d616 100644
--- a/folly/test/StringToFloatBenchmark.cpp
+++ b/folly/test/StringToFloatBenchmark.cpp
@@ -44,7 +44,7 @@
 #include <folly/Traits.h>
 #include <folly/stop_watch.h>
 
-#include <double-conversion/double-conversion.h>
+#include "double-conversion/double-conversion.h"
 #include <fast_float/fast_float.h> // @manual=fbsource//third-party/fast_float:fast_float
 
 template <typename T, typename... A>
