diff --git a/rsvg/build.rs b/rsvg/build.rs
--- a/rsvg/build.rs
+++ b/rsvg/build.rs
@@ -3,21 +3,7 @@
 use std::io::{BufWriter, Write};
 use std::path::Path;
 
-#[cfg(docsrs)]
-fn probe_system_deps() {
-    // do not probe libraries since the docs.rs environment doesn't have them
-}
-
-#[cfg(not(docsrs))]
-fn probe_system_deps() {
-    if let Err(e) = system_deps::Config::new().probe() {
-        eprintln!("{e}");
-        std::process::exit(1);
-    }
-}
-
 fn main() {
-    probe_system_deps();
     generate_srgb_tables();
 }
 
