diff --git a/src/common/getcpu.h b/src/common/getcpu.h
index 24e684c7..2d8afc1e 100644
--- a/src/common/getcpu.h
+++ b/src/common/getcpu.h
@@ -45,7 +45,7 @@ int lttng_ust_get_cpu_internal(void)
 #ifdef __linux__
 
 #if !HAVE_SCHED_GETCPU
-#include <sys/syscall.h>
+#include <unistd.h> // For the standard syscall function
 #define __getcpu(cpu, node, cache)	syscall(__NR_getcpu, cpu, node, cache)
 /*
  * If getcpu is not implemented in the kernel, use cpu 0 as fallback.
