--- MODULE.bazel
+++ MODULE.bazel
@@ -3,30 +3,3 @@
     version = "0.2",
     compatibility_level = 0,
 )
-
-# Here are the files being defined for each OS
-http_appimage = use_repo_rule("//:http_appimage.bzl", "http_appimage")
-http_appimage(
-    name = "openscad_linux_x86_64",
-    url = "https://files.openscad.org/snapshots/OpenSCAD-2024.09.20.ai20452-x86_64.AppImage",
-    sha256 = "eb5bfcc0a19df3016e3e00f3e135695020371c24f2a8629bf1e096009bef8a47",
-)
-http_appimage(
-    name = "openscad_linux_aarch64",
-    url = "https://files.openscad.org/snapshots/OpenSCAD-2023.09.11.ai-aarch64.AppImage",
-    sha256 = "84d7bb1c71e14b4e248a84fbe0a4b02f58bcbf5326f0ee81c8a4de3653a3b568",
-)
-# This is the code that will be required once we support MacOS and Windows
-# http_file = use_repo_rule("@bazel_tools//tools/build_defs/repo:http.bzl", "http_file")
-# http_file(
-#     name = "openscad_windows_",
-#     executable = True,
-#     url = "https://files.openscad.org/snapshots/OpenSCAD-2024.09.20-x86-64.zip",
-#     sha256 = "6ae4f92fc3fb4bf38efe1a7029f03f018707b376f954d886a635dc41bb6b0ccd",
-# )
-# http_file(
-#     name = "openscad_darwin",
-#     executable = True,
-#     url = "https://files.openscad.org/snapshots/OpenSCAD-2024.09.20.dmg",
-#     sha256 = "8ca81fe343d359b52b058d3c8cff2ac0673f508691a3270c64d8c22ef4beb3f0",
-# )
