--- icu4c/source/common/BUILD.bazel
+++ icu4c/source/common/BUILD.bazel
@@ -68,11 +68,19 @@ cc_library(
     deps = [
         ":headers",
         # omit other deps b/c they are sys symbols
+        "@rules_cc//cc/runfiles",
+    ],
+    data = [
+        "@icu_dat",
     ],
     local_defines = [
         "U_COMMON_IMPLEMENTATION",
+        "ICU_DATA_DIR_BAZEL=\\\"$(rlocationpath @icu_dat)\\\"",
     ],
-    linkopts = ["-ldl"],
+    linkopts = select({
+        "@platforms//os:linux": ["-ldl"],
+        "//conditions:default": [],
+    }),
 )
 
 cc_library(
