Index: MODULE.bazel
===================================================================
--- MODULE.bazel
+++ MODULE.bazel
@@ -1,9 +1,9 @@
 "Bazel module definition, see https://bazel.build/external/overview#bzlmod"
+
 module(
     name = "rules_container_structure_test",
+    version = "1.15.0",
     compatibility_level = 1,
-    # Replaced dynamically when published
-    version = "0.0.0",
 )
 
 # To run yq