--- .bazelversion
+++ .bazelversion
@@ -1 +1 @@
-7.3.1
+7.4.0

--- BUILD.bazel
+++ BUILD.bazel
@@ -1,3 +1,5 @@
+load("@buildifier_prebuilt//:rules.bzl", "buildifier", "buildifier_test")
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_gazebo//gazebo:headers.bzl", "gz_configure_header", "gz_export_header", "gz_include_header")
 load("@rules_license//rules:license.bzl", "license")

@@ -151,3 +153,19 @@ test_sources = glob(
         "@gz-plugin//:loader",
     ],
 ) for src in test_sources]
+
+buildifier(
+    name = "buildifier.fix",
+    exclude_patterns = ["./.git/*"],
+    lint_mode = "fix",
+    mode = "fix",
+)
+
+buildifier_test(
+    name = "buildifier.test",
+    exclude_patterns = ["./.git/*"],
+    lint_mode = "warn",
+    mode = "diff",
+    no_sandbox = True,
+    workspace = "//:MODULE.bazel",
+)

--- bazel/gz_physics_plugin_libraries.bzl
+++ bazel/gz_physics_plugin_libraries.bzl
@@ -1,3 +1,7 @@
+"""
+Rules to add gz-physics plugin libraries.
+"""
+
 load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 load("@rules_cc//cc:cc_library.bzl", "cc_library")


--- bullet-featherstone/BUILD.bazel
+++ bullet-featherstone/BUILD.bazel
@@ -1,3 +1,4 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
 load("@rules_gazebo//gazebo:headers.bzl", "gz_export_header")
 load("//bazel:gz_physics_plugin_libraries.bzl", "gz_physics_plugin_libraries")


--- bullet/BUILD.bazel
+++ bullet/BUILD.bazel
@@ -1,3 +1,4 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
 load("@rules_gazebo//gazebo:headers.bzl", "gz_export_header")
 load("//bazel:gz_physics_plugin_libraries.bzl", "gz_physics_plugin_libraries")


--- dartsim/BUILD.bazel
+++ dartsim/BUILD.bazel
@@ -1,3 +1,4 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_gazebo//gazebo:headers.bzl", "gz_export_header")
 load("//bazel:gz_physics_plugin_libraries.bzl", "gz_physics_plugin_libraries")

@@ -70,11 +71,11 @@ cc_library(

 gz_physics_plugin_libraries(
     srcs = ["src/plugin.cc"] + private_headers,
-    so_lib_name = "libgz-physics-dartsim-plugin.so",
-    static_lib_name = "libgz-physics-dartsim-plugin-static",
     defines = [
         "BAZEL_DARTSIM_DISABLE_UNREGISTER_COLLISION_DETECTORS",
     ],
+    so_lib_name = "libgz-physics-dartsim-plugin.so",
+    static_lib_name = "libgz-physics-dartsim-plugin-static",
     visibility = ["//visibility:public"],
     deps = [
         ":dartsim",

--- test/BUILD.bazel
+++ test/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
+
 package(
     default_applicable_licenses = ["//:license"],
     features = [

--- tpe/BUILD.bazel
+++ tpe/BUILD.bazel
@@ -1,4 +1,5 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_gazebo//gazebo:headers.bzl", "gz_export_header")
 load("//bazel:gz_physics_plugin_libraries.bzl", "gz_physics_plugin_libraries")

 package(
