diff --git a/folly/io/async/test/IoUringBackendTest.cpp b/folly/io/async/test/IoUringBackendTest.cpp
index 75cd85ff4..f32f566f9 100644
--- a/folly/io/async/test/IoUringBackendTest.cpp
+++ b/folly/io/async/test/IoUringBackendTest.cpp
@@ -710,7 +710,7 @@ TEST(IoUringBackend, Rename) {
   auto newPath = dirPath / newName;
 
   int fd = folly::fileops::open(
-      oldPath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC);
+      oldPath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC, 0600);
   CHECK_GE(fd, 0);
 
   SCOPE_EXIT {
@@ -747,7 +747,7 @@ TEST(IoUringBackend, RenameDstExists) {
 
   {
     int oldFd = folly::fileops::open(
-        oldPath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC);
+        oldPath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC, 0600);
     CHECK_GE(oldFd, 0);
     folly::fileops::close(oldFd);
   }
@@ -758,7 +758,7 @@ TEST(IoUringBackend, RenameDstExists) {
 
   {
     int newFd = folly::fileops::open(
-        newPath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC);
+        newPath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC, 0600);
     CHECK_GE(newFd, 0);
     folly::fileops::close(newFd);
   }
@@ -817,7 +817,7 @@ TEST(IoUringBackend, Statx) {
       dirPath.string().c_str(), O_DIRECTORY | O_RDONLY, 0666);
   CHECK_GE(dfd, 0);
   int fd = folly::fileops::open(
-      filePath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC);
+      filePath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC, 0600);
   CHECK_GE(fd, 0);
 
   SCOPE_EXIT {
@@ -850,7 +850,7 @@ TEST(IoUringBackend, StatxAbsolute) {
   auto filePath = dirPath / path;
 
   int fd = folly::fileops::open(
-      filePath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC);
+      filePath.string().c_str(), O_CREAT | O_WRONLY | O_TRUNC, 0600);
   CHECK_GE(fd, 0);
 
   SCOPE_EXIT {
