--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,29 @@
+licenses(["notice"])  # Apache 2
+
+config_setting(
+    name = "windows_x86_64",
+    constraint_values = [
+        "@platforms//os:windows",
+        "@platforms//cpu:x86_64",
+    ],
+)
+
+cc_library(
+    name = "circlhist",
+    srcs = ["src/circllhist.c"],
+    hdrs = [
+        "src/circllhist.h",
+    ],
+    copts = select({
+        "//:windows_x86_64": ["-DWIN32"],
+        "//conditions:default": [],
+    }),
+    includes = ["src"],
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "libcirclhist",
+    actual = "//:circlhist",
+    visibility = ["//visibility:public"],
+)
