diff --git a/src/ua_types_encoding_json.c b/src/ua_types_encoding_json.c
index e4a7466a0..3391930b5 100644
--- a/src/ua_types_encoding_json.c
+++ b/src/ua_types_encoding_json.c
@@ -17,11 +17,11 @@
 #include <float.h>
 #include <math.h>
 
-#include "../deps/itoa.h"
-#include "../deps/dtoa.h"
-#include "../deps/parse_num.h"
-#include "../deps/base64.h"
-#include "../deps/libc_time.h"
+#include "itoa.h"
+#include "dtoa.h"
+#include "parse_num.h"
+#include "base64.h"
+#include "libc_time.h"
 
 #ifndef UA_ENABLE_PARSING
 #error UA_ENABLE_PARSING required for JSON encoding
diff --git a/src/ua_types_encoding_json.h b/src/ua_types_encoding_json.h
index 0013e9eca..ffb32e170 100644
--- a/src/ua_types_encoding_json.h
+++ b/src/ua_types_encoding_json.h
@@ -13,7 +13,7 @@
 
 #include "ua_util_internal.h"
 
-#include "../deps/cj5.h"
+#include "cj5.h"
 
 _UA_BEGIN_DECLS
 
