===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -5,9 +5,9 @@
 # Defines the rules_ios bzlmod module.
 # Version is updated during release to the registry.
 module(
     name = "rules_ios",
-    version = "0",
+    version = "4.3.0",
     bazel_compatibility = [
         ">=6.0.0",
         # Temporarily limit the usage of Bazel 7+ until
         # https://github.com/bazel-ios/rules_ios/issues/795
