diff --git a/proxygen/lib/services/WorkerThread.cpp b/proxygen/lib/services/WorkerThread.cpp
index 3c422db04..a0ac27346 100644
--- a/proxygen/lib/services/WorkerThread.cpp
+++ b/proxygen/lib/services/WorkerThread.cpp
@@ -14,6 +14,7 @@
 #include <folly/String.h>
 #include <folly/experimental/io/IoUringBackend.h>
 #include <folly/io/async/EventBaseManager.h>
+#include <folly/portability/GFlags.h>
 #include <glog/logging.h>
 #include <signal.h>
 
