diff --git a/proxygen/httpserver/samples/push/PushRequestHandler.cpp b/proxygen/httpserver/samples/push/PushRequestHandler.cpp
index dd46b308c..77698c1d7 100644
--- a/proxygen/httpserver/samples/push/PushRequestHandler.cpp
+++ b/proxygen/httpserver/samples/push/PushRequestHandler.cpp
@@ -8,6 +8,8 @@
 
 #include "PushRequestHandler.h"
 
+#include <cstdlib>
+
 #include "proxygen/httpserver/samples/push/PushStats.h"
 #include <folly/FileUtil.h>
 #include <proxygen/httpserver/PushHandler.h>
@@ -32,8 +34,13 @@ std::string createLargeBody() {
 
 PushRequestHandler::PushRequestHandler(PushStats* stats) : stats_(stats) {
   if (gPushBody.empty()) {
-    CHECK(folly::readFile(kPushFileName.c_str(), gPushBody))
-        << "Failed to read push file=" << kPushFileName;
+    std::string push_file_name = kPushFileName;
+    const char* push_file_name_env_var_value = std::getenv("PUSH_FILE_NAME");
+    if (push_file_name_env_var_value != nullptr) {
+      push_file_name = push_file_name_env_var_value;
+    }
+    CHECK(folly::readFile(push_file_name.c_str(), gPushBody))
+        << "Failed to read push file=" << push_file_name;
   }
 }
 
