diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..e282a46
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,42 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "dll",
+    deps = [
+        "@boost//:assert",
+        "@boost//:config",
+        "@boost//:core",
+        "@boost//:filesystem",
+        "@boost//:function",
+        "@boost//:move",
+        "@boost//:predef",
+        "@boost//:smart_ptr",
+        "@boost//:spirit",
+        "@boost//:static_assert",
+        "@boost//:system",
+        "@boost//:throw_exception",
+        "@boost//:type_index",
+        "@boost//:type_traits",
+        "@boost//:winapi",
+    ],
+)
+
+boost_library(
+    name = "b2_workarounds",
+    hdrs = ["example/b2_workarounds.hpp"],
+    include_prefix = "dll/",
+    deps = [
+        "@boost//:dll",
+    ],
+)
+
+boost_library(
+    name = "static_plugin",
+    hdrs = ["example/tutorial4/static_plugin.hpp"],
+    include_prefix = "dll/",
+    deps = [
+        "@boost//:dll",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..09ed33a
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.dll",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..6133f70
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,15 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:dll",
+    "@boost//:config",
+    "@boost.dll//:b2_workarounds",
+    "@boost.dll//:static_plugin",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        deps = DEPS,
+    ),
+)
diff --git a/test/cpp_import_class_test.cpp b/test/cpp_import_class_test.cpp
index 97a79a5..78bd47a 100644
--- a/test/cpp_import_class_test.cpp
+++ b/test/cpp_import_class_test.cpp
@@ -11,7 +11,7 @@
 
 #if (__cplusplus >= 201402L) || (defined(_MSVC_LANG) && _MSVC_LANG >= 201402L)
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 
 #include <iostream>
 
diff --git a/test/cpp_import_test.cpp b/test/cpp_import_test.cpp
index a32df87..637d5ef 100644
--- a/test/cpp_import_test.cpp
+++ b/test/cpp_import_test.cpp
@@ -10,7 +10,7 @@
 
 #if (__cplusplus >= 201402L) || (defined(_MSVC_LANG) && _MSVC_LANG >= 201402L)
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 
 #include <boost/dll/smart_library.hpp>
 #include <boost/dll/import_mangled.hpp>
diff --git a/test/cpp_load_test.cpp b/test/cpp_load_test.cpp
index 1c729e0..440919e 100644
--- a/test/cpp_load_test.cpp
+++ b/test/cpp_load_test.cpp
@@ -17,7 +17,7 @@
 
 #if (__cplusplus >= 201402L) || (BOOST_COMP_MSVC >= BOOST_VERSION_NUMBER(14,0,0))
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 
 #include <boost/core/lightweight_test.hpp>
 #include <boost/filesystem.hpp>
diff --git a/test/cpp_mangle_test.cpp b/test/cpp_mangle_test.cpp
index 9c4b54b..1fd3253 100644
--- a/test/cpp_mangle_test.cpp
+++ b/test/cpp_mangle_test.cpp
@@ -10,7 +10,7 @@
 
 #if (__cplusplus >= 201402L) || (defined(_MSVC_LANG) && _MSVC_LANG >= 201402L)
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 
 #include <boost/dll/smart_library.hpp>
 #include <boost/core/lightweight_test.hpp>
diff --git a/test/ctti_type_name_parser_test.cpp b/test/ctti_type_name_parser_test.cpp
index d374608..3191d8d 100644
--- a/test/ctti_type_name_parser_test.cpp
+++ b/test/ctti_type_name_parser_test.cpp
@@ -9,7 +9,7 @@
 #include <boost/config.hpp>
 
 #if (__cplusplus >= 201402L) || (defined(_MSVC_LANG) && _MSVC_LANG >= 201402L)
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 
 #include <boost/core/lightweight_test.hpp>
 #include <boost/dll/smart_library.hpp>
diff --git a/test/empty_library_info_test.cpp b/test/empty_library_info_test.cpp
index 52c18e6..937902b 100644
--- a/test/empty_library_info_test.cpp
+++ b/test/empty_library_info_test.cpp
@@ -6,7 +6,7 @@
 
 // For more information, see http://www.boost.org
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 
 #include <boost/dll/library_info.hpp>
 #include <boost/core/lightweight_test.hpp>
diff --git a/test/library_info_test.cpp b/test/library_info_test.cpp
index 1d8ffc4..93c3458 100644
--- a/test/library_info_test.cpp
+++ b/test/library_info_test.cpp
@@ -9,11 +9,11 @@
 
 // MinGW related workaround
 #define BOOST_DLL_FORCE_ALIAS_INSTANTIATION
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 
 #include <boost/dll/library_info.hpp>
 #include <boost/core/lightweight_test.hpp>
-#include "../example/tutorial4/static_plugin.hpp"
+#include "dll/example/tutorial4/static_plugin.hpp"
 
 // Unit Tests
 
diff --git a/test/shared_library_concurrent_load_test.cpp b/test/shared_library_concurrent_load_test.cpp
index f4d6b28..21aab71 100644
--- a/test/shared_library_concurrent_load_test.cpp
+++ b/test/shared_library_concurrent_load_test.cpp
@@ -14,7 +14,7 @@ int main() {
 
 #else // #ifdef BOOST_TRAVISCI_BUILD
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 #include <boost/dll.hpp>
 #include <boost/filesystem/path.hpp>
 #include <boost/thread/thread.hpp>
diff --git a/test/shared_library_errors.cpp b/test/shared_library_errors.cpp
index eae5462..8aa8e39 100644
--- a/test/shared_library_errors.cpp
+++ b/test/shared_library_errors.cpp
@@ -8,7 +8,7 @@
 
 // For more information, see http://www.boost.org
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 #include <boost/dll/shared_library.hpp>
 #include <boost/dll/library_info.hpp>
 #include <boost/core/lightweight_test.hpp>
diff --git a/test/shared_library_get_symbol_test.cpp b/test/shared_library_get_symbol_test.cpp
index e7a7c0b..eb023ba 100644
--- a/test/shared_library_get_symbol_test.cpp
+++ b/test/shared_library_get_symbol_test.cpp
@@ -8,7 +8,7 @@
 
 // For more information, see http://www.boost.org
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 #include <boost/dll.hpp>
 #include <boost/core/lightweight_test.hpp>
 #include <boost/function.hpp>
diff --git a/test/shared_library_load_test.cpp b/test/shared_library_load_test.cpp
index 32b9b0e..8f99001 100644
--- a/test/shared_library_load_test.cpp
+++ b/test/shared_library_load_test.cpp
@@ -7,7 +7,7 @@
 
 // For more information, see http://www.boost.org
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 #include <boost/dll.hpp>
 #include <boost/core/lightweight_test.hpp>
 // Unit Tests
diff --git a/test/shared_library_search_symbol_test.cpp b/test/shared_library_search_symbol_test.cpp
index 93310c2..78dc9f7 100644
--- a/test/shared_library_search_symbol_test.cpp
+++ b/test/shared_library_search_symbol_test.cpp
@@ -8,7 +8,7 @@
 
 // For more information, see http://www.boost.org
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 #include <boost/dll.hpp>
 #include <boost/core/lightweight_test.hpp>
 
diff --git a/test/symbol_runtime_info_test.cpp b/test/symbol_runtime_info_test.cpp
index 2a661e3..92046d2 100644
--- a/test/symbol_runtime_info_test.cpp
+++ b/test/symbol_runtime_info_test.cpp
@@ -7,7 +7,7 @@
 
 // For more information, see http://www.boost.org
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 #include <boost/dll.hpp>
 #include <boost/dll/runtime_symbol_info.hpp>
 #include <boost/filesystem/operations.hpp>
diff --git a/test/template_method_linux_test.cpp b/test/template_method_linux_test.cpp
index e64f18b..dd80f83 100644
--- a/test/template_method_linux_test.cpp
+++ b/test/template_method_linux_test.cpp
@@ -15,7 +15,7 @@
 #include <iostream>
 #include <string>
 
-#include "../example/b2_workarounds.hpp"
+#include "dll/example/b2_workarounds.hpp"
 #include <boost/core/lightweight_test.hpp>
 
 
