diff --git a/libs/cxxopts/BUILD.bazel b/libs/cxxopts/BUILD.bazel
deleted file mode 100644
index a6cd021..0000000
--- a/libs/cxxopts/BUILD.bazel
+++ /dev/null
@@ -1,16 +0,0 @@
-load("@rules_cc//cc:defs.bzl", "cc_library")
-
-cc_library(
-    name = "cxxopts",
-    hdrs = ["include/cxxopts.hpp"],
-    strip_include_prefix = "include",
-    visibility = ["//visibility:public"],
-)
-
-load("@rules_fuzzing//fuzzing:cc_defs.bzl", "cc_fuzz_test")
-
-cc_fuzz_test(
-    name = "cxxopts_fuzz_test",
-    srcs = ["test/fuzz.cpp"],
-    deps = [":cxxopts"],
-)
\ No newline at end of file
