diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..60be244
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,12 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "system",
+    deps = [
+        "@boost.assert//:assert",
+        "@boost.config//:config",
+        "@boost.throw_exception//:throw_exception",
+        "@boost.variant2//:variant2",
+        "@boost.winapi//:winapi",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..827c7a6
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,13 @@
+module(
+    name = "boost.system",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.assert", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.throw_exception", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.variant2", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.winapi", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
