diff --git a/MODULE.bazel b/MODULE.bazel
index 2c1f801c..9de59405 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,5 +1,6 @@
 module(
     name = "riegeli",
+    version = "0.0.0-20250717-5b2e77e",
     repo_name = "com_google_riegeli",
 )
 
@@ -74,33 +75,13 @@ bazel_dep(
 )
 bazel_dep(
     name = "google_cloud_cpp",
-    version = "2.38.0",
-)
-git_override(
-    module_name = "google_cloud_cpp",
-    commit = "e9eb0d1d8295966e7bca38c0ddf6529299db4f2b",  # 2.38.0
-    remote = "https://github.com/googleapis/google-cloud-cpp.git",
+    version = "3.0.0-rc0",
 )
 
 # Below rules are needed to build google-cloud-cpp 2.38.0.
-bazel_dep(name = "googleapis", version = "0.0.0", repo_name = "com_google_googleapis")
-archive_override(
-    module_name = "googleapis",
-    integrity = "sha256-6FrMqUyMsAgBxSwG2fdSxQPx2XNg1wsVmfTxTOji900=",
-    patch_strip = 1,
-    patches = ["//riegeli/bazel:googleapis.modules.patch"],
-    strip_prefix = "googleapis-de157ca34fa487ce248eb9130293d630b501e4ad",
-    urls = [
-        "https://github.com/googleapis/googleapis/archive/de157ca34fa487ce248eb9130293d630b501e4ad.tar.gz",
-    ],
-)
-
-switched_rules = use_extension("@com_google_googleapis//:extensions.bzl", "switched_rules")
-switched_rules.use_languages(
-    cc = True,
-    grpc = True,
-)
-use_repo(switched_rules, "com_google_googleapis_imports")
+bazel_dep(name = "googleapis", version = "0.0.0-20250703-f9d6fe4a", repo_name = "com_google_googleapis")
+bazel_dep(name = "googleapis-cc", version = "1.0.0")
+bazel_dep(name = "googleapis-grpc-cc", version = "1.0.0")
 
 # Configure hermetic Python toolchain
 SUPPORTED_PYTHON_VERSIONS = [
