diff --git include/boost/thread/pthread/pthread_helpers.hpp include/boost/thread/pthread/pthread_helpers.hpp
index 68099941..2d60f829 100644
--- include/boost/thread/pthread/pthread_helpers.hpp
+++ include/boost/thread/pthread/pthread_helpers.hpp
@@ -7,6 +7,7 @@
 //  accompanying file LICENSE_1_0.txt or copy at
 //  http://www.boost.org/LICENSE_1_0.txt)
 
+#include <boost/assert.hpp>
 #include <boost/thread/detail/config.hpp>
 #include <boost/throw_exception.hpp>
 #include <pthread.h>
