--- MODULE.bazel
+++ MODULE.bazel
@@ -1,7 +1,8 @@
 ## MODULE.bazel
 module(
     name = "gz-common",
-    repo_name = "org_gazebosim_gz-common",
+    version = "6.2.0",
+    compatibility_level = 6,
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
@@ -18,17 +19,5 @@
 
 # Gazebo Dependencies
 bazel_dep(name = "rules_gazebo", version = "0.0.2")
-bazel_dep(name = "gz-utils")
-bazel_dep(name = "gz-math")
-
-archive_override(
-    module_name = "gz-utils",
-    strip_prefix = "gz-utils-gz-utils3",
-    urls = ["https://github.com/gazebosim/gz-utils/archive/refs/heads/gz-utils3.tar.gz"],
-)
-
-archive_override(
-    module_name = "gz-math",
-    strip_prefix = "gz-math-gz-math8",
-    urls = ["https://github.com/gazebosim/gz-math/archive/refs/heads/gz-math8.tar.gz"],
-)
+bazel_dep(name = "gz-utils", version = "3.1.0")
+bazel_dep(name = "gz-math", version = "8.1.1")
