diff --git a/MODULE.bazel b/MODULE.bazel
index 883fad7..6c2b705 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -4,6 +4,8 @@ bazel_dep(name = "platforms", version = "0.0.8")
 
 bazel_dep(name = "bazel_skylib", version = "1.6.1")
 
+bazel_dep(name = "rules_python", version = "0.22.1")
+
 # Note: rules_cc is special-cased repository; a dependency on rules_cc in a
 # module will not ensure that the root Bazel module has that same version of
 # rules_cc. For that reason, this primarily acts as a FYI. You'll still need
@@ -95,4 +97,6 @@ register_toolchains(
     "//bazel/toolchain:arm_gcc_win-x86_64",
     "//bazel/toolchain:arm_gcc_mac-x86_64",
     "//bazel/toolchain:arm_gcc_mac-aarch64",
+    # Require users to opt-in to the Pico SDK's toolchains.
+    dev_dependency = True,
 )
