diff --git a/bazel/example/.bazelrc b/bazel/example/.bazelrc
new file mode 100644
index 0000000..d30eca5
--- /dev/null
+++ b/bazel/example/.bazelrc
@@ -0,0 +1 @@
+build --@rules_cc//cc/toolchains:experimental_enable_rule_based_toolchains
diff --git a/bazel/example/BUILD.bazel b/bazel/example/BUILD.bazel
new file mode 100644
index 0000000..1457241
--- /dev/null
+++ b/bazel/example/BUILD.bazel
@@ -0,0 +1,5 @@
+cc_binary(
+    name = "hello_world",
+    srcs = ["hello_world.c"],
+    deps = ["@pico-sdk//src/common/pico_stdlib:pico_stdlib"],
+)
diff --git a/bazel/example/MODULE.bazel b/bazel/example/MODULE.bazel
new file mode 100644
index 0000000..d50c4bd
--- /dev/null
+++ b/bazel/example/MODULE.bazel
@@ -0,0 +1,28 @@
+bazel_dep(name = "pico-sdk", version = "1.6.0-rc1")
+
+# DO NOT copy this line to your project, this is just to make it so this example
+# can reference the rest of the repository.
+local_path_override(module_name = "pico-sdk", path = "../..")
+
+# Note: rules_cc is special-cased repository; a dependency on rules_cc in a
+# module will not ensure that the root Bazel module has that same version of
+# rules_cc. For that reason, this primarily acts as a FYI. You'll still need
+# to explicitly list this dependency in your own project's MODULE.bazel file.
+bazel_dep(name = "rules_cc", version = "0.0.9")
+
+# rules_cc v0.0.10 is not yet cut, so manually pull in the desired version.
+# This does not apply to dependent projects, so it needs to be copied to your
+# project's MODULE.bazel too.
+archive_override(
+    module_name = "rules_cc",
+    urls = "https://github.com/bazelbuild/rules_cc/archive/1acf5213b6170f1f0133e273cb85ede0e732048f.zip",
+    strip_prefix = "rules_cc-1acf5213b6170f1f0133e273cb85ede0e732048f",
+    integrity = "sha256-NddP6xi6LzsIHT8bMSVJ2NtoURbN+l3xpjvmIgB6aSg=",
+)
+
+register_toolchains(
+    "@pico-sdk//bazel/toolchain:arm_gcc_linux-x86_64",
+    "@pico-sdk//bazel/toolchain:arm_gcc_win-x86_64",
+    "@pico-sdk//bazel/toolchain:arm_gcc_mac-x86_64",
+    "@pico-sdk//bazel/toolchain:arm_gcc_mac-aarch64",
+)
diff --git a/bazel/example/hello_world.c b/bazel/example/hello_world.c
new file mode 100644
index 0000000..cf3dfd2
--- /dev/null
+++ b/bazel/example/hello_world.c
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) 2024 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdio.h>
+#include "pico/stdlib.h"
+
+int main() {
+    stdio_init_all();
+    while (true) {
+        printf("Hello, world!\n");
+        sleep_ms(1000);
+    }
+}
