diff --git a/tools/pioasm/BUILD.bazel b/tools/pioasm/BUILD.bazel
index c9fd601..c9a252e 100644
--- a/tools/pioasm/BUILD.bazel
+++ b/tools/pioasm/BUILD.bazel
@@ -4,6 +4,12 @@ package(default_visibility = ["//visibility:public"])
 
 cc_library(
     name = "pioasm_core",
+    # TODO: Only required for MSVC, but this will apply whenever Windows is
+    # targeted.
+    defines = select({
+        "@platforms//os:windows": ["YY_NO_UNISTD_H=1"],
+        "//conditions:default": [],
+    }),
     srcs = [
         "gen/lexer.cpp",
         "gen/location.h",
