--- /dev/null
+++ BUILD.bazel
<<<<<<< HEAD
@@ -0,0 +1,6 @@
+cc_library(
+    name = "robin-map",
+    hdrs = glob(["include/tsl/*.h"]),
+    strip_include_prefix = "include/",
+    visibility = ["//visibility:public"],
=======
@@ -0,0 +1,5 @@
+cc_library(
+    name = "robin_map",
+    hdrs = glob(["ext/robin_map/include/**/*.h"]),
+    strip_include_prefix = "ext/robin_map/include",
>>>>>>> 71a26d1667b06c634e6af87a04af97dce71f01e7
+)
