diff --git a/third_party/jitterentropy/jitterentropy-library/jitterentropy.h b/third_party/jitterentropy/jitterentropy-library/jitterentropy.h
index ec6f5bc28..cbcc5e438 100644
--- third_party/jitterentropy/jitterentropy-library/jitterentropy.h
+++ third_party/jitterentropy/jitterentropy-library/jitterentropy.h
@@ -162,7 +162,7 @@ extern "C" {
  * Jitter RNG State Definition Section
  ***************************************************************************/
 
-#if defined(_MSC_VER)
+#if defined(_WIN32) || defined(_WIN64)
 #include "arch/jitterentropy-base-windows.h"
 #else
 #include "jitterentropy-base-user.h"
