diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..d42d4b2
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,24 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "signals2",
+    deps = [
+        "@boost//:assert",
+        "@boost//:bind",
+        "@boost//:config",
+        "@boost//:core",
+        "@boost//:function",
+        "@boost//:iterator",
+        "@boost//:mpl",
+        "@boost//:optional",
+        "@boost//:parameter",
+        "@boost//:preprocessor",
+        "@boost//:smart_ptr",
+        "@boost//:throw_exception",
+        "@boost//:tuple",
+        "@boost//:type_traits",
+        "@boost//:variant",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..130b1b8
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.signals2",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..27483d2
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:signals2",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "signals2_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/signals2_test.cc b/bazel_test/signals2_test.cc
new file mode 100644
index 0000000..32c9e3a
--- /dev/null
+++ b/bazel_test/signals2_test.cc
@@ -0,0 +1,31 @@
+#include <boost/signals2.hpp>
+
+class FlagSetter
+{
+ public:
+  FlagSetter(bool *flag) : flag(flag) {}
+
+  void operator()()
+  {
+    *flag = true;
+  }
+
+  bool *flag;
+};
+
+int main()
+{
+  bool called = false;
+
+  boost::signals2::signal<void ()> sig;
+  FlagSetter fs (&called);
+  sig.connect(fs);
+
+  sig();
+
+  if (!called) {
+    return 1;
+  }
+
+  return 0;
+}
