diff --git a/tests/BUILD b/tests/BUILD
index f9c6c06..2bbcdcf 100644
--- a/tests/BUILD
+++ b/tests/BUILD
@@ -74,6 +74,9 @@ py_binary(
     ],
     env = {"MY_BINARY_ENV": "propagated only in Bazel 7+"},
     main = "test.py",
+    #  Tagging this no-remote because the Buildbuddy Remote Execution on the BCR CI complains
+    #  AppImage tests/appimage_py failed: Exec failed due to IOException: The file type of 'tests/dir/link_to_bin_sh' is not supported.
+    tags = ["no-remote"]
 )
 
 filegroup(
