diff --git a/modules/rules_python_gazelle_plugin/1.7.0-rc6/MODULE.bazel b/modules/rules_python_gazelle_plugin/1.7.0-rc6/MODULE.bazel
index 95a84e90..76e3c41b 100644
--- a/modules/rules_python_gazelle_plugin/1.7.0-rc6/MODULE.bazel
+++ b/modules/rules_python_gazelle_plugin/1.7.0-rc6/MODULE.bazel
@@ -7,7 +7,7 @@ module(
 bazel_dep(name = "bazel_skylib", version = "1.8.2")
 bazel_dep(name = "rules_python", version = "0.18.0")
 bazel_dep(name = "rules_go", version = "0.55.1", repo_name = "io_bazel_rules_go")
-bazel_dep(name = "gazelle", version = "0.36.0", repo_name = "bazel_gazelle")
+bazel_dep(name = "gazelle", version = "0.42.0", repo_name = "bazel_gazelle")
 bazel_dep(name = "rules_cc", version = "0.0.16")
 
 local_path_override(
