Index: MODULE.bazel
===================================================================
--- MODULE.bazel
+++ MODULE.bazel
@@ -4,9 +4,9 @@
 VERSION = "v1.34.2"
 
 module(
     name = "aspect_bazel_lib",
-    version = "0.0.0" if VERSION.startswith("$Format") else VERSION,
+    version = "1.34.2" if VERSION.startswith("$Format") else VERSION,
     compatibility_level = 1,
 )
 
 # Lower-bound versions of our dependencies
