diff --git dart/dynamics/detail/EntityNode.hpp dart/dynamics/detail/EntityNode.hpp
index 14dfd19..f532759 100644
--- dart/dynamics/detail/EntityNode.hpp
+++ dart/dynamics/detail/EntityNode.hpp
@@ -51,28 +51,25 @@ void EntityNode<Base>::setAspectProperties(
 template <class Base>
 const std::string& EntityNode<Base>::setName(const std::string& newName)
 {
-  using NameImpl = detail::EntityNodeAspectBase<Base>;
+  if (this->mAspectProperties.mName == newName && !newName.empty())
+    return this->mAspectProperties.mName;
 
-  if (NameImpl::mAspectProperties.mName == newName && !newName.empty())
-    return NameImpl::mAspectProperties.mName;
+  const std::string oldName = this->mAspectProperties.mName;
 
-  const std::string oldName = NameImpl::mAspectProperties.mName;
-
-  NameImpl::mAspectProperties.mName = Node::registerNameChange(newName);
+  this->mAspectProperties.mName = Node::registerNameChange(newName);
 
   Node::incrementVersion();
   Entity::mNameChangedSignal.raise(
-      this, oldName, NameImpl::mAspectProperties.mName);
+      this, oldName, this->mAspectProperties.mName);
 
-  return NameImpl::mAspectProperties.mName;
+  return this->mAspectProperties.mName;
 }
 
 //==============================================================================
 template <class Base>
 const std::string& EntityNode<Base>::getName() const
 {
-  using NameImpl = detail::EntityNodeAspectBase<Base>;
-  return NameImpl::mAspectProperties.mName;
+  return this->mAspectProperties.mName;
 }
 
 } // namespace dynamics
