===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,7 +1,9 @@
 """Bazel Gazelle extension for rules_cc"""
 
-module(name = "gazelle_cc")
+module(name = "gazelle_cc",
+    version = "0.4.0",
+)
 
 bazel_dep(name = "gazelle", version = "0.46.0")
 bazel_dep(name = "rules_go", version = "0.59.0")
 bazel_dep(name = "rules_proto", version = "7.1.0")
