--- MODULE.bazel
+++ MODULE.bazel
@@ -25,21 +25,7 @@
     version = "0.2.5",
 )
 
-# Override the module with the latest rules_cuda
-git_override(
-    module_name = "rules_cuda",
-    remote = "https://github.com/bazel-contrib/rules_cuda.git",
-    commit = "3692ecb",
-)
-
-# Load the extension
-cuda = use_extension("@rules_cuda//cuda:extensions.bzl", "toolchain")
-cuda.toolkit(
-    name = "cuda",
-)
-
-# Use 'cuda' as the repository name
-# use_repo(cuda, "cuda")
-
-# Alias local_cuda
-use_repo(cuda, local_cuda = "cuda")
+# cuda toolchain
+local_cuda = use_extension("@rules_cuda//cuda:extensions.bzl", "toolchain")
+local_cuda.local_toolchain()
+use_repo(local_cuda, "local_cuda")
