diff --git a/MODULE.bazel b/MODULE.bazel
index df0011b..3481fc8 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,10 +1,11 @@
 module(
     name = "picotool",
-    version = "2.2.1-develop",
+    version = "2.2.1-develop.bcr.20260407-d1a8d35",
+    bazel_compatibility = [">=8.0.0"],
 )
 
 bazel_dep(name = "rules_libusb", version = "0.1.0-rc2")
-bazel_dep(name = "pico-sdk", version = "2.1.0")
+bazel_dep(name = "pico-sdk", version = "2.2.2-develop.bcr.20260420-79b38e95")
 bazel_dep(name = "rules_cc", version = "0.0.9")
 bazel_dep(name = "bazel_skylib", version = "1.6.1")
 bazel_dep(name = "rules_python", version = "0.22.1")
