diff --git a/include/boost/charconv/detail/type_traits.hpp b/include/boost/charconv/detail/type_traits.hpp
index 4160555..54b6d54 100644
--- a/include/boost/charconv/detail/type_traits.hpp
+++ b/include/boost/charconv/detail/type_traits.hpp
@@ -6,6 +6,7 @@
 #define BOOST_CHARCONV_DETAIL_TYPE_TRAITS_HPP
 
 #include <boost/charconv/detail/config.hpp>
+#include "emulated128.hpp"
 #include <type_traits>
 
 namespace boost { namespace charconv { namespace detail {
