From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matt Clarkson <mattyclarkson@gmail.com>
Date: Thu, 9 May 2024 09:48:34 +0100
Subject: [PATCH] build: add floating point header to C dependencies

---
 dh/x25519/BUILD.bazel | 2 ++
 dh/x448/BUILD.bazel   | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/dh/x25519/BUILD.bazel b/dh/x25519/BUILD.bazel
index bc34c40..39ec6da 100644
--- a/dh/x25519/BUILD.bazel
+++ b/dh/x25519/BUILD.bazel
@@ -13,6 +13,8 @@ go_library(
         "key.go",
         "table.go",
     ],
+    cgo = True,
+    cdeps = ["//math/fp25519:header"],
     importpath = "github.com/cloudflare/circl/dh/x25519",
     visibility = ["//visibility:public"],
     deps = [
diff --git a/dh/x448/BUILD.bazel b/dh/x448/BUILD.bazel
index c9c94aa..e8a73c4 100644
--- a/dh/x448/BUILD.bazel
+++ b/dh/x448/BUILD.bazel
@@ -13,6 +13,8 @@ go_library(
         "key.go",
         "table.go",
     ],
+    cgo = True,
+    cdeps = ["//math/fp448:header"],
     importpath = "github.com/cloudflare/circl/dh/x448",
     visibility = ["//visibility:public"],
     deps = [
