From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matt Clarkson <mattyclarkson@gmail.com>
Date: Thu, 9 May 2024 09:53:05 +0100
Subject: [PATCH] test: add `pure = "on"` tests

---
 dh/x25519/BUILD.bazel | 16 ++++++++++++++++
 dh/x448/BUILD.bazel   | 16 ++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/dh/x25519/BUILD.bazel b/dh/x25519/BUILD.bazel
index 39ec6da..49efad5 100644
--- a/dh/x25519/BUILD.bazel
+++ b/dh/x25519/BUILD.bazel
@@ -41,3 +41,19 @@ go_test(
         "//math/fp25519",
     ],
 )
+
+go_test(
+    name = "x25519_pure_test",
+    srcs = [
+        "curve_test.go",
+        "key_test.go",
+    ],
+    pure = "on",
+    data = glob(["testdata/**"]),
+    embed = [":x25519"],
+    deps = [
+        "//internal/conv",
+        "//internal/test",
+        "//math/fp25519",
+    ],
+)
diff --git a/dh/x448/BUILD.bazel b/dh/x448/BUILD.bazel
index e8a73c4..7fd50f2 100644
--- a/dh/x448/BUILD.bazel
+++ b/dh/x448/BUILD.bazel
@@ -41,3 +41,19 @@ go_test(
         "//math/fp448",
     ],
 )
+
+go_test(
+    name = "x448_pure_test",
+    srcs = [
+        "curve_test.go",
+        "key_test.go",
+    ],
+    pure = "on",
+    data = glob(["testdata/**"]),
+    embed = [":x448"],
+    deps = [
+        "//internal/conv",
+        "//internal/test",
+        "//math/fp448",
+    ],
+)
