--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,65 @@
+""" Builds libparserutils.
+"""
+
+_LIBPARSERUTILS_PUB_HEADERS = [
+    "include/parserutils/parserutils.h",
+    "include/parserutils/errors.h",
+    "include/parserutils/functypes.h",
+    "include/parserutils/types.h",
+    "include/parserutils/charset/mibenum.h",
+    "include/parserutils/charset/codec.h",
+    "include/parserutils/charset/utf8.h",
+    "include/parserutils/charset/utf16.h",
+    "include/parserutils/input/inputstream.h",
+    "include/parserutils/utils/buffer.h",
+    "include/parserutils/utils/stack.h",
+    "include/parserutils/utils/vector.h",
+]
+
+cc_library(
+    name = "parserutils_local_includes",
+    srcs = [
+        "src/charset/aliases.h",
+        "src/charset/aliases.inc",
+        "src/charset/aliases.c",
+        "src/charset/codec.c",
+        "src/charset/codecs/codec_impl.h",
+        "src/charset/codecs/codec_ascii.c",
+        "src/charset/codecs/codec_8859.c",
+        "src/charset/codecs/8859_tables.h",
+        "src/charset/codecs/codec_ext8.c",
+        "src/charset/codecs/ext8_tables.h",
+        "src/charset/codecs/codec_utf8.c",
+        "src/charset/codecs/codec_utf16.c",
+        "src/charset/encodings/utf8impl.h",
+        "src/charset/encodings/utf8.c",
+        "src/charset/encodings/utf16.c",
+        "src/utils/endian.h",
+        "src/utils/utils.h",
+        "src/utils/buffer.c",
+        "src/utils/stack.c",
+        "src/utils/errors.c",
+        "src/utils/vector.c",
+        "src/input/filter.c",
+        "src/input/filter.h",
+        "src/input/inputstream.c",
+    ],
+    hdrs = _LIBPARSERUTILS_PUB_HEADERS,
+    includes = ["src", "include"],
+    copts = ["-w"],
+    visibility = ["//visibility:private"],
+)
+
+cc_library(
+    name = "parserutils",
+    implementation_deps = ["parserutils_local_includes"],
+    hdrs = _LIBPARSERUTILS_PUB_HEADERS,
+    includes = ["include"],
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "libparserutils",
+    actual = ":parserutils",
+    visibility = ["//visibility:public"],
+)
