From 0957fcccf6618cba3bfc548968d3bc712d529db8 Mon Sep 17 00:00:00 2001
From: Corentin Le Molgat <corentinl@google.com>
Date: Fri, 10 Jul 2026 14:49:13 +0200
Subject: [PATCH 2/2] fix_build_bazel

---
 BUILD.bazel | 1 +
 1 file changed, 1 insertion(+)

diff --git a/BUILD.bazel b/BUILD.bazel
index 6a6589bc46..50ccd4ccd1 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -118,6 +118,7 @@ cc_library(
     ],
     linkopts = select({
         "@rules_cc//cc/compiler:msvc-cl": ["-DEFAULTLIB:shell32.lib"],
+        "@platforms//os:linux": ["-Wl,--no-as-needed -ldl"],
         "//conditions:default": ["-lpthread"],
     }),
     visibility = ["//visibility:public"],
