--- BUILD.bazel
+++ BUILD.bazel
@@ -1,3 +1,5 @@
+load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
 load("@rules_cc//cc:defs.bzl", "cc_library")
 
 cc_library(
@@ -18,26 +20,23 @@
     ],
 )
 
-genrule(
+write_file(
+    name = "generate_config_h",
+    out = "include/opentracing/config.h",
+    content = [
+        "#pragma once",
+        "",
+        "#define OPENTRACING_BUILD_DYNAMIC_LOADING",
+        "",
+    ],
+)
+
+expand_template(
     name = "generate_version_h",
-    srcs = glob([
-        "*",
-        "cmake/*",
-        "src/**/*.cpp",
-    ]),
-    outs = [
-      "include/opentracing/config.h",
-      "include/opentracing/version.h"
-    ],
-    cmd = """
-    TEMP_DIR=$$(mktemp -d)
-    CONFIG_H_OUT=$${PWD}/$(location :include/opentracing/config.h)
-    VERSION_H_OUT=$${PWD}/$(location :include/opentracing/version.h)
-    OPENTRACING_ROOT=$$(dirname $${PWD}/$(location :CMakeLists.txt))
-    cd $$TEMP_DIR
-    cmake -DBUILD_TESTING=OFF -DBUILD_MOCKTRACER=OFF -L $$OPENTRACING_ROOT
-    mv include/opentracing/config.h $$CONFIG_H_OUT
-    mv include/opentracing/version.h $$VERSION_H_OUT
-    rm -rf $$TEMP_DIR
-    """,
+    out = "include/opentracing/version.h",
+    substitutions = {
+        "${OPENTRACING_ABI_VERSION}": "3",
+        "${OPENTRACING_VERSION_STRING}": "1.6.0",
+    },
+    template = "version.h.in",
 )
