--- /dev/null
+++ BUILD
@@ -0,0 +1,41 @@
+cc_library(
+    name = "utf8proc",
+    srcs = ["utf8proc.c"],
+    hdrs = ["utf8proc.h"],
+    defines = ["UTF8PROC_STATIC"],
+    # includes = ["."],
+    textual_hdrs = ["utf8proc_data.c"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "utf8proc_testlib",
+    testonly = True,
+    srcs = ["test/tests.c"],
+    hdrs = ["test/tests.h"],
+    deps = [":utf8proc"],
+)
+
+[
+    cc_test(
+        name = testcase + "_test",
+        srcs = ["test/" + testcase + ".c"],
+        deps = [
+            ":utf8proc",
+            ":utf8proc_testlib",
+        ],
+    )
+    for testcase in [
+        "case",
+        # "charwidth",
+        "custom",
+        # "graphemetest",
+        # "iscase",
+        "iterate",
+        "maxdecomposition",
+        "misc",
+        # "normtest",
+        "printproperty",
+        "valid",
+    ]
+]
