diff --git a/icetime/timings.py b/icetime/timings.py
index 136f604..ef78ccd 100644
--- a/icetime/timings.py
+++ b/icetime/timings.py
@@ -1,5 +1,6 @@
 #!/usr/bin/env python3
 
+import os
 import re
 import sys
 
@@ -56,6 +57,10 @@ if len(sys.argv) >= 2:
 else:
 	chips = "lp384 lp1k lp8k hx1k hx8k up5k".split()
 
-for db in chips:
-    with open("../icefuzz/timings_%s.txt" % db, "r") as f:
-        timings_to_c(db, f);
+if "ICESTORM_TIMINGS_TEMPLATE" in os.environ:
+    with open(os.environ["ICESTORM_TIMINGS_TEMPLATE"], "r") as f:
+        timings_to_c(chips[0], f);
+else:
+    for db in chips:
+        with open("../icefuzz/timings_%s.txt" % db, "r") as f:
+            timings_to_c(db, f);
