From 9a8ba6ed1c60b77f05c8163e0b6a5f23aeecb993 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bj=C3=B6rn=20Esser?= <besser82@fedoraproject.org>
Date: Tue, 18 Aug 2020 21:29:22 +0200
Subject: [PATCH] test/short-outbuf.c: Do not include <crypt.h> directly.

---
 test/short-outbuf.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/test/short-outbuf.c b/test/short-outbuf.c
index 89422923..3f708a2e 100644
--- a/test/short-outbuf.c
+++ b/test/short-outbuf.c
@@ -17,7 +17,6 @@
  */
 
 #include "crypt-port.h"
-#include <crypt.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
