--- a/bazel/private/oss/toolchains/prebuilt/tool_integrity.bzl
+++ b/bazel/private/oss/toolchains/prebuilt/tool_integrity.bzl
@@ -1,4 +1,13 @@
 """Pre-computed during release"""
 RELEASE_VERSION = "v36.1"
 RELEASED_BINARY_INTEGRITY = {
+  "protoc-36.1-linux-aarch_64.zip": "237a68856edf1bd28b6204bddd0596c1cf46d298bc29c620012540b2e44c73e7",
+  "protoc-36.1-linux-ppcle_64.zip": "ecf722a3df8446af59c3c2f01f52978271eff396fccd48d8e793a87b551518cc",
+  "protoc-36.1-linux-s390_64.zip": "dd1bc36cd2efa7ecfa21ae03507d2e00db46f1f05e817d9bd339eb5efe6ab1e8",
+  "protoc-36.1-linux-x86_32.zip": "b8875ab09a8e9f72de78d257e250c8e7e5d041dfbded649ef5f5320c686ee994",
+  "protoc-36.1-linux-x86_64.zip": "c4bc672d9d49214dc8cafdceadf4df92182d6ca8e3ec65a56b2d7de5602669b4",
+  "protoc-36.1-osx-aarch_64.zip": "de56d57afe30c5d191b11d24ff93dd4025728d7fb43b773886b2d3613e0bdbb2",
+  "protoc-36.1-osx-x86_64.zip": "ee2c5496e4af0aa6a224894bc0f7025145260e004d890487d510725ce8b473eb",
+  "protoc-36.1-win32.zip": "9f1da00ed49ee2eab0520c354c7edbfa69f567a9247cf81a7995fe5db27ceb9e",
+  "protoc-36.1-win64.zip": "390e515cb456e6a978553bdb57baf087b054885077fd6da7f7ff0160279c07d6",
 }
