--- MODULE.bazel
+++ MODULE.bazel
@@ -1,7 +1,8 @@
 ## MODULE.bazel
 module(
     name = "gz-common",
-    repo_name = "org_gazebosim_gz-common",
+    version = "7.0.0-pre1",
+    compatibility_level = 6,
 )

 bazel_dep(name = "bazel_skylib", version = "1.7.1")
@@ -18,17 +19,5 @@

 # Gazebo Dependencies
 bazel_dep(name = "rules_gazebo", version = "0.0.2")
-bazel_dep(name = "gz-utils")
-bazel_dep(name = "gz-math")
-
-archive_override(
-    module_name = "gz-utils",
-    strip_prefix = "gz-utils-main",
-    urls = ["https://github.com/gazebosim/gz-utils/archive/refs/heads/main.tar.gz"],
-)
-
-archive_override(
-    module_name = "gz-math",
-    strip_prefix = "gz-math-main",
-    urls = ["https://github.com/gazebosim/gz-math/archive/refs/heads/main.tar.gz"],
-)
+bazel_dep(name = "gz-utils", version = "4.0.0-pre1")
+bazel_dep(name = "gz-math", version = "9.0.0-pre1")
