From 7250d519d78844b2739fd8e836dea2976363decf Mon Sep 17 00:00:00 2001
From: Patrick Balestra <me@patrickbalestra.com>
Date: Tue, 27 Sep 2022 13:20:25 -0700
Subject: [PATCH] Skip `swizzle_absolute_xcttestsourcelocation` on non-macOS
 platforms

Use platforms
---
 lib/BUILD | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lib/BUILD b/lib/BUILD
index 024dc9b..c2ecf78 100644
--- a/lib/BUILD
+++ b/lib/BUILD
@@ -11,6 +11,12 @@ objc_library(
     name = "swizzle_absolute_xcttestsourcelocation",
     testonly = True,
     srcs = ["swizzle_absolute_xcttestsourcelocation.m"],
+    target_compatible_with = [
+        "@platforms//os:macos",
+        "@platforms//os:watchos",
+        "@platforms//os:ios",
+        "@platforms//os:tvos",
+    ],
     visibility = ["//visibility:public"],
     alwayslink = True,
 )
-- 
2.32.0 (Apple Git-132)

