--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,69 @@
+SRCS = [
+    "build_log",
+    "build",
+    "clean",
+    "clparser",
+    "debug_flags",
+    "depfile_parser",  # from re2c
+    "deps_log",
+    "disk_interface",
+    "dyndep_parser",
+    "dyndep",
+    "edit_distance",
+    "eval_env",
+    "graph",
+    "graphviz",
+    "json",
+    "lexer",  # from re2c
+    "line_printer",
+    "manifest_parser",
+    "metrics",
+    "missing_deps",
+    "parser",
+    "state",
+    "status",
+    "string_piece_util",
+    "util",
+    "version",
+]
+
+CC_SRCS = ["src/%s.cc" % (src,) for src in SRCS]
+
+H_SRCS = ["src/%s.h" % (src,) for src in SRCS]
+
+cc_binary(
+    name = "ninja",
+    srcs = [
+        "src/ninja.cc",
+    ] + CC_SRCS + H_SRCS + [
+        "src/browse.h",
+        "src/exit_status.h",
+        "src/hash_map.h",
+        "src/includes_normalize.h",
+        "src/load_status.h",
+        "src/string_piece.h",
+        "src/subprocess.h",
+        "src/timestamp.h",
+    ] + select({
+        "@platforms//os:windows": [
+            "src/getopt.c",
+            "src/getopt.h",
+            "src/includes_normalize-win32.cc",
+            "src/minidump-win32.cc",
+            "src/msvc_helper.h",
+            "src/msvc_helper-win32.cc",
+            "src/msvc_helper_main-win32.cc",
+            "src/subprocess-win32.cc",
+            "src/win32port.h",
+        ],
+        "//conditions:default": [
+            "src/subprocess-posix.cc",
+        ],
+    }),
+    copts = select({
+        "@platforms//os:freebsd": ["-DUSE_PPOLL"],
+        "@platforms//os:linux": ["-DUSE_PPOLL"],
+        "@platforms//os:openbsd": ["-DUSE_PPOLL"],
+        "//conditions:default": [],
+    }),
+)
