diff --git a/MODULE.bazel b/MODULE.bazel
index c20056986..061c9ee83 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -94,3 +94,13 @@ pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
 use_repo(pip, "grpc_python_dependencies")
 
 bazel_dep(name = "cython", version = "3.0.11-1")
+
+http_archive = use_repo_rule("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "typing_extensions",
+    build_file = "//third_party:typing_extensions.BUILD",
+    sha256 = "bf6f56b36d8bc9156e518eb1cc37a146284082fa53522033f772aefbecfd15fc",
+    strip_prefix = "typing_extensions-4.12.2",
+    url = "https://github.com/python/typing_extensions/archive/4.12.2.tar.gz",
+)
