--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,14 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
+
+cc_library(
+    name = "dylib",
+    hdrs = ["include/dylib.hpp"],
+    copts = ["-std=c++11"],
+    includes = ["include"],
+    linkopts = select({
+        "@platforms//os:linux": ["-ldl"],
+        "@platforms//os:macos": ["-ldl"],
+        "//conditions:default": [],
+    }),
+    visibility = ["//visibility:public"],
+)
