--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,39 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
+
+cc_library(
+    name = "dylib",
+    hdrs = ["include/dylib.hpp"],
+    copts = ["-std=c++11"],
+    includes = ["include"],
+    linkopts = select({
+        "@platforms//os:linux": ["-ldl"],
+        "//conditions:default": [],
+    }),
+    visibility = ["//visibility:public"],
+)
+
+
+# Example usage and tests
+cc_binary(
+    name = "dynamic_lib",
+    srcs = ["tests/lib.cpp"],
+    copts = [
+        "-std=c++17",
+        "-fPIC",
+    ],
+    linkshared = True,
+    visibility = ["//visibility:private"],
+)
+
+cc_test(
+    name = "unit_tests",
+    srcs = ["tests/tests.cpp"],
+    copts = ["-std=c++17"],
+    data = [":dynamic_lib"],
+    visibility = ["//visibility:private"],
+    deps = [
+        ":dylib",
+        "@googletest//:gtest",
+        "@googletest//:gtest_main",
+    ],
+)
