diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..2b97efb
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,29 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "math",
+    srcs = [
+        # See https://github.com/boostorg/math/blob/boost-1.56.0/doc/overview/roadmap.qbk#L11-L14
+        # "libs/math/include_private/boost/math/tools/remez.hpp", # TODO...
+        # "libs/math/include_private/boost/math/constants/generate.hpp",
+        # "libs/math/include_private/boost/math/tools/solve.hpp",
+        # "libs/math/include_private/boost/math/tools/test.hpp",
+    ],
+    # includes = ["include_private"], # TODO
+    deps = [
+        # https://github.com/boostorg/lexical_cast/issues/27
+        #":lexical_cast",
+        # https://github.com/boostorg/math/issues/201
+        #":multiprecision",
+        "@boost.assert//:assert",
+        "@boost.concept_check//:concept_check",
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.integer//:integer",
+        "@boost.lexical_cast//:lexical_cast",
+        "@boost.predef//:predef",
+        "@boost.random//:random",
+        "@boost.static_assert//:static_assert",
+        "@boost.throw_exception//:throw_exception",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..48ff446
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,18 @@
+module(
+    name = "boost.math",
+    version = "1.83.0.bzl.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.concept_check", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.integer", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.lexical_cast", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.predef", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.random", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.throw_exception", version = "1.83.0.bzl.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
