diff --git a/MODULE.bazel b/MODULE.bazel
index 184d526..0ac710f 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,6 @@
 module(
-    name = "ra",
-    version = "2.4.0",
+    name = "rabbitmq_ra",
+    version = "2.6.2",
 )
 
 bazel_dep(
@@ -94,7 +94,7 @@ erlang_dev_package = use_extension(
 erlang_package.hex_package(
     name = "eunit_formatters",
     version = "0.5.0",
-    build_file = "@ra//bzl:BUILD.eunit_formatters",
+    build_file = "@rabbitmq_ra//bzl:BUILD.eunit_formatters",
 )
 
 use_repo(
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
index 51fb71d..3bad5fd 100644
--- a/WORKSPACE.bazel
+++ b/WORKSPACE.bazel
@@ -1,3 +1,5 @@
+workspace(name = "rabbitmq_ra")
+
 load(
     "@bazel_tools//tools/build_defs/repo:http.bzl",
     "http_archive",
