Index: MODULE.bazel
===================================================================
--- MODULE.bazel
+++ MODULE.bazel
@@ -2,9 +2,9 @@
 
 module(
     name = "aspect_bazel_lib",
     compatibility_level = 1,
-    version = "0.0.0",
+    version = "1.31.0",
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.4.1")
 bazel_dep(name = "bazel_skylib_gazelle_plugin", version = "1.4.1", dev_dependency = True)
