--- /dev/null
+++ expand_template.bzl
@@ -0,0 +1,19 @@
+def expand_template_impl(ctx):
+  ctx.actions.expand_template(
+      template = ctx.file.template,
+      output = ctx.outputs.out,
+      substitutions = ctx.attr.substitutions,
+      is_executable = False,
+  )
+
+expand_template = rule(
+    attrs = {
+        "template": attr.label(
+            mandatory = True,
+            allow_single_file = True,
+        ),
+        "substitutions": attr.string_dict(mandatory = True),
+        "out": attr.output(mandatory = True),
+    },
+    implementation = expand_template_impl,
+)
