commit 63f89cd3d471e7e81b51dc7e7205b201fed70fc1
Author: Douglas Thor <doug.thor@gmail.com>
Date:   Thu Nov 21 04:55:12 2024 +0000

    Update BUILD files to fully work

diff --git a/BUILD.bazel b/BUILD.bazel
index 699c796..ff03bee 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -4,6 +4,16 @@ load("@gazelle//:def.bzl", "gazelle")
 # gazelle:prefix github.com/smacker/go-tree-sitter
 gazelle(name = "gazelle")
 
+filegroup(
+    name = "common_libs",
+    srcs = [
+        "alloc.h",
+        "api.h",
+        "array.h",
+    ],
+    visibility = [":__subpackages__"],
+)
+
 go_library(
     name = "go-tree-sitter",
     srcs = [
diff --git a/bash/BUILD.bazel b/bash/BUILD.bazel
index 3e627b6..f40cda0 100644
--- a/bash/BUILD.bazel
+++ b/bash/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/bash",
diff --git a/cpp/BUILD.bazel b/cpp/BUILD.bazel
index 4dd649a..321c9ce 100644
--- a/cpp/BUILD.bazel
+++ b/cpp/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/cpp",
diff --git a/csharp/BUILD.bazel b/csharp/BUILD.bazel
index d8995a6..5f42dc7 100644
--- a/csharp/BUILD.bazel
+++ b/csharp/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     copts = ["-Wno-trigraphs"],
diff --git a/css/BUILD.bazel b/css/BUILD.bazel
index 2d26337..a5cf63e 100644
--- a/css/BUILD.bazel
+++ b/css/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/css",
diff --git a/cue/BUILD.bazel b/cue/BUILD.bazel
index cc673af..69389ad 100644
--- a/cue/BUILD.bazel
+++ b/cue/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/cue",
diff --git a/dockerfile/BUILD.bazel b/dockerfile/BUILD.bazel
index 16fb48c..9fc9dad 100644
--- a/dockerfile/BUILD.bazel
+++ b/dockerfile/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/dockerfile",
diff --git a/elixir/BUILD.bazel b/elixir/BUILD.bazel
index 5a9e7bb..6770a81 100644
--- a/elixir/BUILD.bazel
+++ b/elixir/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/elixir",
diff --git a/elm/BUILD.bazel b/elm/BUILD.bazel
index d4e7a85..cfc3e0a 100644
--- a/elm/BUILD.bazel
+++ b/elm/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     cxxopts = ["-std=gnu++11"],
diff --git a/groovy/BUILD.bazel b/groovy/BUILD.bazel
index 6d319e8..198b4b2 100644
--- a/groovy/BUILD.bazel
+++ b/groovy/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/groovy",
diff --git a/hcl/BUILD.bazel b/hcl/BUILD.bazel
index da246d9..2a948a1 100644
--- a/hcl/BUILD.bazel
+++ b/hcl/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/hcl",
diff --git a/html/BUILD.bazel b/html/BUILD.bazel
index b0c80b9..1c79570 100644
--- a/html/BUILD.bazel
+++ b/html/BUILD.bazel
@@ -8,6 +8,7 @@ go_library(
         "parser.h",
         "scanner.c",
         "tag.h",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/html",
diff --git a/javascript/BUILD.bazel b/javascript/BUILD.bazel
index 2e96709..1251c4d 100644
--- a/javascript/BUILD.bazel
+++ b/javascript/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/javascript",
diff --git a/kotlin/BUILD.bazel b/kotlin/BUILD.bazel
index 37b959d..5ea86c8 100644
--- a/kotlin/BUILD.bazel
+++ b/kotlin/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/kotlin",
diff --git a/lua/BUILD.bazel b/lua/BUILD.bazel
index 2f44982..e259c36 100644
--- a/lua/BUILD.bazel
+++ b/lua/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/lua",
diff --git a/markdown/tree-sitter-markdown-inline/BUILD.bazel b/markdown/tree-sitter-markdown-inline/BUILD.bazel
index 2fa6825..edd9bd5 100644
--- a/markdown/tree-sitter-markdown-inline/BUILD.bazel
+++ b/markdown/tree-sitter-markdown-inline/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/markdown/tree-sitter-markdown-inline",
diff --git a/markdown/tree-sitter-markdown/BUILD.bazel b/markdown/tree-sitter-markdown/BUILD.bazel
index b7ee9a7..3877890 100644
--- a/markdown/tree-sitter-markdown/BUILD.bazel
+++ b/markdown/tree-sitter-markdown/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/markdown/tree-sitter-markdown",
diff --git a/php/BUILD.bazel b/php/BUILD.bazel
index 56775f0..e66c6d9 100644
--- a/php/BUILD.bazel
+++ b/php/BUILD.bazel
@@ -8,6 +8,9 @@ go_library(
         "parser.h",
         "scanner.c",
         "scanner.h",
+        "tree_sitter/alloc.h",
+        "tree_sitter/array.h",
+        "tree_sitter/parser.h",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/php",
diff --git a/python/BUILD.bazel b/python/BUILD.bazel
index 52c6fbc..00245e3 100644
--- a/python/BUILD.bazel
+++ b/python/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/python",
diff --git a/ruby/BUILD.bazel b/ruby/BUILD.bazel
index bd5ac03..794ea52 100644
--- a/ruby/BUILD.bazel
+++ b/ruby/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/ruby",
diff --git a/rust/BUILD.bazel b/rust/BUILD.bazel
index 58bfb5d..20f8c8b 100644
--- a/rust/BUILD.bazel
+++ b/rust/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
         "parser.c",
         "parser.h",
         "scanner.c",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/rust",
diff --git a/scala/BUILD.bazel b/scala/BUILD.bazel
index e06084c..b0d400b 100644
--- a/scala/BUILD.bazel
+++ b/scala/BUILD.bazel
@@ -8,6 +8,7 @@ go_library(
         "parser.h",
         "scanner.c",
         "stack.h",
+        "//:common_libs",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/scala",
diff --git a/sql/BUILD.bazel b/sql/BUILD.bazel
index bd9b93b..005eb99 100644
--- a/sql/BUILD.bazel
+++ b/sql/BUILD.bazel
@@ -6,6 +6,7 @@ go_library(
         "binding.go",
         "parser.c",
         "scanner.c",
+        "tree_sitter/parser.h",
     ],
     cgo = True,
     importpath = "github.com/smacker/go-tree-sitter/sql",
