commit 8470bc707f806291efd6f4e1f199c5c43df55000
Author: Douglas Thor <doug.thor@gmail.com>
Date:   Thu Nov 21 20:55:56 2024 +0000

    Add exports_files

diff --git a/BUILD.bazel b/BUILD.bazel
index ff03bee..9af537d 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -4,6 +4,11 @@ load("@gazelle//:def.bzl", "gazelle")
 # gazelle:prefix github.com/smacker/go-tree-sitter
 gazelle(name = "gazelle")
 
+exports_files([
+    "go.mod",
+    "go.sum",
+])
+
 filegroup(
     name = "common_libs",
     srcs = [
