diff --git a/BUILD b/BUILD
index 9a1eba0a3..17faf762c 100644
--- a/BUILD
+++ b/BUILD
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("@bazel_skylib//lib:selects.bzl", "selects")
 load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
 load(
diff --git a/bazel/cc_grpc_library.bzl b/bazel/cc_grpc_library.bzl
index 9f7e21be0..164aec93e 100644
--- a/bazel/cc_grpc_library.bzl
+++ b/bazel/cc_grpc_library.bzl
@@ -29,6 +29,7 @@ def cc_grpc_library(
         allow_deprecated = False,
         use_external = False,  # @unused
         grpc_only = False,
+        plugin_flags = [],
         **kwargs):
     """Generates C++ grpc classes for services defined in a proto file.
 
@@ -110,6 +111,7 @@ def cc_grpc_library(
             well_known_protos = well_known_protos,
             generate_mocks = generate_mocks,
             allow_deprecated = allow_deprecated,
+            flags = plugin_flags,
             **kwargs
         )
 
diff --git a/examples/cpp/auth/BUILD b/examples/cpp/auth/BUILD
index 127046e72..3d989cdd9 100644
--- a/examples/cpp/auth/BUILD
+++ b/examples/cpp/auth/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 licenses(["notice"])
 
 cc_library(
diff --git a/examples/cpp/cancellation/BUILD b/examples/cpp/cancellation/BUILD
index 314c02911..d6889dde5 100644
--- a/examples/cpp/cancellation/BUILD
+++ b/examples/cpp/cancellation/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/compression/BUILD b/examples/cpp/compression/BUILD
index 1f122c955..3419b429e 100644
--- a/examples/cpp/compression/BUILD
+++ b/examples/cpp/compression/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/csm/BUILD b/examples/cpp/csm/BUILD
index 27d8bbace..b5fbdf764 100644
--- a/examples/cpp/csm/BUILD
+++ b/examples/cpp/csm/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/csm/observability/BUILD b/examples/cpp/csm/observability/BUILD
index 7dbb43432..7b795a644 100644
--- a/examples/cpp/csm/observability/BUILD
+++ b/examples/cpp/csm/observability/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/deadline/BUILD b/examples/cpp/deadline/BUILD
index 314c02911..d6889dde5 100644
--- a/examples/cpp/deadline/BUILD
+++ b/examples/cpp/deadline/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/debugging/BUILD b/examples/cpp/debugging/BUILD
index 4fe57c9be..ff0ed9a09 100644
--- a/examples/cpp/debugging/BUILD
+++ b/examples/cpp/debugging/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/default_event_engine/BUILD b/examples/cpp/default_event_engine/BUILD
index 9f0e75a00..c787dfcdd 100644
--- a/examples/cpp/default_event_engine/BUILD
+++ b/examples/cpp/default_event_engine/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/error_details/BUILD b/examples/cpp/error_details/BUILD
index 1374d5b6f..569d85a16 100644
--- a/examples/cpp/error_details/BUILD
+++ b/examples/cpp/error_details/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/error_handling/BUILD b/examples/cpp/error_handling/BUILD
index 0ba1bb73a..d30f38bc5 100644
--- a/examples/cpp/error_handling/BUILD
+++ b/examples/cpp/error_handling/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/flow_control/BUILD b/examples/cpp/flow_control/BUILD
index 050dc1a5d..410641627 100644
--- a/examples/cpp/flow_control/BUILD
+++ b/examples/cpp/flow_control/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/gcp_observability/helloworld/BUILD b/examples/cpp/gcp_observability/helloworld/BUILD
index 283f862fe..ae75be533 100644
--- a/examples/cpp/gcp_observability/helloworld/BUILD
+++ b/examples/cpp/gcp_observability/helloworld/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/generic_api/BUILD b/examples/cpp/generic_api/BUILD
index 46c155b60..faa48494c 100644
--- a/examples/cpp/generic_api/BUILD
+++ b/examples/cpp/generic_api/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/health/BUILD b/examples/cpp/health/BUILD
index 1f273885e..8f802b1ae 100644
--- a/examples/cpp/health/BUILD
+++ b/examples/cpp/health/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/helloworld/BUILD b/examples/cpp/helloworld/BUILD
index 595802873..213f10c30 100644
--- a/examples/cpp/helloworld/BUILD
+++ b/examples/cpp/helloworld/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/interceptors/BUILD b/examples/cpp/interceptors/BUILD
index 3983d3135..72e04e458 100644
--- a/examples/cpp/interceptors/BUILD
+++ b/examples/cpp/interceptors/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/keepalive/BUILD b/examples/cpp/keepalive/BUILD
index 238bc9c32..3523fedc9 100644
--- a/examples/cpp/keepalive/BUILD
+++ b/examples/cpp/keepalive/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/load_balancing/BUILD b/examples/cpp/load_balancing/BUILD
index 60cbc0be0..255b35306 100644
--- a/examples/cpp/load_balancing/BUILD
+++ b/examples/cpp/load_balancing/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/metadata/BUILD b/examples/cpp/metadata/BUILD
index c1aabb535..a040ed00a 100644
--- a/examples/cpp/metadata/BUILD
+++ b/examples/cpp/metadata/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/multiplex/BUILD b/examples/cpp/multiplex/BUILD
index 7cf89711a..bdad84f18 100644
--- a/examples/cpp/multiplex/BUILD
+++ b/examples/cpp/multiplex/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/orca/BUILD b/examples/cpp/orca/BUILD
index 79cebf95a..27af503d2 100644
--- a/examples/cpp/orca/BUILD
+++ b/examples/cpp/orca/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/otel/BUILD b/examples/cpp/otel/BUILD
index ae8eaa116..c967dc66e 100644
--- a/examples/cpp/otel/BUILD
+++ b/examples/cpp/otel/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 licenses(["notice"])
 
 package(
diff --git a/examples/cpp/otel/codelab/BUILD b/examples/cpp/otel/codelab/BUILD
index 6447b167a..c65871a33 100644
--- a/examples/cpp/otel/codelab/BUILD
+++ b/examples/cpp/otel/codelab/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 licenses(["notice"])
 
 package(
diff --git a/examples/cpp/otel/ostream/BUILD b/examples/cpp/otel/ostream/BUILD
index b488633db..e62f85abf 100644
--- a/examples/cpp/otel/ostream/BUILD
+++ b/examples/cpp/otel/ostream/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/reflection/BUILD b/examples/cpp/reflection/BUILD
index 9c0f8512b..d8e40c110 100644
--- a/examples/cpp/reflection/BUILD
+++ b/examples/cpp/reflection/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/retry/BUILD b/examples/cpp/retry/BUILD
index 6be50dfed..9fae299b8 100644
--- a/examples/cpp/retry/BUILD
+++ b/examples/cpp/retry/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/route_guide/BUILD b/examples/cpp/route_guide/BUILD
index 4315ad137..d24570aba 100644
--- a/examples/cpp/route_guide/BUILD
+++ b/examples/cpp/route_guide/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 licenses(["notice"])
 
 cc_library(
diff --git a/examples/cpp/systemd_socket_activation/BUILD b/examples/cpp/systemd_socket_activation/BUILD
index 525780292..80866c224 100644
--- a/examples/cpp/systemd_socket_activation/BUILD
+++ b/examples/cpp/systemd_socket_activation/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/unix_abstract_sockets/BUILD b/examples/cpp/unix_abstract_sockets/BUILD
index 8163cd96f..095087851 100644
--- a/examples/cpp/unix_abstract_sockets/BUILD
+++ b/examples/cpp/unix_abstract_sockets/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/wait_for_ready/BUILD b/examples/cpp/wait_for_ready/BUILD
index aa89a5d8d..96521579c 100644
--- a/examples/cpp/wait_for_ready/BUILD
+++ b/examples/cpp/wait_for_ready/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/examples/cpp/xds/BUILD b/examples/cpp/xds/BUILD
index 15e8028d1..3e1d2ccfa 100644
--- a/examples/cpp/xds/BUILD
+++ b/examples/cpp/xds/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/src/compiler/BUILD b/src/compiler/BUILD
index 019c421ba..496161a1e 100644
--- a/src/compiler/BUILD
+++ b/src/compiler/BUILD
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_library",
diff --git a/src/core/BUILD b/src/core/BUILD
index c6ecae074..e33f88f2c 100644
--- a/src/core/BUILD
+++ b/src/core/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("@bazel_skylib//lib:selects.bzl", "selects")
 load(
     "//bazel:grpc_build_system.bzl",
diff --git a/src/cpp/ext/csm/BUILD b/src/cpp/ext/csm/BUILD
index 6165627b3..f5d585691 100644
--- a/src/cpp/ext/csm/BUILD
+++ b/src/cpp/ext/csm/BUILD
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_library",
diff --git a/src/cpp/ext/gcp/BUILD b/src/cpp/ext/gcp/BUILD
index 5fafab03f..bfbe0179e 100644
--- a/src/cpp/ext/gcp/BUILD
+++ b/src/cpp/ext/gcp/BUILD
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_library",
diff --git a/src/cpp/ext/otel/BUILD b/src/cpp/ext/otel/BUILD
index 761e87193..3ca0cbd63 100644
--- a/src/cpp/ext/otel/BUILD
+++ b/src/cpp/ext/otel/BUILD
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_library",
diff --git a/src/python/grpcio/grpc/_cython/_cygrpc/private_key_signing/BUILD.bazel b/src/python/grpcio/grpc/_cython/_cygrpc/private_key_signing/BUILD.bazel
index cf24414c6..3d30289eb 100644
--- a/src/python/grpcio/grpc/_cython/_cygrpc/private_key_signing/BUILD.bazel
+++ b/src/python/grpcio/grpc/_cython/_cygrpc/private_key_signing/BUILD.bazel
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_library",
diff --git a/src/python/grpcio_observability/grpc_observability/BUILD.bazel b/src/python/grpcio_observability/grpc_observability/BUILD.bazel
index 45a2e4076..1f2b1d160 100644
--- a/src/python/grpcio_observability/grpc_observability/BUILD.bazel
+++ b/src/python/grpcio_observability/grpc_observability/BUILD.bazel
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("@grpc_python_dependencies//:requirements.bzl", "requirement")
 load("@rules_python//python:defs.bzl", "py_library")
 load("//bazel:cython_library.bzl", "pyx_library")
diff --git a/test/core/address_utils/BUILD b/test/core/address_utils/BUILD
index 72a35488d..7f3d1f954 100644
--- a/test/core/address_utils/BUILD
+++ b/test/core/address_utils/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/bad_connection/BUILD b/test/core/bad_connection/BUILD
index 59dae7ccd..49250457f 100644
--- a/test/core/bad_connection/BUILD
+++ b/test/core/bad_connection/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/call/BUILD b/test/core/call/BUILD
index aac95626c..3a424097b 100644
--- a/test/core/call/BUILD
+++ b/test/core/call/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "HISTORY", "MONITORING", "grpc_cc_benchmark")
diff --git a/test/core/call/yodel/BUILD b/test/core/call/yodel/BUILD
index 8f38e5d02..72752f85b 100644
--- a/test/core/call/yodel/BUILD
+++ b/test/core/call/yodel/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_library",
diff --git a/test/core/channel/BUILD b/test/core/channel/BUILD
index 6f5e29e32..5f30cb194 100644
--- a/test/core/channel/BUILD
+++ b/test/core/channel/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/channelz/BUILD b/test/core/channelz/BUILD
index 6f3320ed8..ab6e26e2a 100644
--- a/test/core/channelz/BUILD
+++ b/test/core/channelz/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/channelz/zviz/BUILD b/test/core/channelz/zviz/BUILD
index 269040cda..6fedcb96c 100644
--- a/test/core/channelz/zviz/BUILD
+++ b/test/core/channelz/zviz/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/client_channel/BUILD b/test/core/client_channel/BUILD
index e80153daf..1de0464cf 100644
--- a/test/core/client_channel/BUILD
+++ b/test/core/client_channel/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "HISTORY", "grpc_cc_benchmark")
diff --git a/test/core/client_idle/BUILD b/test/core/client_idle/BUILD
index a3672b53f..6b27d7abe 100644
--- a/test/core/client_idle/BUILD
+++ b/test/core/client_idle/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/compiler_bugs/BUILD b/test/core/compiler_bugs/BUILD
index d05c835f0..02bcce44e 100644
--- a/test/core/compiler_bugs/BUILD
+++ b/test/core/compiler_bugs/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/compression/BUILD b/test/core/compression/BUILD
index ab61ae7c8..61d71b744 100644
--- a/test/core/compression/BUILD
+++ b/test/core/compression/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_test",
diff --git a/test/core/config/BUILD b/test/core/config/BUILD
index fde21062d..d710bd285 100644
--- a/test/core/config/BUILD
+++ b/test/core/config/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/call/BUILD b/test/core/credentials/call/BUILD
index f687e8e40..1f982b5db 100644
--- a/test/core/credentials/call/BUILD
+++ b/test/core/credentials/call/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/call/external/BUILD b/test/core/credentials/call/external/BUILD
index 6ecd2ac80..128642aa5 100644
--- a/test/core/credentials/call/external/BUILD
+++ b/test/core/credentials/call/external/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/call/jwt/BUILD b/test/core/credentials/call/jwt/BUILD
index 6156508ef..09b1fdf8d 100644
--- a/test/core/credentials/call/jwt/BUILD
+++ b/test/core/credentials/call/jwt/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/call/oauth2/BUILD b/test/core/credentials/call/oauth2/BUILD
index 3bb88b356..3fcae6768 100644
--- a/test/core/credentials/call/oauth2/BUILD
+++ b/test/core/credentials/call/oauth2/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/transport/BUILD b/test/core/credentials/transport/BUILD
index a72d5ab55..2234b6759 100644
--- a/test/core/credentials/transport/BUILD
+++ b/test/core/credentials/transport/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/transport/alts/BUILD b/test/core/credentials/transport/alts/BUILD
index 994d2e34c..326126eb2 100644
--- a/test/core/credentials/transport/alts/BUILD
+++ b/test/core/credentials/transport/alts/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/credentials/transport/composite/BUILD b/test/core/credentials/transport/composite/BUILD
index df2b78c07..a621c5322 100644
--- a/test/core/credentials/transport/composite/BUILD
+++ b/test/core/credentials/transport/composite/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/transport/google_default/BUILD b/test/core/credentials/transport/google_default/BUILD
index b009bbc20..1b1adfa93 100644
--- a/test/core/credentials/transport/google_default/BUILD
+++ b/test/core/credentials/transport/google_default/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/transport/insecure/BUILD b/test/core/credentials/transport/insecure/BUILD
index f3e62feab..71fed025a 100644
--- a/test/core/credentials/transport/insecure/BUILD
+++ b/test/core/credentials/transport/insecure/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/transport/local/BUILD b/test/core/credentials/transport/local/BUILD
index bf009d3f6..b477e8be9 100644
--- a/test/core/credentials/transport/local/BUILD
+++ b/test/core/credentials/transport/local/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/transport/ssl/BUILD b/test/core/credentials/transport/ssl/BUILD
index ce58900c1..db7d19d53 100644
--- a/test/core/credentials/transport/ssl/BUILD
+++ b/test/core/credentials/transport/ssl/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/credentials/transport/tls/BUILD b/test/core/credentials/transport/tls/BUILD
index b0db0a754..ed936c217 100644
--- a/test/core/credentials/transport/tls/BUILD
+++ b/test/core/credentials/transport/tls/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/credentials/transport/xds/BUILD b/test/core/credentials/transport/xds/BUILD
index 1243bfb9d..2e1e3c2a2 100644
--- a/test/core/credentials/transport/xds/BUILD
+++ b/test/core/credentials/transport/xds/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/end2end/BUILD b/test/core/end2end/BUILD
index 890c297e4..c99118d11 100644
--- a/test/core/end2end/BUILD
+++ b/test/core/end2end/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 load("//test/core/end2end:grpc_core_end2end_test.bzl", "grpc_core_end2end_test_suite")
 
diff --git a/test/core/end2end/fuzzers/BUILD b/test/core/end2end/fuzzers/BUILD
index d5bd25c1c..349feaa14 100644
--- a/test/core/end2end/fuzzers/BUILD
+++ b/test/core/end2end/fuzzers/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_proto_library", "grpc_internal_proto_library", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/event_engine/BUILD b/test/core/event_engine/BUILD
index b1cf87c10..f98e0c491 100644
--- a/test/core/event_engine/BUILD
+++ b/test/core/event_engine/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/event_engine/cf/BUILD b/test/core/event_engine/cf/BUILD
index e6a5f2436..85869cfdd 100644
--- a/test/core/event_engine/cf/BUILD
+++ b/test/core/event_engine/cf/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test")
 
 package(features = [
diff --git a/test/core/event_engine/fuzzing_event_engine/BUILD b/test/core/event_engine/fuzzing_event_engine/BUILD
index ffb4883b5..dd05da971 100644
--- a/test/core/event_engine/fuzzing_event_engine/BUILD
+++ b/test/core/event_engine/fuzzing_event_engine/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/event_engine/posix/BUILD b/test/core/event_engine/posix/BUILD
index 2a9c93015..d08f3199a 100644
--- a/test/core/event_engine/posix/BUILD
+++ b/test/core/event_engine/posix/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "grpc_cc_benchmark")
 
diff --git a/test/core/event_engine/test_suite/BUILD b/test/core/event_engine/test_suite/BUILD
index 361d43250..0d12ed3a8 100644
--- a/test/core/event_engine/test_suite/BUILD
+++ b/test/core/event_engine/test_suite/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/event_engine/test_suite/posix/BUILD b/test/core/event_engine/test_suite/posix/BUILD
index 1ef11ad73..976a06a03 100644
--- a/test/core/event_engine/test_suite/posix/BUILD
+++ b/test/core/event_engine/test_suite/posix/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/event_engine/test_suite/tests/BUILD b/test/core/event_engine/test_suite/tests/BUILD
index 0eb2f8e44..d0b8f278b 100644
--- a/test/core/event_engine/test_suite/tests/BUILD
+++ b/test/core/event_engine/test_suite/tests/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/event_engine/test_suite/tools/BUILD b/test/core/event_engine/test_suite/tools/BUILD
index 3d003c51a..1e69c087a 100644
--- a/test/core/event_engine/test_suite/tools/BUILD
+++ b/test/core/event_engine/test_suite/tools/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/event_engine/windows/BUILD b/test/core/event_engine/windows/BUILD
index 646c806c3..b7e4fab66 100644
--- a/test/core/event_engine/windows/BUILD
+++ b/test/core/event_engine/windows/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test")
 
 package(features = [
diff --git a/test/core/event_engine/work_queue/BUILD b/test/core/event_engine/work_queue/BUILD
index e23be8e48..14cc3f441 100644
--- a/test/core/event_engine/work_queue/BUILD
+++ b/test/core/event_engine/work_queue/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/experiments/BUILD b/test/core/experiments/BUILD
index eae7e167a..c5b9e0648 100644
--- a/test/core/experiments/BUILD
+++ b/test/core/experiments/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "HISTORY", "grpc_cc_benchmark")
 
diff --git a/test/core/ext/filters/rbac/BUILD b/test/core/ext/filters/rbac/BUILD
index eeeaa87bc..063bf863c 100644
--- a/test/core/ext/filters/rbac/BUILD
+++ b/test/core/ext/filters/rbac/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/filters/BUILD b/test/core/filters/BUILD
index fc530a72e..6dfb2a6e6 100644
--- a/test/core/filters/BUILD
+++ b/test/core/filters/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "HISTORY", "grpc_cc_benchmark")
 
diff --git a/test/core/handshake/BUILD b/test/core/handshake/BUILD
index e9f3a2f14..82d287dad 100644
--- a/test/core/handshake/BUILD
+++ b/test/core/handshake/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/iomgr/BUILD b/test/core/iomgr/BUILD
index 29a34c886..4a30a4744 100644
--- a/test/core/iomgr/BUILD
+++ b/test/core/iomgr/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:custom_exec_properties.bzl", "LARGE_MACHINE")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
diff --git a/test/core/load_balancing/BUILD b/test/core/load_balancing/BUILD
index b787443a0..29c253646 100644
--- a/test/core/load_balancing/BUILD
+++ b/test/core/load_balancing/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_library",
diff --git a/test/core/memory_usage/BUILD b/test/core/memory_usage/BUILD
index 117891c35..1a4a2ec0f 100644
--- a/test/core/memory_usage/BUILD
+++ b/test/core/memory_usage/BUILD
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 grpc_package(
diff --git a/test/core/message_size/BUILD b/test/core/message_size/BUILD
index 98e38a42b..8a26d9c73 100644
--- a/test/core/message_size/BUILD
+++ b/test/core/message_size/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 grpc_package(
diff --git a/test/core/network_benchmarks/BUILD b/test/core/network_benchmarks/BUILD
index 3ef821d14..6b681dd9b 100644
--- a/test/core/network_benchmarks/BUILD
+++ b/test/core/network_benchmarks/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_package")
 
 grpc_package(
diff --git a/test/core/promise/BUILD b/test/core/promise/BUILD
index b0d9a1766..0536fd48a 100644
--- a/test/core/promise/BUILD
+++ b/test/core/promise/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "HISTORY", "grpc_cc_benchmark")
diff --git a/test/core/resolver/BUILD b/test/core/resolver/BUILD
index c4f35a453..36ff919c4 100644
--- a/test/core/resolver/BUILD
+++ b/test/core/resolver/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 grpc_package(
diff --git a/test/core/resource_quota/BUILD b/test/core/resource_quota/BUILD
index 0c298004e..e495695af 100644
--- a/test/core/resource_quota/BUILD
+++ b/test/core/resource_quota/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/resource_tracker/BUILD b/test/core/resource_tracker/BUILD
index a7c82f713..3df71b864 100644
--- a/test/core/resource_tracker/BUILD
+++ b/test/core/resource_tracker/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/security/BUILD b/test/core/security/BUILD
index 7f5192b6d..d714751ad 100644
--- a/test/core/security/BUILD
+++ b/test/core/security/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/server/BUILD b/test/core/server/BUILD
index a4b918023..b4cf60c80 100644
--- a/test/core/server/BUILD
+++ b/test/core/server/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/service_config/BUILD b/test/core/service_config/BUILD
index d89635ea4..3771a2a21 100644
--- a/test/core/service_config/BUILD
+++ b/test/core/service_config/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 grpc_package(
diff --git a/test/core/slice/BUILD b/test/core/slice/BUILD
index e12869062..5ccbb1bb6 100644
--- a/test/core/slice/BUILD
+++ b/test/core/slice/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/surface/BUILD b/test/core/surface/BUILD
index e66bb4d2a..d09badd1c 100644
--- a/test/core/surface/BUILD
+++ b/test/core/surface/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "grpc_cc_benchmark")
 
diff --git a/test/core/telemetry/BUILD b/test/core/telemetry/BUILD
index 0419f680c..3bd770c18 100644
--- a/test/core/telemetry/BUILD
+++ b/test/core/telemetry/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "HISTORY", "grpc_cc_benchmark")
diff --git a/test/core/test_util/BUILD b/test/core/test_util/BUILD
index 334929f84..7e2d23317 100644
--- a/test/core/test_util/BUILD
+++ b/test/core/test_util/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/transport/BUILD b/test/core/transport/BUILD
index 8eb84a2f4..59979dde6 100644
--- a/test/core/transport/BUILD
+++ b/test/core/transport/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "HISTORY", "grpc_cc_benchmark")
 
diff --git a/test/core/transport/chaotic_good/BUILD b/test/core/transport/chaotic_good/BUILD
index c1283986e..6cf18fc9c 100644
--- a/test/core/transport/chaotic_good/BUILD
+++ b/test/core/transport/chaotic_good/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/transport/chaotic_good_legacy/BUILD b/test/core/transport/chaotic_good_legacy/BUILD
index b22fd450b..8b8409c2b 100644
--- a/test/core/transport/chaotic_good_legacy/BUILD
+++ b/test/core/transport/chaotic_good_legacy/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/transport/chttp2/BUILD b/test/core/transport/chttp2/BUILD
index 58e705f77..75a74240d 100644
--- a/test/core/transport/chttp2/BUILD
+++ b/test/core/transport/chttp2/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:custom_exec_properties.bzl", "LARGE_MACHINE")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
diff --git a/test/core/transport/test_suite/BUILD b/test/core/transport/test_suite/BUILD
index 6ca64e210..2cc4fc3e9 100644
--- a/test/core/transport/test_suite/BUILD
+++ b/test/core/transport/test_suite/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_library",
diff --git a/test/core/transport/util/BUILD b/test/core/transport/util/BUILD
index 6d6304272..6fe17497a 100644
--- a/test/core/transport/util/BUILD
+++ b/test/core/transport/util/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/tsi/BUILD b/test/core/tsi/BUILD
index ab6579b98..36cd1fc82 100644
--- a/test/core/tsi/BUILD
+++ b/test/core/tsi/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/tsi/alts/crypt/BUILD b/test/core/tsi/alts/crypt/BUILD
index 96dcf89d4..47ab3e321 100644
--- a/test/core/tsi/alts/crypt/BUILD
+++ b/test/core/tsi/alts/crypt/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/tsi/alts/fake_handshaker/BUILD b/test/core/tsi/alts/fake_handshaker/BUILD
index db3c76a8f..5b79cfad2 100644
--- a/test/core/tsi/alts/fake_handshaker/BUILD
+++ b/test/core/tsi/alts/fake_handshaker/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_grpc_library", "grpc_cc_library", "grpc_cc_proto_library", "grpc_internal_proto_library", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/tsi/alts/frame_protector/BUILD b/test/core/tsi/alts/frame_protector/BUILD
index 4d4dafb8b..99c942225 100644
--- a/test/core/tsi/alts/frame_protector/BUILD
+++ b/test/core/tsi/alts/frame_protector/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/tsi/alts/handshaker/BUILD b/test/core/tsi/alts/handshaker/BUILD
index 378bf9911..8693f7a77 100644
--- a/test/core/tsi/alts/handshaker/BUILD
+++ b/test/core/tsi/alts/handshaker/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/tsi/alts/zero_copy_frame_protector/BUILD b/test/core/tsi/alts/zero_copy_frame_protector/BUILD
index c274621a5..80e35285b 100644
--- a/test/core/tsi/alts/zero_copy_frame_protector/BUILD
+++ b/test/core/tsi/alts/zero_copy_frame_protector/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/core/util/BUILD b/test/core/util/BUILD
index 28a90c642..422991ffd 100644
--- a/test/core/util/BUILD
+++ b/test/core/util/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:custom_exec_properties.bzl", "LARGE_MACHINE")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_proto_library", "grpc_cc_test", "grpc_internal_proto_library", "grpc_package")
 load(
diff --git a/test/core/util/http_client/BUILD b/test/core/util/http_client/BUILD
index dbc1c7070..223972ff6 100644
--- a/test/core/util/http_client/BUILD
+++ b/test/core/util/http_client/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/util/json/BUILD b/test/core/util/json/BUILD
index 475326c9e..420e99df1 100644
--- a/test/core/util/json/BUILD
+++ b/test/core/util/json/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 load("//test/core/test_util:grpc_fuzzer.bzl", "grpc_fuzz_test")
 
diff --git a/test/core/xds/BUILD b/test/core/xds/BUILD
index bcdf8d483..a83d38bf3 100644
--- a/test/core/xds/BUILD
+++ b/test/core/xds/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//bazel:grpc_build_system.bzl",
     "grpc_cc_library",
diff --git a/test/cpp/client/BUILD b/test/cpp/client/BUILD
index 951a275fd..818e4b0a1 100644
--- a/test/cpp/client/BUILD
+++ b/test/cpp/client/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/codegen/BUILD b/test/cpp/codegen/BUILD
index c52e35bf8..7c2dfc115 100644
--- a/test/cpp/codegen/BUILD
+++ b/test/cpp/codegen/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_test", "grpc_package", "grpc_sh_test")
 
 licenses(["notice"])
diff --git a/test/cpp/common/BUILD b/test/cpp/common/BUILD
index 597c11da0..74823e58b 100644
--- a/test/cpp/common/BUILD
+++ b/test/cpp/common/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/end2end/BUILD b/test/cpp/end2end/BUILD
index 11898b0a3..f4a6c1302 100644
--- a/test/cpp/end2end/BUILD
+++ b/test/cpp/end2end/BUILD
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/end2end/xds/BUILD b/test/cpp/end2end/xds/BUILD
index 9e5dd3647..6c6791c2c 100644
--- a/test/cpp/end2end/xds/BUILD
+++ b/test/cpp/end2end/xds/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/ext/csm/BUILD b/test/cpp/ext/csm/BUILD
index 01f927e4d..0f198805a 100644
--- a/test/cpp/ext/csm/BUILD
+++ b/test/cpp/ext/csm/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/ext/filters/census/BUILD b/test/cpp/ext/filters/census/BUILD
index 7d52cfd7b..f9e9c93e6 100644
--- a/test/cpp/ext/filters/census/BUILD
+++ b/test/cpp/ext/filters/census/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/ext/filters/logging/BUILD b/test/cpp/ext/filters/logging/BUILD
index fd9231cce..0f6b0bafc 100644
--- a/test/cpp/ext/filters/logging/BUILD
+++ b/test/cpp/ext/filters/logging/BUILD
@@ -14,6 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/ext/gcp/BUILD b/test/cpp/ext/gcp/BUILD
index cb4739dda..e3cc3c100 100644
--- a/test/cpp/ext/gcp/BUILD
+++ b/test/cpp/ext/gcp/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/ext/otel/BUILD b/test/cpp/ext/otel/BUILD
index a6181b366..6595b42ac 100644
--- a/test/cpp/ext/otel/BUILD
+++ b/test/cpp/ext/otel/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/grpclb/BUILD b/test/cpp/grpclb/BUILD
index a23204255..d56d653ae 100644
--- a/test/cpp/grpclb/BUILD
+++ b/test/cpp/grpclb/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/interop/BUILD b/test/cpp/interop/BUILD
index 166dd00c6..4c6903050 100644
--- a/test/cpp/interop/BUILD
+++ b/test/cpp/interop/BUILD
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/microbenchmarks/BUILD b/test/cpp/microbenchmarks/BUILD
index 78eee5aab..a378435f7 100644
--- a/test/cpp/microbenchmarks/BUILD
+++ b/test/cpp/microbenchmarks/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_package")
 load("//test/cpp/microbenchmarks:grpc_benchmark_config.bzl", "HISTORY", "grpc_cc_benchmark")
 
diff --git a/test/cpp/microbenchmarks/huffman_geometries/BUILD b/test/cpp/microbenchmarks/huffman_geometries/BUILD
index 6607cd93e..8018707ff 100644
--- a/test/cpp/microbenchmarks/huffman_geometries/BUILD
+++ b/test/cpp/microbenchmarks/huffman_geometries/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/naming/BUILD b/test/cpp/naming/BUILD
index 6033960a4..24825e1da 100644
--- a/test/cpp/naming/BUILD
+++ b/test/cpp/naming/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_py_binary")
 load(":generate_resolver_component_tests.bzl", "generate_resolver_component_tests")
 
diff --git a/test/cpp/performance/BUILD b/test/cpp/performance/BUILD
index 42ba1cc45..b1ac4df3a 100644
--- a/test/cpp/performance/BUILD
+++ b/test/cpp/performance/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/qps/BUILD b/test/cpp/qps/BUILD
index 0c9ae8d4a..18db248fb 100644
--- a/test/cpp/qps/BUILD
+++ b/test/cpp/qps/BUILD
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:custom_exec_properties.bzl", "LARGE_MACHINE")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_cc_test", "grpc_package", "grpc_py_binary")
 load("//test/cpp/qps:qps_benchmark_script.bzl", "json_run_localhost_batch", "qps_json_driver_batch")
diff --git a/test/cpp/security/BUILD b/test/cpp/security/BUILD
index dbd72a506..de956cfa1 100644
--- a/test/cpp/security/BUILD
+++ b/test/cpp/security/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/server/BUILD b/test/cpp/server/BUILD
index ec62f8929..ed6c2c112 100644
--- a/test/cpp/server/BUILD
+++ b/test/cpp/server/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/server/load_reporter/BUILD b/test/cpp/server/load_reporter/BUILD
index bee05b0db..4bde30012 100644
--- a/test/cpp/server/load_reporter/BUILD
+++ b/test/cpp/server/load_reporter/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/sleuth/BUILD b/test/cpp/sleuth/BUILD
index 6c5dc51ff..9451cab22 100644
--- a/test/cpp/sleuth/BUILD
+++ b/test/cpp/sleuth/BUILD
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/test/BUILD b/test/cpp/test/BUILD
index e5dd1786b..538f21dd9 100644
--- a/test/cpp/test/BUILD
+++ b/test/cpp/test/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/thread_manager/BUILD b/test/cpp/thread_manager/BUILD
index 2f1ac6672..b4ae967bd 100644
--- a/test/cpp/thread_manager/BUILD
+++ b/test/cpp/thread_manager/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/util/BUILD b/test/cpp/util/BUILD
index e3dceab68..7e2f2a5a2 100644
--- a/test/cpp/util/BUILD
+++ b/test/cpp/util/BUILD
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_cc_test", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/cpp/util/windows/BUILD b/test/cpp/util/windows/BUILD
index d86f9e23d..0ee8d66dd 100644
--- a/test/cpp/util/windows/BUILD
+++ b/test/cpp/util/windows/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_package")
 
 licenses(["notice"])
diff --git a/test/distrib/bazel/cpp/BUILD b/test/distrib/bazel/cpp/BUILD
index 0869e63f2..91213ee97 100644
--- a/test/distrib/bazel/cpp/BUILD
+++ b/test/distrib/bazel/cpp/BUILD
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 licenses(["notice"])
 
 cc_binary(
diff --git a/third_party/address_sorting/BUILD b/third_party/address_sorting/BUILD
index 2f4f6b7fd..b9b025f6e 100644
--- a/third_party/address_sorting/BUILD
+++ b/third_party/address_sorting/BUILD
@@ -28,6 +28,7 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(":address_sorting.bzl", "address_sorting_cc_library")
 
 package(
diff --git a/third_party/address_sorting/address_sorting.bzl b/third_party/address_sorting/address_sorting.bzl
index 05b4c6d84..93717d0eb 100644
--- a/third_party/address_sorting/address_sorting.bzl
+++ b/third_party/address_sorting/address_sorting.bzl
@@ -28,8 +28,9 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 def address_sorting_cc_library(name, srcs, hdrs, copts, includes, linkopts=[]):
-    native.cc_library(
+    cc_library(
         name = name,
         srcs = srcs,
         hdrs = hdrs,
diff --git a/third_party/toolchains/rbe_ubuntu2004_bazel7/cc/BUILD b/third_party/toolchains/rbe_ubuntu2004_bazel7/cc/BUILD
index 4398be4d0..c37e6040e 100644
--- a/third_party/toolchains/rbe_ubuntu2004_bazel7/cc/BUILD
+++ b/third_party/toolchains/rbe_ubuntu2004_bazel7/cc/BUILD
@@ -14,6 +14,7 @@
 
 # This becomes the BUILD file for @local_config_cc// under non-BSD unixes.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(":cc_toolchain_config.bzl", "cc_toolchain_config")
 load(":armeabi_cc_toolchain_config.bzl", "armeabi_cc_toolchain_config")
 load("@rules_cc//cc:defs.bzl", "cc_toolchain", "cc_toolchain_suite")
diff --git a/third_party/toolchains/rbe_windows_vs2022_bazel7/cc/BUILD b/third_party/toolchains/rbe_windows_vs2022_bazel7/cc/BUILD
index d47b0c00d..07f971385 100755
--- a/third_party/toolchains/rbe_windows_vs2022_bazel7/cc/BUILD
+++ b/third_party/toolchains/rbe_windows_vs2022_bazel7/cc/BUILD
@@ -14,6 +14,7 @@
 
 # This becomes the BUILD file for @local_config_cc// under Windows.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("@rules_cc//cc:defs.bzl", "cc_toolchain", "cc_toolchain_suite", "cc_library")
 load(":windows_cc_toolchain_config.bzl", "cc_toolchain_config")
 load(":armeabi_cc_toolchain_config.bzl", "armeabi_cc_toolchain_config")
diff --git a/third_party/upb/upb/BUILD b/third_party/upb/upb/BUILD
index 7745404b6..daaf1bc1f 100644
--- a/third_party/upb/upb/BUILD
+++ b/third_party/upb/upb/BUILD
@@ -5,6 +5,7 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
 load("@rules_python//python:defs.bzl", "py_binary")
 load("//bazel/private:upb_proto_library_internal/copts.bzl", "upb_proto_library_copts")
diff --git a/third_party/upb/upb/base/BUILD b/third_party/upb/upb/base/BUILD
index 198131a6a..6bcd93fdd 100644
--- a/third_party/upb/upb/base/BUILD
+++ b/third_party/upb/upb/base/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//upb/bazel:build_defs.bzl", "UPB_DEFAULT_COPTS")
 
 package(default_applicable_licenses = ["//:license"])
diff --git a/third_party/upb/upb/conformance/BUILD b/third_party/upb/upb/conformance/BUILD
index 1b58234c5..98f5e133b 100644
--- a/third_party/upb/upb/conformance/BUILD
+++ b/third_party/upb/upb/conformance/BUILD
@@ -5,6 +5,7 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 load("@rules_shell//shell:sh_test.bzl", "sh_test")
 load(
     "//bazel:upb_proto_library.bzl",
diff --git a/third_party/upb/upb/hash/BUILD b/third_party/upb/upb/hash/BUILD
index 1a7c574aa..99333ca34 100644
--- a/third_party/upb/upb/hash/BUILD
+++ b/third_party/upb/upb/hash/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//upb/bazel:build_defs.bzl", "UPB_DEFAULT_COPTS")
 
 package(default_applicable_licenses = ["//:license"])
diff --git a/third_party/upb/upb/io/BUILD b/third_party/upb/upb/io/BUILD
index d00a18a44..1995cf0ef 100644
--- a/third_party/upb/upb/io/BUILD
+++ b/third_party/upb/upb/io/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 package(default_applicable_licenses = ["//:license"])
 
 cc_library(
diff --git a/third_party/upb/upb/json/BUILD b/third_party/upb/upb/json/BUILD
index 3e1e8daf0..ce936980b 100644
--- a/third_party/upb/upb/json/BUILD
+++ b/third_party/upb/upb/json/BUILD
@@ -1,5 +1,7 @@
 # TODO: describe this package.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:proto_library.bzl", "proto_library")
 load(
     "//bazel:upb_proto_library.bzl",
diff --git a/third_party/upb/upb/lex/BUILD b/third_party/upb/upb/lex/BUILD
index 0ae91b27c..b9c4d4d3e 100644
--- a/third_party/upb/upb/lex/BUILD
+++ b/third_party/upb/upb/lex/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//upb/bazel:build_defs.bzl", "UPB_DEFAULT_COPTS")
 
 package(default_applicable_licenses = ["//:license"])
diff --git a/third_party/upb/upb/mem/BUILD b/third_party/upb/upb/mem/BUILD
index 30941c0e3..50f294e99 100644
--- a/third_party/upb/upb/mem/BUILD
+++ b/third_party/upb/upb/mem/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//upb/bazel:build_defs.bzl", "UPB_DEFAULT_COPTS")
 
 package(default_applicable_licenses = ["//:license"])
diff --git a/third_party/upb/upb/message/BUILD b/third_party/upb/upb/message/BUILD
index 3ef0cf127..7d71ab582 100644
--- a/third_party/upb/upb/message/BUILD
+++ b/third_party/upb/upb/message/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:proto_library.bzl", "proto_library")
 load(
     "//bazel:upb_minitable_proto_library.bzl",
diff --git a/third_party/upb/upb/mini_descriptor/BUILD b/third_party/upb/upb/mini_descriptor/BUILD
index 75dfb915f..b3e7d72e6 100644
--- a/third_party/upb/upb/mini_descriptor/BUILD
+++ b/third_party/upb/upb/mini_descriptor/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//upb/bazel:build_defs.bzl",
     "UPB_DEFAULT_COPTS",
diff --git a/third_party/upb/upb/mini_table/BUILD b/third_party/upb/upb/mini_table/BUILD
index 9b25826fa..3ecf0821b 100644
--- a/third_party/upb/upb/mini_table/BUILD
+++ b/third_party/upb/upb/mini_table/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:proto_library.bzl", "proto_library")
 load(
     "//bazel:upb_minitable_proto_library.bzl",
diff --git a/third_party/upb/upb/port/BUILD b/third_party/upb/upb/port/BUILD
index 9088825d2..a8e9a13f3 100644
--- a/third_party/upb/upb/port/BUILD
+++ b/third_party/upb/upb/port/BUILD
@@ -5,6 +5,7 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//upb/bazel:build_defs.bzl", "UPB_DEFAULT_COPTS")
 
 package(default_applicable_licenses = ["//:license"])
diff --git a/third_party/upb/upb/reflection/BUILD b/third_party/upb/upb/reflection/BUILD
index 81844c4ba..974d87bb7 100644
--- a/third_party/upb/upb/reflection/BUILD
+++ b/third_party/upb/upb/reflection/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(
     "//editions:defaults.bzl",
     "compile_edition_defaults",
diff --git a/third_party/upb/upb/test/BUILD b/third_party/upb/upb/test/BUILD
index b3f750fe0..6bd45dfe5 100644
--- a/third_party/upb/upb/test/BUILD
+++ b/third_party/upb/upb/test/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:proto_library.bzl", "proto_library")
 load(
     "//bazel:upb_minitable_proto_library.bzl",
diff --git a/third_party/upb/upb/text/BUILD b/third_party/upb/upb/text/BUILD
index 1063be802..1258428ab 100644
--- a/third_party/upb/upb/text/BUILD
+++ b/third_party/upb/upb/text/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//upb/bazel:build_defs.bzl", "UPB_DEFAULT_COPTS")
 
 package(default_applicable_licenses = ["//:license"])
diff --git a/third_party/upb/upb/util/BUILD b/third_party/upb/upb/util/BUILD
index 097f58cf7..861bdcdc6 100644
--- a/third_party/upb/upb/util/BUILD
+++ b/third_party/upb/upb/util/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:proto_library.bzl", "proto_library")
 load(
     "//bazel:upb_proto_library.bzl",
diff --git a/third_party/upb/upb/wire/BUILD b/third_party/upb/upb/wire/BUILD
index 3480552cb..887e6ef66 100644
--- a/third_party/upb/upb/wire/BUILD
+++ b/third_party/upb/upb/wire/BUILD
@@ -5,6 +5,8 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//upb/bazel:build_defs.bzl", "UPB_DEFAULT_COPTS")
 
 package(default_applicable_licenses = ["//:license"])
diff --git a/third_party/utf8_range/BUILD.bazel b/third_party/utf8_range/BUILD.bazel
index ffaf7b1f0..fcbd69348 100644
--- a/third_party/utf8_range/BUILD.bazel
+++ b/third_party/utf8_range/BUILD.bazel
@@ -4,6 +4,8 @@
 # license that can be found in the LICENSE file or at
 # https://opensource.org/licenses/MIT.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("@rules_license//rules:license.bzl", "license")
 
 DEFAULT_VISIBILITY = [
diff --git a/third_party/xxhash/BUILD b/third_party/xxhash/BUILD
index 91d7bd056..c4f49aec6 100644
--- a/third_party/xxhash/BUILD
+++ b/third_party/xxhash/BUILD
@@ -1,4 +1,5 @@
 # TODO(donnadionne): describe this package.
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 cc_library(
     name = "xxhash",
     hdrs = [
diff --git a/tools/artifact_gen/BUILD b/tools/artifact_gen/BUILD
index a4881e49d..90617c808 100644
--- a/tools/artifact_gen/BUILD
+++ b/tools/artifact_gen/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 cc_library(
     name = "extract_metadata_from_bazel_xml",
     srcs = ["extract_metadata_from_bazel_xml.cc"],
diff --git a/tools/codegen/core/BUILD b/tools/codegen/core/BUILD
index 056ffe262..4e14fe647 100644
--- a/tools/codegen/core/BUILD
+++ b/tools/codegen/core/BUILD
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_cc_test")
 
 package(features = [
diff --git a/tools/codegen/core/gen_experiments/BUILD b/tools/codegen/core/gen_experiments/BUILD
index 094381c3c..02633f816 100644
--- a/tools/codegen/core/gen_experiments/BUILD
+++ b/tools/codegen/core/gen_experiments/BUILD
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_library", "grpc_cc_test")
 
 package(features = [
diff --git a/tools/codegen/core/gen_stats/BUILD b/tools/codegen/core/gen_stats/BUILD
index d695ac721..330433049 100644
--- a/tools/codegen/core/gen_stats/BUILD
+++ b/tools/codegen/core/gen_stats/BUILD
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//bazel:grpc_build_system.bzl", "grpc_cc_binary", "grpc_cc_library", "grpc_cc_test")
 
 package(features = [
diff --git a/tools/distrib/python/grpcio_tools/BUILD.bazel b/tools/distrib/python/grpcio_tools/BUILD.bazel
index 3a2871353..044210cf6 100644
--- a/tools/distrib/python/grpcio_tools/BUILD.bazel
+++ b/tools/distrib/python/grpcio_tools/BUILD.bazel
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("@rules_python//python:defs.bzl", "py_library")
 load("//bazel:cython_library.bzl", "pyx_library")
 load("grpcio_tools.bzl", "internal_copied_filegroup")
