--- a/MODULE.bazel	2026-04-06 16:19:29.891996127 -0700
+++ b/MODULE.bazel	2026-04-06 16:19:29.893996093 -0700
@@ -14,7 +14,7 @@
 
 module(
     name = "grpc",
-    version = "1.80.0",
+    version = "1.80.0.bcr.1",
     compatibility_level = 1,
     repo_name = "com_github_grpc_grpc",
 )
@@ -24,7 +24,7 @@
 
 # Upgrade version of rules_swift to avoid an error in BCR CI of not
 # finding switfc.exe on Windows with bazel 7.
-bazel_dep(name = "rules_swift", version = "2.5.0")
+bazel_dep(name = "rules_swift", version = "3.1.2")
 
 # -- Abseil library.
 # Pinning version to be consistent with third_party/abseil-cpp
