diff --git a/MODULE.bazel b/MODULE.bazel
index 18203077a7..3c9d2b4d8f 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -14,7 +14,8 @@
 
 module(
     name = "grpc",
-    version = "1.80.0",
+    version = "1.80.0.bcr.1",
+    bazel_compatibility = [">=7.2.1"],
     compatibility_level = 1,
     repo_name = "com_github_grpc_grpc",
 )
@@ -43,8 +44,8 @@ bazel_dep(name = "envoy_api", version = "0.0.0-20251216-6ef568c")
 bazel_dep(name = "googleapis-cc", version = "1.0.0")
 bazel_dep(name = "googleapis-grpc-cc", version = "1.0.0")
 bazel_dep(name = "googleapis-python", version = "1.0.0")
-bazel_dep(name = "googletest", version = "1.17.0", repo_name = "com_google_googletest")
-bazel_dep(name = "opencensus-cpp", version = "0.0.0-20230502-50eb5de.bcr.2", repo_name = "io_opencensus_cpp")
+bazel_dep(name = "googletest", version = "1.17.0.bcr.2", repo_name = "com_google_googletest")
+bazel_dep(name = "opencensus-cpp", version = "0.0.0-20230502-50eb5de.bcr.3", repo_name = "io_opencensus_cpp")
 bazel_dep(name = "openssl", version = "3.3.1.bcr.1")
 bazel_dep(name = "opentelemetry-cpp", version = "1.19.0", repo_name = "io_opentelemetry_cpp")
 bazel_dep(name = "platforms", version = "0.0.11")
