--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -48,14 +48,20 @@
 
 bazel_dep(name = "bazel_jar_jar", version = "0.1.7")
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
-bazel_dep(name = "googleapis", repo_name = "com_google_googleapis", version = "0.0.0-20240326-1c8d509c5")
+bazel_dep(name = "googleapis", repo_name = "com_google_googleapis", version = "0.0.0-20240819-fe8ba054a")
+
+bazel_dep(name = "envoy_api", version = "0.0.0-20241214-918efc9")
+bazel_dep(name = "grpc", repo_name = "com_github_grpc_grpc", version = "1.69.0")
 bazel_dep(name = "grpc-proto", repo_name = "io_grpc_grpc_proto", version = "0.0.0-20240627-ec30f58")
-# Protobuf 25.5+ is incompatible with Bazel 7 with bzlmod
-bazel_dep(name = "protobuf", repo_name = "com_google_protobuf", version = "24.4")
-bazel_dep(name = "rules_cc", version = "0.0.9")
-bazel_dep(name = "rules_java", version = "5.3.5")
+bazel_dep(name = "opencensus-proto", repo_name = "opencensus_proto", version = "0.4.1.bcr.2")
+bazel_dep(name = "protobuf", repo_name = "com_google_protobuf", version = "29.0")
+bazel_dep(name = "protoc-gen-validate", repo_name = "com_envoyproxy_protoc_gen_validate", version = "1.2.1.bcr.1")
+bazel_dep(name = "rules_cc", version = "0.0.17")
+bazel_dep(name = "rules_java", version = "8.5.1")
+bazel_dep(name = "rules_go", repo_name = "io_bazel_rules_go", version = "0.46.0")
 bazel_dep(name = "rules_jvm_external", version = "6.0")
-bazel_dep(name = "rules_proto", version = "5.3.0-21.7")
+bazel_dep(name = "rules_proto", version = "7.0.2")
+bazel_dep(name = "xds", repo_name = "com_github_cncf_xds", version = "0.0.0-20240423-555b57e")
 
 maven = use_extension("@rules_jvm_external//:extensions.bzl", "maven")
 
@@ -183,3 +189,7 @@
     coordinates = "io.grpc:grpc-util",
     target = "@io_grpc_grpc_java//util",
 )
+
+switched_rules = use_extension("@com_google_googleapis//:extensions.bzl", "switched_rules")
+
+switched_rules.use_languages(java = True)

--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -14,6 +15,7 @@
 load("@rules_jvm_external//:defs.bzl", "artifact")
 load(":java_grpc_library.bzl", "java_grpc_library")
+load("@com_google_protobuf//bazel:java_proto_library.bzl", "java_proto_library")
 
 java_proto_library(
     name = "api_proto_java",
     deps = ["@com_google_protobuf//:api_proto"],
 )
