--- a/gutil/BUILD.bazel	2026-02-26 22:07:27
+++ b/gutil/BUILD.bazel	2026-02-26 22:08:22
@@ -56,6 +56,7 @@
 
 cc_test(
     name = "proto_matchers_test",
+    visibility = ["//visibility:private"],
     srcs = ["proto_matchers_test.cc"],
     deps = [
         ":proto_matchers",
@@ -80,6 +81,7 @@
 
 cc_test(
     name = "status_matchers_test",
+    visibility = ["//visibility:private"],
     srcs = ["status_matchers_test.cc"],
     deps = [
         ":status_matchers",
@@ -104,6 +106,7 @@
 
 cc_test(
     name = "testing_test",
+    visibility = ["//visibility:private"],
     srcs = ["testing_test.cc"],
     deps = [
         ":testing",
@@ -169,6 +172,7 @@
 
 cc_test(
     name = "io_test",
+    visibility = ["//visibility:private"],
     srcs = [
         "io_test.cc",
     ],
@@ -192,6 +196,7 @@
 
 cc_test(
     name = "collections_test",
+    visibility = ["//visibility:private"],
     srcs = ["collections_test.cc"],
     deps = [
         ":collections",
@@ -204,6 +209,7 @@
 
 cc_test(
     name = "proto_test",
+    visibility = ["//visibility:private"],
     srcs = ["proto_test.cc"],
     deps = [
         ":proto",
@@ -254,6 +260,7 @@
 
 cc_test(
     name = "version_test",
+    visibility = ["//visibility:private"],
     srcs = ["version_test.cc"],
     deps = [
         ":status_matchers",
@@ -323,6 +330,7 @@
 
 cc_test(
     name = "proto_ordering_test",
+    visibility = ["//visibility:private"],
     srcs = ["proto_ordering_test.cc"],
     deps = [
         ":proto_matchers",
@@ -336,6 +344,7 @@
 
 cc_test(
     name = "test_artifact_writer_test",
+    visibility = ["//visibility:private"],
     srcs = ["test_artifact_writer_test.cc"],
     deps = [
         ":status_matchers",
@@ -351,6 +360,7 @@
 
 cc_test(
     name = "timer_test",
+    visibility = ["//visibility:private"],
     srcs = ["timer_test.cc"],
     deps = [
         ":timer",
