From 1f9b4c0bbbe7980220c1d3baa75200a3ae2a2340 Mon Sep 17 00:00:00 2001
From: Andrew Bylard <andrew.bylard@dexterity.ai>
Date: Tue, 28 Nov 2023 14:42:27 -0800
Subject: add direct boost is_floating_point import

---
 include/pinocchio/math/fwd.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git include/pinocchio/math/fwd.hpp include/pinocchio/math/fwd.hpp
index d10fe48c8c..deaa12e6eb 100644
--- include/pinocchio/math/fwd.hpp
+++ include/pinocchio/math/fwd.hpp
@@ -8,6 +8,7 @@
 #include "pinocchio/fwd.hpp"
 #include <math.h>
 #include <boost/math/constants/constants.hpp>
+#include <boost/type_traits/is_floating_point.hpp>

 namespace pinocchio
 {
