--- interchange/DeviceResources.capnp
+++ interchange/DeviceResources.capnp
@@ -13,11 +13,8 @@
 # limitations under the License.
 
 @0x9d262c6ba6512325;
-using Java = import "/capnp/java.capnp";
 using Ref = import "References.capnp";
 using Dir = import "LogicalNetlist.capnp";
-$Java.package("com.xilinx.rapidwright.interchange");
-$Java.outerClassname("DeviceResources");
 
 using Cxx = import "/capnp/c++.capnp";
 $Cxx.namespace("DeviceResources");
--- interchange/LogicalNetlist.capnp
+++ interchange/LogicalNetlist.capnp
@@ -13,10 +13,7 @@
 # limitations under the License.
 
 @0xcb2ccd67aa912967;
-using Java = import "/capnp/java.capnp";
 using Ref = import "References.capnp";
-$Java.package("com.xilinx.rapidwright.interchange");
-$Java.outerClassname("LogicalNetlist");
 
 using Cxx = import "/capnp/c++.capnp";
 $Cxx.namespace("LogicalNetlist");
--- interchange/PhysicalNetlist.capnp
+++ interchange/PhysicalNetlist.capnp
@@ -12,9 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 @0xcb2ccd67aa912968;
-using Java = import "/capnp/java.capnp";
-$Java.package("com.xilinx.rapidwright.interchange");
-$Java.outerClassname("PhysicalNetlist");
 
 using Cxx = import "/capnp/c++.capnp";
 $Cxx.namespace("PhysicalNetlist");
--- interchange/References.capnp
+++ interchange/References.capnp
@@ -13,9 +13,6 @@
 # limitations under the License.
 
 @0xa0082c6893796a86;
-using Java = import "/capnp/java.capnp";
-$Java.package("com.xilinx.rapidwright.interchange");
-$Java.outerClassname("References");
 
 enum ReferenceType {
     root       @0;
