--- MODULE.bazel
+++ MODULE.bazel
@@ -2,7 +2,7 @@
 
 module(
     name = "muda",
-    version = "0.1.0",
+    version = "0.1.0-beta",
     compatibility_level = 1,
 )
 
@@ -17,20 +17,9 @@
 
 bazel_dep(name = "bazel_skylib", version = "1.8.2")
 
-# Optional for debug: Hedron's Compile Commands Extractor for Bazel
-# https://github.com/hedronvision/bazel-compile-commands-extractor
-bazel_dep(name = "hedron_compile_commands", dev_dependency = True)
-git_override(
-    module_name = "hedron_compile_commands",
-    remote = "https://github.com/hedronvision/bazel-compile-commands-extractor.git",
-    commit = "abb61a688167623088f8768cc9264798df6a9d10",
-    # Replace the commit hash (above) with the latest (https://github.com/hedronvision/bazel-compile-commands-extractor/commits/main).
-    # Even better, set up Renovate and let it do the work for you (see "Suggestion: Updates" in the README).
-)
-
- cuda = use_extension("@rules_cuda//cuda:extensions.bzl", "toolchain")
- cuda.local_toolchain(
-     name = "local_cuda",
-    toolkit_path = "/usr/local/cuda",
- )
- use_repo(cuda, "local_cuda")
