diff --git a/examples/tests_and_lints/MODULE.bazel b/examples/tests_and_lints/MODULE.bazel
index 69ef1f2..eca5013 100644
--- a/examples/tests_and_lints/MODULE.bazel
+++ b/examples/tests_and_lints/MODULE.bazel
@@ -16,9 +16,17 @@ local_path_override(
     path = "../..",
 )
 
+bazel_dep(
+    name = "rules_java",
+    version = "8.5.1",
+)
 bazel_dep(
     name = "rules_jvm_external",
-    version = "6.6",
+    version = "6.7",
+)
+bazel_dep(
+    name = "bazel_skylib",
+    version = "1.8.1",
 )
 
 linter = use_extension("@apple_rules_lint//lint:extensions.bzl", "linter")
@@ -49,6 +57,6 @@ use_repo(
 )
 
 register_toolchains(
-    "@@bazel_skylib//toolchains/unittest:cmd_toolchain",
-    "@@bazel_skylib//toolchains/unittest:bash_toolchain",
+    "@bazel_skylib//toolchains/unittest:cmd_toolchain",
+    "@bazel_skylib//toolchains/unittest:bash_toolchain",
 )
