diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..51e78e2
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,13 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "crc",
+    deps = [
+        "@boost//:array",
+        "@boost//:config",
+        "@boost//:integer",
+        "@boost//:type_traits",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..1e7b6bc
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.crc",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..3f58e72
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,22 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:crc",
+    "@boost//:detail",
+    "@boost//:predef",
+    "@boost//:random",
+    "@boost//:timer",
+    "@boost//:typeof",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        defines = ["BOOST_TIMER_ENABLE_DEPRECATED"],  # crc_test.cpp currently uses depricated timer header
+        names = [
+            "crc_test",
+            "crc_test2",
+        ],
+        deps = DEPS,
+    ),
+)
