diff --git a/src/google/protobuf/BUILD.bazel b/src/google/protobuf/BUILD.bazel
index d986479..9876481 100644
--- a/src/google/protobuf/BUILD.bazel
+++ b/src/google/protobuf/BUILD.bazel
@@ -373,7 +373,7 @@ cc_library(
         "@com_google_absl//absl/strings:internal",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/time",
-        "@utf8_range//:utf8_validity",
+        "//third_party/utf8_range:utf8_validity",
     ],
 )
 
@@ -446,7 +446,7 @@ cc_library(
         "@com_google_absl//absl/strings:internal",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/time",
-        "@utf8_range//:utf8_validity",
+        "//third_party/utf8_range:utf8_validity",
     ],
 )
 
diff --git a/src/google/protobuf/generated_message_tctable_lite.cc b/src/google/protobuf/generated_message_tctable_lite.cc
index 86d067b..9c6f7a8 100644
--- a/src/google/protobuf/generated_message_tctable_lite.cc
+++ b/src/google/protobuf/generated_message_tctable_lite.cc
@@ -40,7 +40,7 @@
 #include "google/protobuf/message_lite.h"
 #include "google/protobuf/parse_context.h"
 #include "google/protobuf/wire_format_lite.h"
-#include "utf8_validity.h"
+#include "third_party/utf8_range/utf8_validity.h"
 
 
 // clang-format off
diff --git a/src/google/protobuf/json/BUILD.bazel b/src/google/protobuf/json/BUILD.bazel
index ea363ac..898bb77 100644
--- a/src/google/protobuf/json/BUILD.bazel
+++ b/src/google/protobuf/json/BUILD.bazel
@@ -115,7 +115,7 @@ cc_library(
         "@com_google_absl//absl/types:optional",
         "@com_google_absl//absl/types:span",
         "@com_google_absl//absl/types:variant",
-        "@utf8_range//:utf8_validity",
+        "//third_party/utf8_range:utf8_validity",
     ],
 )
 
diff --git a/src/google/protobuf/json/internal/untyped_message.cc b/src/google/protobuf/json/internal/untyped_message.cc
index 5c3c0e8..b3a6bf2 100644
--- a/src/google/protobuf/json/internal/untyped_message.cc
+++ b/src/google/protobuf/json/internal/untyped_message.cc
@@ -54,7 +54,7 @@
 #include "google/protobuf/port.h"
 #include "google/protobuf/util/type_resolver.h"
 #include "google/protobuf/wire_format_lite.h"
-#include "utf8_validity.h"
+#include "third_party/utf8_range/utf8_validity.h"
 #include "google/protobuf/stubs/status_macros.h"
 
 // Must be included last.
diff --git a/src/google/protobuf/parse_context.cc b/src/google/protobuf/parse_context.cc
index 11d2181..0a5eed1 100644
--- a/src/google/protobuf/parse_context.cc
+++ b/src/google/protobuf/parse_context.cc
@@ -38,7 +38,7 @@
 #include "google/protobuf/message_lite.h"
 #include "google/protobuf/repeated_field.h"
 #include "google/protobuf/wire_format_lite.h"
-#include "utf8_validity.h"
+#include "third_party/utf8_range/utf8_validity.h"
 
 
 // Must be included last.
diff --git a/src/google/protobuf/wire_format_lite.cc b/src/google/protobuf/wire_format_lite.cc
index 2a4adb1..48bee19 100644
--- a/src/google/protobuf/wire_format_lite.cc
+++ b/src/google/protobuf/wire_format_lite.cc
@@ -44,7 +44,7 @@
 #include "absl/strings/cord.h"
 #include "absl/strings/str_cat.h"
 #include "absl/strings/str_format.h"
-#include "utf8_validity.h"
+#include "third_party/utf8_range/utf8_validity.h"
 
 
 // Must be included last.
