--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,37 @@
+""" Builds libzmq.
+"""
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "srcs",
+    srcs = glob(
+        ["**"],
+        exclude = ["bazel-*"],
+    ),
+)
+
+cmake(
+    name = "libzmq",
+    cache_entries = {
+        # Perftools fail to link on some Linux distributions and turning off
+        # shared library build also disables perftools from building.
+        "BUILD_SHARED": "OFF",
+        # Tests fail to link on some Linux distributions.
+        "ZMQ_BUILD_TESTS": "OFF",
+        # Gnu TLS is not available in BCR.
+        "WITH_TLS": "OFF",
+    },
+    copts = ["-w"],
+    env = {"CMAKE_BUILD_TYPE": "Release"},
+    lib_source = ":srcs",
+    out_include_dir = "include",
+    out_static_libs = ["libzmq.a"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "libzmq_headers_only",
+    hdrs = glob(["include/*.h"]),
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+)
