diff --git a/src/include/common/openssl.h b/src/include/common/openssl.h
index fcc5b50..dfab707 100644
--- a/src/include/common/openssl.h
+++ b/src/include/common/openssl.h
@@ -40,8 +40,6 @@

 /* src/common/protocol_openssl.c */
 #ifndef SSL_CTX_set_min_proto_version
-extern int	SSL_CTX_set_min_proto_version(SSL_CTX *ctx, int version);
-extern int	SSL_CTX_set_max_proto_version(SSL_CTX *ctx, int version);
 #endif

 #endif							/* USE_OPENSSL */
diff --git a/src/interfaces/libpq/fe-secure-openssl.c b/src/interfaces/libpq/fe-secure-openssl.c
index 5f34049..97a04e1 100644
--- a/src/interfaces/libpq/fe-secure-openssl.c
+++ b/src/interfaces/libpq/fe-secure-openssl.c
@@ -1389,12 +1389,9 @@ open_client_SSL(PGconn *conn)
 							 * codes listed here, but a hint about protocol
 							 * versions seems like it's appropriate for all.
 							 */
-						case SSL_R_NO_PROTOCOLS_AVAILABLE:
 						case SSL_R_UNSUPPORTED_PROTOCOL:
-						case SSL_R_BAD_PROTOCOL_VERSION_NUMBER:
 						case SSL_R_UNKNOWN_PROTOCOL:
 						case SSL_R_UNKNOWN_SSL_VERSION:
-						case SSL_R_UNSUPPORTED_SSL_VERSION:
 						case SSL_R_WRONG_SSL_VERSION:
 						case SSL_R_WRONG_VERSION_NUMBER:
 						case SSL_R_TLSV1_ALERT_PROTOCOL_VERSION:

