===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,8 +1,9 @@
 module(
     name = "depend_on_what_you_use",
     # Keep in sync with .bcr/presubmit.yml and the README.md, test/aspect/execute_tests.py and test/workspace_integration/test.py
     bazel_compatibility = [">=7.6.0"],
+    version = "1.0.0",
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.8.2")
 bazel_dep(name = "buildozer", version = "8.5.1")
