diff --git a/BUILD.bazel b/BUILD.bazel
index e6622ff..1da0d55 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -30,6 +30,8 @@ package(default_visibility = ["//visibility:public"])
 
 licenses(["notice"])
 
+exports_files(["COPYING"])
+
 SNAPPY_VERSION = (1, 2, 2)
 
 config_setting(
diff --git a/MODULE.bazel b/MODULE.bazel
index 4f80d95..2686580 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,6 @@
 module(
     name = "snappy",
-    version = "1.2.2",
+    version = "1.2.2.bcr.1",
     compatibility_level = 1,
 )
 
