--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,38 @@
+""" Builds libxml2.
+"""
+
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]),
+)
+
+cache_entries = {
+    "CMAKE_INSTALL_LIBDIR": "lib",
+    "CMAKE_POSITION_INDEPENDENT_CODE": "ON",
+    "BUILD_SHARED_LIBS": "OFF",
+    #libxml2 specific options.
+    "LIBXML2_WITH_PYTHON": "OFF",
+    "LIBXML2_WITH_LZMA": "OFF",
+    "LIBXML2_WITH_ICONV": "OFF",
+    "LIBXML2_WITH_TESTS": "OFF",
+    "LIBXML2_WITH_PROGRAMS": "OFF",
+}
+
+cmake(
+    name = "libxml2",
+    env = {"CMAKE_BUILD_TYPE": "Release"},
+    lib_source = ":srcs",
+    cache_entries = cache_entries,
+    out_static_libs = select({
+        "@platforms//os:windows": [
+            "libxml2s.lib",
+        ],
+        "//conditions:default": [
+            "libxml2.a",
+        ],
+    }),
+    out_include_dir = "include/libxml2",
+    visibility = ["//visibility:public"],
+)
