diff --git a/MODULE.bazel b/MODULE.bazel
index 71ae8e7b6..8c7426cdd 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -18,6 +18,7 @@ IO_GRPC_GRPC_JAVA_ARTIFACTS = [
     "com.google.errorprone:error_prone_annotations:2.28.0",
     "com.google.guava:failureaccess:1.0.1",
     "com.google.guava:guava:33.2.1-android",
+    "com.google.j2objc:j2objc-annotations:3.0.0",
     "com.google.re2j:re2j:1.7",
     "com.google.truth:truth:1.4.2",
     "com.squareup.okhttp:okhttp:2.7.5",
@@ -46,8 +47,8 @@ IO_GRPC_GRPC_JAVA_ARTIFACTS = [
 # GRPC_DEPS_END
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
-bazel_dep(name = "googleapis", repo_name = "com_google_googleapis", version = "0.0.0-20240326-1c8d509c5")
-bazel_dep(name = "grpc", repo_name = "com_github_grpc_grpc", version = "1.56.3.bcr.1")
+bazel_dep(name = "googleapis", repo_name = "com_google_googleapis", version = "0.0.0-20240819-fe8ba054a")
+bazel_dep(name = "grpc", repo_name = "com_github_grpc_grpc", version = "1.66.0.bcr.2")
 bazel_dep(name = "grpc-proto", repo_name = "io_grpc_grpc_proto", version = "0.0.0-20240627-ec30f58")
 bazel_dep(name = "protobuf", repo_name = "com_google_protobuf", version = "23.1")
 bazel_dep(name = "rules_cc", version = "0.0.9")