===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -4,9 +4,9 @@
     name = "rules_scala",
     # The publish-to-bcr GitHub workflow supplies the actual `version`. See:
     # - https://bazel.build/external/faq#module-versioning-best-practices
     # - https://github.com/bazel-contrib/publish-to-bcr/blob/v1.1.0/src/domain/create-entry.ts#L176-L216
-    version = "",
+    version = "7.2.3",
     bazel_compatibility = [">=7.1.0"],
 )
 
 SCALA_VERSION = "2.12.21"
