--- /dev/null
+++ BUILD.bazel
@@ -0,0 +0,37 @@
+""" Builds libyaml.
+"""
+
+load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+expand_template(
+    name = "config_h",
+    out = "include/config.h",
+    substitutions = {
+        "@YAML_VERSION_MAJOR@": "0",
+        "@YAML_VERSION_MINOR@": "2",
+        "@YAML_VERSION_PATCH@": "5",
+        "@YAML_VERSION_STRING@": "0.2.5",
+    },
+    template = "cmake/config.h.in",
+)
+
+cc_library(
+    name = "yaml",
+    srcs = [
+        "src/api.c",
+        "src/dumper.c",
+        "src/emitter.c",
+        "src/loader.c",
+        "src/parser.c",
+        "src/reader.c",
+        "src/scanner.c",
+        "src/writer.c",
+        "src/yaml_private.h",
+    ] + [":config_h"],
+    hdrs = ["include/yaml.h"],
+    copts = ["-w"],
+    includes = ["include"],
+    local_defines = ["HAVE_CONFIG_H"],
+    visibility = ["//visibility:public"],
+)
