diff --git a/MODULE.bazel b/MODULE.bazel
index 1e500de..380b6d3 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,5 +1,7 @@
 module(
     name = "rules_ml_toolchain",
+    version = "0.4.0-20260506-398d613",
+    bazel_compatibility = [">=7.2.1"],
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.8.2")
@@ -8,7 +10,7 @@ bazel_dep(name = "rules_proto", version = "7.1.0")
 bazel_dep(name = "pybind11_bazel", version = "2.13.6")
 bazel_dep(name = "rules_cc", version = "0.2.0")
 bazel_dep(name = "abseil-cpp", version = "20250814.1", repo_name = "com_google_absl")
-bazel_dep(name = "googletest", version = "1.17.0", repo_name = "gtest")
+bazel_dep(name = "googletest", version = "1.17.0.bcr.2", repo_name = "gtest")
 bazel_dep(name = "platforms", version = "1.0.0")
 bazel_dep(name = "rules_python", version = "1.8.5")
 
@@ -179,4 +179,3 @@ use_repo(
 
 local_sysroot = use_extension("//extensions:local_sysroot_configure.bzl", "local_sysroot_ext")
 use_repo(local_sysroot, "local_sysroot_config")
-
