--- MODULE.bazel.orig	2023-09-01 09:50:45.729641178 +0000
+++ MODULE.bazel	2023-09-01 09:54:43.625118522 +0000
@@ -13,10 +13,10 @@
 bazel_dep(name = "platforms", version = "0.0.7")
 bazel_dep(name = "rules_cc", version = "0.0.8")
 bazel_dep(name = "abseil-cpp", version = "20230802.0", repo_name = "com_google_absl")
-bazel_dep(name = "google_benchmark", version = "1.8.2", repo_name = "com_github_google_benchmark")
-bazel_dep(name = "googletest", version = "1.14.0", repo_name = "com_google_googletest")
+bazel_dep(name = "google_benchmark", version = "1.8.2", repo_name = "com_github_google_benchmark", dev_dependency = True)
+bazel_dep(name = "googletest", version = "1.14.0", repo_name = "com_google_googletest", dev_dependency = True)
 bazel_dep(name = "rules_python", version = "0.25.0")
-bazel_dep(name = "abseil-py", version = "1.4.0", repo_name = "io_abseil_py")
+bazel_dep(name = "abseil-py", version = "1.4.0", repo_name = "io_abseil_py", dev_dependency = True)
 bazel_dep(name = "pybind11_bazel", version = "2.11.1")
 
 python_configure = use_extension("@pybind11_bazel//:python_configure.bzl", "extension")
