--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,27 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "rapidcheck",
+    srcs = glob([
+        "src/**/*.cpp",
+        "src/**/*.h",
+        "include/**/*.cpp",
+    ]),
+    hdrs = glob([
+        "include/**/*.h",
+        "include/**/*.hpp",
+    ]),
+    includes = [
+        "extras/gtest/include",
+        "include",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "rapidcheck_gtest",
+    hdrs = ["extras/gtest/include/rapidcheck/gtest.h"],
+    includes = ["extras/gtest/include"],
+    visibility = ["//visibility:public"],
+    deps = [":rapidcheck"],
+)
