--- MODULE.bazel
+++ MODULE.bazel
@@ -2,7 +2,7 @@
     name = "grpc-java",
     compatibility_level = 0,
     repo_name = "io_grpc_grpc_java",
-    version = "0",
+    version = "1.64.0.bcr.1",
 )

 # GRPC_DEPS_START
@@ -18,6 +18,7 @@ IO_GRPC_GRPC_JAVA_ARTIFACTS = [
     "com.google.errorprone:error_prone_annotations:2.23.0",
     "com.google.guava:failureaccess:1.0.1",
     "com.google.guava:guava:32.1.3-android",
+    "com.google.j2objc:j2objc-annotations:1.3",
     "com.google.re2j:re2j:1.7",
     "com.google.truth:truth:1.1.5",
     "com.squareup.okhttp:okhttp:2.7.5",
@@ -45,7 +46,7 @@
 ]
 # GRPC_DEPS_END

-bazel_dep(name = "googleapis", repo_name = "com_google_googleapis", version = "0.0.0-20240326-1c8d509c5")
-bazel_dep(name = "grpc", repo_name = "com_github_grpc_grpc", version = "1.56.3.bcr.1")
+bazel_dep(name = "googleapis", repo_name = "com_google_googleapis", version = "0.0.0-20240326-1c8d509c5.bcr.1")
+bazel_dep(name = "grpc", repo_name = "com_github_grpc_grpc", version = "1.65.0.bcr.1")
 bazel_dep(name = "protobuf", repo_name = "com_google_protobuf", version = "23.1")
 bazel_dep(name = "rules_cc", version = "0.0.9")
@@ -199,8 +200,5 @@
     target = "@io_grpc_grpc_java//util",
 )

-switched_rules = use_extension("@com_google_googleapis//:extensions.bzl", "switched_rules")
-
-switched_rules.use_languages(java = True)
-
-use_repo(switched_rules, "com_google_googleapis_imports")
+googleapis_injections = use_extension("@com_google_googleapis//:injections.bzl", "injections")
+googleapis_injections.inject(grpc_java = "@io_grpc_grpc_java")
