===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -6,7 +6,7 @@ Defines all the external repositories and dependencies for rules_ios.
 # Version is updated during release to the registry.
 module(
     name = "rules_ios",
-    version = "0",
+    version = "3.0.1",
     bazel_compatibility = [">=5.0.0"],
     compatibility_level = 1,
     repo_name = "build_bazel_rules_ios",
