diff --git a/quic/tools/tperf/TperfTcp.cpp b/quic/tools/tperf/TperfTcp.cpp
index 1588ed7..ec9da8d 100644
--- a/quic/tools/tperf/TperfTcp.cpp
+++ b/quic/tools/tperf/TperfTcp.cpp
@@ -7,6 +7,8 @@
 
 #include <quic/tools/tperf/TperfTcp.h>
 
+#include <folly/CppAttributes.h>
+
 #include <algorithm>
 #include <array>
 #include <chrono>
@@ -131,8 +133,8 @@ class TPerfTcpClient::Impl : public folly::AsyncSocket::ConnectCallback,
   }
 
   void getReadBuffer(
-      void* _Nonnull* _Nonnull bufReturn,
-      size_t* _Nonnull lenReturn) override {
+      void* FOLLY_NONNULL* FOLLY_NONNULL bufReturn,
+      size_t* FOLLY_NONNULL lenReturn) override {
     *bufReturn = readBuffer_.data();
     *lenReturn = readBuffer_.size();
   }
@@ -212,7 +214,7 @@ class TcpServerConnection
       public folly::AsyncTransportWrapper::WriteCallback {
  public:
   TcpServerConnection(
-      folly::EventBase* _Nonnull eventBase,
+      folly::EventBase* FOLLY_NONNULL eventBase,
       folly::NetworkSocket fd,
       std::shared_ptr<const fizz::server::FizzServerContext> serverCtx,
       const TPerfTcpServerConfig& config,
@@ -252,8 +254,8 @@ class TcpServerConnection
   }
 
   void getReadBuffer(
-      void* _Nonnull* _Nonnull bufReturn,
-      size_t* _Nonnull lenReturn) override {
+      void* FOLLY_NONNULL* FOLLY_NONNULL bufReturn,
+      size_t* FOLLY_NONNULL lenReturn) override {
     *bufReturn = readBuffer_.data();
     *lenReturn = readBuffer_.size();
   }
@@ -325,7 +327,7 @@ class TcpServerConnection
     }
   }
 
-  folly::EventBase* _Nonnull eventBase_;
+  folly::EventBase* FOLLY_NONNULL eventBase_;
   std::function<void(TcpServerConnection*)> onClose_;
   fizz::server::AsyncFizzServer::UniquePtr tcpServer_;
   std::array<uint8_t, kReadBufferSize> readBuffer_{};
