--- a/src/V3EmitMk.cpp
+++ b/src/V3EmitMk.cpp
@@ -814,7 +814,7 @@ class EmitMkHierVerilation final {
         of.puts("VM_HIER_RUN_DIR := " + cwd + "\n");
         of.puts("# Common options for hierarchical blocks\n");
         const string fullpath_bin = V3Os::filenameRealPath(v3Global.opt.buildDepBin());
-        const string verilator_wrapper = V3Os::filenameDir(fullpath_bin) + "/verilator";
+        const string verilator_wrapper = V3Os::filenameDir(fullpath_bin) + "/verilator_bin";
         of.puts("VM_HIER_VERILATOR := " + verilator_wrapper + "\n");
         of.puts("VM_HIER_INPUT_FILES := \\\n");
         const V3StringList& vFiles = v3Global.opt.vFiles();
