diff --git pcl_config.h.in pcl_config.h.in
index 2e93d63d6..5557dccf3 100644
--- pcl_config.h.in
+++ pcl_config.h.in
@@ -35,7 +35,13 @@
   #endif //PCL_MINOR_VERSION
 #endif 
 
-#define PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC ${PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC}
+#if (defined(EIGEN_DEFAULT_ALIGN_BYTES) && EIGEN_DEFAULT_ALIGN_BYTES==0) || (defined(EIGEN_MALLOC_ALREADY_ALIGNED) && EIGEN_MALLOC_ALREADY_ALIGNED)
+  #define PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC 1
+#else
+  #define PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC 0
+#endif
+
+#cmakedefine PCL_SILENCE_MALLOC_WARNING
 
 #cmakedefine HAVE_OPENNI 1
 
