diff --git a/tar/bsdtar.c b/tar/bsdtar.c
index 42baab28..95d53097 100644
--- a/tar/bsdtar.c
+++ b/tar/bsdtar.c
@@ -1070,7 +1070,11 @@ static const char *long_help_msg =
 	"  -c Create  -r Add/Replace  -t List  -u Update  -x Extract\n"
 	"Common Options:\n"
 	"  -b #  Use # 512-byte records per I/O block\n"
+#ifdef _PATH_DEFTAPE 
 	"  -f <filename>  Location of archive (default " _PATH_DEFTAPE ")\n"
+#else
+	"  -f <filename>  Location of archive \n"
+#endif
 	"  -v    Verbose\n"
 	"  -w    Interactive\n"
 	"Create: %p -c [options] [<file> | <dir> | @<archive> | -C <dir> ]\n"