===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -11,9 +11,11 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-module(name = "google_bazel_common")
+module(name = "google_bazel_common",
+    version = "0.0.1",
+)
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
 bazel_dep(name = "rules_java", version = "8.6.1")
 bazel_dep(name = "rules_jvm_external", version = "6.6")
