diff --git BUILD.bazel BUILD.bazel
index 0306468e..d958b862 100644
--- BUILD.bazel
+++ BUILD.bazel
@@ -30,6 +30,9 @@
 #
 #   Bazel Build for Google C++ Testing Framework(Google Test)
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 package(default_visibility = ["//visibility:public"])
 
 licenses(["notice"])
diff --git googlemock/test/BUILD.bazel googlemock/test/BUILD.bazel
index d4297c80..625cff96 100644
--- googlemock/test/BUILD.bazel
+++ googlemock/test/BUILD.bazel
@@ -30,6 +30,8 @@
 #
 #   Bazel Build for Google C++ Testing Framework(Google Test)-googlemock
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("@rules_python//python:defs.bzl", "py_library", "py_test")
 
 licenses(["notice"])
diff --git googletest/test/BUILD.bazel googletest/test/BUILD.bazel
index 234b1a8f..65d1fe5c 100644
--- googletest/test/BUILD.bazel
+++ googletest/test/BUILD.bazel
@@ -30,6 +30,8 @@
 #
 # Bazel BUILD for The Google C++ Testing Framework (Google Test)
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("@rules_python//python:defs.bzl", "py_library", "py_test")
 
 licenses(["notice"])
