diff --git a/MODULE.bazel b/MODULE.bazel
index 2a1c5aec1e..4e3e877b85 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -14,7 +14,7 @@
 
 module(
     name = "grpc",
-    version = "1.78.0",
+    version = "1.78.0.bcr.1",
     compatibility_level = 1,
     repo_name = "com_github_grpc_grpc",
 )
@@ -60,7 +60,7 @@ bazel_dep(name = "rules_shell", version = "0.4.0", dev_dependency = True)
 # Python dependencies
 # ===================
 
-bazel_dep(name = "rules_python", version = "1.5.4")
+bazel_dep(name = "rules_python", version = "1.6.3")
 
 PYTHON_VERSIONS = [
     "3.9",
@@ -68,7 +68,7 @@ PYTHON_VERSIONS = [
     "3.11",
     "3.12",
     "3.13",
-    "3.14.0b2",
+    "3.14",
 ]
 
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
diff --git a/bazel/grpc_python_deps.bzl b/bazel/grpc_python_deps.bzl
index 4c3def0332..a07079057c 100644
--- a/bazel/grpc_python_deps.bzl
+++ b/bazel/grpc_python_deps.bzl
@@ -22,9 +22,9 @@ def grpc_python_deps():
     if "rules_python" not in native.existing_rules():
         http_archive(
             name = "rules_python",
-            sha256 = "13671d304cfe43350302213a60d93a5fc0b763b0a6de17397e3e239253b61b73",
-            strip_prefix = "rules_python-1.5.4",
-            url = "https://github.com/bazel-contrib/rules_python/releases/download/1.5.4/rules_python-1.5.4.tar.gz",
+            sha256 = "2f5c284fbb4e86045c2632d3573fc006facbca5d1fa02976e89dc0cd5488b590",
+            strip_prefix = "rules_python-1.6.3",
+            url = "https://github.com/bazel-contrib/rules_python/releases/download/1.6.3/rules_python-1.6.3.tar.gz",
         )
 
     python_configure(name = "local_config_python")
diff --git a/templates/MODULE.bazel.inja b/templates/MODULE.bazel.inja
index 53108d9382..e2b88a77b7 100644
--- a/templates/MODULE.bazel.inja
+++ b/templates/MODULE.bazel.inja
@@ -60,7 +60,7 @@ bazel_dep(name = "rules_shell", version = "0.4.0", dev_dependency = True)
 # Python dependencies
 # ===================
 
-bazel_dep(name = "rules_python", version = "1.5.4")
+bazel_dep(name = "rules_python", version = "1.6.3")
 
 PYTHON_VERSIONS = [
     "3.9",
@@ -68,7 +68,7 @@ PYTHON_VERSIONS = [
     "3.11",
     "3.12",
     "3.13",
-    "3.14.0b2",
+    "3.14",
 ]
 
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
