===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,9 +1,9 @@
 module(
     name = "windows_support",
 
     # NOTE:
-    version = "",
+    version = "0.1.1",
     #
     # Always leave version unset or set to "" (the default). The default value
     # can prevent issues when the module is used via non-registry overrides
     # (e.g. https://github.com/bazel-contrib/rules_go/issues/4380).
