diff --git a/build.rs b/build.rs
index 129ec36..24a7fc0 100644
--- a/build.rs
+++ b/build.rs
@@ -82,13 +82,13 @@ fn sync_opaque_types_lockfile() {
 }
 
 fn main() {
-    sync_opaque_types_lockfile();
-    buildrs::opaque_types_generator::generate_opaque_types();
+    // sync_opaque_types_lockfile();
+    // buildrs::opaque_types_generator::generate_opaque_types();
     buildrs::cbindgen_generator::generate_c_headers();
-    if let Some(out_path) = env::var_os("ZENOHC_DUMP_SRC_DIR") {
-        let out_path = std::path::Path::new(&out_path);
-        dump_rust_sources(out_path);
-    }
+    // if let Some(out_path) = env::var_os("ZENOHC_DUMP_SRC_DIR") {
+    //     let out_path = std::path::Path::new(&out_path);
+    //     dump_rust_sources(out_path);
+    // }
     println!("cargo:rerun-if-changed=build.rs");
     println!("cargo:rerun-if-changed=buildrs");
     println!("cargo:rerun-if-changed=src");
diff --git a/buildrs/cbindgen_generator.rs b/buildrs/cbindgen_generator.rs
index d3ad7da..9634c5b 100644
--- a/buildrs/cbindgen_generator.rs
+++ b/buildrs/cbindgen_generator.rs
@@ -36,19 +36,19 @@ static RUST_TO_C_FEATURES: phf::Map<&'static str, &'static str> = phf_map! {
 };
 
 pub fn generate_c_headers() {
-    let crate_dir = std::env::var("CARGO_MANIFEST_DIR").unwrap();
-    let config = cbindgen::Config::from_root_or_default(crate_dir.clone());
-    cbindgen::Builder::new()
-        .with_config(config)
-        .with_only_target_dependencies(true)
-        .with_crate(crate_dir)
-        .with_src(get_out_rs_path().join("opaque_types.rs"))
-        .generate()
-        .expect("Unable to generate bindings")
-        .write_to_file(BUGGY_GENERATION_PATH);
+    // let crate_dir = std::env::var("CARGO_MANIFEST_DIR").unwrap();
+    // let config = cbindgen::Config::from_root_or_default(crate_dir.clone());
+    // cbindgen::Builder::new()
+    //     .with_config(config)
+    //     .with_only_target_dependencies(true)
+    //     .with_crate(crate_dir)
+    //     .with_src(get_out_rs_path().join("opaque_types.rs"))
+    //     .generate()
+    //     .expect("Unable to generate bindings")
+    //     .write_to_file(BUGGY_GENERATION_PATH);
 
     fix_cbindgen(BUGGY_GENERATION_PATH, GENERATION_PATH);
-    std::fs::remove_file(BUGGY_GENERATION_PATH).unwrap();
+    // std::fs::remove_file(BUGGY_GENERATION_PATH).unwrap();
 
     preprocess_header(GENERATION_PATH, PREPROCESS_PATH);
     create_generics_header(PREPROCESS_PATH, "include/zenoh_macros.h");
diff --git a/buildrs/common_helpers.rs b/buildrs/common_helpers.rs
index 4f4dae7..5c20f0e 100644
--- a/buildrs/common_helpers.rs
+++ b/buildrs/common_helpers.rs
@@ -32,20 +32,22 @@ pub fn test_feature(feature: &str) -> bool {
 pub fn cargo_target_dir() -> PathBuf {
     // OUT_DIR is a path to the directory where the build script writes its output.
     let out_dir = PathBuf::from(env::var("OUT_DIR").expect("OUT_DIR should be set"));
-    // PROFILE is the profile that the build script is being run for. This will be one of "debug" or "release".
-    let profile = env::var("PROFILE").expect("PROFILE should be set");
+    out_dir.to_path_buf()
 
-    let mut target_dir = None;
-    let mut out_dir_path = out_dir.as_path();
-    while let Some(parent) = out_dir_path.parent() {
-        if parent.ends_with(&profile) {
-            target_dir = Some(parent);
-            break;
-        }
-        out_dir_path = parent;
-    }
+    // // PROFILE is the profile that the build script is being run for. This will be one of "debug" or "release".
+    // let profile = env::var("PROFILE").expect("PROFILE should be set");
 
-    target_dir
-        .expect("OUT_DIR should be a child of a PROFILE directory")
-        .to_path_buf()
+    // let mut target_dir = None;
+    // let mut out_dir_path = out_dir.as_path();
+    // while let Some(parent) = out_dir_path.parent() {
+    //     if parent.ends_with(&profile) {
+    //         target_dir = Some(parent);
+    //         break;
+    //     }
+    //     out_dir_path = parent;
+    // }
+
+    // target_dir
+    //     .expect("OUT_DIR should be a child of a PROFILE directory")
+    //     .to_path_buf()
 }
diff --git a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -59,6 +59,8 @@
     "transport_udp",
     "transport_unixsock-stream",
     "transport_ws",
+    "shared-memory",
+    "unstable",
 ]
 
 [badges]
@@ -76,8 +78,8 @@
 unwrap-infallible = "0.1.5"
 const_format = "0.2.32"
 flume = "*"
-zenoh = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0", default-features = false, features = ["internal"] }
-zenoh-ext = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0", default-features = false, features=["internal"] }
+zenoh = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0", default-features = false, features = ["internal", "shared-memory", "unstable"] }
+zenoh-ext = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0", default-features = false, features = ["internal", "unstable"] }
 zenoh-runtime = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0" }
 zenoh-util = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0" }
 # zenoh-pinned-deps-1-75 = { version = "1.7.2", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
diff --git a/build-resources/opaque-types/Cargo.toml b/build-resources/opaque-types/Cargo.toml
--- a/build-resources/opaque-types/Cargo.toml
+++ b/build-resources/opaque-types/Cargo.toml
@@ -25,14 +25,21 @@
 
 panic = [
 ] # The whole purpose of this project is to generate set of compilation panic messages with calculated structure sizes. To do it the "panic" feature should be set. By default we just want to check if build is successful.
-default = ["zenoh/default"]
+default = [
+    "zenoh/default",
+    "shared-memory",
+    "unstable",
+]
 
 [dependencies]
 zenoh = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0", default-features = false, features = [
     "internal",
+    "shared-memory",
+    "unstable",
 ] }
 zenoh-ext = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0", default-features = false, features = [
     "internal",
+    "unstable",
 ] }
 zenoh-protocol = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0" }
 const_format = "0.2.32"
