===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -2,13 +2,13 @@
 
 module(
     name = "aspect_rules_js",
-    version = "0.0.0",
+    version = "1.40.0",
     compatibility_level = 1,
 )
 
 # Lower-bounds for runtime dependencies.
 # Do not bump these unless rules_js requires a newer version to function.
-bazel_dep(name = "aspect_bazel_lib", version = "1.42.0")
+bazel_dep(name = "aspect_bazel_lib", version = "1.42.2")
 bazel_dep(name = "aspect_rules_lint", version = "0.12.0")
 bazel_dep(name = "bazel_features", version = "0.1.0")
 bazel_dep(name = "bazel_skylib", version = "1.5.0")
