diff --git a/include/rawrtcc/config.h b/include/rawrtcc/config.h
new file mode 100644
index 000000000..e69de29bb
--- /dev/null
+++ b/include/rawrtcc/config.h
@@ -0,0 +1,1 @@
+
diff --git a/src/message_buffer/buffer.c b/src/message_buffer/buffer.c
index 75004f1aa..24c854736 100644
--- a/src/message_buffer/buffer.c
+++ b/src/message_buffer/buffer.c
@@ -79,12 +79,13 @@ enum rawrtc_code rawrtc_message_buffer_clear(
 
         // Handle message
         unlink = message_handler(buffered_message->buffer, buffered_message->context, arg);
+        struct le* next = le->next;
         if (unlink) {
             list_unlink(le);
         }
 
         // Get next message
-        le = le->next;
+        le = next;
 
         // Remove message
         if (unlink) {
