--- a/fake_lib/fakeethercat.h	2026-03-19 11:41:57.239780394 +0000
+++ b/fake_lib/fakeethercat.h	2026-03-19 11:41:57.246274415 +0000
@@ -128,7 +128,7 @@
         uint32_t vendor_id;    /**< Expected vendor ID. */
         uint32_t product_code; /**< Expected product code. */
         std::map<unsigned int, syncManager> sync_managers;
-        std::map<sdo_address, std::basic_string<uint8_t>> sdos;
+        std::map<sdo_address, std::string> sdos;
 
         ec_slave_config(
                 ec_address address,
