--- a/fake_lib/fakeethercat.cpp	2026-03-19 11:41:57.243012895 +0000
+++ b/fake_lib/fakeethercat.cpp	2026-03-19 11:41:57.257586856 +0000
@@ -845,7 +845,7 @@
 )
 {
     sc->sdos[sdo_address {index, subindex}] =
-            std::basic_string<uint8_t>(data, data + size);
+            std::string(reinterpret_cast<const char*>(data), size);
     return 0;
 }
 
@@ -1493,11 +1493,11 @@
     map2Json(
             out,
             sdos,
-            [](std::ostream &out, const std::basic_string<uint8_t> &value) {
+            [](std::ostream &out, const std::string &value) {
                 out << "\"0x";
                 for (const auto s : value) {
                     out << std::hex << std::setfill('0') << std::setw(2)
-                        << (unsigned) s;
+                        << (unsigned char) s;
                 }
                 out << '"';
             },
