diff --git a/bazel/glog.bzl b/bazel/glog.bzl
index 40833b4..bcbdc8a 100644
--- a/bazel/glog.bzl
+++ b/bazel/glog.bzl
@@ -33,7 +33,7 @@ def dict_union(x, y):
 
 def glog_library(namespace = "google", with_gflags = 1, **kwargs):
     if native.repository_name() != "@":
-        repo_name = native.repository_name().lstrip("@")
+        repo_name = native.repository_name()[1:] # Strip the first leading @
         gendir = "$(GENDIR)/external/" + repo_name
         src_windows = "external/%s/src/windows" % repo_name
     else:
