--- MODULE.bazel
+++ MODULE.bazel
@@ -1,29 +1,6 @@
-"""
-A C++17 port of the JavaScript pixelmatch library, providing a small pixel-level image comparison library.
-
-To use, add the following to your MODULE.bazel file:
-```
-bazel_dep(name = "pixelmatch-cpp17", version = "1.0.3")
-```
-"""
-
 module(
     name = "pixelmatch-cpp17",
     version = "1.0.3",
     compatibility_level = 0,
 )
-
-#
-# Build dependencies
-#
 bazel_dep(name = "rules_cc", version = "0.1.1")
-
-#
-# Test dependencies
-#
-bazel_dep(name = "googletest", version = "1.16.0.bcr.1", dev_dependency = True)
-
-#
-# Fuzzing dependencies
-#
-bazel_dep(name = "rules_fuzzing", version = "0.5.2", dev_dependency = True)
