--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,52 @@
+# Copyright 2010-2024 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cc_library(
+    name = "glpk",
+    srcs = glob(
+        [
+            "src/*/*.c",
+            "src/*.h",
+            "src/*/*.h",
+        ],
+        exclude = [
+            "src/proxy/main.c",
+            "src/zlib/*",
+        ],
+    ),
+    hdrs = [
+        "src/glpk.h",
+    ],
+    copts = [
+        "-w",
+    ],
+    deps = ["@zlib"],
+    includes = [
+        "src",
+        # TODO find a way to determine "external/glpk~VERSION" for use via copts "-I"
+        "src/amd",
+        "src/api",
+        "src/bflib",
+        "src/colamd",
+        "src/draft",
+        "src/env",
+        "src/intopt",
+        "src/minisat",
+        "src/misc",
+        "src/mpl",
+        "src/npp",
+        "src/proxy",
+        "src/simplex",
+    ],
+    visibility = ["//visibility:public"],
+)

