diff --git include/boost/log/detail/adaptive_mutex.hpp include/boost/log/detail/adaptive_mutex.hpp
index fa06a0b..072a806 100644
--- include/boost/log/detail/adaptive_mutex.hpp
+++ include/boost/log/detail/adaptive_mutex.hpp
@@ -17,6 +17,7 @@
 #define BOOST_LOG_DETAIL_ADAPTIVE_MUTEX_HPP_INCLUDED_
 
 #include <boost/log/detail/config.hpp>
+#include <boost/throw_exception.hpp>
 
 #ifdef BOOST_HAS_PRAGMA_ONCE
 #pragma once
