diff --git a/include/behaviortree_cpp/blackboard.h b/include/behaviortree_cpp/blackboard.h
index 1c3aa96c..71495e7a 100644
--- a/include/behaviortree_cpp/blackboard.h
+++ b/include/behaviortree_cpp/blackboard.h
@@ -6,7 +6,7 @@
 #include <mutex>
 
 #include "behaviortree_cpp/basic_types.h"
-#include "behaviortree_cpp/contrib/json.hpp"
+#include "nlohmann/json.hpp"
 #include "behaviortree_cpp/utils/safe_any.hpp"
 #include "behaviortree_cpp/exceptions.h"
 #include "behaviortree_cpp/utils/locked_reference.hpp"
diff --git a/include/behaviortree_cpp/bt_factory.h b/include/behaviortree_cpp/bt_factory.h
index 88009912..7aa13a91 100644
--- a/include/behaviortree_cpp/bt_factory.h
+++ b/include/behaviortree_cpp/bt_factory.h
@@ -21,7 +21,7 @@
 #include <set>
 #include <vector>
 
-#include "behaviortree_cpp/contrib/magic_enum.hpp"
+#include "magic_enum/magic_enum.hpp"
 #include "behaviortree_cpp/behavior_tree.h"
 
 namespace BT
diff --git a/include/behaviortree_cpp/json_export.h b/include/behaviortree_cpp/json_export.h
index 3eeee714..20976a30 100644
--- a/include/behaviortree_cpp/json_export.h
+++ b/include/behaviortree_cpp/json_export.h
@@ -5,7 +5,7 @@
 #include "behaviortree_cpp/basic_types.h"
 
 // Use the version nlohmann::json embedded in BT.CPP
-#include "behaviortree_cpp/contrib/json.hpp"
+#include "nlohmann/json.hpp"
 
 namespace BT
 {
diff --git a/include/behaviortree_cpp/loggers/groot2_protocol.h b/include/behaviortree_cpp/loggers/groot2_protocol.h
index 507f2a00..c97dbe9f 100644
--- a/include/behaviortree_cpp/loggers/groot2_protocol.h
+++ b/include/behaviortree_cpp/loggers/groot2_protocol.h
@@ -9,7 +9,7 @@
 #include <condition_variable>
 #include <mutex>
 #include "behaviortree_cpp/basic_types.h"
-#include "behaviortree_cpp/contrib/json.hpp"
+#include "nlohmann/json.hpp"
 
 namespace BT::Monitor
 {
