--- include/avro/Types.hh
+++ include/avro/Types.hh
@@ -113,8 +113,8 @@ std::ostream &operator<<(std::ostream &os, const Null &null);
 template<>
 struct fmt::formatter<avro::Type> : fmt::formatter<std::string> {
     template<typename FormatContext>
-    auto format(avro::Type t, FormatContext &ctx) {
-        return fmt::formatter<std::string>::format(avro::toString(t), ctx);
+    auto format(avro::Type t, FormatContext &ctx) const {
+        return fmt::format_to(ctx.out(), "{}", avro::toString(const_cast<avro::Type&>(t)));
     }
 };

