diff --git a/src/scip/buildflags.h b/src/scip/buildflags.h
index c686abcd20..1ea6a4fbf4 100644
--- a/src/scip/buildflags.h
+++ b/src/scip/buildflags.h
@@ -29,10 +29,44 @@
 
 /*---+----1----+----2----+----3----+----4----+----5----+----6----+----7----+----8----+----9----+----0----+----1----+----2*/
 
+
+#if defined(__GNUC__) && defined(__linux__)
+#define SCIP_OSTYPE "Linux"
+#define SCIP_COMP "GCC"
+#elif defined(__APPLE__) && defined(__GNUC__)  // MacOS
+#define SCIP_OSTYPE "Darwin"
+#define SCIP_COMP "AppleLang"
+#elif defined(__FreeBSD__)
+#define SCIP_OSTYPE "FreeBSD"
+#define SCIP_COMP "GCC"
+#elif defined(__NetBSD__)
+#define SCIP_OSTYPE "NetBSD"
+#define SCIP_COMP "GCC"
+#elif defined(__OpenBSD__)
+#define SCIP_OSTYPE "OpenBSD"
+#define SCIP_COMP "GCC"
+#elif defined(_MSC_VER) || defined(__MINGW32__) || defined(__MINGW64__)  // Windows
+#define SCIP_OSTYPE "Windows"
+#define SCIP_COMP "VisualStudio"
+#else
+#define SCIP_OSTYPE "UnknownArch"
+#define SCIP_COMP "UnknownCompiler"
+#endif
+
+#if defined(__x86_64__) || defined(_M_X64)
+#define SCIP_ARCH "x86_64"
+#elif defined(__aarch64__)
+#define SCIP_ARCH "arm64"
+#else
+#define SCIP_ARCH "UnknownArch"
+#endif
+
+
+
 #define SCIP_BUILDFLAGS				\
-  "ARCH=arm64\n "                               \
-  "OSTYPE=Darwin-23.6.0\n "			\
-  "COMP=AppleClang 16.0.0.16000026\n "		\
+  "ARCH=" SCIP_ARCH "\n "                       \
+  "OSTYPE=" SCIP_OSTYPE "\n "			\
+  "COMP=" SCIP_COMP "\n "		\
   "BUILD=Release\n "				\
   "DEBUGSOL=OFF\n "				\
   "EXPRINT=none\n "				\
