diff --git a/src/symmetry/compute_symmetry_bliss.cpp b/src/symmetry/compute_symmetry_bliss.cpp
index 26117a8026..2f164d133b 100644
--- a/src/compute_symmetry_bliss.cpp	2024-11-08 16:36:13
+++ b/src/symmetry/compute_symmetry_bliss.cpp	2024-11-08 16:37:32
@@ -34,8 +34,9 @@
 #include "compute_symmetry.h"

 /* include bliss graph */
-#include <bliss/defs.hh>
-#include <bliss/graph.hh>
+#define BLISS_VERSION "0.73"
+#include <defs.hh>
+#include <graph.hh>

 #include <string.h>
 #include <vector>
