--- MODULE.bazel
+++ MODULE.bazel
@@ -3,11 +3,7 @@
 module(
     name = "aspect_bazel_lib",
     compatibility_level = 1,
-    toolchains_to_register = [
-        "@jq_toolchains//:all",
-        "@yq_toolchains//:all",
-    ],
-    version = "0.0.0",
+    version = "1.5.1",
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.1.1")
@@ -17,5 +13,8 @@
 ext = use_extension("@aspect_bazel_lib//lib:extensions.bzl", "ext")
 
 use_repo(ext, "jq_toolchains")
-
 use_repo(ext, "yq_toolchains")
+register_toolchains(
+    "@jq_toolchains//:all",
+    "@yq_toolchains//:all",
+)
