--- a/bazel/private/oss/toolchains/prebuilt/tool_integrity.bzl
+++ b/bazel/private/oss/toolchains/prebuilt/tool_integrity.bzl
@@ -6,8 +6,8 @@
   "protoc-36.0-linux-s390_64.zip": "27556b282e33d6e04300d05ed4c38bf85e5a92d7080efedd5c41fd02cf9e3f5b",
   "protoc-36.0-linux-x86_32.zip": "60fc1d382cbbad06584761e1bf1a79e4eeefa33a8667aad5a4184e4d31c45ddb",
   "protoc-36.0-linux-x86_64.zip": "bc8211ce760bd43ee21ddc145d6d9dbaeeabae205267a79d9054a240e367d4b4",
-  "protoc-36.0-osx-aarch_64.zip": "726141c541c84a01cdc501ae6c0c845d2dda26414ccf42e641ed096e7491344f",
-  "protoc-36.0-osx-x86_64.zip": "98c3ebfca2e2e15eb5c22eb2ee97457a1573739377532a9e36cc861afb12a2d6",
+  "protoc-36.0-osx-aarch_64.zip": "b6bc4afdcb880124bf342851d05155b6e3d9b6e661236d87b9c614250d26ae00",
+  "protoc-36.0-osx-x86_64.zip": "2847d952ecd1c466769ae3ca319c9cd34c3613542eba335dc9b02c49537f6c70",
   "protoc-36.0-win32.zip": "50b904fe09980eb44d566755eb65b4693a0deac30998ea3323c6901d9540e60d",
   "protoc-36.0-win64.zip": "510fb2369a4720adb457783768c5e1481a0e1137d9e7694478dfe9b8fe445dc1",
 }
