diff --git a/cudd/cudd.h b/cudd/cudd.h
index 7d6f10e..de8f082 100644
--- a/cudd/cudd.h
+++ b/cudd/cudd.h
@@ -58,6 +58,8 @@
 /*---------------------------------------------------------------------------*/
 
 #include <inttypes.h>
+#include <stdio.h>
+#include <stddef.h>
 
 /*---------------------------------------------------------------------------*/
 /* Constant declarations                                                     */
diff --git a/mtr/mtrInt.h b/mtr/mtrInt.h
index 09aaac9..934728e 100644
--- a/mtr/mtrInt.h
+++ b/mtr/mtrInt.h
@@ -52,6 +52,7 @@
 
 #include "config.h"
 #include "mtr.h"
+#include <stdint.h>
 
 /*---------------------------------------------------------------------------*/
 /* Constant declarations                                                     */
diff --git a/util/cstringstream.h b/util/cstringstream.h
index b5a85cd..24cc627 100644
--- a/util/cstringstream.h
+++ b/util/cstringstream.h
@@ -47,6 +47,8 @@
 #ifndef CSTRINGSTREAM_H_
 #define CSTRINGSTREAM_H_
 
+#include <stddef.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
