--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -52,17 +52,9 @@ python.toolchain(
     python_version = "3.11",
 )
 
-bazel_dep(name = "googleapis", version = "0.0.0", repo_name = "com_google_googleapis")
-archive_override(
-    module_name = "googleapis",
-    integrity = "sha256-6FrMqUyMsAgBxSwG2fdSxQPx2XNg1wsVmfTxTOji900=",
-    patch_strip = 1,
-    patches = ["//bazel:googleapis.modules.patch"],
-    strip_prefix = "googleapis-de157ca34fa487ce248eb9130293d630b501e4ad",
-    urls = [
-        "https://github.com/googleapis/googleapis/archive/de157ca34fa487ce248eb9130293d630b501e4ad.tar.gz",
-    ],
-)
+bazel_dep(name = "googleapis", version = "0.0.0-20250604-de157ca3", repo_name = "com_google_googleapis")
+bazel_dep(name = "googleapis-cc", version = "1.0.0")
+bazel_dep(name = "googleapis-grpc-cc", version = "1.0.0")
 
 switched_rules = use_extension("@com_google_googleapis//:extensions.bzl", "switched_rules")
 switched_rules.use_languages(
