diff --git a/tests/import/import_test.bzl b/tests/import/import_test.bzl
index c78437c..9b8e883 100644
--- a/tests/import/import_test.bzl
+++ b/tests/import/import_test.bzl
@@ -1,5 +1,5 @@
 load("@rules_sh//sh:import.bzl", "create_shim")
-load(":get_cpu_value.bzl", "get_cpu_value")
+load("@rules_sh//sh/private:get_cpu_value.bzl", "get_cpu_value")
 
 # create shim ########################################################
 
