diff --git examples/.bazelrc examples/.bazelrc
new file mode 100644
index 0000000..c39c658
--- /dev/null
+++ examples/.bazelrc
@@ -0,0 +1,2 @@
+import %workspace%/../tools/bazel.rc
+build --workspace_status_command=../tools/stamping/workspace_status.sh
diff --git examples/MODULE.bazel examples/MODULE.bazel
new file mode 100644
index 0000000..c2990d2
--- /dev/null
+++ examples/MODULE.bazel
@@ -0,0 +1,14 @@
+module(
+    name = "examples",
+    version = "0.0.0",
+)
+
+bazel_dep(
+    name = "rules_jsonnet",
+    version = "0.0.0",
+    repo_name = "io_bazel_rules_jsonnet",
+)
+local_path_override(
+    module_name = "rules_jsonnet",
+    path = "..",
+)
diff --git examples/invalid.out examples/invalid.out
index f449612..d8e6fb8 100644
--- examples/invalid.out
+++ examples/invalid.out
@@ -1,3 +1,3 @@
 ^RUNTIME ERROR: Foo\.
-	../examples/invalid\.jsonnet:15:1-13	($|\$
+	.*invalid\.jsonnet:15:1-13	($|\$
 	During evaluation	)
