diff --git a/include/boost/process/v2/windows/default_launcher.hpp b/include/boost/process/v2/windows/default_launcher.hpp
index d0ca19b..d8740e2 100644
--- a/include/boost/process/v2/windows/default_launcher.hpp
+++ b/include/boost/process/v2/windows/default_launcher.hpp
@@ -21,6 +21,7 @@
 #include <numeric>
 #include <memory>
 #include <type_traits>
+#include <vector>
 #include <windows.h>
 
 #if defined(BOOST_PROCESS_V2_STANDALONE)
