--- BUILD.bazel
+++ BUILD.bazel
@@ -74,6 +74,7 @@ cc_library(
     deps = [
         "@gz-utils//:ImplPtr",
         "@gz-utils//:NeverDestroyed",
+        "@gz-utils//:SuppressWarning",
     ],
 )
 
@@ -113,6 +114,7 @@ test_sources = glob(
             ":gz-math",
             "@googletest//:gtest",
             "@googletest//:gtest_main",
+            "@gz-utils//:SuppressWarning",
         ],
     )
     for src in test_sources
