--- BUILD.bazel
+++ BUILD.bazel
@@ -1,5 +1,5 @@
 load("@buildifier_prebuilt//:rules.bzl", "buildifier", "buildifier_test")
-load("@rules_gazebo//gazebo:headers.bzl", "gz_configure_header", "gz_export_header")
+load("@rules_gazebo//gazebo:headers.bzl", "gz_configure_header", "gz_export_header", "gz_include_header")
 load("@rules_license//rules:license.bzl", "license")

 package(
@@ -36,13 +36,22 @@ gz_configure_header(
     package_xml = "package.xml",
 )

-public_headers = glob(
+public_headers_no_gen = glob(
     include = [
         "include/gz/math/*.hh",
         "include/gz/math/detail/*.hh",
         "include/gz/math/graph/*.hh",
     ],
-) + [
+)
+
+gz_include_header(
+    name = "Include",
+    out = "include/gz/math.hh",
+    hdrs = public_headers_no_gen + ["include/gz/math/config.hh"],
+)
+
+public_headers = public_headers_no_gen + [
+    "include/gz/math.hh",
     "include/gz/math/Export.hh",
     "include/gz/math/config.hh",
 ]
