--- eigen3/BUILD.bazel
+++ eigen3/BUILD.bazel
@@ -1,3 +1,4 @@
+load("@rules_gazebo//gazebo:headers.bzl", "gz_include_header")
 load("@rules_license//rules:license.bzl", "license")

 package(
@@ -13,12 +14,22 @@ license(
     package_name = "gz-math-eigen3",
 )

-public_headers = glob(
+public_headers_no_gen = glob(
     include = [
         "include/gz/math/eigen3/*.hh",
     ],
 )

+gz_include_header(
+    name = "Include",
+    out = "include/gz/math/eigen3.hh",
+    hdrs = public_headers_no_gen,
+)
+
+public_headers = public_headers_no_gen + [
+    "include/gz/math/eigen3.hh",
+]
+
 cc_library(
     name = "eigen3",
     hdrs = public_headers,
