diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..d14c607
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,65 @@
+load("@bazel_skylib//lib:selects.bzl", "selects")
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "stacktrace",
+    srcs = selects.with_or({
+        "@platforms//os:android": [
+            "src/basic.cpp",
+            #"src/noop.cpp",
+        ],
+        "@boost.rules.tools//:linux_arm": [
+            "src/basic.cpp",
+            "src/noop.cpp",
+        ],
+        "@boost.rules.tools//:linux_aarch64": [
+            "src/basic.cpp",
+            # "src/noop.cpp",
+        ],
+        "@boost.rules.tools//:linux_ppc": [
+            "src/backtrace.cpp",
+        ],
+        "@boost.rules.tools//:linux_x86_64": [
+            "src/backtrace.cpp",
+        ],
+        ("@platforms//os:osx", "@platforms//os:ios", "@platforms//os:watchos", "@platforms//os:tvos"): [
+            "src/addr2line.cpp",
+        ],
+        "@boost.rules.tools//:windows_x86_64": [
+            "src/windbg.cpp",
+            "src/windbg_cached.cpp",
+        ],
+        "//conditions:default": [],
+    }),
+    defines = selects.with_or({
+        ("@platforms//os:osx", "@platforms//os:ios", "@platforms//os:watchos", "@platforms//os:tvos"): [
+            "BOOST_STACKTRACE_GNU_SOURCE_NOT_REQUIRED",
+        ],
+        "//conditions:default": [],
+    }),
+    exclude_src = ["src/*.cpp"],
+    linkopts = select({
+        "@boost.rules.tools//:linux_ppc": [
+            "-lbacktrace -ldl",
+        ],
+        "@boost.rules.tools//:linux_x86_64": [
+            "-lbacktrace -ldl",
+        ],
+        "@boost.rules.tools//:linux_aarch64": [
+            "-lbacktrace -ldl",
+        ],
+        "//conditions:default": [],
+    }),
+    deps = [
+        "@boost//:array",
+        "@boost//:config",
+        "@boost//:container_hash",
+        "@boost//:core",
+        "@boost//:predef",
+        "@boost//:static_assert",
+        "@boost//:type_traits",
+        "@boost//:winapi",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..abdc8d5
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,23 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.stacktrace",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "bazel_skylib", version = "1.5.0")
+
+
+
+
+
+
+
+
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..8bf79d5
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,12 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        negative_test_names = [],
+        deps = DEPS,
+    ),
+)
