diff --git a/MODULE.bazel b/MODULE.bazel
index d343487..f32d869 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -38,9 +38,7 @@ python = use_extension("@rules_python//python/extensions:python.bzl", "python")
 
 use_repo(python, python = "python_versions")
 
-#### DEV ONLY DEPENDENCIES BELOW HERE ####
-
-pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip", dev_dependency = True)
+pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
 
 [
     pip.parse(
