--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,49 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+package(default_visibility = ["//visibility:public"])
+
+constraint_setting(
+    name = "cpu",
+)
+
+constraint_value(
+    name = "armv8-a",
+    constraint_setting = ":cpu",
+)
+
+constraint_value(
+    name = "armv8-a-32",
+    constraint_setting = ":cpu",
+)
+
+constraint_value(
+    name = "armv7-a",
+    constraint_setting = ":cpu",
+)
+
+config_setting(
+    name = "is_armv8_a",
+    constraint_values = [":armv8-a"],
+)
+
+config_setting(
+    name = "is_armv8_a_32",
+    constraint_values = [":armv8-a-32"],
+)
+
+config_setting(
+    name = "is_armv7_a",
+    constraint_values = [":armv7-a"],
+)
+
+cc_library(
+    name = "sse2neon",
+    hdrs = ["sse2neon.h"],
+    copts = select({
+        ":is_armv8_a": ["-march=armv8-a+fp+simd+crypto+crc"],
+        ":is_armv8_a_32": ["-mfpu=neon-fp-armv8"],
+        ":is_armv7_a": ["-mfpu=neon"],
+        "//conditions:default": [],
+    }),
+    visibility = ["//visibility:public"],
+)
