===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,9 +1,9 @@
 """rules_tsickle provides bazel rules for transforming ts files to closure compatible js"""
 
 module(
     name = "rules_tsickle",
-    version = "0.0.0",
+    version = "1.1.0",
     compatibility_level = 1,
 )
 
 # -----------------------------------------------------------------
