--- a/MODULE.bazel	2025-11-23 09:29:26.079299974 +0000
+++ b/MODULE.bazel	2025-11-23 09:29:15.092315062 +0000
@@ -17,9 +17,9 @@

 module(
     name = "skywalking-data-collect-protocol",
-    version = "10.2.1-dev",
-    repo_name = "skywalking_data_collect_protocol",
+    compatibility_level = 1,
+    version = "10.3.0",
 )

 bazel_dep(name = "grpc", version = "1.74.1", repo_name = "com_github_grpc_grpc")
 bazel_dep(name = "protobuf", version = "32.0", repo_name = "com_google_protobuf")
