--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,27 @@
+""" Builds gperf.
+"""
+
+cc_library(
+    name = "gperf_c_lib",
+    srcs = glob([
+        "lib/*.c",
+        "lib/*.h",
+    ]),
+    includes = ["lib"],
+    visibility = ["//visibility:private"],
+)
+
+cc_binary(
+    name = "gperf",
+    srcs = glob([
+        "src/*.inc",
+        "src/*.cc",
+        "src/*.h",
+        "lib/*.cc",
+        "lib/*.h",
+    ]) + ["config.h"],
+    includes = ["lib", "src"],
+    copts = ["-std=c++11","-w"],
+    deps = ["gperf_c_lib"],
+    visibility = ["//visibility:public"],
+)
