diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 00000000..4911889c
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -0,0 +1,93 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(intel-xlink VERSION 1.0.0 LANGUAGES CXX;C)
+
+include(GNUInstallDirs)
+
+unset(ENV{MV_COMMON_BASE})
+
+include("XLink/XLink.cmake")
+
+add_definitions(-D__PC__)
+add_definitions(-DUSE_USB_VSC)
+add_definitions(-DCONFIG_XLINK_USB_VSC)
+add_definitions(-DXLINK_USE_BUS)
+
+include_directories(
+  ${XLINK_INCLUDE_DIRECTORIES}
+  ${XLINK_PLATFORM_INCLUDE}
+)
+
+ADD_LIBRARY(intel-xlink SHARED ${XLINK_SOURCES})
+target_link_libraries(intel-xlink PRIVATE ${LIBUSB_LIBRARY} Threads::Threads)
+
+target_include_directories(intel-xlink 
+                          PUBLIC
+                          "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+                          "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+                          )
+
+install(TARGETS intel-xlink
+        EXPORT intel-xlinkTargets
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/intel-xlink
+        )
+
+install(FILES 
+		    XLink/shared/include/XLinkPublicDefines.h
+        XLink/shared/include/XLinkVersion.h
+        XLink/shared/include/XLink.h
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/intel-xlink
+        )
+
+install(EXPORT intel-xlinkTargets
+        FILE intel-xlinkTargets.cmake
+        NAMESPACE intel-xlink::
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/intel-xlink
+        )
+
+include(CMakePackageConfigHelpers)
+
+configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in
+  "${CMAKE_CURRENT_BINARY_DIR}/intel-xlinkConfig.cmake"
+  INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/intel-xlink
+)
+
+set(version 1.0)
+
+set_property(TARGET intel-xlink PROPERTY VERSION ${version})
+set_property(TARGET intel-xlink PROPERTY SOVERSION 0)
+set_property(TARGET intel-xlink PROPERTY
+  INTERFACE_intel-xlink_MAJOR_VERSION 3)
+set_property(TARGET intel-xlink APPEND PROPERTY
+  COMPATIBLE_INTERFACE_STRING intel-xlink_MAJOR_VERSION
+)
+
+# generate the version file for the config file
+write_basic_package_version_file(
+  "${CMAKE_CURRENT_BINARY_DIR}/intel-xlinkConfigVersion.cmake"
+  VERSION "${version}"
+  COMPATIBILITY AnyNewerVersion
+)
+
+install(FILES
+          "${CMAKE_CURRENT_BINARY_DIR}/intel-xlinkConfig.cmake"
+          "${CMAKE_CURRENT_BINARY_DIR}/intel-xlinkConfigVersion.cmake"
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/intel-xlink
+)
+
+include(InstallRequiredSystemLibraries)
+
+set(CPACK_GENERATOR "DEB")
+set(CPACK_RPM_COMPONENT_INSTALL ON)
+set(CPACK_PACKAGE_INSTALL_DIRECTORY "CMake ${CMake_VERSION_MAJOR}.${CMake_VERSION_MINOR}")
+set(CPACK_DEBIAN_PACKAGE_MAINTAINER "Akasha Imaging")
+
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Intel XLink Wrapper")
+set(CPACK_PACKAGE_VENDOR "Intrinsic Innovation LLC")
+set(CPACK_PACKAGE_VERSION_MAJOR "1")
+set(CPACK_PACKAGE_VERSION_MINOR "0")
+set(CPACK_PACKAGE_VERSION_PATCH "0-alpha1")
+include(CPack)
