diff --git a/BUILD b/BUILD
new file mode 100644
index 00000000..9cf96eb7
--- a/BUILD
+++ b/BUILD
@@ -0,0 +1,65 @@
+# intel-xlink: XLink component of the Intel Myriad Development Kit (MDK) for Intel(R) Movidius(TM)
+# Myriad(TM) X Vision Processing Units (VPUs).
+# XLink implements a USB communication protocol between Intel VPUs and a PC.
+
+load("@bazel_skylib//rules:build_test.bzl", "build_test")
+load("@rules_license//rules:license.bzl", "license")
+
+package(
+    default_applicable_licenses = [":license"],
+    default_visibility = ["//visibility:public"],
+)
+
+license(
+    name = "license",
+    package_name = "intel_xlink",
+    license_kinds = [
+      "@rules_license//licenses/spdx:Apache-2.0",
+    ],
+    license_text = "LICENSE",
+    package_url = "https://github.com/openvinotoolkit/openvino",
+)
+
+cc_library(
+    name = "intel-xlink",
+    srcs = glob(
+        [
+            "**/*.c",
+        ],
+        exclude = [
+            "XLink/pc/Win/**/*.c",
+        ],
+    ),
+    hdrs = glob(
+        [
+            "**/*.h",
+        ],
+        exclude = [
+            "XLink/pc/Win/**/*.h",
+        ],
+    ),
+    copts = [
+        "-Ilibusb/libusb1",
+        "-D__PC__",
+        "-DUSE_USB_VSC",
+        "-DCONFIG_XLINK_USB_VSC",
+        "-DXLINK_USE_BUS",
+        "-Wno-enum-conversion",
+    ],
+    include_prefix = "intel-xlink",
+    includes = [
+        "XLink/pc/MacOS",
+        "XLink/pc/protocols",
+        "XLink/shared/include",
+    ],
+    linkopts = ["-lpthread"],
+    deps = [
+        "@libusb",
+    ],
+)
+
+build_test(
+    name = "intel_xlink_build_test",
+    targets = [":intel-xlink"],
+)
