diff --git a/folly/debugging/symbolizer/DwarfUtil.cpp b/folly/debugging/symbolizer/DwarfUtil.cpp
index da6c2f2..139f3b6 100644
--- a/folly/debugging/symbolizer/DwarfUtil.cpp
+++ b/folly/debugging/symbolizer/DwarfUtil.cpp
@@ -27,7 +27,7 @@
 
 #if FOLLY_HAVE_DWARF && FOLLY_HAVE_ELF
 
-#include <dwarf.h> // @manual
+#include <libdwarf/dwarf.h>
 
 // We need a single dwarf5 tag, but may not be building against
 // a new enough libdwarf, so just define it ourselves.
