diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..9acc490
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,20 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "assign",
+    deps = [
+        "@boost//:array",
+        "@boost//:config",
+        "@boost//:core",
+        "@boost//:move",
+        "@boost//:preprocessor",
+        "@boost//:ptr_container",
+        "@boost//:range",
+        "@boost//:static_assert",
+        "@boost//:throw_exception",
+        "@boost//:tuple",
+        "@boost//:type_traits",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..2890fe9
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.assign",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..726f81c
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:assign",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "assign_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/assign_test.cc b/bazel_test/assign_test.cc
new file mode 100644
index 0000000..67f2691
--- /dev/null
+++ b/bazel_test/assign_test.cc
@@ -0,0 +1,11 @@
+#include <boost/assign.hpp>
+#include <boost/assign/ptr_map_inserter.hpp>
+#include <boost/assign/ptr_list_inserter.hpp>
+#include <boost/assign/ptr_list_of.hpp>
+#include <list>
+ 
+int main()
+{
+    const std::list<int> two = boost::assign::list_of(1)(2);
+    return two.size() == 2 ? 0 : 1;
+}   
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..79a2e04
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,32 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:assign",
+    "@boost//:test",
+    "@boost//:test",
+    "@boost//:multi_index",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        names = [
+            "array",
+            "basic",
+            # "compile/decl_header", # TODO Does this need to work? Where does the declared file come from?
+            "email_example",
+            "list_inserter",
+            "list_of",
+            "list_of_workaround",
+            "multi_index_container",
+            "my_vector_example",
+            "ptr_list_inserter",
+            "ptr_list_of",
+            "ptr_map_inserter",
+            "static_list_of",
+            "std",
+            "tuple_list_of",
+        ],
+        deps = DEPS,
+    ),
+)
