diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..f6e910e
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,70 @@
+cc_library(
+    name = "xplane_sdk",
+    srcs = glob(["CHeaders/**/*.cpp"]) + select({
+        ":windows": ["windows_extras.cpp"],
+        "//conditions:default": [],
+    }),
+    hdrs = glob(["CHeaders/**/*.h"]),
+    additional_linker_inputs = select({
+        ":windows": [
+            "Libraries/Win/XPLM_64.lib",
+            "Libraries/Win/XPWidgets_64.lib",
+        ],
+        ":apple": [
+            "Libraries/Mac/XPLM.framework",
+            "Libraries/Mac/XPWidgets.framework",
+        ],
+        "//conditions:default": [],
+    }),
+    copts = select({
+        ":linux": ["-fvisibility=hidden"],
+        "//conditions:default": [],
+    }),
+    defines = select({
+        ":apple": ["APL=1"],
+        ":windows": ["IBM=1"],
+        "//conditions:default": ["LIN=1"],
+    }) + [
+        "XPLM200",  # API 2.0 Compatible
+        "XPLM210",  # API 2.1 Compatible
+        "XPLM300",  # API 3.0 Compatible
+    ],
+    includes = [
+        "CHeaders/Widgets",
+        "CHeaders/Wrappers",
+        "CHeaders/XPLM",
+    ],
+    linkopts = select({
+        ":linux": [
+            "-shared",
+            "-rdynamic",
+            "-nodefaultlibs",
+            "-undefined_warning",
+            "-Wl",
+            "--version-script=exports.txt",
+        ],
+        "//conditions:default": [],
+    }),
+    visibility = ["//visibility:public"],
+)
+
+config_setting(
+    name = "apple",
+    constraint_values = [
+        "@platforms//os:osx",
+    ],
+)
+
+config_setting(
+    name = "windows",
+    constraint_values = [
+        "@platforms//os:windows",
+    ],
+)
+
+config_setting(
+    name = "linux",
+    constraint_values = [
+        "@platforms//os:linux",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..ad9bde3
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,7 @@
+module(
+    name = "xplane_sdk",
+    version = "4.0.1.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/exports.txt b/exports.txt
new file mode 100644
index 0000000..2c48b97
--- /dev/null
+++ b/exports.txt
@@ -0,0 +1,12 @@
+{
+ global:
+  _init;
+  _fini;
+  XPluginStart;
+  XPluginStop;
+  XPluginEnable;
+  XPluginDisable;
+  XPluginReceiveMessage;
+ local:
+   *;
+};
\ No newline at end of file
diff --git a/windows_extras.cpp b/windows_extras.cpp
new file mode 100644
index 0000000..20e2b21
--- /dev/null
+++ b/windows_extras.cpp
@@ -0,0 +1,12 @@
+#include <windows.h>
+BOOL APIENTRY DllMain(HANDLE hModule, DWORD ul_reason_for_call,
+                      LPVOID lpReserved) {
+  switch (ul_reason_for_call) {
+  case DLL_PROCESS_ATTACH:
+  case DLL_THREAD_ATTACH:
+  case DLL_THREAD_DETACH:
+  case DLL_PROCESS_DETACH:
+    break;
+  }
+  return TRUE;
+}
