diff --git a/c++/WORKSPACE b/c++/WORKSPACE
deleted file mode 100644
index 0dea9c21..00000000
--- a/c++/WORKSPACE
+++ /dev/null
@@ -1,54 +0,0 @@
-workspace(name = "capnp-cpp")
-
-load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
-load("//:build/load_br.bzl", "load_brotli")
-
-http_archive(
-    name = "bazel_skylib",
-    sha256 = "f7be3474d42aae265405a592bb7da8e171919d74c16f082a5457840f06054728",
-    urls = [
-        "https://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/releases/download/1.2.1/bazel-skylib-1.2.1.tar.gz",
-        "https://github.com/bazelbuild/bazel-skylib/releases/download/1.2.1/bazel-skylib-1.2.1.tar.gz",
-    ],
-)
-
-load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
-
-bazel_skylib_workspace()
-
-http_archive(
-    name = "ssl",
-    sha256 = "873ec711658f65192e9c58554ce058d1cfa4e57e13ab5366ee16f76d1c757efc",
-    strip_prefix = "google-boringssl-ed2e74e",
-    type = "tgz",
-    # from master-with-bazel branch
-    urls = ["https://github.com/google/boringssl/tarball/ed2e74e737dc802ed9baad1af62c1514430a70d6"],
-)
-
-# Based on https://github.com/bazelbuild/bazel/blob/master/third_party/zlib/BUILD.
-_zlib_build = """
-cc_library(
-    name = "zlib",
-    srcs = glob(["*.c"]),
-    hdrs = glob(["*.h"]),
-    # Workaround for zlib warnings and mac compilation. Some issues were resolved in v1.3, but there are still implicit function declarations.
-    copts = [
-        "-w",
-        "-Dverbose=-1",
-    ] + select({
-        "@platforms//os:macos": [ "-Wno-implicit-function-declaration" ],
-        "//conditions:default": [],
-    }),
-    visibility = ["//visibility:public"],
-)
-"""
-
-http_archive(
-    name = "zlib",
-    build_file_content = _zlib_build,
-    sha256 = "ff0ba4c292013dbc27530b3a81e1f9a813cd39de01ca5e0f8bf355702efa593e",
-    strip_prefix = "zlib-1.3",
-    urls = ["https://zlib.net/fossils/zlib-1.3.tar.gz"],
-)
-
-load_brotli()
diff --git a/c++/src/capnp/BUILD.bazel b/c++/src/capnp/BUILD.bazel
index f11a7a24..c4ed647d 100644
--- a/c++/src/capnp/BUILD.bazel
+++ b/c++/src/capnp/BUILD.bazel
@@ -1,4 +1,4 @@
-load("@capnp-cpp//src/capnp:cc_capnp_library.bzl", "cc_capnp_library")
+load("@capnproto//src/capnp:cc_capnp_library.bzl", "cc_capnp_library")
 
 cc_library(
     name = "capnp",
diff --git a/c++/src/capnp/cc_capnp_library.bzl b/c++/src/capnp/cc_capnp_library.bzl
index 9e4acd35..a08bbf34 100644
--- a/c++/src/capnp/cc_capnp_library.bzl
+++ b/c++/src/capnp/cc_capnp_library.bzl
@@ -72,10 +72,10 @@ _capnp_gen = rule(
         "data": attr.label_list(allow_files = True),
         "outs": attr.output_list(),
         "src_prefix": attr.string(),
-        "_capnpc": attr.label(executable = True, allow_single_file = True, cfg = "exec", default = "@capnp-cpp//src/capnp:capnp_tool"),
-        "_capnpc_cxx": attr.label(executable = True, allow_single_file = True, cfg = "exec", default = "@capnp-cpp//src/capnp:capnpc-c++"),
-        "_capnpc_capnp": attr.label(executable = True, allow_single_file = True, cfg = "exec", default = "@capnp-cpp//src/capnp:capnpc-capnp"),
-        "_capnp_system": attr.label(default = "@capnp-cpp//src/capnp:capnp_system_library"),
+        "_capnpc": attr.label(executable = True, allow_single_file = True, cfg = "exec", default = "@capnproto//src/capnp:capnp_tool"),
+        "_capnpc_cxx": attr.label(executable = True, allow_single_file = True, cfg = "exec", default = "@capnproto//src/capnp:capnpc-c++"),
+        "_capnpc_capnp": attr.label(executable = True, allow_single_file = True, cfg = "exec", default = "@capnproto//src/capnp:capnpc-capnp"),
+        "_capnp_system": attr.label(default = "@capnproto//src/capnp:capnp_system_library"),
     },
     output_to_genfiles = True,
     implementation = _capnp_gen_impl,
@@ -121,7 +121,7 @@ def cc_capnp_library(
         name = name,
         srcs = srcs_cpp,
         hdrs = hdrs,
-        deps = deps + ["@capnp-cpp//src/capnp:capnp_runtime"],
+        deps = deps + ["@capnproto//src/capnp:capnp_runtime"],
         visibility = visibility,
         target_compatible_with = target_compatible_with,
         **kwargs
diff --git a/c++/src/capnp/compat/BUILD.bazel b/c++/src/capnp/compat/BUILD.bazel
index fcaecfa3..8dc7f5ee 100644
--- a/c++/src/capnp/compat/BUILD.bazel
+++ b/c++/src/capnp/compat/BUILD.bazel
@@ -1,4 +1,4 @@
-load("@capnp-cpp//src/capnp:cc_capnp_library.bzl", "cc_capnp_library")
+load("@capnproto//src/capnp:cc_capnp_library.bzl", "cc_capnp_library")
 
 exports_files([
     "json.capnp",
diff --git a/c++/src/kj/BUILD.bazel b/c++/src/kj/BUILD.bazel
index f5527bea..6fd960ff 100644
--- a/c++/src/kj/BUILD.bazel
+++ b/c++/src/kj/BUILD.bazel
@@ -1,4 +1,4 @@
-load("//:build/configure.bzl", "kj_configure")
+load("//build:configure.bzl", "kj_configure")
 
 kj_configure()
 
diff --git a/c++/src/kj/compat/BUILD.bazel b/c++/src/kj/compat/BUILD.bazel
index f9e31ce6..2350724a 100644
--- a/c++/src/kj/compat/BUILD.bazel
+++ b/c++/src/kj/compat/BUILD.bazel
@@ -18,7 +18,7 @@ cc_library(
     visibility = ["//visibility:public"],
     deps = [
         "//src/kj:kj-async",
-        "@ssl",
+        "@boringssl//:ssl",
     ],
 )
 
