diff --git BUILD.bazel BUILD.bazel
new file mode 100644
index 0000000..396de67
--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,72 @@
+""" Builds foonathan_memory.
+"""
+
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake", "cmake_variant")
+
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["**"]),
+)
+
+cache_entries = {
+    "CMAKE_POSITION_INDEPENDENT_CODE": "ON",  # Must be set!
+    "BUILD_SHARED_LIBS": "OFF",
+    # foonathan_memory specific options.
+    "FOONATHAN_MEMORY_BUILD_EXAMPLES": "OFF",
+    "FOONATHAN_MEMORY_BUILD_TESTS": "OFF",
+}
+
+config_setting(
+    name = "osx_arm64",
+    constraint_values = [
+        "@platforms//os:osx",
+        "@platforms//cpu:aarch64",
+    ],
+)
+
+alias(
+    name = "foonathan_memory",
+    actual = select({
+        ":osx_arm64": "foonathan_memory_osx_arm64",
+        "@platforms//os:windows": "foonathan_memory_windows",
+        "//conditions:default": "foonathan_memory_default",
+    }),
+)
+
+cmake(
+    name = "foonathan_memory_default",
+    build_args = [
+        "--",
+        "-j4",
+    ],
+    cache_entries = cache_entries,
+    lib_source = ":all_srcs",
+    lib_name = "libfoonathan_memory-0.7.3",
+    visibility = ["//visibility:public"],
+)
+
+cmake_variant(
+    name = "foonathan_memory_osx_arm64",
+    build_args = [
+        "--",
+        "-j4",
+    ],
+    cache_entries = cache_entries,
+    lib_source = ":all_srcs",
+    lib_name = "libfoonathan_memory-0.7.3",
+    visibility = ["//visibility:public"],
+    toolchain = "@rules_foreign_cc//toolchains:built_ninja_toolchain",
+)
+
+cmake_variant(
+    name = "foonathan_memory_windows",
+    build_args = [
+        "--",
+        "-j4",
+    ],
+    cache_entries = cache_entries,
+    lib_source = ":all_srcs",
+    lib_name = "libfoonathan_memory-0.7.3",
+    visibility = ["//visibility:public"],
+    toolchain = "@rules_foreign_cc//toolchains:preinstalled_nmake_toolchain",
+)
