diff --git BUILD.bazel BUILD.bazel
new file mode 100644
index 0000000..1e43776
--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,47 @@
+""" Builds foonathan_memory.
+"""
+
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["**"]),
+)
+
+cache_entries = {
+    "CMAKE_POSITION_INDEPENDENT_CODE": "ON",  # Must be set!
+    "BUILD_SHARED_LIBS": "OFF",
+    # foonathan_memory specific options.
+    "FOONATHAN_MEMORY_BUILD_EXAMPLES": "OFF",
+    "FOONATHAN_MEMORY_BUILD_TESTS": "OFF",
+}
+
+cmake(
+    name = "foonathan_memory",
+    build_args = [
+        "--",
+        "-j4",
+    ],
+    cache_entries = cache_entries,
+    generate_args = select({
+        "@platforms//os:osx": [],
+        "@platforms//os:linux": [],
+        "@platforms//os:windows": [
+            "-G \"NMake Makefiles\"",
+        ],
+    }),
+    lib_source = ":all_srcs",
+    out_lib_dir = select({
+        "@platforms//os:windows": "bin",
+        "//conditions:default": "lib",
+    }),
+    out_static_libs = select({
+        "@platforms//os:windows": [
+            "libfoonathan_memory-0.7.3.lib",
+        ],
+        "//conditions:default": [
+            "libfoonathan_memory-0.7.3.a",
+        ],
+    }),
+    visibility = ["//visibility:public"],
+)
