diff --git MODULE.bazel MODULE.bazel
new file mode 100644
index 0000000..2f73726
--- /dev/null
+++ MODULE.bazel
@@ -0,0 +1,15 @@
+module(
+    name = "foonathan_memory",
+    version = "0.7.3",
+    compatibility_level = 1,
+)
+
+bazel_dep(
+    name = "rules_foreign_cc",
+    version = "0.10.1",
+)
+
+register_toolchains(
+    "@rules_foreign_cc//toolchains:built_ninja_toolchain",
+    "@rules_foreign_cc//toolchains:preinstalled_nmake_toolchain",
+)
