--- MODULE.bazel
+++ MODULE.bazel
@@ -1,10 +1,11 @@
-## MODULE.bazel
 module(
     name = "gz-physics",
-    repo_name = "org_gazebosim_gz-physics",
+    version = "9.0.0-pre2",
+    compatibility_level = 8,
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "buildifier_prebuilt", version = "8.2.0.2")
 bazel_dep(name = "bullet", version = "3.26.0-rc0")
 bazel_dep(name = "dartsim", version = "6.13.2.bcr.1")
 bazel_dep(name = "eigen", version = "3.4.0.bcr.3")
@@ -14,38 +15,8 @@
 
 # Gazebo Dependencies
 bazel_dep(name = "rules_gazebo", version = "0.0.6")
-bazel_dep(name = "gz-common")
-bazel_dep(name = "gz-math")
-bazel_dep(name = "gz-plugin")
-bazel_dep(name = "gz-utils")
-bazel_dep(name = "sdformat")
-
-archive_override(
-    module_name = "gz-common",
-    strip_prefix = "gz-common-gz-common7",
-    urls = ["https://github.com/gazebosim/gz-common/archive/refs/heads/gz-common7.tar.gz"],
-)
-
-archive_override(
-    module_name = "gz-math",
-    strip_prefix = "gz-math-gz-math9",
-    urls = ["https://github.com/gazebosim/gz-math/archive/refs/heads/gz-math9.tar.gz"],
-)
-
-archive_override(
-    module_name = "gz-plugin",
-    strip_prefix = "gz-plugin-gz-plugin4",
-    urls = ["https://github.com/gazebosim/gz-plugin/archive/refs/heads/gz-plugin4.tar.gz"],
-)
-
-archive_override(
-    module_name = "gz-utils",
-    strip_prefix = "gz-utils-gz-utils4",
-    urls = ["https://github.com/gazebosim/gz-utils/archive/refs/heads/gz-utils4.tar.gz"],
-)
-
-archive_override(
-    module_name = "sdformat",
-    strip_prefix = "sdformat-sdf16",
-    urls = ["https://github.com/gazebosim/sdformat/archive/refs/heads/sdf16.tar.gz"],
-)
+bazel_dep(name = "gz-common", version = "7.0.0-pre1")
+bazel_dep(name = "gz-math", version = "9.0.0-pre1")
+bazel_dep(name = "gz-plugin", version = "4.0.0-pre1")
+bazel_dep(name = "gz-utils", version = "4.0.0-pre1")
+bazel_dep(name = "sdformat", version = "16.0.0-pre2")
