diff --git a/deps/fmt/include/fmt/format.h b/deps/fmt/include/fmt/format.h
--- a/deps/fmt/include/fmt/format.h
+++ b/deps/fmt/include/fmt/format.h
@@ -40,6 +40,7 @@
 #  include <cmath>    // std::signbit
 #  include <cstddef>  // std::byte
 #  include <cstdint>  // uint32_t
+#  include <cstdlib>  // malloc, free
 #  include <cstring>  // std::memcpy
 #  include <limits>   // std::numeric_limits
 #  include <new>      // std::bad_alloc

diff --git a/deps/spdlog/include/spdlog/fmt/bundled/format.h b/deps/spdlog/include/spdlog/fmt/bundled/format.h
--- a/deps/spdlog/include/spdlog/fmt/bundled/format.h
+++ b/deps/spdlog/include/spdlog/fmt/bundled/format.h
@@ -40,6 +40,7 @@
 #  include <cmath>    // std::signbit
 #  include <cstddef>  // std::byte
 #  include <cstdint>  // uint32_t
+#  include <cstdlib>  // malloc, free
 #  include <cstring>  // std::memcpy
 #  include <limits>   // std::numeric_limits
 #  include <new>      // std::bad_alloc

diff --git a/src/dstring.h b/src/dstring.h
--- a/src/dstring.h
+++ b/src/dstring.h
@@ -115,15 +115,17 @@
 
     /** For converting a JavaCC string */
+#ifndef DOXYGEN_JAVACC_CHAR_TYPE_IS_CHAR
 #ifndef DISABLE_JAVACC
     DString( const JavaCCString &s)
     {
       m_rep.resize(s.size());
       std::memcpy(m_rep.data(),s.data(),s.size());
     }
     DString &operator=( const JavaCCString &s)
     {
       m_rep.resize(s.size());
       std::memcpy(m_rep.data(),s.data(),s.size());
       return *this;
     }
 #endif
+#endif
