--- BUILD.bazel
+++ BUILD.bazel
@@ -50,7 +50,7 @@
         "src/arch/**/*.h",
         "src/*.h",
     ]),
-    hdrs = glob(["src/include/**/*.h"]),
+    hdrs = ["config-host.h"] + glob(["src/include/**/*.h"]),
 
     # cflags aligned with upstream:
     # https://github.com/axboe/liburing/blob/master/src/Makefile#L13
@@ -63,7 +63,6 @@
         "-Wall",
         "-Wextra",
         "-fno-stack-protector",
-        "-include config-host.h",
     ],
     includes = ["src/include"],
     visibility = ["//visibility:public"],
