diff --git a/MODULE.bazel b/MODULE.bazel
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,21 +1,22 @@
 module(
     name = "envoy_api",
-    version = "1.37.0-dev",
+    version = "0.0.0-20260901-005c18a",
 )

 # All bazel_dep entries are organized alphabetically for improved readability.

 # Core dependencies from Bazel Central Registry
 bazel_dep(name = "abseil-cpp", version = "20250814.1")
+bazel_dep(name = "aspect_bazel_lib", version = "2.22.5")
 bazel_dep(name = "bazel_skylib", version = "1.8.2")
 bazel_dep(name = "gazelle", version = "0.47.0", repo_name = "gazelle")
 bazel_dep(name = "googleapis", version = "0.0.0-20251003-2193a2bf")
 bazel_dep(name = "googleapis-cc", version = "1.0.0")
 bazel_dep(name = "googleapis-go", version = "1.0.0")
 bazel_dep(name = "googleapis-java", version = "1.0.0")
-bazel_dep(name = "grpc", version = "1.76.0.bcr.1", repo_name = "com_github_grpc_grpc")
+bazel_dep(name = "grpc", version = "1.76.0.bcr.1")
 bazel_dep(name = "opentelemetry-proto", version = "1.8.0")
-bazel_dep(name = "protobuf", version = "33.1", repo_name = "com_google_protobuf")
+bazel_dep(name = "protobuf", version = "33.1")
 bazel_dep(name = "protoc-gen-validate", version = "1.2.1.bcr.2")
 bazel_dep(name = "re2", version = "2024-07-02.bcr.1")
 bazel_dep(name = "rules_cc", version = "0.2.22")
