--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,7 +1,10 @@
-module(name = "com_github_mvukov_rules_ros2")
+module(
+    name = "com_github_mvukov_rules_ros2",
+    version = "0.0.0-20250612-f0d04fe",
+)
 
 bazel_dep(name = "asio", version = "1.32.0")
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
 bazel_dep(name = "boringssl", version = "0.20250415.0")
 bazel_dep(name = "curl", version = "8.8.0.bcr.3")
 bazel_dep(name = "eigen", version = "3.4.0.bcr.3")
