diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..3ef1895
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,29 @@
+cc_library(
+    name = "safeint",
+    hdrs = ["SafeInt.hpp"],
+    visibility = ["//visibility:public"],
+)
+
+cc_test(
+    name = "tests",
+    srcs = glob(
+        [
+            "Test/*.cpp",
+            "Test/*.h",
+        ],
+        exclude = [
+            "Test/CompileTest.cpp",
+        ],
+    ),
+    deps = [
+        ":safeint",
+    ],
+)
+
+cc_test(
+    name = "compile_test",
+    srcs = ["Test/CompileTest.cpp"],
+    deps = [
+        ":safeint",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..7285595
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,4 @@
+module(
+    name = "safeint",
+    version = "3.0.28a",
+)
