--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,29 @@
+package(
+    default_visibility = ["//visibility:public"],
+)
+
+exports_files([
+  "LICENSE",
+])
+
+
+cc_library(
+  name = "osqp++",
+  srcs = ["src/osqp++.cc"],
+  hdrs = [
+    "include/osqp++.h",
+    "@osqp//:include/ctrlc.h",
+    "@osqp//:include/osqp.h",
+  ],
+  strip_include_prefix = "include",
+  visibility = ["//visibility:public"],
+  deps = [
+    "@abseil-cpp//absl/base:core_headers",
+    "@abseil-cpp//absl/status",
+    "@abseil-cpp//absl/status:statusor",
+    "@abseil-cpp//absl/strings",
+    "@abseil-cpp//absl/strings:str_format",
+    "@eigen//:eigen",
+    "@osqp",
+  ]
+)
+