--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,41 @@
+load("@rules_license//rules:license.bzl", "license")
+
+package(
+    default_applicable_licenses = [":license"],
+    default_visibility = ["//visibility:public"],
+)
+
+exports_files([
+  "LICENSE",
+])
+
+license(
+    name = "license",
+    package_name = "osqp_cpp",
+    license_kinds = [
+      "@rules_license//licenses/spdx:Apache-2.0",
+    ],
+    license_text = "LICENSE",
+    package_url = "https://github.com/google/osqp-cpp",
+)
+
+cc_library(
+  name = "osqp++",
+  srcs = ["src/osqp++.cc"],
+  hdrs = [
+    "include/osqp++.h",
+    "@osqp//:include/ctrlc.h",
+    "@osqp//:include/osqp.h",
+  ],
+  strip_include_prefix = "include",
+  visibility = ["//visibility:public"],
+  deps = [
+    "@abseil-cpp//absl/base:core_headers",
+    "@abseil-cpp//absl/status",
+    "@abseil-cpp//absl/status:statusor",
+    "@abseil-cpp//absl/strings",
+    "@abseil-cpp//absl/strings:str_format",
+    "@eigen//:eigen",
+    "@osqp",
+  ]
+)
