--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,96 @@
+genrule(
+    name = "prebuilt_pnglibconf",
+    srcs = ["scripts/pnglibconf.h.prebuilt"],
+    outs = ["pnglibconf.h"],
+    cmd = "cp $< $@",
+)
+
+cc_library(
+    name = "libpng",
+    srcs = [
+        "png.c",
+        "pngerror.c",
+        "pngget.c",
+        "pngmem.c",
+        "pngpread.c",
+        "pngread.c",
+        "pngrio.c",
+        "pngrtran.c",
+        "pngrutil.c",
+        "pngset.c",
+        "pngtrans.c",
+        "pngwio.c",
+        "pngwrite.c",
+        "pngwtran.c",
+        "pngwutil.c",
+    ] + [
+        "pngconf.h",
+        "pngdebug.h",
+        "pnginfo.h",
+        "pngpriv.h",
+        "pngstruct.h",
+        ":prebuilt_pnglibconf",
+    ],
+    hdrs = ["png.h"],
+    visibility = ["//visibility:public"],
+    deps = ["@zlib"],
+)
+
+cc_test(
+    name = "pngtest_single_file",
+    size = "small",
+    srcs = ["pngtest.c"],
+    args = ["$(rootpath pngtest.png)"],
+    data = ["pngtest.png"],
+    # Bazel doesn't create the runfiles tree for Windows.
+    target_compatible_with = [
+        "@platforms//os:linux",
+        "@platforms//os:macos",
+    ],
+    deps = [":libpng"],
+)
+
+filegroup(
+    name = "contrib_passing_files",
+    srcs = glob(["contrib/testpngs/*.png"]),
+)
+
+cc_test(
+    name = "pngtest_contrib_passing",
+    size = "small",
+    srcs = ["pngtest.c"],
+    args = [
+        "-m",
+        "$(locations :contrib_passing_files)",
+    ],
+    data = [":contrib_passing_files"],
+    # Bazel doesn't create the runfiles tree for Windows.
+    target_compatible_with = [
+        "@platforms//os:linux",
+        "@platforms//os:macos",
+    ],
+    deps = [":libpng"],
+)
+
+filegroup(
+    name = "contrib_crashers_files",
+    srcs = glob(["contrib/testpngs/crashers/*.png"]),
+)
+
+cc_test(
+    name = "pngtest_contrib_crashers",
+    size = "small",
+    srcs = ["pngtest.c"],
+    args = [
+        "--xfail",
+        "-m",
+        "$(locations :contrib_crashers_files)",
+    ],
+    data = [":contrib_crashers_files"],
+    # Bazel doesn't create the runfiles tree for Windows.
+    target_compatible_with = [
+        "@platforms//os:linux",
+        "@platforms//os:macos",
+    ],
+    deps = [":libpng"],
+)
