--- extensions.bzl
+++ extensions.bzl
@@ -0,0 +1,25 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def _non_module_dependencies_impl(_ctx):
+    http_archive(
+        name = "rules_ruby",
+        urls = [
+            "https://github.com/protocolbuffers/rules_ruby/archive/b7f3e9756f3c45527be27bc38840d5a1ba690436.zip",
+        ],
+        sha256 = "347927fd8de6132099fcdc58e8f7eab7bde4eb2fd424546b9cd4f1c6f8f8bad8",
+        strip_prefix = "rules_ruby-b7f3e9756f3c45527be27bc38840d5a1ba690436",
+    )
+
+    http_archive(
+        name = "upb",
+        urls = [
+	     "https://github.com/protocolbuffers/upb/archive/636b8f0c9be4091ec3c2d1b4b46536d62d922544.zip",
+        ],
+        sha256 = "00d658bb274a3f3f9fac7d385208813d011e588fd2cb5748b64df1f5419cd43f",
+        strip_prefix = "upb-636b8f0c9be4091ec3c2d1b4b46536d62d922544",
+        patches = ["@com_google_protobuf//build_defs:upb.patch"],
+    )
+
+non_module_dependencies = module_extension(
+    implementation = _non_module_dependencies_impl,
+)
