diff --git a/bant/util/grep-highlighter.cc b/bant/util/grep-highlighter.cc
index 1b29503..4d70e3b 100644
--- a/bant/util/grep-highlighter.cc
+++ b/bant/util/grep-highlighter.cc
@@ -32,7 +32,7 @@
 #include "re2/re2.h"
 
 namespace bant {
-static constexpr std::initializer_list<std::string_view> kDefaultColors = {
+inline constexpr std::array<std::string_view, 5> kDefaultColors = {
   "\033[7m",   // Invers
   "\033[41m",  // red background
   // avoid green as that is a typical terminal color
@@ -43,7 +43,7 @@ static constexpr std::initializer_list<std::string_view> kDefaultColors = {
 
 GrepHighlighter::GrepHighlighter(bool do_highlight, bool and_semantics)
     : do_highlight_(do_highlight), and_semantics_(and_semantics) {
-  SetHighlightStart(kDefaultColors);
+  SetHighlightStart({kDefaultColors.begin(), kDefaultColors.end()});
   SetHighlightEnd("\033[0m");
 }
 
