--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,26 @@
+licenses(["notice"])  # Apache 2
+
+cc_library(
+    name = "protobuf-mutator",
+    srcs = glob(
+        [
+            "src/**/*.cc",
+            "src/**/*.h",
+        ],
+        exclude = ["**/*_test.cc"],
+    ) + ["port/protobuf.h"],
+    hdrs = ["src/libfuzzer/libfuzzer_macro.h"],
+    include_prefix = "libprotobuf_mutator",
+    includes = [
+        "port",
+        "src",
+    ],
+    visibility = ["//visibility:public"],
+    deps = ["@com_google_protobuf//:protobuf"],
+)
+
+alias(
+    name = "libprotobuf-mutator",
+    actual = ":protobuf-mutator",
+    visibility = ["//visibility:public"],
+)
