diff --git a/aom_dsp/reduce_sum_hwy.h b/aom_dsp/reduce_sum_hwy.h
index 9f4c005..4d27298 100644
--- a/aom_dsp/reduce_sum_hwy.h
+++ b/aom_dsp/reduce_sum_hwy.h
@@ -12,7 +12,7 @@
 #define AOM_AOM_DSP_REDUCE_SUM_HWY_H_
 
 #include <type_traits>
-#include "third_party/highway/hwy/highway.h"
+#include "hwy/highway.h"
 
 HWY_BEFORE_NAMESPACE();
 
diff --git a/aom_dsp/sad_hwy.h b/aom_dsp/sad_hwy.h
index 7d35ce5..c5d2bfd 100644
--- a/aom_dsp/sad_hwy.h
+++ b/aom_dsp/sad_hwy.h
@@ -12,7 +12,7 @@
 #define AOM_AOM_DSP_SAD_HWY_H_
 
 #include "aom_dsp/reduce_sum_hwy.h"
-#include "third_party/highway/hwy/highway.h"
+#include "hwy/highway.h"
 
 HWY_BEFORE_NAMESPACE();
 
diff --git a/av1/common/selfguided_hwy.h b/av1/common/selfguided_hwy.h
index 33433e4..2e38caa 100644
--- a/av1/common/selfguided_hwy.h
+++ b/av1/common/selfguided_hwy.h
@@ -15,7 +15,7 @@
 #include "av1/common/restoration.h"
 #include "config/aom_config.h"
 #include "config/av1_rtcd.h"
-#include "third_party/highway/hwy/highway.h"
+#include "hwy/highway.h"
 
 HWY_BEFORE_NAMESPACE();
 
diff --git a/av1/encoder/av1_fwd_txfm2d_hwy.h b/av1/encoder/av1_fwd_txfm2d_hwy.h
index 65a3668..412db53 100644
--- a/av1/encoder/av1_fwd_txfm2d_hwy.h
+++ b/av1/encoder/av1_fwd_txfm2d_hwy.h
@@ -16,7 +16,7 @@
 
 #include "config/aom_config.h"
 #include "config/av1_rtcd.h"
-#include "third_party/highway/hwy/highway.h"
+#include "hwy/highway.h"
 #include "aom_dsp/txfm_common.h"
 #include "av1/common/av1_txfm.h"
 #include "av1/common/enums.h"
