diff --git a/common/webmdec.cc b/common/webmdec.cc
index 099e4a4..98177ff 100644
--- a/common/webmdec.cc
+++ b/common/webmdec.cc
@@ -15,8 +15,8 @@
 #include <cstring>
 #include <cstdio>
 
-#include "third_party/libwebm/mkvparser/mkvparser.h"
-#include "third_party/libwebm/mkvparser/mkvreader.h"
+#include "mkvparser/mkvparser.h"
+#include "mkvparser/mkvreader.h"
 
 namespace {
 
diff --git a/common/webmenc.cc b/common/webmenc.cc
index 0b0ff65..380ef45 100644
--- a/common/webmenc.cc
+++ b/common/webmenc.cc
@@ -19,9 +19,9 @@
 #include <string>
 
 #include "common/av1_config.h"
-#include "third_party/libwebm/mkvmuxer/mkvmuxer.h"
-#include "third_party/libwebm/mkvmuxer/mkvmuxerutil.h"
-#include "third_party/libwebm/mkvmuxer/mkvwriter.h"
+#include "mkvmuxer/mkvmuxer.h"
+#include "mkvmuxer/mkvmuxerutil.h"
+#include "mkvmuxer/mkvwriter.h"
 
 namespace {
 const uint64_t kDebugTrackUid = 0xDEADBEEF;
