diff --git a/bazel/example/BUILD.bazel b/bazel/example/BUILD.bazel
index 05ab0f3..505911e 100644
--- a/bazel/example/BUILD.bazel
+++ b/bazel/example/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "main",
     size = "small",
diff --git a/bazel/glog.bzl b/bazel/glog.bzl
index a311d0d..15d1666 100644
--- a/bazel/glog.bzl
+++ b/bazel/glog.bzl
@@ -9,6 +9,8 @@
 #
 # Known issue: the namespace parameter is not supported on Win32.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 def expand_template_impl(ctx):
     ctx.actions.expand_template(
         template = ctx.file.template,
@@ -165,7 +167,7 @@ def glog_library(with_gflags = 1, **kwargs):
         ]
     )
 
-    native.cc_library(
+    cc_library(
         name = "glog",
         visibility = ["//visibility:public"],
         srcs = [
@@ -266,7 +268,7 @@ def glog_library(with_gflags = 1, **kwargs):
 
     # Workaround https://github.com/bazelbuild/bazel/issues/6337 by declaring
     # the dependencies without strip_include_prefix.
-    native.cc_library(
+    cc_library(
         name = "strip_include_prefix_hack",
         hdrs = [
             "src/glog/flags.h",
