diff --git a/third_party/address_sorting/BUILD b/third_party/address_sorting/BUILD
index 2f4f6b7fd..b9b025f6e 100644
--- a/third_party/address_sorting/BUILD
+++ b/third_party/address_sorting/BUILD
@@ -28,6 +28,7 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load(":address_sorting.bzl", "address_sorting_cc_library")
 
 package(
diff --git a/third_party/address_sorting/address_sorting.bzl b/third_party/address_sorting/address_sorting.bzl
index 05b4c6d84..93717d0eb 100644
--- a/third_party/address_sorting/address_sorting.bzl
+++ b/third_party/address_sorting/address_sorting.bzl
@@ -28,8 +28,9 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 def address_sorting_cc_library(name, srcs, hdrs, copts, includes, linkopts=[]):
-    native.cc_library(
+    cc_library(
         name = name,
         srcs = srcs,
         hdrs = hdrs,
