diff --git a/tests/json_patch_tests.c b/tests/json_patch_tests.c
index 7c3d6ae..731da6e 100644
--- a/tests/json_patch_tests.c
+++ b/tests/json_patch_tests.c
@@ -31,0 +32,4 @@
+#ifndef TEST_DIR_PATH
+#define TEST_DIR_PATH "json-patch-tests/"
+#endif
+
@@ -185 +189 @@ static void cjson_utils_should_pass_json_patch_test_tests(void)
-    cJSON *tests = parse_test_file("json-patch-tests/tests.json");
+    cJSON *tests = parse_test_file(TEST_DIR_PATH "tests.json");
@@ -202 +206 @@ static void cjson_utils_should_pass_json_patch_test_spec_tests(void)
-    cJSON *tests = parse_test_file("json-patch-tests/spec_tests.json");
+    cJSON *tests = parse_test_file(TEST_DIR_PATH "spec_tests.json");
@@ -219 +223 @@ static void cjson_utils_should_pass_json_patch_test_cjson_utils_tests(void)
-    cJSON *tests = parse_test_file("json-patch-tests/cjson-utils-tests.json");
+    cJSON *tests = parse_test_file(TEST_DIR_PATH "cjson-utils-tests.json");
diff --git a/tests/parse_examples.c b/tests/parse_examples.c
index d35d6cf..1e12e89 100644
--- a/tests/parse_examples.c
+++ b/tests/parse_examples.c
@@ -60,0 +61 @@ static void do_test(const char *test_name)
+#ifndef TEST_DIR_PATH
@@ -61,0 +63 @@ static void do_test(const char *test_name)
+#endif
@@ -139 +141,6 @@ static void file_test6_should_not_be_parsed(void)
-    test6 = read_file("inputs/test6");
+    char *test_path = NULL;
+    const char * test_name = "test6";
+    test_path = (char*)malloc(sizeof(TEST_DIR_PATH) + strlen(test_name));
+    TEST_ASSERT_NOT_NULL_MESSAGE(test_path, "Failed to allocate test_path buffer.");
+    sprintf(test_path, TEST_DIR_PATH"%s", test_name);
+    test6 = read_file(test_path);
