diff --git a/MODULE.bazel b/MODULE.bazel
index fc3751d..3813d18 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,5 +1,5 @@
 module(
-    name = "com_github_rabbitmq_osiris",
+    name = "rabbitmq_osiris",
     version = "1.6.3",
 )
 
@@ -69,7 +69,7 @@ erlang_dev_package = use_extension(
 erlang_package.hex_package(
     name = "eunit_formatters",
     version = "0.5.0",
-    build_file = "@com_github_rabbitmq_osiris//bzl:BUILD.eunit_formatters",
+    build_file = "@rabbitmq_osiris//bzl:BUILD.eunit_formatters",
 )
 
 use_repo(
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
index 7a70a32..8d3b94b 100644
--- a/WORKSPACE.bazel
+++ b/WORKSPACE.bazel
@@ -1,4 +1,4 @@
-workspace(name = "com_github_rabbitmq_osiris")
+workspace(name = "rabbitmq_osiris")
 
 load(
     "@bazel_tools//tools/build_defs/repo:http.bzl",
