From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander Khabarov <alexander.khabarov@arm.com>
Date: Thu, 19 Oct 2023 15:05:58 +0000
Subject: [PATCH] Add `BUILD.bazel`

---
 BUILD.bazel | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 BUILD.bazel

diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 000000000..05c507654
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,4 @@
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
