diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..35cb791
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,23 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "url",
+    srcs = glob([
+        "src/detail/**/*.cpp",
+        "src/grammar/**/*.cpp",
+        "src/rfc/**/*.cpp",
+    ]),
+    deps = [
+        "@boost.align//:align",
+        "@boost.assert//:assert",
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.mp11//:mp11",
+        "@boost.optional//:optional",
+        "@boost.static_assert//:static_assert",
+        "@boost.system//:system",
+        "@boost.throw_exception//:throw_exception",
+        "@boost.type_traits//:type_traits",
+        "@boost.variant2//:variant2",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..cd85537
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,19 @@
+module(
+    name = "boost.url",
+    version = "1.83.0.bzl.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.align", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.mp11", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.optional", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.system", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.throw_exception", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.variant2", version = "1.83.0.bzl.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..d20a22b
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost.url//:url",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "url_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/url_test.cc b/bazel_test/url_test.cc
new file mode 100644
index 0000000..0b601f9
--- /dev/null
+++ b/bazel_test/url_test.cc
@@ -0,0 +1,9 @@
+#include <boost/url.hpp>
+#include <cassert>
+
+int main()
+{
+    boost::urls::url_view u2 = boost::urls::parse_uri( "http://www.example.com/#" ).value();
+    assert(u2.fragment().empty());
+    assert(u2.has_fragment());
+}
