diff --git a/main.cpp b/main.cpp
index 091aae7..6d9f65b 100644
--- a/main.cpp
+++ b/main.cpp
@@ -18,7 +18,7 @@ int main(int argc, char *argv[])
   Params params;
 
   // Model files
-  string input_model;
+  std::string input_model;
   params.seed = (unsigned)time(NULL);
 
   // args
@@ -117,7 +117,7 @@ int main(int argc, char *argv[])
     }
   }
 
-  string ext;
+  std::string ext;
   if (params.input_model.length() > 4)
   {
     ext = params.input_model.substr(params.input_model.length() - 4);
@@ -207,8 +207,8 @@ int main(int argc, char *argv[])
 
   RecoverParts(parts, bbox, rot, params);
 
-  string objName = regex_replace(params.output_name, regex("wrl"), "obj");
-  string wrlName = regex_replace(params.output_name, regex("obj"), "wrl");
+  std::string objName = regex_replace(params.output_name, regex("wrl"), "obj");
+  std::string wrlName = regex_replace(params.output_name, regex("obj"), "wrl");
 
   SaveVRML(wrlName, parts, params);
   SaveOBJ(objName, parts, params);
diff --git a/src/io.cpp b/src/io.cpp
index 7267714..b64e121 100644
--- a/src/io.cpp
+++ b/src/io.cpp
@@ -29,7 +29,7 @@ namespace coacd
         logger::info("\tRandom Seed:                               {}", params.seed);
     }
 
-    void SaveMesh(const string &filename, Model &mesh)
+    void SaveMesh(const std::string &filename, Model &mesh)
     {
         std::ofstream os(filename);
         for (int i = 0; i < (int)mesh.points.size(); ++i)
@@ -45,7 +45,7 @@ namespace coacd
         os.close();
     }
 
-    void SaveOBJ(const string &filename, vector<Model> parts, Params &params)
+    void SaveOBJ(const std::string &filename, vector<Model> parts, Params &params)
     {
         vector<int> v_numbers;
         v_numbers.push_back(0);
@@ -68,7 +68,7 @@ namespace coacd
         os.close();
     }
 
-    void SaveOBJs(const string &foldername, const string &filename, vector<Model> parts, Params &params)
+    void SaveOBJs(const std::string &foldername, const std::string &filename, vector<Model> parts, Params &params)
     {
         int n_zero = 3;
         for (int n = 0; n < (int)parts.size(); n++)
@@ -168,7 +168,7 @@ namespace coacd
         }
     }
 
-    void SaveVRML(const string &fileName, vector<Model>& meshes, Params &params)
+    void SaveVRML(const std::string &fileName, vector<Model>& meshes, Params &params)
     {
         ofstream foutCH(fileName);
         if (foutCH.is_open())
diff --git a/src/io.h b/src/io.h
index f3f0302..8a10206 100644
--- a/src/io.h
+++ b/src/io.h
@@ -29,10 +29,10 @@ namespace coacd
 {
 
     //////////////// IO ////////////////
-    void SaveMesh(const string &filename, Model &mesh);
+    void SaveMesh(const std::string &filename, Model &mesh);
     void SaveConfig(Params params);
-    void SaveOBJ(const string &filename, vector<Model> parts, Params &params);
-    void SaveOBJs(const string &foldername, const string &filename, vector<Model> parts, Params &params);
+    void SaveOBJ(const std::string &filename, vector<Model> parts, Params &params);
+    void SaveOBJs(const std::string &foldername, const std::string &filename, vector<Model> parts, Params &params);
     bool WriteVRML(ofstream &fout, Model mesh);
-    void SaveVRML(const string &fileName, vector<Model>& meshes, Params &params);
+    void SaveVRML(const std::string &fileName, vector<Model>& meshes, Params &params);
 }
diff --git a/src/logger.h b/src/logger.h
index 62947c4..6163389 100644
--- a/src/logger.h
+++ b/src/logger.h
@@ -26,7 +26,7 @@ namespace coacd
         inline void debug(std::string_view fmt, const Args &...args)
         {
             #ifndef DISABLE_SPDLOG
-            get()->debug(fmt, args...);
+            get()->debug(fmt::runtime(fmt), args...);
             #else
             doPrint(std::cout, fmt, args...);
             #endif
@@ -36,7 +36,7 @@ namespace coacd
         inline void info(std::string_view fmt, const Args &...args)
         {
             #ifndef DISABLE_SPDLOG
-            get()->info(fmt, args...);
+            get()->info(fmt::runtime(fmt), args...);
             #else
             doPrint(std::cout, fmt, args...);
             #endif
@@ -46,7 +46,7 @@ namespace coacd
         inline void warn(std::string_view fmt, const Args &...args)
         {
             #ifndef DISABLE_SPDLOG
-            get()->warn(fmt, args...);
+            get()->warn(fmt::runtime(fmt), args...);
             #else
             doPrint(std::cout, fmt, args...);
             #endif
@@ -56,7 +56,7 @@ namespace coacd
         inline void error(std::string_view fmt, const Args &...args)
         {
             #ifndef DISABLE_SPDLOG
-            get()->error(fmt, args...);
+            get()->error(fmt::runtime(fmt), args...);
             #else
             doPrint(std::cout, fmt, args...);
             #endif
@@ -66,7 +66,7 @@ namespace coacd
         inline void critical(std::string_view fmt, const Args &...args)
         {
             #ifndef DISABLE_SPDLOG
-            get()->critical(fmt, args...);
+            get()->critical(fmt::runtime(fmt), args...);
             #else
             doPrint(std::cout, fmt, args...);
             #endif
diff --git a/src/process.cpp b/src/process.cpp
index 4d18a48..824f3c0 100644
--- a/src/process.cpp
+++ b/src/process.cpp
@@ -105,7 +105,7 @@ namespace coacd
         return sqrt(pow(pt[0] - p[0], 2) + pow(pt[1] - p[1], 2) + pow(pt[2] - p[2], 2));
     }
 
-    double compute_edge_cost(Model &ch, string apx_mode, int tri_i, int tri_j, vector<int> &rm_pt_idxs)
+    double compute_edge_cost(Model &ch, std::string apx_mode, int tri_i, int tri_j, vector<int> &rm_pt_idxs)
     {
         // Compute the edge length
         double cost = pts_norm(ch.points[tri_i], ch.points[tri_j]);
@@ -138,7 +138,7 @@ namespace coacd
         return cost;
     }
 
-    void DecimateCH(Model &ch, int tgt_pts, string apx_mode)
+    void DecimateCH(Model &ch, int tgt_pts, std::string apx_mode)
     {
         if (tgt_pts >= (int)ch.points.size())
             return;
diff --git a/src/process.h b/src/process.h
index 63e2905..a041da9 100644
--- a/src/process.h
+++ b/src/process.h
@@ -25,7 +25,7 @@ namespace coacd
 {
   extern thread_local std::mt19937 random_engine;
 
-  void DecimateCH(Model &ch, int tgt_pts, string apx_mode);
+  void DecimateCH(Model &ch, int tgt_pts, std::string apx_mode);
   void DecimateConvexHulls(vector<Model> &cvxs, Params &params);
   void MergeCH(Model &ch1, Model &ch2, Model &ch, Params &params);
   double MergeConvexHulls(Model &m, vector<Model> &meshs, vector<Model> &cvxs, Params &params, double epsilon = 0.02, double threshold = 0.01);