--- MODULE.bazel
+++ MODULE.bazel
@@ -2,7 +2,7 @@
 """
 module(
     name = "highs",
-    version = "1.13.1",
+    version = "1.14.0",
 )
 
 bazel_dep(
@@ -25,14 +25,7 @@
     version = "0.3.0",
 )
 
-# Load the extension
-cuda = use_extension("@rules_cuda//cuda:extensions.bzl", "toolchain")
-cuda.toolkit(
-    name = "cuda",
-)
-
-# Use 'cuda' as the repository name
-# use_repo(cuda, "cuda")
-
-# Alias local_cuda
-use_repo(cuda, local_cuda = "cuda")
+# cuda toolchain
+local_cuda = use_extension("@rules_cuda//cuda:extensions.bzl", "toolchain")
+local_cuda.local_toolchain()
+use_repo(local_cuda, "local_cuda")
