--- BUILD.bazel
+++ BUILD.bazel
@@ -58,7 +58,6 @@ sources = glob(
     ],
     exclude = [
         "src/*_TEST.cc",
-        "src/InstallationDirectories.cc",
     ],
 )

@@ -72,6 +71,7 @@ cc_library(
     name = "gz-physics",
     srcs = sources,
     hdrs = public_headers,
+    defines = ["GZ_PHYSICS_ENGINE_RELATIVE_INSTALL_DIR='\"\"'"],
     includes = ["include"],
     visibility = ["//visibility:public"],
     deps = [