diff --git a/generator/dawn_gpu_info_generator.py b/generator/dawn_gpu_info_generator.py
index 820f706fe6..b9148d5eba 100644
--- a/generator/dawn_gpu_info_generator.py
+++ b/generator/dawn_gpu_info_generator.py
@@ -29,7 +29,7 @@
 import json, os, sys
 from collections import namedtuple
 
-from generator_lib import Generator, run_generator, FileRender, GeneratorOutput
+from generator.generator_lib import Generator, run_generator, FileRender, GeneratorOutput
 
 
 def parse_mask(mask):
diff --git a/generator/dawn_json_generator.py b/generator/dawn_json_generator.py
index bc095be482..abcec0bc34 100644
--- a/generator/dawn_json_generator.py
+++ b/generator/dawn_json_generator.py
@@ -30,8 +30,8 @@ import json, os, sys
 from collections import namedtuple, defaultdict
 from copy import deepcopy
 
-from generator_lib import Generator, run_generator, FileRender, GeneratorOutput
-from webgpu_docs_utility import build_doc_map, load_json_data
+from generator.generator_lib import Generator, run_generator, FileRender, GeneratorOutput
+from generator.webgpu_docs_utility import build_doc_map, load_json_data
 
 ############################################################
 # OBJECT MODEL
diff --git a/generator/dawn_version_generator.py b/generator/dawn_version_generator.py
index 4117747387..0e64b52238 100644
--- a/generator/dawn_version_generator.py
+++ b/generator/dawn_version_generator.py
@@ -28,7 +28,7 @@
 
 import os, subprocess, sys, shutil
 
-from generator_lib import Generator, run_generator, FileRender, GeneratorOutput
+from generator.generator_lib import Generator, run_generator, FileRender, GeneratorOutput
 
 
 def get_git():
diff --git a/generator/generator_lib.py b/generator/generator_lib.py
index 45d90ab5a..d30a1eb3d 100644
--- a/generator/generator_lib.py
+++ b/generator/generator_lib.py
@@ -339,0 +339,5 @@ def run_generator(generator):
+    parser.add_argument(
+        '--output-filter',
+        default=[],
+        action='append',
+        help='Only write a generated output with this repository-relative path.')
@@ -344,0 +349,9 @@ def run_generator(generator):
+    if args.output_filter:
+        output_filter = set(args.output_filter)
+        output = GeneratorOutput(
+            renders=[
+                render for render in output.renders
+                if render.output in output_filter
+            ],
+            imported_templates=output.imported_templates)
+
