diff --git a/src/tint/api/BUILD.bazel b/src/tint/api/BUILD.bazel
--- a/src/tint/api/BUILD.bazel
+++ b/src/tint/api/BUILD.bazel
@@ -115,6 +115,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
+  include_prefix = "src/tint/api",
   visibility = ["//visibility:public"],
 )
 cc_library(
diff --git a/src/tint/api/common/BUILD.bazel b/src/tint/api/common/BUILD.bazel
--- a/src/tint/api/common/BUILD.bazel
+++ b/src/tint/api/common/BUILD.bazel
@@ -63,6 +63,7 @@
     "//src/utils",
   ],
   copts = COPTS,
+  include_prefix = "src/tint/api/common",
   visibility = ["//visibility:public"],
 )
 cc_library(
diff --git a/src/tint/lang/core/type/BUILD.bazel b/src/tint/lang/core/type/BUILD.bazel
--- a/src/tint/lang/core/type/BUILD.bazel
+++ b/src/tint/lang/core/type/BUILD.bazel
@@ -159,6 +159,7 @@
     "//src/utils",
   ],
   copts = COPTS,
+  include_prefix = "src/tint/lang/core/type",
   visibility = ["//visibility:public"],
 )
 cc_library(
diff --git a/src/tint/lang/msl/writer/BUILD.bazel b/src/tint/lang/msl/writer/BUILD.bazel
--- a/src/tint/lang/msl/writer/BUILD.bazel
+++ b/src/tint/lang/msl/writer/BUILD.bazel
@@ -72,6 +72,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
+  include_prefix = "src/tint/lang/msl/writer",
   visibility = ["//visibility:public"],
 )
 cc_library(
diff --git a/src/tint/lang/wgsl/BUILD.bazel b/src/tint/lang/wgsl/BUILD.bazel
--- a/src/tint/lang/wgsl/BUILD.bazel
+++ b/src/tint/lang/wgsl/BUILD.bazel
@@ -63,6 +63,7 @@
     "//src/utils",
   ],
   copts = COPTS,
+  include_prefix = "src/tint/lang/wgsl",
   visibility = ["//visibility:public"],
 )
 cc_library(
diff --git a/src/tint/lang/wgsl/inspector/BUILD.bazel b/src/tint/lang/wgsl/inspector/BUILD.bazel
--- a/src/tint/lang/wgsl/inspector/BUILD.bazel
+++ b/src/tint/lang/wgsl/inspector/BUILD.bazel
@@ -73,6 +73,7 @@
     "//src/utils",
   ],
   copts = COPTS,
+  include_prefix = "src/tint/lang/wgsl/inspector",
   visibility = ["//visibility:public"],
 )
 cc_library(
diff --git a/src/tint/lang/wgsl/reader/BUILD.bazel b/src/tint/lang/wgsl/reader/BUILD.bazel
--- a/src/tint/lang/wgsl/reader/BUILD.bazel
+++ b/src/tint/lang/wgsl/reader/BUILD.bazel
@@ -77,6 +77,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
+  include_prefix = "src/tint/lang/wgsl/reader",
   visibility = ["//visibility:public"],
 )
 cc_library(
diff --git a/src/tint/utils/diagnostic/BUILD.bazel b/src/tint/utils/diagnostic/BUILD.bazel
--- a/src/tint/utils/diagnostic/BUILD.bazel
+++ b/src/tint/utils/diagnostic/BUILD.bazel
@@ -60,6 +60,7 @@
     "//src/utils",
   ],
   copts = COPTS,
+  include_prefix = "src/tint/utils/diagnostic",
   visibility = ["//visibility:public"],
 )
 cc_library(
diff --git a/src/tint/utils/text/BUILD.bazel b/src/tint/utils/text/BUILD.bazel
--- a/src/tint/utils/text/BUILD.bazel
+++ b/src/tint/utils/text/BUILD.bazel
@@ -86,6 +86,7 @@
     "//src/utils",
   ],
   copts = COPTS,
+  include_prefix = "src/tint/utils/text",
   visibility = ["//visibility:public"],
 )
 cc_library(
