diff --git a/src/tint/flags.bzl b/src/tint/flags.bzl
--- a/src/tint/flags.bzl
+++ b/src/tint/flags.bzl
@@ -15,14 +15,14 @@
         default: boolean, if the flag should default to on or off.
     """
 
-    bool_flag(name = name, build_setting_default = default)
+    bool_flag(name = name, build_setting_default = default, visibility = ["//visibility:public"])
 
     native.config_setting(
         name = name + "_true",
         flag_values = {
             ":" + name: "True",
         },
-        visibility = ["//visibility:public"],
+        visibility = ["//:dawn_internal"],
     )
 
     native.config_setting(
@@ -30,7 +30,7 @@
         flag_values = {
             ":" + name: "False",
         },
-        visibility = ["//visibility:public"],
+        visibility = ["//:dawn_internal"],
     )
 
 def declare_os_flag():
@@ -51,18 +51,19 @@
         name = "os",
         build_setting_default = "other",
         values = OSes,
+        visibility = ["//visibility:public"],
     )
 
     for os in OSes:
         native.config_setting(
             name = "tint_build_is_{}_true".format(os),
             flag_values = { ":os": os },
-            visibility = ["//visibility:public"],
+            visibility = ["//:dawn_internal"],
         )
         selects.config_setting_group(
             name = "tint_build_is_{}_false".format(os),
             match_any = [ "tint_build_is_{}_true".format(other) for other in OSes if other != os],
-            visibility = ["//visibility:public"],
+            visibility = ["//:dawn_internal"],
         )
 
 COPTS = select({
diff --git a/tools/src/cmd/gen/build/BUILD.bazel.tmpl b/tools/src/cmd/gen/build/BUILD.bazel.tmpl
--- a/tools/src/cmd/gen/build/BUILD.bazel.tmpl
+++ b/tools/src/cmd/gen/build/BUILD.bazel.tmpl
@@ -118,7 +118,7 @@
 {{-     end}}
 {{-   end}},
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = [{{Eval "TargetVisibility" $}}],
 )
 {{/* newline */}}
 {{- end}}
@@ -126,6 +126,22 @@
 
 {{- /*
 --------------------------------------------------------------------------------
+-- Emits the visibility for generated targets
+-------------------------------------------------------------------------------
+*/ -}}
+{{- define "TargetVisibility" -}}
+{{-   if and $.Kind.IsLib (or (eq $.Directory.Path "api") (eq $.Directory.Path "api/common")) -}}
+"//visibility:public"
+{{-   else if and $.Kind.IsCmd (eq $.Directory.Path "cmd/tint") -}}
+"//visibility:public"
+{{-   else -}}
+"//:dawn_internal"
+{{-   end -}}
+{{- end}}
+
+
+{{- /*
+--------------------------------------------------------------------------------
 -- Emits a dependency target name
 --------------------------------------------------------------------------------
 */ -}}

diff --git a/src/tint/api/BUILD.bazel b/src/tint/api/BUILD.bazel
--- a/src/tint/api/BUILD.bazel
+++ b/src/tint/api/BUILD.bazel
@@ -147,7 +147,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/api/common/BUILD.bazel b/src/tint/api/common/BUILD.bazel
--- a/src/tint/api/common/BUILD.bazel
+++ b/src/tint/api/common/BUILD.bazel
@@ -88,6 +88,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/api/helpers/BUILD.bazel b/src/tint/api/helpers/BUILD.bazel
--- a/src/tint/api/helpers/BUILD.bazel
+++ b/src/tint/api/helpers/BUILD.bazel
@@ -64,6 +64,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/cmd/bench/BUILD.bazel b/src/tint/cmd/bench/BUILD.bazel
--- a/src/tint/cmd/bench/BUILD.bazel
+++ b/src/tint/cmd/bench/BUILD.bazel
@@ -76,7 +76,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_binary(
   name = "bench_cmd",
@@ -137,7 +137,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/bench/glsl/BUILD.bazel b/src/tint/cmd/bench/glsl/BUILD.bazel
--- a/src/tint/cmd/bench/glsl/BUILD.bazel
+++ b/src/tint/cmd/bench/glsl/BUILD.bazel
@@ -80,7 +80,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/bench/hlsl/BUILD.bazel b/src/tint/cmd/bench/hlsl/BUILD.bazel
--- a/src/tint/cmd/bench/hlsl/BUILD.bazel
+++ b/src/tint/cmd/bench/hlsl/BUILD.bazel
@@ -80,7 +80,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/bench/msl/BUILD.bazel b/src/tint/cmd/bench/msl/BUILD.bazel
--- a/src/tint/cmd/bench/msl/BUILD.bazel
+++ b/src/tint/cmd/bench/msl/BUILD.bazel
@@ -80,7 +80,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/bench/spirv/BUILD.bazel b/src/tint/cmd/bench/spirv/BUILD.bazel
--- a/src/tint/cmd/bench/spirv/BUILD.bazel
+++ b/src/tint/cmd/bench/spirv/BUILD.bazel
@@ -79,7 +79,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/bench/wgsl/BUILD.bazel b/src/tint/cmd/bench/wgsl/BUILD.bazel
--- a/src/tint/cmd/bench/wgsl/BUILD.bazel
+++ b/src/tint/cmd/bench/wgsl/BUILD.bazel
@@ -80,7 +80,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/common/BUILD.bazel b/src/tint/cmd/common/BUILD.bazel
--- a/src/tint/cmd/common/BUILD.bazel
+++ b/src/tint/cmd/common/BUILD.bazel
@@ -91,7 +91,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/fuzz/ir/as/BUILD.bazel b/src/tint/cmd/fuzz/ir/as/BUILD.bazel
--- a/src/tint/cmd/fuzz/ir/as/BUILD.bazel
+++ b/src/tint/cmd/fuzz/ir/as/BUILD.bazel
@@ -86,7 +86,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/fuzz/ir/dis/BUILD.bazel b/src/tint/cmd/fuzz/ir/dis/BUILD.bazel
--- a/src/tint/cmd/fuzz/ir/dis/BUILD.bazel
+++ b/src/tint/cmd/fuzz/ir/dis/BUILD.bazel
@@ -97,7 +97,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/fuzz/ir/helpers/BUILD.bazel b/src/tint/cmd/fuzz/ir/helpers/BUILD.bazel
--- a/src/tint/cmd/fuzz/ir/helpers/BUILD.bazel
+++ b/src/tint/cmd/fuzz/ir/helpers/BUILD.bazel
@@ -65,7 +65,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -95,6 +95,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/cmd/info/BUILD.bazel b/src/tint/cmd/info/BUILD.bazel
--- a/src/tint/cmd/info/BUILD.bazel
+++ b/src/tint/cmd/info/BUILD.bazel
@@ -73,7 +73,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/test/BUILD.bazel b/src/tint/cmd/test/BUILD.bazel
--- a/src/tint/cmd/test/BUILD.bazel
+++ b/src/tint/cmd/test/BUILD.bazel
@@ -171,7 +171,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/cmd/tintd/BUILD.bazel b/src/tint/cmd/tintd/BUILD.bazel
--- a/src/tint/cmd/tintd/BUILD.bazel
+++ b/src/tint/cmd/tintd/BUILD.bazel
@@ -60,7 +60,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/core/BUILD.bazel b/src/tint/lang/core/BUILD.bazel
--- a/src/tint/lang/core/BUILD.bazel
+++ b/src/tint/lang/core/BUILD.bazel
@@ -68,7 +68,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -91,6 +91,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/core/constant/BUILD.bazel b/src/tint/lang/core/constant/BUILD.bazel
--- a/src/tint/lang/core/constant/BUILD.bazel
+++ b/src/tint/lang/core/constant/BUILD.bazel
@@ -79,7 +79,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -113,6 +113,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/core/intrinsic/BUILD.bazel b/src/tint/lang/core/intrinsic/BUILD.bazel
--- a/src/tint/lang/core/intrinsic/BUILD.bazel
+++ b/src/tint/lang/core/intrinsic/BUILD.bazel
@@ -69,7 +69,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -98,6 +98,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/core/ir/BUILD.bazel b/src/tint/lang/core/ir/BUILD.bazel
--- a/src/tint/lang/core/ir/BUILD.bazel
+++ b/src/tint/lang/core/ir/BUILD.bazel
@@ -181,7 +181,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -267,6 +267,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/core/ir/analysis/BUILD.bazel b/src/tint/lang/core/ir/analysis/BUILD.bazel
--- a/src/tint/lang/core/ir/analysis/BUILD.bazel
+++ b/src/tint/lang/core/ir/analysis/BUILD.bazel
@@ -71,7 +71,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -106,6 +106,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/core/ir/binary/BUILD.bazel b/src/tint/lang/core/ir/binary/BUILD.bazel
--- a/src/tint/lang/core/ir/binary/BUILD.bazel
+++ b/src/tint/lang/core/ir/binary/BUILD.bazel
@@ -71,7 +71,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -106,7 +106,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/core/ir/transform/BUILD.bazel b/src/tint/lang/core/ir/transform/BUILD.bazel
--- a/src/tint/lang/core/ir/transform/BUILD.bazel
+++ b/src/tint/lang/core/ir/transform/BUILD.bazel
@@ -138,7 +138,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -203,6 +203,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/core/ir/type/BUILD.bazel b/src/tint/lang/core/ir/type/BUILD.bazel
--- a/src/tint/lang/core/ir/type/BUILD.bazel
+++ b/src/tint/lang/core/ir/type/BUILD.bazel
@@ -57,6 +57,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/core/type/BUILD.bazel b/src/tint/lang/core/type/BUILD.bazel
--- a/src/tint/lang/core/type/BUILD.bazel
+++ b/src/tint/lang/core/type/BUILD.bazel
@@ -161,7 +161,7 @@
   ],
   copts = COPTS,
   include_prefix = "src/tint/lang/core/type",
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -220,6 +220,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/glsl/BUILD.bazel b/src/tint/lang/glsl/BUILD.bazel
--- a/src/tint/lang/glsl/BUILD.bazel
+++ b/src/tint/lang/glsl/BUILD.bazel
@@ -57,6 +57,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/glsl/intrinsic/BUILD.bazel b/src/tint/lang/glsl/intrinsic/BUILD.bazel
--- a/src/tint/lang/glsl/intrinsic/BUILD.bazel
+++ b/src/tint/lang/glsl/intrinsic/BUILD.bazel
@@ -66,6 +66,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/glsl/ir/BUILD.bazel b/src/tint/lang/glsl/ir/BUILD.bazel
--- a/src/tint/lang/glsl/ir/BUILD.bazel
+++ b/src/tint/lang/glsl/ir/BUILD.bazel
@@ -71,7 +71,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -106,6 +106,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/glsl/validate/BUILD.bazel b/src/tint/lang/glsl/validate/BUILD.bazel
--- a/src/tint/lang/glsl/validate/BUILD.bazel
+++ b/src/tint/lang/glsl/validate/BUILD.bazel
@@ -68,7 +68,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/glsl/writer/BUILD.bazel b/src/tint/lang/glsl/writer/BUILD.bazel
--- a/src/tint/lang/glsl/writer/BUILD.bazel
+++ b/src/tint/lang/glsl/writer/BUILD.bazel
@@ -73,7 +73,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -134,7 +134,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/glsl/writer/common/BUILD.bazel b/src/tint/lang/glsl/writer/common/BUILD.bazel
--- a/src/tint/lang/glsl/writer/common/BUILD.bazel
+++ b/src/tint/lang/glsl/writer/common/BUILD.bazel
@@ -68,7 +68,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -95,7 +95,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/glsl/writer/helpers/BUILD.bazel b/src/tint/lang/glsl/writer/helpers/BUILD.bazel
--- a/src/tint/lang/glsl/writer/helpers/BUILD.bazel
+++ b/src/tint/lang/glsl/writer/helpers/BUILD.bazel
@@ -69,7 +69,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/glsl/writer/printer/BUILD.bazel b/src/tint/lang/glsl/writer/printer/BUILD.bazel
--- a/src/tint/lang/glsl/writer/printer/BUILD.bazel
+++ b/src/tint/lang/glsl/writer/printer/BUILD.bazel
@@ -74,7 +74,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/glsl/writer/raise/BUILD.bazel b/src/tint/lang/glsl/writer/raise/BUILD.bazel
--- a/src/tint/lang/glsl/writer/raise/BUILD.bazel
+++ b/src/tint/lang/glsl/writer/raise/BUILD.bazel
@@ -88,7 +88,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -131,7 +131,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/hlsl/BUILD.bazel b/src/tint/lang/hlsl/BUILD.bazel
--- a/src/tint/lang/hlsl/BUILD.bazel
+++ b/src/tint/lang/hlsl/BUILD.bazel
@@ -57,6 +57,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/hlsl/intrinsic/BUILD.bazel b/src/tint/lang/hlsl/intrinsic/BUILD.bazel
--- a/src/tint/lang/hlsl/intrinsic/BUILD.bazel
+++ b/src/tint/lang/hlsl/intrinsic/BUILD.bazel
@@ -68,6 +68,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/hlsl/ir/BUILD.bazel b/src/tint/lang/hlsl/ir/BUILD.bazel
--- a/src/tint/lang/hlsl/ir/BUILD.bazel
+++ b/src/tint/lang/hlsl/ir/BUILD.bazel
@@ -71,7 +71,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -107,6 +107,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/hlsl/type/BUILD.bazel b/src/tint/lang/hlsl/type/BUILD.bazel
--- a/src/tint/lang/hlsl/type/BUILD.bazel
+++ b/src/tint/lang/hlsl/type/BUILD.bazel
@@ -67,7 +67,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -92,6 +92,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/hlsl/validate/BUILD.bazel b/src/tint/lang/hlsl/validate/BUILD.bazel
--- a/src/tint/lang/hlsl/validate/BUILD.bazel
+++ b/src/tint/lang/hlsl/validate/BUILD.bazel
@@ -70,7 +70,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/hlsl/writer/BUILD.bazel b/src/tint/lang/hlsl/writer/BUILD.bazel
--- a/src/tint/lang/hlsl/writer/BUILD.bazel
+++ b/src/tint/lang/hlsl/writer/BUILD.bazel
@@ -69,7 +69,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -122,7 +122,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/hlsl/writer/common/BUILD.bazel b/src/tint/lang/hlsl/writer/common/BUILD.bazel
--- a/src/tint/lang/hlsl/writer/common/BUILD.bazel
+++ b/src/tint/lang/hlsl/writer/common/BUILD.bazel
@@ -69,7 +69,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -102,6 +102,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/hlsl/writer/printer/BUILD.bazel b/src/tint/lang/hlsl/writer/printer/BUILD.bazel
--- a/src/tint/lang/hlsl/writer/printer/BUILD.bazel
+++ b/src/tint/lang/hlsl/writer/printer/BUILD.bazel
@@ -73,6 +73,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/hlsl/writer/raise/BUILD.bazel b/src/tint/lang/hlsl/writer/raise/BUILD.bazel
--- a/src/tint/lang/hlsl/writer/raise/BUILD.bazel
+++ b/src/tint/lang/hlsl/writer/raise/BUILD.bazel
@@ -97,7 +97,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -145,6 +145,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/msl/BUILD.bazel b/src/tint/lang/msl/BUILD.bazel
--- a/src/tint/lang/msl/BUILD.bazel
+++ b/src/tint/lang/msl/BUILD.bazel
@@ -58,6 +58,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/msl/intrinsic/BUILD.bazel b/src/tint/lang/msl/intrinsic/BUILD.bazel
--- a/src/tint/lang/msl/intrinsic/BUILD.bazel
+++ b/src/tint/lang/msl/intrinsic/BUILD.bazel
@@ -68,6 +68,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/msl/ir/BUILD.bazel b/src/tint/lang/msl/ir/BUILD.bazel
--- a/src/tint/lang/msl/ir/BUILD.bazel
+++ b/src/tint/lang/msl/ir/BUILD.bazel
@@ -73,7 +73,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -107,6 +107,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/msl/type/BUILD.bazel b/src/tint/lang/msl/type/BUILD.bazel
--- a/src/tint/lang/msl/type/BUILD.bazel
+++ b/src/tint/lang/msl/type/BUILD.bazel
@@ -65,7 +65,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -88,6 +88,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/msl/validate/BUILD.bazel b/src/tint/lang/msl/validate/BUILD.bazel
--- a/src/tint/lang/msl/validate/BUILD.bazel
+++ b/src/tint/lang/msl/validate/BUILD.bazel
@@ -66,7 +66,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/msl/writer/BUILD.bazel b/src/tint/lang/msl/writer/BUILD.bazel
--- a/src/tint/lang/msl/writer/BUILD.bazel
+++ b/src/tint/lang/msl/writer/BUILD.bazel
@@ -74,7 +74,7 @@
   }),
   copts = COPTS,
   include_prefix = "src/tint/lang/msl/writer",
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -122,7 +122,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/msl/writer/common/BUILD.bazel b/src/tint/lang/msl/writer/common/BUILD.bazel
--- a/src/tint/lang/msl/writer/common/BUILD.bazel
+++ b/src/tint/lang/msl/writer/common/BUILD.bazel
@@ -70,7 +70,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -100,7 +100,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/msl/writer/printer/BUILD.bazel b/src/tint/lang/msl/writer/printer/BUILD.bazel
--- a/src/tint/lang/msl/writer/printer/BUILD.bazel
+++ b/src/tint/lang/msl/writer/printer/BUILD.bazel
@@ -75,7 +75,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/msl/writer/raise/BUILD.bazel b/src/tint/lang/msl/writer/raise/BUILD.bazel
--- a/src/tint/lang/msl/writer/raise/BUILD.bazel
+++ b/src/tint/lang/msl/writer/raise/BUILD.bazel
@@ -96,7 +96,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -142,7 +142,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/null/writer/BUILD.bazel b/src/tint/lang/null/writer/BUILD.bazel
--- a/src/tint/lang/null/writer/BUILD.bazel
+++ b/src/tint/lang/null/writer/BUILD.bazel
@@ -66,7 +66,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/null/writer/common/BUILD.bazel b/src/tint/lang/null/writer/common/BUILD.bazel
--- a/src/tint/lang/null/writer/common/BUILD.bazel
+++ b/src/tint/lang/null/writer/common/BUILD.bazel
@@ -59,6 +59,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/null/writer/raise/BUILD.bazel b/src/tint/lang/null/writer/raise/BUILD.bazel
--- a/src/tint/lang/null/writer/raise/BUILD.bazel
+++ b/src/tint/lang/null/writer/raise/BUILD.bazel
@@ -66,6 +66,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/spirv/BUILD.bazel b/src/tint/lang/spirv/BUILD.bazel
--- a/src/tint/lang/spirv/BUILD.bazel
+++ b/src/tint/lang/spirv/BUILD.bazel
@@ -57,6 +57,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/spirv/intrinsic/BUILD.bazel b/src/tint/lang/spirv/intrinsic/BUILD.bazel
--- a/src/tint/lang/spirv/intrinsic/BUILD.bazel
+++ b/src/tint/lang/spirv/intrinsic/BUILD.bazel
@@ -68,6 +68,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/spirv/ir/BUILD.bazel b/src/tint/lang/spirv/ir/BUILD.bazel
--- a/src/tint/lang/spirv/ir/BUILD.bazel
+++ b/src/tint/lang/spirv/ir/BUILD.bazel
@@ -71,7 +71,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -105,6 +105,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/spirv/reader/BUILD.bazel b/src/tint/lang/spirv/reader/BUILD.bazel
--- a/src/tint/lang/spirv/reader/BUILD.bazel
+++ b/src/tint/lang/spirv/reader/BUILD.bazel
@@ -71,7 +71,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -114,7 +114,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/spirv/reader/common/BUILD.bazel b/src/tint/lang/spirv/reader/common/BUILD.bazel
--- a/src/tint/lang/spirv/reader/common/BUILD.bazel
+++ b/src/tint/lang/spirv/reader/common/BUILD.bazel
@@ -57,7 +57,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -78,7 +78,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/spirv/reader/lower/BUILD.bazel b/src/tint/lang/spirv/reader/lower/BUILD.bazel
--- a/src/tint/lang/spirv/reader/lower/BUILD.bazel
+++ b/src/tint/lang/spirv/reader/lower/BUILD.bazel
@@ -86,7 +86,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -128,6 +128,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/spirv/reader/parser/BUILD.bazel b/src/tint/lang/spirv/reader/parser/BUILD.bazel
--- a/src/tint/lang/spirv/reader/parser/BUILD.bazel
+++ b/src/tint/lang/spirv/reader/parser/BUILD.bazel
@@ -82,7 +82,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -144,7 +144,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/spirv/type/BUILD.bazel b/src/tint/lang/spirv/type/BUILD.bazel
--- a/src/tint/lang/spirv/type/BUILD.bazel
+++ b/src/tint/lang/spirv/type/BUILD.bazel
@@ -69,7 +69,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -101,6 +101,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/spirv/validate/BUILD.bazel b/src/tint/lang/spirv/validate/BUILD.bazel
--- a/src/tint/lang/spirv/validate/BUILD.bazel
+++ b/src/tint/lang/spirv/validate/BUILD.bazel
@@ -63,7 +63,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -86,7 +86,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/spirv/writer/BUILD.bazel b/src/tint/lang/spirv/writer/BUILD.bazel
--- a/src/tint/lang/spirv/writer/BUILD.bazel
+++ b/src/tint/lang/spirv/writer/BUILD.bazel
@@ -79,7 +79,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -144,7 +144,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/spirv/writer/analysis/BUILD.bazel b/src/tint/lang/spirv/writer/analysis/BUILD.bazel
--- a/src/tint/lang/spirv/writer/analysis/BUILD.bazel
+++ b/src/tint/lang/spirv/writer/analysis/BUILD.bazel
@@ -69,7 +69,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -104,6 +104,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/spirv/writer/common/BUILD.bazel b/src/tint/lang/spirv/writer/common/BUILD.bazel
--- a/src/tint/lang/spirv/writer/common/BUILD.bazel
+++ b/src/tint/lang/spirv/writer/common/BUILD.bazel
@@ -78,7 +78,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -128,7 +128,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/spirv/writer/printer/BUILD.bazel b/src/tint/lang/spirv/writer/printer/BUILD.bazel
--- a/src/tint/lang/spirv/writer/printer/BUILD.bazel
+++ b/src/tint/lang/spirv/writer/printer/BUILD.bazel
@@ -81,7 +81,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/spirv/writer/raise/BUILD.bazel b/src/tint/lang/spirv/writer/raise/BUILD.bazel
--- a/src/tint/lang/spirv/writer/raise/BUILD.bazel
+++ b/src/tint/lang/spirv/writer/raise/BUILD.bazel
@@ -106,7 +106,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -158,7 +158,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/BUILD.bazel b/src/tint/lang/wgsl/BUILD.bazel
--- a/src/tint/lang/wgsl/BUILD.bazel
+++ b/src/tint/lang/wgsl/BUILD.bazel
@@ -65,7 +65,7 @@
   ],
   copts = COPTS,
   include_prefix = "src/tint/lang/wgsl",
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -125,7 +125,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/ast/BUILD.bazel b/src/tint/lang/wgsl/ast/BUILD.bazel
--- a/src/tint/lang/wgsl/ast/BUILD.bazel
+++ b/src/tint/lang/wgsl/ast/BUILD.bazel
@@ -213,7 +213,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -305,6 +305,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/wgsl/inspector/BUILD.bazel b/src/tint/lang/wgsl/inspector/BUILD.bazel
--- a/src/tint/lang/wgsl/inspector/BUILD.bazel
+++ b/src/tint/lang/wgsl/inspector/BUILD.bazel
@@ -75,7 +75,7 @@
   ],
   copts = COPTS,
   include_prefix = "src/tint/lang/wgsl/inspector",
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -113,7 +113,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/intrinsic/BUILD.bazel b/src/tint/lang/wgsl/intrinsic/BUILD.bazel
--- a/src/tint/lang/wgsl/intrinsic/BUILD.bazel
+++ b/src/tint/lang/wgsl/intrinsic/BUILD.bazel
@@ -68,7 +68,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -108,7 +108,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/ir/BUILD.bazel b/src/tint/lang/wgsl/ir/BUILD.bazel
--- a/src/tint/lang/wgsl/ir/BUILD.bazel
+++ b/src/tint/lang/wgsl/ir/BUILD.bazel
@@ -71,7 +71,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -135,7 +135,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/ls/BUILD.bazel b/src/tint/lang/wgsl/ls/BUILD.bazel
--- a/src/tint/lang/wgsl/ls/BUILD.bazel
+++ b/src/tint/lang/wgsl/ls/BUILD.bazel
@@ -104,7 +104,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -156,7 +156,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/program/BUILD.bazel b/src/tint/lang/wgsl/program/BUILD.bazel
--- a/src/tint/lang/wgsl/program/BUILD.bazel
+++ b/src/tint/lang/wgsl/program/BUILD.bazel
@@ -68,7 +68,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -100,6 +100,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/wgsl/reader/BUILD.bazel b/src/tint/lang/wgsl/reader/BUILD.bazel
--- a/src/tint/lang/wgsl/reader/BUILD.bazel
+++ b/src/tint/lang/wgsl/reader/BUILD.bazel
@@ -79,7 +79,7 @@
   }),
   copts = COPTS,
   include_prefix = "src/tint/lang/wgsl/reader",
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -107,7 +107,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/reader/lower/BUILD.bazel b/src/tint/lang/wgsl/reader/lower/BUILD.bazel
--- a/src/tint/lang/wgsl/reader/lower/BUILD.bazel
+++ b/src/tint/lang/wgsl/reader/lower/BUILD.bazel
@@ -68,7 +68,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -102,6 +102,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/wgsl/reader/parser/BUILD.bazel b/src/tint/lang/wgsl/reader/parser/BUILD.bazel
--- a/src/tint/lang/wgsl/reader/parser/BUILD.bazel
+++ b/src/tint/lang/wgsl/reader/parser/BUILD.bazel
@@ -74,7 +74,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -178,7 +178,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/reader/program_to_ir/BUILD.bazel b/src/tint/lang/wgsl/reader/program_to_ir/BUILD.bazel
--- a/src/tint/lang/wgsl/reader/program_to_ir/BUILD.bazel
+++ b/src/tint/lang/wgsl/reader/program_to_ir/BUILD.bazel
@@ -72,7 +72,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -126,7 +126,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/resolver/BUILD.bazel b/src/tint/lang/wgsl/resolver/BUILD.bazel
--- a/src/tint/lang/wgsl/resolver/BUILD.bazel
+++ b/src/tint/lang/wgsl/resolver/BUILD.bazel
@@ -83,7 +83,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -209,7 +209,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/sem/BUILD.bazel b/src/tint/lang/wgsl/sem/BUILD.bazel
--- a/src/tint/lang/wgsl/sem/BUILD.bazel
+++ b/src/tint/lang/wgsl/sem/BUILD.bazel
@@ -131,7 +131,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -170,6 +170,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/wgsl/writer/BUILD.bazel b/src/tint/lang/wgsl/writer/BUILD.bazel
--- a/src/tint/lang/wgsl/writer/BUILD.bazel
+++ b/src/tint/lang/wgsl/writer/BUILD.bazel
@@ -79,7 +79,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -121,7 +121,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/writer/ast_printer/BUILD.bazel b/src/tint/lang/wgsl/writer/ast_printer/BUILD.bazel
--- a/src/tint/lang/wgsl/writer/ast_printer/BUILD.bazel
+++ b/src/tint/lang/wgsl/writer/ast_printer/BUILD.bazel
@@ -70,7 +70,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -140,7 +140,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/writer/common/BUILD.bazel b/src/tint/lang/wgsl/writer/common/BUILD.bazel
--- a/src/tint/lang/wgsl/writer/common/BUILD.bazel
+++ b/src/tint/lang/wgsl/writer/common/BUILD.bazel
@@ -59,6 +59,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/lang/wgsl/writer/ir_to_program/BUILD.bazel b/src/tint/lang/wgsl/writer/ir_to_program/BUILD.bazel
--- a/src/tint/lang/wgsl/writer/ir_to_program/BUILD.bazel
+++ b/src/tint/lang/wgsl/writer/ir_to_program/BUILD.bazel
@@ -73,7 +73,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -121,7 +121,7 @@
     "//conditions:default": [],
   }),
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/lang/wgsl/writer/raise/BUILD.bazel b/src/tint/lang/wgsl/writer/raise/BUILD.bazel
--- a/src/tint/lang/wgsl/writer/raise/BUILD.bazel
+++ b/src/tint/lang/wgsl/writer/raise/BUILD.bazel
@@ -73,7 +73,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -106,6 +106,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/BUILD.bazel b/src/tint/utils/BUILD.bazel
--- a/src/tint/utils/BUILD.bazel
+++ b/src/tint/utils/BUILD.bazel
@@ -59,7 +59,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -80,6 +80,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/bytes/BUILD.bazel b/src/tint/utils/bytes/BUILD.bazel
--- a/src/tint/utils/bytes/BUILD.bazel
+++ b/src/tint/utils/bytes/BUILD.bazel
@@ -60,7 +60,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -83,6 +83,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/command/BUILD.bazel b/src/tint/utils/command/BUILD.bazel
--- a/src/tint/utils/command/BUILD.bazel
+++ b/src/tint/utils/command/BUILD.bazel
@@ -77,7 +77,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -101,7 +101,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/utils/containers/BUILD.bazel b/src/tint/utils/containers/BUILD.bazel
--- a/src/tint/utils/containers/BUILD.bazel
+++ b/src/tint/utils/containers/BUILD.bazel
@@ -67,7 +67,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -100,6 +100,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/diagnostic/BUILD.bazel b/src/tint/utils/diagnostic/BUILD.bazel
--- a/src/tint/utils/diagnostic/BUILD.bazel
+++ b/src/tint/utils/diagnostic/BUILD.bazel
@@ -62,7 +62,7 @@
   ],
   copts = COPTS,
   include_prefix = "src/tint/utils/diagnostic",
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -86,6 +86,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/file/BUILD.bazel b/src/tint/utils/file/BUILD.bazel
--- a/src/tint/utils/file/BUILD.bazel
+++ b/src/tint/utils/file/BUILD.bazel
@@ -66,7 +66,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -80,7 +80,7 @@
     "@gtest",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/utils/ice/BUILD.bazel b/src/tint/utils/ice/BUILD.bazel
--- a/src/tint/utils/ice/BUILD.bazel
+++ b/src/tint/utils/ice/BUILD.bazel
@@ -52,7 +52,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -67,6 +67,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/macros/BUILD.bazel b/src/tint/utils/macros/BUILD.bazel
--- a/src/tint/utils/macros/BUILD.bazel
+++ b/src/tint/utils/macros/BUILD.bazel
@@ -54,7 +54,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -69,6 +69,6 @@
     "@gtest",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/math/BUILD.bazel b/src/tint/utils/math/BUILD.bazel
--- a/src/tint/utils/math/BUILD.bazel
+++ b/src/tint/utils/math/BUILD.bazel
@@ -52,7 +52,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -73,6 +73,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/memory/BUILD.bazel b/src/tint/utils/memory/BUILD.bazel
--- a/src/tint/utils/memory/BUILD.bazel
+++ b/src/tint/utils/memory/BUILD.bazel
@@ -56,7 +56,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -74,6 +74,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/rtti/BUILD.bazel b/src/tint/utils/rtti/BUILD.bazel
--- a/src/tint/utils/rtti/BUILD.bazel
+++ b/src/tint/utils/rtti/BUILD.bazel
@@ -57,7 +57,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -76,7 +76,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "bench",
@@ -93,6 +93,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/strconv/BUILD.bazel b/src/tint/utils/strconv/BUILD.bazel
--- a/src/tint/utils/strconv/BUILD.bazel
+++ b/src/tint/utils/strconv/BUILD.bazel
@@ -57,7 +57,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -71,6 +71,6 @@
     "@gtest",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/symbol/BUILD.bazel b/src/tint/utils/symbol/BUILD.bazel
--- a/src/tint/utils/symbol/BUILD.bazel
+++ b/src/tint/utils/symbol/BUILD.bazel
@@ -59,7 +59,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -80,6 +80,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
diff --git a/src/tint/utils/system/BUILD.bazel b/src/tint/utils/system/BUILD.bazel
--- a/src/tint/utils/system/BUILD.bazel
+++ b/src/tint/utils/system/BUILD.bazel
@@ -88,7 +88,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/utils/text/BUILD.bazel b/src/tint/utils/text/BUILD.bazel
--- a/src/tint/utils/text/BUILD.bazel
+++ b/src/tint/utils/text/BUILD.bazel
@@ -88,7 +88,7 @@
   ],
   copts = COPTS,
   include_prefix = "src/tint/utils/text",
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 cc_library(
   name = "test",
@@ -113,7 +113,7 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
 alias(
diff --git a/src/tint/utils/text_generator/BUILD.bazel b/src/tint/utils/text_generator/BUILD.bazel
--- a/src/tint/utils/text_generator/BUILD.bazel
+++ b/src/tint/utils/text_generator/BUILD.bazel
@@ -58,6 +58,6 @@
     "//src/utils",
   ],
   copts = COPTS,
-  visibility = ["//visibility:public"],
+  visibility = ["//:dawn_internal"],
 )
 
