diff --git a/src/tint/flags.bzl b/src/tint/flags.bzl
--- a/src/tint/flags.bzl
+++ b/src/tint/flags.bzl
@@ -65,35 +65,49 @@
             visibility = ["//visibility:public"],
         )
 
-COPTS = [
-    "-fno-rtti",
-    "-fno-exceptions",
-    "--std=c++20",
-] + select({
-    "//src/tint:tint_build_glsl_writer_true": [ "-DTINT_BUILD_GLSL_WRITER" ],
-    "//conditions:default": [],
+COPTS = select({
+    "@rules_cc//cc/compiler:clang-cl": [
+        "/GR-",
+        "/D_HAS_EXCEPTIONS=0",
+        "/std:c++20",
+        "/Zc:preprocessor",
+    ],
+    "@rules_cc//cc/compiler:msvc-cl": [
+        "/GR-",
+        "/D_HAS_EXCEPTIONS=0",
+        "/std:c++20",
+        "/Zc:preprocessor",
+    ],
+    "//conditions:default": [
+        "-fno-rtti",
+        "-fno-exceptions",
+        "--std=c++20",
+    ],
 }) + select({
-    "//src/tint:tint_build_hlsl_writer_true": [ "-DTINT_BUILD_HLSL_WRITER" ],
-    "//conditions:default": [],
+    "//src/tint:tint_build_glsl_writer_true": [ "-DTINT_BUILD_GLSL_WRITER=1" ],
+    "//conditions:default": [ "-DTINT_BUILD_GLSL_WRITER=0" ],
 }) + select({
-    "//src/tint:tint_build_ir_true":          [ "-DTINT_BUILD_IR" ],
-    "//conditions:default": [],
+    "//src/tint:tint_build_hlsl_writer_true": [ "-DTINT_BUILD_HLSL_WRITER=1" ],
+    "//conditions:default": [ "-DTINT_BUILD_HLSL_WRITER=0" ],
 }) + select({
-    "//src/tint:tint_build_msl_writer_true":  [ "-DTINT_BUILD_MSL_WRITER" ],
-    "//conditions:default": [],
+    "//src/tint:tint_build_ir_true":          [ "-DTINT_BUILD_IR=1" ],
+    "//conditions:default": [ "-DTINT_BUILD_IR=0" ],
 }) + select({
-    "//src/tint:tint_build_spv_reader_true":  [ "-DTINT_BUILD_SPV_READER" ],
-    "//conditions:default": [],
+    "//src/tint:tint_build_msl_writer_true":  [ "-DTINT_BUILD_MSL_WRITER=1" ],
+    "//conditions:default": [ "-DTINT_BUILD_MSL_WRITER=0" ],
 }) + select({
-    "//src/tint:tint_build_spv_writer_true":  [ "-DTINT_BUILD_SPV_WRITER" ],
-    "//conditions:default": [],
+    "//src/tint:tint_build_spv_reader_true":  [ "-DTINT_BUILD_SPV_READER=1" ],
+    "//conditions:default": [ "-DTINT_BUILD_SPV_READER=0" ],
 }) + select({
-    "//src/tint:tint_build_wgsl_reader_true": [ "-DTINT_BUILD_WGSL_READER" ],
-    "//conditions:default": [],
+    "//src/tint:tint_build_spv_writer_true":  [ "-DTINT_BUILD_SPV_WRITER=1" ],
+    "//conditions:default": [ "-DTINT_BUILD_SPV_WRITER=0" ],
 }) + select({
-    "//src/tint:tint_build_wgsl_writer_true": [ "-DTINT_BUILD_WGSL_WRITER" ],
-    "//conditions:default": [],
+    "//src/tint:tint_build_wgsl_reader_true": [ "-DTINT_BUILD_WGSL_READER=1" ],
+    "//conditions:default": [ "-DTINT_BUILD_WGSL_READER=0" ],
 }) + select({
-    "//src/tint:tint_build_null_writer_true": [ "-DTINT_BUILD_NULL_WRITER" ],
-    "//conditions:default": [],
+    "//src/tint:tint_build_wgsl_writer_true": [ "-DTINT_BUILD_WGSL_WRITER=1" ],
+    "//conditions:default": [ "-DTINT_BUILD_WGSL_WRITER=0" ],
+}) + select({
+    "//src/tint:tint_build_null_writer_true": [ "-DTINT_BUILD_NULL_WRITER=1" ],
+    "//conditions:default": [ "-DTINT_BUILD_NULL_WRITER=0" ],
 })
