diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..ea7a9ac
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,14 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "format",
+    deps = [
+        "@boost.assert//:assert",
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.optional//:optional",
+        "@boost.smart_ptr//:smart_ptr",
+        "@boost.throw_exception//:throw_exception",
+        "@boost.utility//:utility",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..9c382d0
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,15 @@
+module(
+    name = "boost.format",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.assert", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.optional", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.smart_ptr", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.throw_exception", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.utility", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..2c67ed9
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost.format//:format",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "format_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/format_test.cc b/bazel_test/format_test.cc
new file mode 100644
index 0000000..c81cc3e
--- /dev/null
+++ b/bazel_test/format_test.cc
@@ -0,0 +1,10 @@
+#include <boost/format.hpp>
+
+int main()
+{
+  if ((boost::format("hello %d") % 1).str() != "hello 1") {
+    return 1;
+  }
+
+  return 0;
+}
