--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,36 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "configure_make", "runnable_binary")
+load("@rules_license//rules:license.bzl", "license")
+
+package(
+    default_applicable_licenses = [":license"],
+    default_visibility = ["//visibility:public"],
+)
+
+exports_files(["COPYING"])
+
+license(
+    name = "license",
+    package_name = "gifsicle",
+    license_kinds = [
+        "@rules_license//licenses/spdx:GPL-2.0-or-later",
+    ],
+    license_text = "COPYING",
+    package_url = "http://www.lcdf.org/gifsicle/",
+)
+
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["**"]),
+)
+
+configure_make(
+    name = "gifsicle_make",
+    lib_source = ":all_srcs",
+    out_binaries = ["gifsicle"],
+)
+
+runnable_binary(
+    name = "gifsicle",
+    binary = "gifsicle",
+    foreign_cc_target = ":gifsicle_make",
+)
