--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,62 @@
+config_setting(
+    name = "msvc_compiler",
+    flag_values = {"@bazel_tools//tools/cpp:compiler": "msvc-cl"},
+)
+
+config_setting(
+    name = "macos",
+    constraint_values = ["@platforms//os:macos"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "nanobind",
+    hdrs = glob(
+        include = [
+            "include/nanobind/*.h",
+            "include/nanobind/stl/*.h",
+            "include/nanobind/stl/detail/*.h",
+        ],
+    ),
+    srcs = [
+        "include/nanobind/stl/detail/nb_dict.h",
+        "include/nanobind/stl/detail/nb_list.h",
+        "include/nanobind/stl/detail/traits.h",
+        "src/buffer.h",
+        "src/common.cpp",
+        "src/error.cpp",
+        "src/implicit.cpp",
+        "src/nb_enum.cpp",
+        "src/nb_func.cpp",
+        "src/nb_internals.cpp",
+        "src/nb_internals.h",
+        "src/nb_ndarray.cpp",
+        "src/nb_type.cpp",
+        "src/trampoline.cpp",
+    ],
+    copts = select({
+        ":msvc_compiler": ["/std:c++17"],
+        "//conditions:default": [
+            "--std=c++17",
+            "-fexceptions",
+            "-Os",  # size optimization
+            "-flto", # enable LTO
+        ],
+    }),
+    linkopts = select({
+        ":macos": [
+            "-undefined dynamic_lookup",
+            "-Wl,-no_fixup_chains",
+            "-Wl,-dead_strip",
+        ],
+        "//conditions:default": [],
+    }),
+    includes = ["include"],
+    strip_include_prefix = "include",
+    deps = [
+        "@nanobind_python3_8//:python_headers",
+        "@robin-map//:robin-map",
+    ],
+    visibility = ["//visibility:public"],
+)
+
