--- MODULE.bazel
+++ MODULE.bazel
@@ -0,0 +1,15 @@
+module(
+    name = "nanobind",
+    version = "1.2.0",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "platforms", version = "0.0.6")
+bazel_dep(name = "rules_python", version = "0.21.0")
+bazel_dep(name = "robin-map", version = "1.2.1")
+
+python = use_extension("@rules_python//python:extensions.bzl", "python")
+python.toolchain(name="nanobind_python3_8", python_version="3.8")
+use_repo(python, "nanobind_python3_8")
+
+# No need to register the toolchain for Python, we just want the headers.
