--- MODULE.bazel
+++ MODULE.bazel
@@ -1,6 +1,16 @@
    module(
-  name = "rules_spring",
-  version = "2.3.2",
-  repo_name = "rules_spring",
-  compatibility_level = 1,
+    name = "rules_spring",
+    version = "2.3.2",
+    compatibility_level = 1,
+    repo_name = "rules_spring",
    )
+
+bazel_dep(name = "rules_python", version = "0.31.0")
+
+python = use_extension("@rules_python//python/extensions:python.bzl", "python")
+python.toolchain(
+    # This takes priority over captive python from //tools/python_interpreter unless extra toolchains are specified e.g.
+    # bazel run --extra_toolchains //tools/python_interpreter:captive_python_toolchain <target>
+    is_default = True,
+    python_version = "3.11",
+)
