diff --git a/MODULE.bazel b/MODULE.bazel
index 7b84c2b7..12345678 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -20,6 +20,7 @@ bazel_dep(name = "prometheus-cpp", version = "1.3.0", repo_name = "com_github_ju
 bazel_dep(name = "protobuf", version = "29.0", repo_name = "com_google_protobuf")
 bazel_dep(name = "rapidyaml", version = "0.9.0")
 bazel_dep(name = "rules_cc", version = "0.2.9")
+bazel_dep(name = "rules_foreign_cc", version = "0.15.1")
 bazel_dep(name = "rules_proto", version = "7.0.2")
 bazel_dep(name = "zlib", version = "1.3.1.bcr.5")

