diff --git a/src/soplex/config.h b/src/soplex/config.h
new file mode 100644
index 00000000..e465106c
--- /dev/null
+++ b/src/soplex/config.h
@@ -0,0 +1,18 @@
+#ifndef __SPXCONFIG_H__
+#define __SPXCONFIG_H__
+
+#define CMAKE_BUILD_TYPE "Release"
+
+#define SOPLEX_VERSION_MAJOR 7
+#define SOPLEX_VERSION_MINOR 1
+#define SOPLEX_VERSION_PATCH 4
+#define SOPLEX_VERSION_SUB 0
+#define SOPLEX_WITH_BOOST
+#define SOPLEX_WITH_CPPMPF
+/* #undef SOPLEX_WITH_FLOAT128 */
+/* #undef SOPLEX_WITH_GMP */
+/* #undef SOPLEX_WITH_MPFR */
+/* #undef SOPLEX_WITH_PAPILO */
+#define SOPLEX_WITH_ZLIB
+
+#endif
