--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,143 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+
+licenses(["notice"])  # BSD-3-Clause
+
+###############
+#  Libraries  #
+###############
+
+cc_library(
+    name = "create",
+    srcs = glob(["src/**/*.cpp"]),
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@boost//:asio",
+        "@boost//:filesystem",
+        "@boost//:numeric_ublas",
+        "@boost//:system",
+    ],
+)
+
+alias(
+    name = "libcreate",
+    actual = ":create",
+    visibility = ["//visibility:public"],
+)
+
+################
+#   Examples   #
+################
+
+cc_binary(
+    name = "battery_level",
+    srcs = ["examples/battery_level.cpp"],
+    deps = [":libcreate"],
+)
+
+cc_binary(
+    name = "bumpers",
+    srcs = ["examples/bumpers.cpp"],
+    deps = [":libcreate"],
+)
+
+cc_binary(
+    name = "cliffs",
+    srcs = ["examples/cliffs.cpp"],
+    deps = [":libcreate"],
+)
+
+cc_binary(
+    name = "drive_circle",
+    srcs = ["examples/drive_circle.cpp"],
+    deps = [":libcreate"],
+)
+
+cc_binary(
+    name = "leds",
+    srcs = ["examples/leds.cpp"],
+    deps = [":libcreate"],
+)
+
+cc_binary(
+    name = "packets",
+    srcs = ["examples/packets.cpp"],
+    deps = [":libcreate"],
+)
+
+cc_binary(
+    name = "play_song",
+    srcs = ["examples/play_song.cpp"],
+    deps = [":libcreate"],
+)
+
+cc_binary(
+    name = "wheeldrop",
+    srcs = ["examples/wheeldrop.cpp"],
+    deps = [":libcreate"],
+)
+
+###############
+#    Tests    #
+###############
+
+cc_test(
+    name = "create_test",
+    srcs = ["tests/test_create.cpp"],
+    linkstatic = True,
+    deps = [
+        ":libcreate",
+        "@googletest//:gtest_main",
+    ],
+)
+
+cc_test(
+    name = "data_test",
+    srcs = ["tests/test_data.cpp"],
+    linkstatic = True,
+    deps = [
+        ":libcreate",
+        "@googletest//:gtest_main",
+    ],
+)
+
+cc_test(
+    name = "packet_test",
+    srcs = ["tests/test_packet.cpp"],
+    linkstatic = True,
+    deps = [
+        ":libcreate",
+        "@googletest//:gtest_main",
+    ],
+)
+
+cc_test(
+    name = "robot_model_test",
+    srcs = ["tests/test_robot_model.cpp"],
+    linkstatic = True,
+    deps = [
+        ":libcreate",
+        "@googletest//:gtest_main",
+    ],
+)
+
+cc_test(
+    name = "serial_query_test",
+    srcs = ["tests/test_serial_query.cpp"],
+    linkstatic = True,
+    deps = [
+        ":libcreate",
+        "@googletest//:gtest_main",
+    ],
+)
+
+cc_test(
+    name = "serial_stream_test",
+    srcs = ["tests/test_serial_stream.cpp"],
+    linkstatic = True,
+    deps = [
+        ":libcreate",
+        "@googletest//:gtest_main",
+    ],
+)
