diff --git a/include/create/serial.h b/include/create/serial.h
index ccfdaea..3af4901 100644
--- a/include/create/serial.h
+++ b/include/create/serial.h
@@ -51,7 +51,7 @@ namespace create {
   class Serial : public std::enable_shared_from_this<Serial> {
 
     protected:
-      boost::asio::io_service io;
+      boost::asio::io_context io;
       boost::asio::signal_set signals;
       boost::asio::serial_port port;
 
diff --git a/src/serial.cpp b/src/serial.cpp
index 6617e50..b1576d4 100644
--- a/src/serial.cpp
+++ b/src/serial.cpp
@@ -119,7 +119,7 @@ namespace create {
 
     if (!startSensorStream()) return false;
 
-    io.reset();
+    io.restart();
 
     // Start continuously reading one byte at a time
     boost::asio::async_read(port,
@@ -130,8 +130,8 @@ namespace create {
                                       std::placeholders::_2));
 
     ioThread = std::thread(std::bind(
-        static_cast<std::size_t(boost::asio::io_service::*)(void)>(
-          &boost::asio::io_service::run), &io));
+        static_cast<std::size_t(boost::asio::io_context::*)(void)>(
+          &boost::asio::io_context::run), &io));
 
     // Wait for first complete read to finish
     std::unique_lock<std::mutex> lock(dataReadyMut);
