--- /dev/null
+++ WORKSPACE.bzlmod
@@ -0,0 +1,14 @@
+workspace(name = "libcreate")
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "com_github_nelhage_rules_boost",
+    integrity = "sha256-bBVULQ07b0fNTSoWMqMFSgUCtRAuRxtEUzVDEstHRAc=",
+    strip_prefix = "rules_boost-5d04542e52164931841d06d5a6b3fd2f43c436d0",  # Right before the support for bzlmod.
+    url = "https://github.com/nelhage/rules_boost/archive/5d04542e52164931841d06d5a6b3fd2f43c436d0.tar.gz",
+)
+
+load("@com_github_nelhage_rules_boost//:boost/boost.bzl", "boost_deps")
+
+boost_deps()
