diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..bf2486f
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,36 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "spirit",
+    deps = [
+        "@boost.array//:array",
+        "@boost.assert//:assert",
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.endian//:endian",
+        "@boost.function//:function",
+        "@boost.function_types//:function_types",
+        "@boost.fusion//:fusion",
+        "@boost.integer//:integer",
+        "@boost.io//:io",
+        "@boost.iterator//:iterator",
+        "@boost.move//:move",
+        "@boost.mpl//:mpl",
+        "@boost.optional//:optional",
+        "@boost.phoenix//:phoenix",
+        "@boost.pool//:pool",
+        "@boost.preprocessor//:preprocessor",
+        "@boost.proto//:proto",
+        "@boost.range//:range",
+        "@boost.regex//:regex",
+        "@boost.smart_ptr//:smart_ptr",
+        "@boost.static_assert//:static_assert",
+        "@boost.thread//:thread",
+        "@boost.throw_exception//:throw_exception",
+        "@boost.type_traits//:type_traits",
+        "@boost.typeof//:typeof",
+        "@boost.unordered//:unordered",
+        "@boost.utility//:utility",
+        "@boost.variant//:variant",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..68a5748
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,37 @@
+module(
+    name = "boost.spirit",
+    version = "1.83.0.bzl.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.array", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.endian", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.function", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.function_types", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.fusion", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.integer", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.io", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.iterator", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.move", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.mpl", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.optional", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.phoenix", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.pool", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.preprocessor", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.proto", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.range", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.regex", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.smart_ptr", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.thread", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.throw_exception", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.typeof", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.unordered", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.utility", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.variant", version = "1.83.0.bzl.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
