--- MODULE.bazel
+++ MODULE.bazel
@@ -2,7 +2,7 @@
 
 module(
     name = "hugooole_muda",
-    version = "1.0.0-beta",
+    version = "1.0.0",
     compatibility_level = 1,
 )
 
@@ -12,31 +12,3 @@
 
 bazel_dep(name = "eigen", version = "3.4.1", dev_dependency = True)
 bazel_dep(name = "bazel_skylib", version = "1.8.2")
-
-# rules_cuda is required for CUDA toolchain support
-bazel_dep(name = "rules_cuda", version = "0.2.4")
-git_override(
-    module_name = "rules_cuda",
-    remote = "https://github.com/bazel-contrib/rules_cuda.git",
-    commit = "1bf21ef1eee98042c0b2c5f39365ac39a4217c53",
-)
-
-
-
-cuda = use_extension("@rules_cuda//cuda:extensions.bzl", "toolchain")
-cuda.toolkit(
-    name = "cuda",
-    toolkit_path = "",
-)
-use_repo(cuda, "cuda")
-
-# Optional for debug: Hedron's Compile Commands Extractor for Bazel
-# https://github.com/hedronvision/bazel-compile-commands-extractor
-# bazel_dep(name = "hedron_compile_commands", dev_dependency = True)
-# git_override(
-#     module_name = "hedron_compile_commands",
-#     remote = "https://github.com/hedronvision/bazel-compile-commands-extractor.git",
-#     commit = "abb61a688167623088f8768cc9264798df6a9d10",
-#     # Replace the commit hash (above) with the latest (https://github.com/hedronvision/bazel-compile-commands-extractor/commits/main).
-#     # Even better, set up Renovate and let it do the work for you (see "Suggestion: Updates" in the README).
-# )
