diff --git a/BUILD b/BUILD
index 7e7df80..c34cc31 100644
--- a/BUILD
+++ b/BUILD
@@ -2,7 +2,7 @@ load("@aspect_bazel_lib//lib:jq.bzl", "jq")
 load("//:macros.bzl", "json_data")
 load("//:versions.bzl", "VERSIONS")
 
-exports_files(["jq.sh"])
+exports_files(["jq.sh", "versions.bzl", "macros.bzl", "packages.bzl"])
 
 json_data(
     name = "deps",
@@ -11,6 +11,7 @@ json_data(
 
 jq(
     name = "dependency_versions",
+    visibility = ["//visibility:public"],
     srcs = [":deps"],
     out = "dependency_shas.json",
     filter = """
diff --git a/compile/BUILD b/compile/BUILD
index 8be033d..efb1ac1 100644
--- a/compile/BUILD
+++ b/compile/BUILD
@@ -3,6 +3,8 @@ load("//:versions.bzl", "VERSIONS")
 
 package(default_visibility = ["//visibility:public"])
 
+exports_files(["sanitizer_libs.bzl"])
+
 SANITIZER_ENV = {
     "CXXFLAGS": "-nostdinc++ -nostdlib++",
     "LDFLAGS": "-nostdlib++",
diff --git a/coverage/grcov/BUILD b/coverage/grcov/BUILD
index 779d169..df56d7a 100644
--- a/coverage/grcov/BUILD
+++ b/coverage/grcov/BUILD
@@ -1 +1,3 @@
 licenses(["notice"])  # Apache 2
+
+exports_files(["grcov_repository.bzl"])
diff --git a/dependency/BUILD b/dependency/BUILD
index 7a7a6a9..d5947c9 100644
--- a/dependency/BUILD
+++ b/dependency/BUILD
@@ -2,6 +2,7 @@ load("//dependency:macros.bzl", "updater")
 
 exports_files([
     "bazel-update.sh",
+    "macros.bzl",
 ])
 
 updater(
diff --git a/repository/BUILD b/repository/BUILD
index e69de29..fbe9d5f 100644
--- a/repository/BUILD
+++ b/repository/BUILD
@@ -0,0 +1 @@
+exports_files(["utils.bzl"])
