diff --git ncurses/tinfo/MKcaptab.sh ncurses/tinfo/MKcaptab.sh
--- ncurses/tinfo/MKcaptab.sh
+++ ncurses/tinfo/MKcaptab.sh
@@ -29,6 +29,8 @@
 ##############################################################################
 # $Id: MKcaptab.sh,v 1.19 2020/02/02 23:34:34 tom Exp $
 
+set -e
+
 if test $# != 0
 then
 	AWK="$1"; shift 1
diff --git ncurses/tinfo/MKfallback.sh ncurses/tinfo/MKfallback.sh
--- ncurses/tinfo/MKfallback.sh
+++ ncurses/tinfo/MKfallback.sh
@@ -37,6 +37,8 @@
 # specified list of types generated in.
 #
 
+set -e
+
 terminfo_dir=$1
 shift
 
diff --git ncurses/tinfo/MKkeys_list.sh ncurses/tinfo/MKkeys_list.sh
--- ncurses/tinfo/MKkeys_list.sh
+++ ncurses/tinfo/MKkeys_list.sh
@@ -35,6 +35,9 @@
 #
 # Extract function-key names from the Caps file
 #
+
+set -e
+
 : ${AWK-awk}
 if test $# != 0
 then
diff --git ncurses/tinfo/MKuserdefs.sh ncurses/tinfo/MKuserdefs.sh
--- ncurses/tinfo/MKuserdefs.sh
+++ ncurses/tinfo/MKuserdefs.sh
@@ -27,6 +27,9 @@
 # authorization.                                                             #
 ##############################################################################
 # $Id: MKuserdefs.sh,v 1.10 2020/02/02 23:34:34 tom Exp $
+
+set -e
+
 AWK=${1-awk}; shift 1
 OPT1=${1-0}; shift 1
 
