--- /dev/null	2023-06-26 14:05:31
+++ BUILD.bazel	2023-06-26 14:05:28
@@ -0,0 +1,33 @@
+_COPTS = select({
+        "@bazel_tools//src/conditions:windows": [],
+        "//conditions:default": [
+            "-O3", "-Wall", "-Wextra", "-Wno-unknown-pragmas", "-Wcast-qual"
+        ],
+})
+
+cc_library(
+    name = "try",
+    srcs = ["try.c"],
+    hdrs = ["try.h"],
+    copts = _COPTS,
+)
+
+cc_library(
+    name = "yarn",
+    srcs = ["yarn.c"],
+    hdrs = ["yarn.h"],
+    copts = _COPTS,
+)
+
+cc_library(
+    name = "pigz",
+    srcs = ["pigz.c"],
+    deps = [":try", ":yarn", "@zopfli", "@zlib"],
+    copts = _COPTS,
+)
+
+cc_binary(
+    name = "bin",
+    deps = [":pigz"],
+    linkopts = ["-lm", "-lpthread"],   
+)
