From d37fa4fb115632866ee9ec7416cdd2dbf6a7ef18 Mon Sep 17 00:00:00 2001
From: Patrick Balestra <me@patrickbalestra.com>
Date: Tue, 1 Nov 2022 12:03:24 +0100
Subject: [PATCH] Add MODULE.bazel (#1742)
---
 MODULE.bazel | 10 ++++++++++
 1 file changed, 10 insertions(+)
 create mode 100644 MODULE.bazel

diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 00000000..a0584b4e
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,10 @@
+module(
+    name = "rules_apple",
+    compatibility_level = 1,
+    repo_name = "build_bazel_rules_apple",
+    version = "1.1.2",
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.1.1")
+bazel_dep(name = "apple_support", repo_name = "build_bazel_apple_support", version = "1.3.1")
+bazel_dep(name = "rules_swift", repo_name = "build_bazel_rules_swift", version = "1.2.0")
-- 
2.37.0 (Apple Git-136)

