diff --git a/folly/debugging/symbolizer/DwarfImpl.cpp b/folly/debugging/symbolizer/DwarfImpl.cpp
index ac4eef961..a0b22435b 100644
--- a/folly/debugging/symbolizer/DwarfImpl.cpp
+++ b/folly/debugging/symbolizer/DwarfImpl.cpp
@@ -27,7 +27,7 @@
 
 #if FOLLY_HAVE_DWARF && FOLLY_HAVE_ELF
 
-#include <dwarf.h> // @manual=fbsource//third-party/libdwarf:dwarf
+#include <libdwarf/dwarf.h> // @manual=fbsource//third-party/libdwarf:dwarf
 
 // We need a single dwarf5 tag, but may not be building against
 // a new enough libdwarf, so just define it ourselves.
