diff --git a/folly/compression/Compression.cpp b/folly/compression/Compression.cpp
index 463c103ac..f4c5da408 100644
--- a/folly/compression/Compression.cpp
+++ b/folly/compression/Compression.cpp
@@ -27,8 +27,8 @@
 #include <glog/logging.h>
 
 #if FOLLY_HAVE_LIBSNAPPY
-#include <snappy-sinksource.h>
-#include <snappy.h>
+#include "snappy-sinksource.h"
+#include "snappy.h"
 #endif
 
 #if FOLLY_HAVE_LIBZ
