--- MODULE.bazel
+++ MODULE.bazel
@@ -0,0 +1,14 @@
+module(
+    name = "rules_pkg",
+    version = "0.8.1",  # Must sync with version.bzl.
+    repo_name = "rules_pkg",
+)
+
+# Do not update to newer versions until you need a specific new feature.
+bazel_dep(name = "rules_license", version = "0.0.3")
+bazel_dep(name = "bazel_skylib", version = "1.2.0")
+bazel_dep(name = "rules_python", version = "0.10.2")
+
+# Only for development
+bazel_dep(name = "platforms", version = "0.0.5", dev_dependency = True)
+bazel_dep(name = "stardoc", version = "0.5.3", dev_dependency = True)
