diff --git BUILD.bazel BUILD.bazel
new file mode 100644
index 00000000..d99b09e7
--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,22 @@
+load("@rules_license//rules:license.bzl", "license")
+
+package(
+    default_applicable_licenses = [":license"],
+    default_visibility = ["//visibility:public"],
+)
+
+exports_files(["LICENSE"])
+
+license(
+    name = "license",
+    package_name = "cppregpattern",
+    license_kinds = [
+        "@rules_license//licenses/spdx:MIT",
+    ],
+)
+
+cc_library(
+    name = "registry",
+    hdrs = ["include/cppregpattern/registry.h"],
+    strip_include_prefix = "include/",
+)
\ No newline at end of file
