--- a/MODULE.bazel	2026-05-05 07:49:24.974114676 +0200
+++ b/MODULE.bazel	2026-05-05 07:49:24.974538615 +0200
@@ -14,9 +14,7 @@
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
 
 python.defaults(
-    # Use python.defaults if you have defined multiple toolchain versions.
     python_version = "3.9",
-    python_version_env = "BAZEL_PYTHON_VERSION",
 )
 
 python.toolchain(
@@ -27,7 +25,7 @@
 use_repo(python, "python_versions")
 
 bazel_dep(name = "googletest", version = "1.13.0")
-bazel_dep(name = "trlc", version = "0.0.0")
+bazel_dep(name = "trlc", version = "2.0.4")
 
 pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
 
@@ -48,9 +46,3 @@
 
 bazel_dep(name = "gazelle", version = "0.44.0", repo_name = "bazel_gazelle")
 use_repo(python, "python_3_9")
-
-git_override(
-    commit = "650b51a47264a4f232b3341f473527710fc32669",
-    module_name = "trlc",
-    remote = "https://github.com/bmw-software-engineering/trlc.git",
-)
