diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..c00792f
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,69 @@
+load("@bazel_skylib//lib:selects.bzl", "selects")
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "thread",
+    srcs = selects.with_or({
+        ("@platforms//os:linux", "@platforms//os:android", "@platforms//os:osx", "@platforms//os:ios", "@platforms//os:watchos", "@platforms//os:tvos"): [
+            "src/pthread/once.cpp",
+            "src/pthread/thread.cpp",
+        ],
+        "@boost.rules.tools//:windows_x86_64": [
+            "src/win32/thread.cpp",
+            "src/win32/tss_dll.cpp",
+            "src/win32/tss_pe.cpp",
+        ],
+    }),
+    hdrs = selects.with_or({
+        ("@platforms//os:linux", "@platforms//os:android", "@platforms//os:osx", "@platforms//os:ios", "@platforms//os:watchos", "@platforms//os:tvos"): [
+            "src/pthread/once_atomic.cpp",
+        ],
+        "@boost.rules.tools//:windows_x86_64": [],
+    }),
+    linkopts = selects.with_or({
+        ("@platforms//os:linux", "@platforms//os:osx", "@platforms//os:ios", "@platforms//os:watchos", "@platforms//os:tvos"): ["-lpthread"],
+        "@boost.rules.tools//:windows_x86_64": [],
+        "@platforms//os:android": [],
+    }),
+    local_defines = selects.with_or({
+        ("@platforms//os:linux", "@platforms//os:osx", "@platforms//os:ios", "@platforms//os:watchos", "@platforms//os:tvos"): [],
+        "@boost.rules.tools//:windows_x86_64": [
+            "BOOST_ALL_NO_LIB",
+            "BOOST_THREAD_BUILD_LIB",
+            "BOOST_THREAD_USE_LIB",
+            "BOOST_WIN32_THREAD",
+        ],
+        "//conditions:default": [],
+    }),
+    deps = [
+        "@boost.algorithm//:algorithm",
+        "@boost.assert//:assert",
+        "@boost.atomic//:atomic",
+        "@boost.bind//:bind",
+        "@boost.chrono//:chrono",
+        "@boost.concept_check//:concept_check",
+        "@boost.config//:config",
+        "@boost.container//:container",
+        "@boost.container_hash//:container_hash",
+        "@boost.core//:core",
+        "@boost.date_time//:date_time",
+        "@boost.exception//:exception",
+        "@boost.function//:function",
+        "@boost.intrusive//:intrusive",
+        "@boost.io//:io",
+        "@boost.iterator//:iterator",
+        "@boost.lexical_cast//:lexical_cast",
+        "@boost.move//:move",
+        "@boost.optional//:optional",
+        "@boost.predef//:predef",
+        "@boost.preprocessor//:preprocessor",
+        "@boost.smart_ptr//:smart_ptr",
+        "@boost.static_assert//:static_assert",
+        "@boost.system//:system",
+        "@boost.throw_exception//:throw_exception",
+        "@boost.tuple//:tuple",
+        "@boost.type_traits//:type_traits",
+        "@boost.utility//:utility",
+        "@boost.winapi//:winapi",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..66fe632
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,38 @@
+module(
+    name = "boost.thread",
+    version = "1.83.0.bzl.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "bazel_skylib", version = "1.5.0")
+bazel_dep(name = "boost.algorithm", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.atomic", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.bind", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.chrono", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.concept_check", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.container", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.container_hash", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.date_time", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.exception", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.function", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.intrusive", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.io", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.iterator", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.lexical_cast", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.move", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.optional", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.predef", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.preprocessor", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.smart_ptr", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.system", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.throw_exception", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.tuple", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.utility", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.winapi", version = "1.83.0.bzl.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
