--- /dev/null
+++ a/BUILD.bazel
@@ -0,0 +1,43 @@
+# Copyright 2010-2024 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cc_library(
+    name = "bliss",
+    srcs = [
+        "bliss_C.cc",
+        "defs.cc",
+        "graph.cc",
+        "heap.cc",
+        "orbit.cc",
+        "partition.cc",
+        "timer.cc",
+        "uintseqhash.cc",
+        "utils.cc",
+    ],
+    hdrs = [
+        "bignum.hh",
+        "bliss_C.h",
+        "defs.hh",
+        "graph.hh",
+        "heap.hh",
+        "kqueue.hh",
+        "kstack.hh",
+        "orbit.hh",
+        "partition.hh",
+        "timer.hh",
+        "uintseqhash.hh",
+        "utils.hh",
+    ],
+    includes = ["."],
+    visibility = ["//visibility:public"],
+)
