diff --git libmpdec/mpdecimal.h.in libmpdec/mpdecimal.h
similarity index 97%
rename from libmpdec/mpdecimal.h.in
rename to libmpdec/mpdecimal.h
index b2d9072..e5b1461 100644
--- libmpdec/mpdecimal.h.in
+++ libmpdec/mpdecimal.h
@@ -79,7 +79,48 @@ const char *mpd_version(void);
 /******************************************************************************/
 /*                              Configuration                                 */
 /******************************************************************************/
-@MPD_HEADER_CONFIG@
+/* Mac OS X: support for building universal binaries */
+#if defined(MPD_CONFIG_64) || defined(MPD_CONFIG_32)
+#error "cannot use MPD_CONFIG_64 or MPD_CONFIG_32 with universal header."
+#endif
+
+#if defined(CONFIG_64) || defined(CONFIG_32)
+#error "cannot use CONFIG_64 or CONFIG_32 with universal header."
+#endif
+
+#if defined(__ppc__)
+  #define MPD_CONFIG_32 1
+  #ifdef UNIVERSAL
+    #define CONFIG_32
+    #define ANSI
+  #endif
+#elif defined(__ppc64__)
+  #define MPD_CONFIG_64 1
+  #ifdef UNIVERSAL
+    #define CONFIG_64
+    #define ANSI
+  #endif
+#elif defined(__i386__)
+  #define MPD_CONFIG_32 1
+  #ifdef UNIVERSAL
+    #define CONFIG_32
+    #define ANSI
+  #endif
+#elif defined(__x86_64__)
+  #define MPD_CONFIG_64 1
+  #ifdef UNIVERSAL
+    #define CONFIG_64
+    #define ASM
+  #endif
+#elif defined(__arm64__)
+  #define MPD_CONFIG_64 1
+  #ifdef UNIVERSAL
+    #define CONFIG_64
+    #define ANSI
+  #endif
+#else
+#error "unknown architecture for universal build."
+#endif
 
 
 /* BEGIN MPD_CONFIG_64 */
