diff --git a/test/apple_support_test.bzl b/test/apple_support_test.bzl
index 9efc5cc..aabb373 100644
--- a/test/apple_support_test.bzl
+++ b/test/apple_support_test.bzl
@@ -14,7 +14,7 @@
 """Definition of a test rule to test apple_support."""
 
 load(
-    "@build_bazel_apple_support//lib:apple_support.bzl",
+    "//lib:apple_support.bzl",
     "apple_support",
 )
 
diff --git a/test/xcode_support_test.bzl b/test/xcode_support_test.bzl
index 27deeba..e10a949 100644
--- a/test/xcode_support_test.bzl
+++ b/test/xcode_support_test.bzl
@@ -14,11 +14,11 @@
 """Definition of a test rule to test xcode_support."""
 
 load(
-    "@build_bazel_apple_support//lib:apple_support.bzl",
+    "//lib:apple_support.bzl",
     "apple_support",
 )
 load(
-    "@build_bazel_apple_support//lib:xcode_support.bzl",
+    "//lib:xcode_support.bzl",
     "xcode_support",
 )
 
