--- a/src/utilities/utility.h	2026-04-14 17:02:02.584327944 +0000
+++ b/src/utilities/utility.h	2026-04-14 17:02:02.587074414 +0000
@@ -57,7 +57,7 @@
 
 namespace KDL {
 
-#ifdef __GNUC__
+#if defined(__GNUC__) || defined(_MSC_VER)
     // so that sin,cos can be overloaded and complete 
     // resolution of overloaded functions work.
     using ::sin;
@@ -78,7 +78,7 @@
     using ::tanh;
     using ::atan2;
 #endif
-#ifndef __GNUC__
+#if !defined(__GNUC__) && !defined(_MSC_VER)
     //only real solution : get Rall1d and varia out of namespaces.
     #pragma warning (disable:4786)
 
