--- a/src/main/java/com/github/luben/zstd/util/Native.java
+++ b/src/main/java/com/github/luben/zstd/util/Native.java
@@ -97,7 +97,7 @@ public enum Native {
         if (loaded.get()) {
             return;
         }
-        String resourceName = resourceName();
+        String resourceName = "/libzstd-jni.so";
 
         String overridePath = System.getProperty(nativePathOverride);
         if (overridePath != null) {
