--- MODULE.bazel
+++ MODULE.bazel
@@ -1,7 +1,7 @@
 """muda"""
 
 module(
-    name = "muda",
+    name = "hugooole_muda",
     version = "1.0.0-beta",
     compatibility_level = 1,
 )
@@ -26,7 +26,7 @@
 cuda = use_extension("@rules_cuda//cuda:extensions.bzl", "toolchain")
 cuda.toolkit(
     name = "cuda",
-    toolkit_path = "/usr/local/cuda/",
+    toolkit_path = "",
 )
 use_repo(cuda, "cuda")
 
