diff --git src/numpy/dtype.cpp src/numpy/dtype.cpp
--- src/numpy/dtype.cpp
+++ src/numpy/dtype.cpp
@@ -98,7 +98,7 @@
   return python::detail::new_reference(reinterpret_cast<PyObject*>(obj));
 }
 
-int dtype::get_itemsize() const { return reinterpret_cast<PyArray_Descr*>(ptr())->elsize;}
+int dtype::get_itemsize() const { return static_cast<int>(PyDataType_ELSIZE(reinterpret_cast<PyArray_Descr*>(ptr())));}
 
 bool equivalent(dtype const & a, dtype const & b) {
     // On Windows x64, the behaviour described on 
