--- BUILD.bazel
+++ BUILD.bazel
@@ -0,0 +1,18 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["**"]),
+)
+
+cmake(
+    name = "foonathan_memory",
+    cache_entries = {
+        "CMAKE_C_FLAGS": "-fPIC",
+        "FOONATHAN_MEMORY_BUILD_TESTS": "OFF",
+        "FOONATHAN_MEMORY_BUILD_EXAMPLES": "OFF",
+        "FOONATHAN_MEMORY_BUILD_TOOLS": "OFF"
+    },
+    lib_source = ":all_srcs",
+    out_static_libs = ["libfoonathan_memory-0.7.2.a"],
+)
