diff --git MODULE.bazel MODULE.bazel
new file mode 100644
index 0000000..9a9368e
--- /dev/null
+++ MODULE.bazel
@@ -0,0 +1,40 @@
+module(
+    name = "rules_antlr",
+    version = "0.5.0.bcr.1",
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "rules_java", version = "7.5.0")
+bazel_dep(name = "stardoc", version = "0.7.0", repo_name = "io_bazel_stardoc")
+
+antlr = use_extension("//antlr:extensions.bzl", "antlr")
+antlr.download(version = "4.8")
+use_repo(
+    antlr,
+    "antlr3_runtime",
+    "antlr4_runtime",
+    "antlr4_tool",
+    "javax_json",
+    "stringtemplate4",
+)
+
+http_jar = use_repo_rule("@bazel_tools//tools/build_defs/repo:http.bzl", "http_jar")
+
+http_jar(
+    name = "junit",
+    sha256 = "59721f0805e223d84b90677887d9ff567dc534d7c502ca903c0c2b17f05c116a",
+    url = "https://jcenter.bintray.com/junit/junit/4.12/junit-4.12.jar",
+)
+
+http_jar(
+    name = "jimfs",
+    sha256 = "c4828e28d7c0a930af9387510b3bada7daa5c04d7c25a75c7b8b081f1c257ddd",
+    url = "https://jcenter.bintray.com/com/google/jimfs/jimfs/1.1/jimfs-1.1.jar",
+)
+
+http_jar(
+    name = "guava",
+    sha256 = "4a5aa70cc968a4d137e599ad37553e5cfeed2265e8c193476d7119036c536fe7",
+    url = "https://jcenter.bintray.com/com/google/guava/guava/27.1-jre/guava-27.1-jre.jar",
+)
+
diff --git antlr/extensions.bzl antlr/extensions.bzl
new file mode 100644
index 0000000..3151e01
--- /dev/null
+++ antlr/extensions.bzl
@@ -0,0 +1,17 @@
+load("//antlr:repositories.bzl", "rules_antlr_dependencies")
+
+download = tag_class(attrs = {"version": attr.string()})
+
+def _antlr(module_ctx):
+    rules_antlr_dependencies(
+        max([
+            ([int(part) for part in download.version.split(".")], download.version)
+            for mod in module_ctx.modules
+            for download in mod.tags.download
+        ])[1],
+    )
+
+antlr = module_extension(
+    implementation = _antlr,
+    tag_classes = {"download": download},
+)
diff --git examples/MODULE.bazel examples/MODULE.bazel
new file mode 100644
index 0000000..a05ec91
--- /dev/null
+++ examples/MODULE.bazel
@@ -0,0 +1,9 @@
+module(name = "examples")
+
+bazel_dep(name = "rules_antlr")
+
+local_path_override(
+    module_name = "rules_antlr",
+    path = "..",
+)
+

