diff --git antlr/BUILD antlr/BUILD
index 2d6696f..248cb80 100644
--- antlr/BUILD
+++ antlr/BUILD
@@ -19,9 +19,7 @@ bzl_library(
 stardoc(
     name = "antlr4",
     out = "antlr4.md",
-    func_template = "@stardoc_templates//:func.vm",
     input = "antlr4.bzl",
-    rule_template = "@stardoc_templates//:rule.vm",
     deps = [":shared"],
 )
 
@@ -29,7 +27,6 @@ stardoc(
     name = "antlr3",
     out = "antlr3.md",
     input = "antlr3.bzl",
-    rule_template = "@stardoc_templates//:rule.vm",
     deps = [":shared"],
 )
 
@@ -37,6 +34,5 @@ stardoc(
     name = "antlr2",
     out = "antlr2.md",
     input = "antlr2.bzl",
-    rule_template = "@stardoc_templates//:rule.vm",
     deps = [":shared"],
 )

