diff --git smoke_test.sh smoke_test.sh
new file mode 100755
index 00000000..cca2bdfd
--- /dev/null
+++ smoke_test.sh
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -xeu
+
+cd "$(dirname "$TEST_BINARY")"
+./rsyslogd -i ./pidfile -f ./test.conf -C
+
+trap "kill $(cat ./pidfile)" EXIT
+
+MSG="An application event log entr"
+echo '<165>1 2003-10-11T22:14:15.003002Z mymachine.example.com evntslog - ID47 [exampleSDID@32473 iut="3" eventSource="Application" eventID="1011"]' "$MSG" |
+ nc -u -w 0 127.0.0.1 10514
+
+diff ./logfile <(echo '2003-10-11T22:14:15.003002Z mymachine.example.com evntslog An application event log entr')
