--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,28 @@
+""" Builds zstd.
+"""
+
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "all_srcs",
+    # TODO(mvukov) Minimize srcs.
+    srcs = glob(["**"]),
+)
+
+cmake(
+    name = "zstd",
+    build_args = [
+        "--",
+        "-j4",
+    ],
+    cache_entries = {
+        "ZSTD_BUILD_PROGRAMS": "OFF",
+        "ZSTD_BUILD_SHARED": "OFF",
+        "ZSTD_BUILD_STATIC": "ON",
+    },
+    generate_args = ["-GNinja"],
+    lib_source = ":all_srcs",
+    out_static_libs = ["libzstd.a"],
+    visibility = ["//visibility:public"],
+    working_directory = "build/cmake",
+)
