diff --git a/opencensus/proto/agent/common/v1/BUILD.bazel b/opencensus/proto/agent/common/v1/BUILD.bazel
index d0a0e08..bd1d872 100644
--- a/opencensus/proto/agent/common/v1/BUILD.bazel
+++ b/opencensus/proto/agent/common/v1/BUILD.bazel
@@ -31,7 +31,7 @@ go_proto_library(
     importpath = "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1",
     proto = ":common_proto",
     deps = [
-        "@com_github_golang_protobuf//ptypes/timestamp:go_default_library",
+        "@org_golang_google_protobuf//types/known/timestamppb:go_default_library",
     ],
 )
 
diff --git a/opencensus/proto/metrics/v1/BUILD.bazel b/opencensus/proto/metrics/v1/BUILD.bazel
index 6e8afdc..fd1a951 100644
--- a/opencensus/proto/metrics/v1/BUILD.bazel
+++ b/opencensus/proto/metrics/v1/BUILD.bazel
@@ -44,7 +44,7 @@ go_proto_library(
     proto = ":metrics_proto",
     deps = [
         "//opencensus/proto/resource/v1:resource_proto_go",
-        "@com_github_golang_protobuf//ptypes/timestamp:go_default_library",
-        "@com_github_golang_protobuf//ptypes/wrappers:go_default_library",
+        "@org_golang_google_protobuf//types/known/timestamppb:go_default_library",
+        "@org_golang_google_protobuf//types/known/wrapperspb:go_default_library",
     ],
 )
diff --git a/opencensus/proto/stats/v1/BUILD.bazel b/opencensus/proto/stats/v1/BUILD.bazel
index 5ab569d..2d18cad 100644
--- a/opencensus/proto/stats/v1/BUILD.bazel
+++ b/opencensus/proto/stats/v1/BUILD.bazel
@@ -42,6 +42,6 @@ go_proto_library(
     importpath = "github.com/census-instrumentation/opencensus-proto/gen-go/stats/v1",
     proto = ":stats_proto",
     deps = [
-        "@com_github_golang_protobuf//ptypes/timestamp:go_default_library",
+        "@org_golang_google_protobuf//types/known/timestamppb:go_default_library",
     ],
 )
diff --git a/opencensus/proto/trace/v1/BUILD.bazel b/opencensus/proto/trace/v1/BUILD.bazel
index 0beb6d8..9aa1ad1 100644
--- a/opencensus/proto/trace/v1/BUILD.bazel
+++ b/opencensus/proto/trace/v1/BUILD.bazel
@@ -61,8 +61,8 @@ go_proto_library(
     proto = ":trace_proto",
     deps = [
         "//opencensus/proto/resource/v1:resource_proto_go",
-        "@com_github_golang_protobuf//ptypes/timestamp:go_default_library",
-        "@com_github_golang_protobuf//ptypes/wrappers:go_default_library",
+        "@org_golang_google_protobuf//types/known/timestamppb:go_default_library",
+        "@org_golang_google_protobuf//types/known/wrapperspb:go_default_library",
     ],
 )
 
@@ -77,7 +77,6 @@ py_proto_library(
     srcs = ["trace.proto"],
     deps = [
         "//opencensus/proto/resource/v1:resource_proto_py",
-        "@com_google_protobuf//:protobuf_python",
     ],
 )
