--- MODULE.bazel
+++ MODULE.bazel
@@ -1,7 +1,7 @@
-## MODULE.bazel
 module(
     name = "gz-fuel-tools",
-    repo_name = "org_gazebosim_gz-fuel-tools",
+    version = "11.0.0",
+    compatibility_level = 10,
 )
 
 bazel_dep(name = "curl", version = "8.8.0.bcr.3")
@@ -15,31 +15,7 @@
 
 # Gazebo Dependencies
 bazel_dep(name = "rules_gazebo", version = "0.0.3")
-bazel_dep(name = "gz-common")
-bazel_dep(name = "gz-math")
-bazel_dep(name = "gz-msgs")
-bazel_dep(name = "gz-utils")
-
-archive_override(
-    module_name = "gz-common",
-    strip_prefix = "gz-common-gz-common7",
-    urls = ["https://github.com/gazebosim/gz-common/archive/refs/heads/gz-common7.tar.gz"],
-)
-
-archive_override(
-    module_name = "gz-math",
-    strip_prefix = "gz-math-gz-math9",
-    urls = ["https://github.com/gazebosim/gz-math/archive/refs/heads/gz-math9.tar.gz"],
-)
-
-archive_override(
-    module_name = "gz-msgs",
-    strip_prefix = "gz-msgs-gz-msgs12",
-    urls = ["https://github.com/gazebosim/gz-msgs/archive/refs/heads/gz-msgs12.tar.gz"],
-)
-
-archive_override(
-    module_name = "gz-utils",
-    strip_prefix = "gz-utils-gz-utils4",
-    urls = ["https://github.com/gazebosim/gz-utils/archive/refs/heads/gz-utils4.tar.gz"],
-)
+bazel_dep(name = "gz-common", version = "7.0.0")
+bazel_dep(name = "gz-math", version = "9.0.0")
+bazel_dep(name = "gz-msgs", version = "12.0.0")
+bazel_dep(name = "gz-utils", version = "4.0.0")
