diff --git a/patches/0001-fix-bazel-9-compatibility-67.patch b/patches/0001-fix-bazel-9-compatibility-67.patch
new file mode 100644
index 0000000..85ab09f
--- /dev/null
+++ b/patches/0001-fix-bazel-9-compatibility-67.patch
@@ -0,0 +1,41 @@
+From c5e495c5907f57e8080be8eeb059982a94073442 Mon Sep 17 00:00:00 2001
+From: Mathieu Olivari <molivari@apple.com>
+Date: Sun, 3 May 2026 08:57:43 +0100
+Subject: [PATCH] fix: bazel 9 compatibility (#67)
+
+This fixes an error when the ruleset is used in a bazel 9 project.
+CcToolchainConfigInfo isn't defined in bazel 9 anymore. It must be
+imported from rules_cc.
+---
+ BUILD.bazel                  | 1 +
+ musl_cc_toolchain_config.bzl | 5 +++++
+ 2 files changed, 6 insertions(+)
+
+diff --git BUILD.bazel BUILD.bazel
+index 1fd40c4..22164f3 100644
+--- BUILD.bazel
++++ BUILD.bazel
+@@ -1,3 +1,4 @@
++load("@rules_cc//cc:defs.bzl", "cc_toolchain")
+ load("//:musl_cc_toolchain_config.bzl", "musl_cc_test_toolchain", "musl_cc_toolchain_config")
+ 
+ package(default_visibility = ["//visibility:public"])
+diff --git musl_cc_toolchain_config.bzl musl_cc_toolchain_config.bzl
+index 3c3d06c..413076e 100644
+--- musl_cc_toolchain_config.bzl
++++ musl_cc_toolchain_config.bzl
+@@ -25,6 +25,11 @@ load(
+     "tool_path",
+     "with_feature_set",
+ )
++load(
++    "@rules_cc//cc:defs.bzl",
++    "CcToolchainConfigInfo",
++    "cc_common",
++)
+ 
+ _CcTestInfo = provider(
+     doc = "Toolchain implementation for @bazel_tools//tools/cpp:test_runner_toolchain_type",
+-- 
+2.50.1 (Apple Git-155)
+
diff --git a/repositories.bzl b/repositories.bzl
index acdc63e..b6dca37 100644
--- a/repositories.bzl
+++ b/repositories.bzl
@@ -242,48 +242,56 @@ toolchain_repo = repository_rule(
 def load_musl_toolchains(extra_exec_compatible_with=[], extra_target_compatible_with=[], target_settings=[]):
     http_archive(
         name = "musl-1_2_3-platform-x86_64-unknown-linux-gnu-target-x86_64-linux-musl",
+        patches = ["patches/0001-fix-bazel-9-compatibility-67.patch"],
         sha256 = "51e59d21ea780c830c0e2a38a3a4278126bbd9ea6fbbaac421861c631ab3f122",
         url = "https://github.com/bazel-contrib/musl-toolchain/releases/download/v0.1.27/musl-1.2.3-platform-x86_64-unknown-linux-gnu-target-x86_64-linux-musl.tar.gz",
     )
 
     http_archive(
         name = "musl-1_2_3-platform-aarch64-unknown-linux-gnu-target-x86_64-linux-musl",
+        patches = ["patches/0001-fix-bazel-9-compatibility-67.patch"],
         sha256 = "c052e4b05653cf6f378c55bf600ddb02fa84f43a9bc4faba6764e870fa708627",
         url = "https://github.com/bazel-contrib/musl-toolchain/releases/download/v0.1.27/musl-1.2.3-platform-aarch64-unknown-linux-gnu-target-x86_64-linux-musl.tar.gz",
     )
 
     http_archive(
         name = "musl-1_2_3-platform-x86_64-apple-darwin-target-x86_64-linux-musl",
+        patches = ["patches/0001-fix-bazel-9-compatibility-67.patch"],
         sha256 = "748575629934f5c75091baffb2b93fa2a18bab0d9dc61c364936d209c464bc29",
         url = "https://github.com/bazel-contrib/musl-toolchain/releases/download/v0.1.27/musl-1.2.3-platform-x86_64-apple-darwin-target-x86_64-linux-musl.tar.gz",
     )
 
     http_archive(
         name = "musl-1_2_3-platform-aarch64-apple-darwin-target-x86_64-linux-musl",
+        patches = ["patches/0001-fix-bazel-9-compatibility-67.patch"],
         sha256 = "46d2384102425c99c1a7b03119055a39eb3e9743ac667a8c9617f503040901e9",
         url = "https://github.com/bazel-contrib/musl-toolchain/releases/download/v0.1.27/musl-1.2.3-platform-aarch64-apple-darwin-target-x86_64-linux-musl.tar.gz",
     )
 
     http_archive(
         name = "musl-1_2_3-platform-x86_64-unknown-linux-gnu-target-aarch64-linux-musl",
+        patches = ["patches/0001-fix-bazel-9-compatibility-67.patch"],
         sha256 = "709df48b5830d8426f4a74d5a6dad46d84f30345c17bb37d504d101509a66104",
         url = "https://github.com/bazel-contrib/musl-toolchain/releases/download/v0.1.27/musl-1.2.3-platform-x86_64-unknown-linux-gnu-target-aarch64-linux-musl.tar.gz",
     )
 
     http_archive(
         name = "musl-1_2_3-platform-aarch64-unknown-linux-gnu-target-aarch64-linux-musl",
+        patches = ["patches/0001-fix-bazel-9-compatibility-67.patch"],
         sha256 = "9a9d2d9af7e34e9740cd45249df1bdced19b003305ee4e7ed9f3eca96de68a0a",
         url = "https://github.com/bazel-contrib/musl-toolchain/releases/download/v0.1.27/musl-1.2.3-platform-aarch64-unknown-linux-gnu-target-aarch64-linux-musl.tar.gz",
     )
 
     http_archive(
         name = "musl-1_2_3-platform-x86_64-apple-darwin-target-aarch64-linux-musl",
+        patches = ["patches/0001-fix-bazel-9-compatibility-67.patch"],
         sha256 = "99d444759a52764b118359a2f97abcd1a98b7fd74afd068f6c9d7b6d7a4240ea",
         url = "https://github.com/bazel-contrib/musl-toolchain/releases/download/v0.1.27/musl-1.2.3-platform-x86_64-apple-darwin-target-aarch64-linux-musl.tar.gz",
     )
 
     http_archive(
         name = "musl-1_2_3-platform-aarch64-apple-darwin-target-aarch64-linux-musl",
+        patches = ["patches/0001-fix-bazel-9-compatibility-67.patch"],
         sha256 = "158300e91e672607b9521d80d653afb63e364d909d50f979d20ded684bb83e57",
         url = "https://github.com/bazel-contrib/musl-toolchain/releases/download/v0.1.27/musl-1.2.3-platform-aarch64-apple-darwin-target-aarch64-linux-musl.tar.gz",
     )
