diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..eaacc7d
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,17 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "multi_array",
+    deps = [
+        "@boost.array//:array",
+        "@boost.assert//:assert",
+        "@boost.concept_check//:concept_check",
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.functional//:functional",
+        "@boost.iterator//:iterator",
+        "@boost.mpl//:mpl",
+        "@boost.static_assert//:static_assert",
+        "@boost.type_traits//:type_traits",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..03c30e4
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,18 @@
+module(
+    name = "boost.multi_array",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.array", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.assert", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.concept_check", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.functional", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.iterator", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.mpl", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
