--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,120 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_swift//swift:swift.bzl", "swift_library")
+
+cc_library(
+    name = "CCryptoBoringSSL",
+    srcs = glob(
+        [
+            "Sources/CCryptoBoringSSL/**/*.cc",
+            "Sources/CCryptoBoringSSL/**/*.S",
+        ],
+        exclude = [
+            "Sources/CCryptoBoringSSL/crypto/bio/connect.cc",
+            "Sources/CCryptoBoringSSL/crypto/bio/socket_helper.cc",
+            "Sources/CCryptoBoringSSL/crypto/bio/socket.cc",
+        ],
+    ),
+    hdrs = glob([
+        "Sources/CCryptoBoringSSL/**/*.h",
+        "Sources/CCryptoBoringSSL/**/*.inc",
+    ]),
+    copts = [
+        "-std=c++17",
+    ],
+    includes = [
+        "Sources/CCryptoBoringSSL/include",
+    ],
+    linkstatic = True,
+)
+
+cc_library(
+    name = "CCryptoBoringSSLShims",
+    srcs = glob(["Sources/CCryptoBoringSSLShims/**/*.c"]),
+    hdrs = glob(["Sources/CCryptoBoringSSLShims/include/*.h"]),
+    includes = [
+        "Sources/CCryptoBoringSSLShims/include",
+    ],
+    linkstatic = True,
+    deps = [":CCryptoBoringSSL"],
+)
+
+cc_library(
+    name = "CXKCP",
+    srcs = glob([
+        "Sources/CXKCP/**/*.c",
+    ]),
+    hdrs = glob([
+        "Sources/CXKCP/**/*.h",
+        "Sources/CXKCP/**/*.macros",
+        "Sources/CXKCP/**/*.inc",
+    ]),
+    copts = [
+        "-DXKCP_has_KeccakP1600",
+    ],
+    includes = [
+        "Sources/CXKCP/include",
+        "Sources/CXKCP/FIPS202-opt64",
+    ],
+    linkstatic = True,
+)
+
+cc_library(
+    name = "CXKCPShims",
+    srcs = glob(["Sources/CXKCPShims/**/*.c"]),
+    hdrs = glob(["Sources/CXKCPShims/include/*.h"]),
+    includes = [
+        "Sources/CXKCPShims/include",
+    ],
+    linkstatic = True,
+    deps = [":CXKCP"],
+)
+
+swift_library(
+    name = "CryptoBoringWrapper",
+    srcs = glob(["Sources/CryptoBoringWrapper/**/*.swift"]),
+    module_name = "CryptoBoringWrapper",
+    package_name = "swift-crypto",
+    deps = [
+        ":CCryptoBoringSSL",
+        ":CCryptoBoringSSLShims",
+    ],
+)
+
+swift_library(
+    name = "Crypto",
+    srcs = glob(["Sources/Crypto/**/*.swift"]),
+    defines = [
+        "CRYPTO_IN_SWIFTPM",
+        "CRYPTO_IN_SWIFTPM_FORCE_BUILD_API",
+    ],
+    module_name = "Crypto",
+    package_name = "swift-crypto",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":CCryptoBoringSSL",
+        ":CCryptoBoringSSLShims",
+        ":CryptoBoringWrapper",
+        ":CXKCP",
+        ":CXKCPShims",
+    ],
+)
+
+swift_library(
+    name = "CryptoExtras",
+    srcs = glob(["Sources/CryptoExtras/**/*.swift"]),
+    defines = [
+        "CRYPTO_IN_SWIFTPM",
+        "CRYPTO_IN_SWIFTPM_FORCE_BUILD_API",
+    ],
+    module_name = "CryptoExtras",
+    package_name = "swift-crypto",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":CCryptoBoringSSL",
+        ":CCryptoBoringSSLShims",
+        ":Crypto",
+        ":CryptoBoringWrapper",
+        "@swift-asn1//:SwiftASN1",
+    ],
+)
+
