diff --git a/MODULE.bazel b/MODULE.bazel
index f05c57a..4cb6104 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -28,9 +28,9 @@ register_toolchains(
 # To allow /tools to be built from source
 # NOTE: when publishing to BCR, we patch this to be dev_dependency, as we publish pre-built binaries
 # along with our releases.
-bazel_dep(name = "rules_go", version = "0.41.0", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "rules_go", version = "0.41.0", repo_name = "io_bazel_rules_go", dev_dependency = True)

-go_deps = use_extension("@gazelle//:extensions.bzl", "go_deps")
+go_deps = use_extension("@gazelle//:extensions.bzl", "go_deps", dev_dependency = True)
 go_deps.from_file(go_mod = "//:go.mod")
 use_repo(
     go_deps,
