diff --git a/squashfs-tools/print_pager.c b/squashfs-tools/print_pager.c
index cfdf4ef0..0dafd93f 100644
--- a/squashfs-tools/print_pager.c
+++ b/squashfs-tools/print_pager.c
@@ -77,7 +77,7 @@ static char *get_base(char *pathname)
 }
 
 
-inline int quoted_bs_char(char cur)
+static inline int quoted_bs_char(char cur)
 {
 	/*
 	 * Within double quoted strings Bash allows the characters $, `, ",\ and
