diff --git a/.bazelrc b/.bazelrc
index 2eb88cb..13f4fc3 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -33,7 +33,7 @@ common:asan --test_tag_filters=-no_san  # skip san targets
 common:asan --define san=true  # skip san targets
 common:asan --features=asan  # https://github.com/bazelbuild/bazel/issues/6932
 
-common:clang --incompatible_enable_cc_toolchain_resolution
+common:clang --incompatible_enable_cc_toolchain_resolution --consistent_labels
 common:clang --//mbo/mope:clang_format=
 common:clang --cxxopt=-gmlt
 common:clang --host_cxxopt=-gmlt
