--- MODULE.bazel
+++ MODULE.bazel
@@ -1,12 +1,12 @@
 module(
     name = "rules_ytt",
-    version = "0.0.0",
+    version = "0.3.0",
     compatibility_level = 0,
 )
-
 bazel_dep(name = "platforms", version = "0.0.11")
 bazel_dep(name = "bazel_features", version = "1.19.0")
 bazel_dep(name = "aspect_bazel_lib", version = "2.13.0")
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
 
 bazel_dep(name = "stardoc", version = "0.8.0", dev_dependency = True)
 
