--- MODULE.bazel
+++ MODULE.bazel
@@ -1,16 +1,26 @@
 module(
-    name = "rules_python",
-    compatibility_level = 1,
-    version = "0.0.0",
+  name = "rules_python",
+  version = "0.12.0",
+  compatibility_level = 1,
 )
 
-pip_install = use_extension("@rules_python//python:extensions.bzl", "pip_install")
+pip_install = use_extension("//python:extensions.bzl", "pip_install")
 
-use_repo(
-    pip_install,
-    "pypi__click",
-    "pypi__pip",
-    "pypi__pip_tools",
-    "pypi__setuptools",
-    "pypi__wheel",
+use_repo(pip_install,
+  "pypi__build",
+  "pypi__click",
+  "pypi__colorama",
+  "pypi__importlib_metadata",
+  "pypi__installer",
+  "pypi__more_itertools",
+  "pypi__packaging",
+  "pypi__pep517",
+  "pypi__pip",
+  "pypi__pip_tools",
+  "pypi__pyparsing",
+  "pypi__setuptools",
+  "pypi__tomli",
+  "pypi__wheel",
+  "pypi__zipp",
 )
+register_toolchains("@bazel_tools//tools/python:autodetecting_toolchain")
