--- log/src/main/java/com/dylibso/chicory/log/Logger.java
+++ log/src/main/java/com/dylibso/chicory/log/Logger.java
@@ -1,6 +1,5 @@
 package com.dylibso.chicory.log;
 
-import com.google.errorprone.annotations.FormatMethod;
 import java.util.Objects;
 import java.util.function.Supplier;
 
@@ -75,7 +74,6 @@ public interface Logger {
         }
     }
 
-    @FormatMethod
     default void tracef(String format, Object... args) {
         Objects.requireNonNull(format);
 
@@ -117,7 +115,6 @@ public interface Logger {
         }
     }
 
-    @FormatMethod
     default void debugf(String format, Object... args) {
         Objects.requireNonNull(format);
 
@@ -159,7 +156,6 @@ public interface Logger {
         }
     }
 
-    @FormatMethod
     default void infof(String format, Object... args) {
         Objects.requireNonNull(format);
 
@@ -201,7 +197,6 @@ public interface Logger {
         }
     }
 
-    @FormatMethod
     default void warnf(String format, Object... args) {
         Objects.requireNonNull(format);
 
@@ -243,7 +238,6 @@ public interface Logger {
         }
     }
 
-    @FormatMethod
     default void errorf(String format, Object... args) {
         Objects.requireNonNull(format);
 
