--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -54,7 +54,7 @@
 bazel_dep(name = "protobuf", version = "33.4", repo_name = "com_google_protobuf")
 bazel_dep(name = "rules_cc", version = "0.0.9")
 bazel_dep(name = "rules_java", version = "9.1.0")
-bazel_dep(name = "rules_jvm_external", version = "6.0")
+bazel_dep(name = "rules_jvm_external", version = "6.5")

 maven = use_extension("@rules_jvm_external//:extensions.bzl", "maven")
 maven.install(
