--- MODULE.bazel
+++ MODULE.bazel
@@ -1,10 +1,15 @@
 """Module for the bazel_iwyu repository"""
 
-module(name = "bazel_iwyu")
+module(
+    name = "bazel_iwyu",
+    version = "20240913.0",
+)
 
 bazel_dep(name = "abseil-cpp", version = "20240116.2")
 bazel_dep(name = "bazel_skylib", version = "1.5.0")
+bazel_dep(name = "rules_shell", version = "0.3.0")
 bazel_dep(name = "googletest", version = "1.14.0", dev_dependency = True, repo_name = "com_google_googletest")
+bazel_dep(name = "rules_cc", version = "0.0.9", dev_dependency = True)
 
 prebuilt_pkg = use_repo_rule("//bazel:prebuilt_pkg.bzl", "prebuilt_pkg")
 prebuilt_pkg(
