diff --git BUILD.bazel BUILD.bazel
new file mode 100644
index 0000000..45721ac
--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,30 @@
+""" Builds foonathan_memory.
+"""
+
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["**"]),
+)
+
+cache_entries = {
+    "CMAKE_POSITION_INDEPENDENT_CODE": "ON",  # Must be set!
+    "BUILD_SHARED_LIBS": "OFF",
+    # foonathan_memory specific options.
+    "FOONATHAN_MEMORY_BUILD_EXAMPLES": "OFF",
+    "FOONATHAN_MEMORY_BUILD_TESTS": "OFF",
+}
+
+cmake(
+    name = "foonathan_memory",
+    build_args = [
+        "--",
+        "-j4",
+    ],
+    cache_entries = cache_entries,
+    lib_source = ":all_srcs",
+    lib_name = "libfoonathan_memory-0.7.3",
+    out_include_dir = "include/foonathan_memory",
+    visibility = ["//visibility:public"]
+)
