diff --git a/bazel/python_rules.bzl b/bazel/python_rules.bzl
index 3f940a56ea..0c3636f5b0 100644
--- a/bazel/python_rules.bzl
+++ b/bazel/python_rules.bzl
@@ -190,11 +190,7 @@ def _generate_pb2_grpc_src_impl(context):
     arguments = []
     tools = [context.executable._protoc, context.executable._grpc_plugin]
     out_dir = get_out_dir(protos, context)
-    if out_dir.import_path:
-        # is virtual imports
-        out_path = out_dir.path
-    else:
-        out_path = context.genfiles_dir.path
+    out_path = out_dir.path
     arguments += get_plugin_args(
         context.executable._grpc_plugin,
         plugin_flags,
