diff --git a/lib/locks.h b/lib/locks.h
index f564b980..67b2f028 100644
--- a/lib/locks.h
+++ b/lib/locks.h
@@ -29,6 +29,9 @@
 #ifdef USE_LOCKS
 # include <pthread.h>
 # include <assert.h>
+# ifndef assert_perror
+#  define assert_perror(errnum) assert((errnum) == 0)
+# endif
 # define rwlock_define(class,name)      class pthread_rwlock_t name
 # define once_define(class,name)  class pthread_once_t name = PTHREAD_ONCE_INIT
 # define RWLOCK_CALL(call)              \
