--- a/arrow/cdata/BUILD.bazel
+++ b/arrow/cdata/BUILD.bazel
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_source")
 
 go_library(
     name = "cdata",
@@ -8,6 +8,7 @@ go_library(
         "exports.go",
         "interface.go",
         "trampoline.c",
+        ":headers",
     ],
     cgo = True,
     importpath = "github.com/apache/arrow/go/v14/arrow/cdata",
@@ -31,3 +32,11 @@ alias(
     actual = ":cdata",
     visibility = ["//visibility:public"],
 )
+
+go_source(
+    name = "headers",
+    srcs = [
+        "arrow/c/abi.h",
+        "arrow/c/helpers.h",
+    ],
+)
