diff --git a/BUILD.bazel b/BUILD.bazel
index 8285865..37e192e 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -18,49 +18,83 @@ go_library(
         "transaction.go",
         "types.go",
     ],
+    cdeps = select({
+        "@io_bazel_rules_go//go/platform:android_amd64": [
+            "//deps/linux_amd64:libduckdb",
+        ],
+        "@io_bazel_rules_go//go/platform:android_arm64": [
+            "//deps/linux_arm64:libduckdb",
+        ],
+        "@io_bazel_rules_go//go/platform:darwin_386": [
+        ],
+        "@io_bazel_rules_go//go/platform:darwin_amd64": [
+            "//deps/darwin_amd64:libduckdb",
+        ],
+        "@io_bazel_rules_go//go/platform:darwin_arm": [
+        ],
+        "@io_bazel_rules_go//go/platform:darwin_arm64": [
+            "//deps/darwin_arm64:libduckdb",
+        ],
+        "@io_bazel_rules_go//go/platform:freebsd_amd64": [
+            "//deps/freebsd_amd64:libduckdb",
+        ],
+        "@io_bazel_rules_go//go/platform:ios_amd64": [
+            "//deps/darwin_amd64:libduckdb",
+        ],
+        "@io_bazel_rules_go//go/platform:ios_arm64": [
+            "//deps/darwin_arm64:libduckdb",
+        ],
+        "@io_bazel_rules_go//go/platform:linux_amd64": [
+            "//deps/linux_amd64:libduckdb",
+        ],
+        "@io_bazel_rules_go//go/platform:linux_arm64": [
+            "//deps/linux_arm64:libduckdb",
+        ],
+        "//conditions:default": [],
+    }),
     cgo = True,
     clinkopts = select({
         "@io_bazel_rules_go//go/platform:android_amd64": [
             "-lduckdb",
-            "-lstdc++ -lm -ldl -Ldeps/linux_amd64",
+            "-lstdc++ -lm -ldl -Lexternal/com_github_marcboeker_go_duckdb/deps/linux_amd64",
         ],
         "@io_bazel_rules_go//go/platform:android_arm64": [
             "-lduckdb",
-            "-lstdc++ -lm -ldl -Ldeps/linux_arm64",
+            "-lstdc++ -lm -ldl -Lexternal/com_github_marcboeker_go_duckdb/deps/linux_arm64",
         ],
         "@io_bazel_rules_go//go/platform:darwin_386": [
             "-lduckdb",
         ],
         "@io_bazel_rules_go//go/platform:darwin_amd64": [
-            "-lc++ -Ldeps/darwin_amd64",
+            "-lc++ -Lexternal/com_github_marcboeker_go_duckdb/deps/darwin_amd64",
             "-lduckdb",
         ],
         "@io_bazel_rules_go//go/platform:darwin_arm": [
             "-lduckdb",
         ],
         "@io_bazel_rules_go//go/platform:darwin_arm64": [
-            "-lc++ -Ldeps/darwin_arm64",
+            "-lc++ -Lexternal/com_github_marcboeker_go_duckdb/deps/darwin_arm64",
             "-lduckdb",
         ],
         "@io_bazel_rules_go//go/platform:freebsd_amd64": [
             "-lduckdb",
-            "-lstdc++ -lm -ldl -Ldeps/freebsd_amd64",
+            "-lstdc++ -lm -ldl -Lexternal/com_github_marcboeker_go_duckdb/deps/freebsd_amd64",
         ],
         "@io_bazel_rules_go//go/platform:ios_amd64": [
             "-lc++ -Ldeps/darwin_amd64",
             "-lduckdb",
         ],
         "@io_bazel_rules_go//go/platform:ios_arm64": [
-            "-lc++ -Ldeps/darwin_arm64",
+            "-lc++ -Lexternal/com_github_marcboeker_go_duckdb/deps/darwin_arm64",
             "-lduckdb",
         ],
         "@io_bazel_rules_go//go/platform:linux_amd64": [
             "-lduckdb",
-            "-lstdc++ -lm -ldl -Ldeps/linux_amd64",
+            "-lstdc++ -lm -ldl -Lexternal/com_github_marcboeker_go_duckdb/deps/linux_amd64",
         ],
         "@io_bazel_rules_go//go/platform:linux_arm64": [
             "-lduckdb",
-            "-lstdc++ -lm -ldl -Ldeps/linux_arm64",
+            "-lstdc++ -lm -ldl -Lexternal/com_github_marcboeker_go_duckdb/deps/linux_arm64",
         ],
         "//conditions:default": [],
     }),
diff --git a/deps/darwin_amd64/BUILD.bazel b/deps/darwin_amd64/BUILD.bazel
index 471671e..4f508b6 100644
--- a/deps/darwin_amd64/BUILD.bazel
+++ b/deps/darwin_amd64/BUILD.bazel
@@ -12,3 +12,9 @@ alias(
     actual = ":darwin_amd64",
     visibility = ["//visibility:public"],
 )
+
+cc_library(
+    name = "libduckdb",
+    srcs = ["libduckdb.a"],
+    visibility = ["//visibility:public"],
+)
diff --git a/deps/darwin_arm64/BUILD.bazel b/deps/darwin_arm64/BUILD.bazel
index edc81b3..848fe8b 100644
--- a/deps/darwin_arm64/BUILD.bazel
+++ b/deps/darwin_arm64/BUILD.bazel
@@ -12,3 +12,9 @@ alias(
     actual = ":darwin_arm64",
     visibility = ["//visibility:public"],
 )
+
+cc_library(
+    name = "libduckdb",
+    srcs = ["libduckdb.a"],
+    visibility = ["//visibility:public"],
+)
diff --git a/deps/freebsd_amd64/BUILD.bazel b/deps/freebsd_amd64/BUILD.bazel
index 849a7e3..f347669 100644
--- a/deps/freebsd_amd64/BUILD.bazel
+++ b/deps/freebsd_amd64/BUILD.bazel
@@ -12,3 +12,9 @@ alias(
     actual = ":freebsd_amd64",
     visibility = ["//visibility:public"],
 )
+
+cc_library(
+    name = "libduckdb",
+    srcs = ["libduckdb.a"],
+    visibility = ["//visibility:public"],
+)
diff --git a/deps/linux_amd64/BUILD.bazel b/deps/linux_amd64/BUILD.bazel
index 0acbb38..23be608 100644
--- a/deps/linux_amd64/BUILD.bazel
+++ b/deps/linux_amd64/BUILD.bazel
@@ -12,3 +12,9 @@ alias(
     actual = ":linux_amd64",
     visibility = ["//visibility:public"],
 )
+
+cc_library(
+    name = "libduckdb",
+    srcs = ["libduckdb.a"],
+    visibility = ["//visibility:public"],
+)
diff --git a/deps/linux_arm64/BUILD.bazel b/deps/linux_arm64/BUILD.bazel
index 64200d5..5c7fba0 100644
--- a/deps/linux_arm64/BUILD.bazel
+++ b/deps/linux_arm64/BUILD.bazel
@@ -12,3 +12,9 @@ alias(
     actual = ":linux_arm64",
     visibility = ["//visibility:public"],
 )
+
+cc_library(
+    name = "libduckdb",
+    srcs = ["libduckdb.a"],
+    visibility = ["//visibility:public"],
+)
