===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -12,8 +12,9 @@
 # breaking change affects most use cases and isn't easy to work around, and
 # in the same commit that introduces the incompatible change.
 module(
     name = "rules_dart",
+    version = "0.1.9",
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.9.0")
 bazel_dep(name = "package_metadata", version = "0.0.7")
