diff --git a/MODULE.bazel b/MODULE.bazel
index b4385e038..27fa3b1b2 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -3,7 +3,7 @@
 
 module(
     name = "protobuf",
-    version = "30.0-rc1",  # Automatically updated on release
+    version = "30.0-rc1.bcr.1",  # Automatically updated on release
     compatibility_level = 1,
     repo_name = "com_google_protobuf",
 )
diff --git a/build_defs/BUILD.bazel b/build_defs/BUILD.bazel
index 268ecbb65..394078305 100644
--- a/build_defs/BUILD.bazel
+++ b/build_defs/BUILD.bazel
@@ -28,7 +28,6 @@ platform(
     constraint_values = [
         "@platforms//cpu:x86_64",
         "@platforms//os:windows",
-        "@bazel_tools//tools/cpp:clang-cl",
         # This is necessary for Bazel 7 compatibility with a MODULE.bazel file that still works in
         # Bazel 8.  Using cc_configure_extension from rules_cc produces a @local_config_cc
         # repository that's not compatible with @bazel_tools//tools/cpp:clang-cl from before
@@ -42,7 +41,6 @@ platform(
     constraint_values = [
         "@platforms//cpu:x86_64",
         "@platforms//os:windows",
-        "@bazel_tools//tools/cpp:msvc",
         # This may be necessary in case cc_configure_extension from rules_cc produces a
         # @local_config_cc repository that's not compatible with @bazel_tools//tools/cpp:msvc from
         # before Bazel 8, as with clang-cl above, to avoid silently falling back to clang-cl.
