
From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shantanu Gontia <gontia.shantanu@gmail.com>
Date: Fri, 12 Jun 2026 05:02:35 +0530
Subject: BUILD bazel


diff --git BUILD.bazel BUILD.bazel
index f4d69a747..3b7dd10f7 100644
--- BUILD.bazel
+++ BUILD.bazel
@@ -19,14 +19,22 @@ filegroup(
 cmake(
     name = "z3_dynamic",
     generate_args = [
-        "-G Ninja", 
+        "-G Ninja",
         "-D Z3_EXPORTED_TARGETS=",  # prevents installation, leaving symlinks between dylibs intact on copy
     ],
     lib_source = ":all_files",
     out_binaries = ["z3"],
     out_shared_libs = select({
-        "@platforms//os:linux": ["libz3.so"],
-        # "@platforms//os:osx": ["libz3.dylib"],    # FIXME: this is not working, libz3<version>.dylib is not copied
+        "@platforms//os:linux": [
+            "libz3.so",
+            "libz3.so.4.16",
+            "libz3.so.4.16.0.0",
+        ],
+        "@platforms//os:osx": [
+            "libz3.dylib",
+            "libz3.4.16.dylib",
+            "libz3.4.16.0.0.dylib",
+        ],
         "@platforms//os:windows": ["libz3.dll"],
         "//conditions:default": ["@platforms//:incompatible"],
     }),
@@ -36,7 +44,7 @@ cmake(
 cmake(
     name = "z3_static",
     generate_args = [
-        "-G Ninja", 
+        "-G Ninja",
         "-D BUILD_SHARED_LIBS=OFF",
         "-D Z3_BUILD_LIBZ3_SHARED=OFF",
     ],
-- 
2.54.0
