--- BUILD.bazel	2026-06-12 03:35:44
+++ BUILD.bazel	2026-06-12 03:37:18
@@ -25,8 +25,16 @@
     lib_source = ":all_files",
     out_binaries = ["z3"],
     out_shared_libs = select({
-        "@platforms//os:linux": ["libz3.so"],
-        # "@platforms//os:osx": ["libz3.dylib"],    # FIXME: this is not working, libz3<version>.dylib is not copied
+        "@platforms//os:linux": [
+            "libz3.so",
+            "libz3.so.4.16",
+            "libz3.so.4.16.0.0",
+        ],
+        "@platforms//os:osx": [
+            "libz3.dylib",
+            "libz3.4.16.dylib",
+            "libz3.4.16.0.0.dylib",
+        ],
         "@platforms//os:windows": ["libz3.dll"],
         "//conditions:default": ["@platforms//:incompatible"],
     }),
