diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..f60a43e
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,12 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "hana",
+    deps = [
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.fusion//:fusion",
+        "@boost.mpl//:mpl",
+        "@boost.tuple//:tuple",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..1a18241
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,13 @@
+module(
+    name = "boost.hana",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.config", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.fusion", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.mpl", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.tuple", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..74b4bf1
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost.hana//:hana",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "hana_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/hana_test.cc b/bazel_test/hana_test.cc
new file mode 100644
index 0000000..36a287d
--- /dev/null
+++ b/bazel_test/hana_test.cc
@@ -0,0 +1,12 @@
+#include <string>
+#include <boost/hana.hpp>
+namespace hana = boost::hana;
+
+struct Fish { std::string name; };
+struct Cat  { std::string name; };
+struct Dog  { std::string name; };
+
+int main(){
+  auto animals = hana::make_tuple(Fish{"Nemo"}, Cat{"Garfield"}, Dog{"Snoopy"});
+  return hana::size(animals).value == 3 ? 0 : 1;
+}
