diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..8e4e484
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,14 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "hana",
+    deps = [
+        "@boost//:config",
+        "@boost//:core",
+        "@boost//:fusion",
+        "@boost//:mpl",
+        "@boost//:tuple",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..2fa3c8a
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.hana",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..f23721e
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:hana",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "hana_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/hana_test.cc b/bazel_test/hana_test.cc
new file mode 100644
index 0000000..36a287d
--- /dev/null
+++ b/bazel_test/hana_test.cc
@@ -0,0 +1,12 @@
+#include <string>
+#include <boost/hana.hpp>
+namespace hana = boost::hana;
+
+struct Fish { std::string name; };
+struct Cat  { std::string name; };
+struct Dog  { std::string name; };
+
+int main(){
+  auto animals = hana::make_tuple(Fish{"Nemo"}, Cat{"Garfield"}, Dog{"Snoopy"});
+  return hana::size(animals).value == 3 ? 0 : 1;
+}
