diff --git a/pkg.bzl b/pkg.bzl
index f1d8e42..f54982d 100644
--- a/pkg.bzl
+++ b/pkg.bzl
@@ -47,7 +47,7 @@ SUPPORTED_TAR_COMPRESSIONS = (
 )
 deb_filetype = [".deb", ".udeb"]
 _DEFAULT_MTIME = -1
-_stamp_condition = str(Label("//private:private_stamp_detect"))
+_stamp_condition = Label("//private:private_stamp_detect")
 
 def _remap(remap_paths, path):
     """If path starts with a key in remap_paths, rewrite it."""
