diff --git a/common/BUILD b/common/BUILD
index be69694..8dcc661 100644
--- a/common/BUILD
+++ b/common/BUILD
@@ -82,10 +82,11 @@ cc_library(
     name = "hash",
     srcs = ["hash.cc"],
     hdrs = ["hash.h"],
-    linkopts = ["-Wl,-Bstatic -lcrypto -Wl,-Bdynamic -ldl"],
+    # linkopts = ["-Wl,-Bstatic -lcrypto -Wl,-Bdynamic -ldl"],
     visibility = EXTENDED_API_VISIBILITY,
     deps = [
         ":defs",
+        "@boringssl//:crypto"
     ],
 )
 
