diff --git a/MODULE.bazel b/MODULE.bazel
index 6b90ca6..a5a79ed 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,6 @@
 module(
     name = "fuzztest",
-    version = "head",
+    version = "20240813.0",
     repo_name = "com_google_fuzztest",
 )
 
@@ -39,20 +39,17 @@ bazel_dep(
 bazel_dep(
     name = "protobuf",
     version = "26.0",
-    dev_dependency = True,
     repo_name = "com_google_protobuf",
 )
 
 bazel_dep(
     name = "rules_proto",
     version = "5.3.0-21.7",
-    dev_dependency = True,
 )
 
 bazel_dep(
     name = "riegeli",
     version = "0.0.0-20240606-973b6f0",
-    dev_dependency = True,
     repo_name = "com_google_riegeli",
 )
 
diff --git a/MODULE.bazel b/MODULE.bazel
index a5a79edf..26b9d01f 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -21,6 +21,11 @@ bazel_dep(
     version = "1.7.1",
 )
 
+bazel_dep(
+    name = "boringssl", 
+    version = "0.0.0-20240530-2db0eb3",
+)
+
 # Dev dependencies.
 bazel_dep(
     name = "googletest",
