diff --git a/centipede/distill.cc b/centipede/distill.cc
index 84e8c13..2fdc7da 100644
--- a/centipede/distill.cc
+++ b/centipede/distill.cc
@@ -289,7 +289,7 @@ class DistillingInputFilter {
 
   Stats GetStats() {
     absl::MutexLock lock{&mu_};
-    stats_.coverage_str = (std::stringstream{} << seen_features_).str();
+    stats_.coverage_str = static_cast<std::ostringstream&>(std::stringstream{} << seen_features_).str();
     return stats_;
   }
 
