diff --git MODULE.bazel MODULE.bazel
new file mode 100644
index 00000000..44caadb7
--- /dev/null
+++ MODULE.bazel
@@ -0,0 +1,7 @@
+module(name = "osqp", version = "0.6.3.bcr.2")
+
+bazel_dep(name = "rules_cc", version = "0.0.9")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "qdldl", version = "0.1.7.bcr.1")
+bazel_dep(name = "suitesparse", version = "7.6.0.bcr.1")
+bazel_dep(name = "cmake_configure_file", version = "0.1.0")
