--- MODULE.bazel
+++ MODULE.bazel
@@ -0,0 +1,10 @@
+module(
+    name = "blake3",
+    version = "1.5.1.bcr.1",
+    compatibility_level = 1,
+)
+bazel_dep(name = "rules_cc", version = "0.0.6")
+
+# Required only by `--extra_toolchains=@local_config_cc//:cc-toolchain-arm64_windows` used in CI.
+cc_configure = use_extension("@bazel_tools//tools/cpp:cc_configure.bzl", "cc_configure_extension", dev_dependency = True)
+use_repo(cc_configure, "local_config_cc")
