--- include/pinocchio/parsers/urdf/types.hpp
+++ include/pinocchio/parsers/urdf/types.hpp
@@ -41,6 +41,7 @@ namespace urdf
   PINOCCHIO_URDF_TYPEDEF_CLASS_POINTER(Sphere);
   PINOCCHIO_URDF_TYPEDEF_CLASS_POINTER(Visual);
   
+/*
   template<class T, class U>
   PINOCCHIO_URDF_SHARED_PTR(T) const_pointer_cast(PINOCCHIO_URDF_SHARED_PTR(U) const & r)
   {
@@ -70,6 +71,7 @@ namespace urdf
     return boost::static_pointer_cast<T>(r);
 #endif
   }
+*/
 }
 
 #undef PINOCCHIO_URDF_TYPEDEF_CLASS_POINTER
