diff --git pcl_config.h.in pcl_config.h.in
--- pcl_config.h.in
+++ pcl_config.h.in
@@ -35,7 +35,13 @@
   #endif //PCL_MINOR_VERSION
 #endif 
 
-#define PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC ${PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC}
+#if (defined(EIGEN_DEFAULT_ALIGN_BYTES) && EIGEN_DEFAULT_ALIGN_BYTES!=0) && (defined(EIGEN_MALLOC_ALREADY_ALIGNED) && !EIGEN_MALLOC_ALREADY_ALIGNED)
+  #define PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC 1
+#else
+  #define PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC 0
+#endif
+
+#cmakedefine PCL_SILENCE_MALLOC_WARNING
 
 #cmakedefine HAVE_OPENNI 1
 
diff --git common/include/pcl/memory.h common/include/pcl/memory.h
--- common/include/pcl/memory.h
+++ common/include/pcl/memory.h
@@ -43,11 +43,10 @@
  */
 
 #include <pcl/type_traits.h>  // for has_custom_allocator
-#include <pcl/pcl_config.h> // for PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC
-
-#include <Eigen/Core>  // for EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+#include <Eigen/Core>  // for EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+#include <pcl/pcl_config.h> // for PCL_USES_EIGEN_HANDMADE_ALIGNED_MALLOC
 
 #include <memory>  // for std::allocate_shared, std::dynamic_pointer_cast, std::make_shared, std::shared_ptr, std::static_pointer_cast, std::weak_ptr
 #include <type_traits>  // for std::enable_if_t, std::false_type, std::true_type
