diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..086bfd4
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,48 @@
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+bool_flag(
+    name = "beast_use_std_string_view",
+    build_setting_default = False,
+)
+
+config_setting(
+    name = "beast_std_string_view",
+    flag_values = {
+        ":beast_use_std_string_view": "True",
+    },
+)
+
+boost_library(
+    name = "beast",
+    srcs = [
+        "build_beast.cpp",
+    ],
+    defines = [
+        "BOOST_BEAST_SEPARATE_COMPILATION",
+    ] + select({
+        ":beast_std_string_view": ["BOOST_BEAST_USE_STD_STRING_VIEW"],
+        "//conditions:default": [],
+    }),
+    deps = [
+        "@boost.asio//:asio",
+        "@boost.assert//:assert",
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.detail//:detail",
+        "@boost.endian//:endian",
+        "@boost.intrusive//:intrusive",
+        "@boost.bind//:bind",
+        "@boost.mp11//:mp11",
+        "@boost.optional//:optional",
+        "@boost.preprocessor//:preprocessor",
+        "@boost.smart_ptr//:smart_ptr",
+        "@boost.static_assert//:static_assert",
+        "@boost.static_string//:static_string",
+        "@boost.system//:system",
+        "@boost.throw_exception//:throw_exception",
+        "@boost.tribool//:tribool",
+        "@boost.type_traits//:type_traits",
+        "@boost.utility//:utility",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..4fbf525
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,30 @@
+module(
+    name = "boost.beast",
+    version = "1.83.0.bzl.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "bazel_skylib", version = "1.5.0")
+bazel_dep(name = "boost.asio", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.bind", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.container", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.container_hash", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.endian", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.intrusive", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.logic", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.mp11", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.optional", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.preprocessor", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.smart_ptr", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.static_string", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.system", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.throw_exception", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.type_index", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.winapi", version = "1.83.0.bzl.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/build_beast.cpp b/build_beast.cpp
new file mode 100644
index 0000000..11ef613
--- /dev/null
+++ b/build_beast.cpp
@@ -0,0 +1,2 @@
+// https://www.boost.org/doc/libs/1_77_0/libs/beast/doc/html/beast/config/configuration_preprocessor_defin.html
+#include "boost/beast/src.hpp"
