--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,72 +1,38 @@
-## MODULE.bazel
 module(
     name = "pthreadpool",
+    version = "0.0.0-20260119-9003ee6",
+    bazel_compatibility = [">=7.2.1"],
+    compatibility_level = 1,
 )
 
-# Bazel rule definitions
-bazel_dep(name = "rules_cc", version = "0.1.1")
-bazel_dep(name = "rules_python", version = "1.0.0")
-
-pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
-pip.parse(
-    hub_name = "pip",
-    python_version = "3.11",
-    requirements_lock = "//:requirements_lock.txt",
+bazel_dep(
+    name = "cpuinfo",
+    version = "0.0.0-20260312-7607ca5",
 )
-use_repo(pip, "pip")
-
-# Bazel Skylib.
-bazel_dep(name = "bazel_skylib", version = "1.7.1")
-
-# Bazel Platforms
-bazel_dep(name = "platforms", version = "0.0.10")
-
-# TODO: some (most? all?) of the http_archive() calls below could become bazel_dep() calls,
-# but it would require verifying that the semver provided by the Bazel registry matches the hash
-# that we expect in CMake; it's not clear that it is a big win to do so given the modest
-# complexity of our deps, so I'm leaving it like this for now to ensure that the Bazel and CMake
-# builds are using identical dependencies.
-
-http_archive = use_repo_rule("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
-
-# LINT.IfChange(googletest)
-# Google Test framework, used by most unit-tests.
-http_archive(
-    name = "com_google_googletest",
-    sha256 = "ce7366fe57eb49928311189cb0e40e0a8bf3d3682fca89af30d884c25e983786",
-    strip_prefix = "googletest-release-1.12.0",
-    urls = ["https://github.com/google/googletest/archive/release-1.12.0.zip"],
+bazel_dep(
+    name = "fxdiv",
+    version = "0.0.0-20201209-63058ef",
+    repo_name = "FXdiv",
 )
-# LINT.ThenChange(cmake/DownloadGoogleTest.cmake,WORKSPACE:googletest)
-
-# LINT.IfChange(benchmark)
-# Google Benchmark library, used in micro-benchmarks.
-http_archive(
-    name = "com_google_benchmark",
-    sha256 = "28c7cac12cc25d87d3dcc8cbdefa4b03c32d1a27bd50e37ca466d8127c1688d834800c38f3c587a396188ee5fb7d1bd0971b41a599a5c4787f8742cb39ca47db",
-    strip_prefix = "benchmark-1.5.3",
-    urls = ["https://github.com/google/benchmark/archive/v1.5.3.zip"],
+bazel_dep(
+    name = "google_benchmark",
+    version = "1.9.5",
+    repo_name = "com_google_benchmark",
 )
-# LINT.ThenChange(cmake/DownloadGoogleBenchmark.cmake,WORKSPACE:benchmark)
-
-# LINT.IfChange
-# FXdiv library, used for repeated integer division by the same factor
-http_archive(
-    name = "FXdiv",
-    sha256 = "ab7dfb08829bee33dca38405d647868fb214ac685e379ec7ef2bebcd234cd44d",
-    strip_prefix = "FXdiv-b408327ac2a15ec3e43352421954f5b1967701d1",
-    urls = ["https://github.com/Maratyszcza/FXdiv/archive/b408327ac2a15ec3e43352421954f5b1967701d1.zip"],
+bazel_dep(
+    name = "googletest",
+    version = "1.17.0.bcr.2",
+    repo_name = "com_google_googletest",
 )
-# LINT.ThenChange(cmake/DownloadFXdiv.cmake)
-
-# LINT.IfChange
-# cpuinfo library, used for detecting processor characteristics
-http_archive(
-    name = "cpuinfo",
-    sha256 = "52e0ffd7998d8cb3a927d8a6e1145763744d866d2be09c4eccea27fc157b6bb0",
-    strip_prefix = "cpuinfo-cebb0933058d7f181c979afd50601dc311e1bf8c",
-    urls = [
-        "https://github.com/pytorch/cpuinfo/archive/cebb0933058d7f181c979afd50601dc311e1bf8c.zip",
-    ],
+bazel_dep(
+    name = "rules_cc",
+    version = "0.2.17",
 )
-# LINT.ThenChange(cmake/DownloadCpuinfo.cmake)
+bazel_dep(
+    name = "rules_license",
+    version = "1.0.0",
+)
+bazel_dep(
+    name = "platforms",
+    version = "1.0.0",
+)
