diff --git a/BUILD.bazel b/BUILD.bazel
index 0033f4d..896c368 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -59,12 +59,18 @@ cc_library(
     hdrs = [
         "include/pthreadpool.h",
     ],
-    copts = [
-        "-std=c11",
-    ] + select({
+    copts = select({
         ":optimized_build": ["-O2"],
         "//conditions:default": [],
     }),
+    conlyopts = select({
+        ":windows_clang_cl": ["/std:c11"],
+        ":windows_msvc": [
+            "/std:c11",
+            "/experimental:c11atomics",
+        ],
+        "//conditions:default": ["-std=c11"],
+    }),
     defines = select({
         ":linux_arm": ["PTHREADPOOL_USE_CPUINFO=1"],
         ":linux_armeabi": ["PTHREADPOOL_USE_CPUINFO=1"],
@@ -223,6 +229,18 @@ config_setting(
     },
 )
 
+config_setting(
+    name = "windows_clang_cl",
+    flag_values = {"@bazel_tools//tools/cpp:compiler": "clang-cl"},
+    constraint_values = ["@platforms//os:windows"],
+)
+
+config_setting(
+    name = "windows_msvc",
+    flag_values = {"@bazel_tools//tools/cpp:compiler": "msvc-cl"},
+    constraint_values = ["@platforms//os:windows"],
+)
+
 config_setting(
     name = "linux_x86_64",
     values = {"cpu": "k8"},
