diff --git a/BUILD.bazel b/BUILD.bazel
index 0033f4d..5a306f7 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -10,9 +10,20 @@
 load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
+load("@rules_license//rules:license.bzl", "license")
+
+package(
+    default_applicable_licenses = [":license"],
+)

 licenses(["notice"])

 exports_files(["LICENSE"])

+license(
+    name = "license",
+    license_kinds = ["@rules_license//licenses/spdx:BSD-2-Clause"],
+    license_text = "LICENSE",
+)
+
 ############################## pthreadpool library #############################

 INTERNAL_HDRS = [
@@ -59,12 +70,18 @@ cc_library(
     hdrs = [
         "include/pthreadpool.h",
     ],
-    copts = [
-        "-std=c11",
-    ] + select({
+    copts = select({
         ":optimized_build": ["-O2"],
         "//conditions:default": [],
     }),
+    conlyopts = select({
+        ":windows_clang_cl": ["/std:c11"],
+        ":windows_msvc": [
+            "/std:c11",
+            "/experimental:c11atomics",
+        ],
+        "//conditions:default": ["-std=c11"],
+    }),
     defines = select({
         ":linux_arm": ["PTHREADPOOL_USE_CPUINFO=1"],
         ":linux_armeabi": ["PTHREADPOOL_USE_CPUINFO=1"],
@@ -108,6 +125,12 @@ cc_library(
             "-s ALLOW_BLOCKING_ON_MAIN_THREAD=1",
             "-s PTHREAD_POOL_SIZE=8",
         ],
+        ":linux_x86_64": ["-pthread"],
+        ":linux_arm": ["-pthread"],
+        ":linux_armeabi": ["-pthread"],
+        ":linux_armhf": ["-pthread"],
+        ":linux_armv7a": ["-pthread"],
+        ":linux_aarch64": ["-pthread"],
         "//conditions:default": [],
     }),
     strip_include_prefix = "include",
@@ -223,6 +246,18 @@ config_setting(
     },
 )
 
+config_setting(
+    name = "windows_clang_cl",
+    flag_values = {"@bazel_tools//tools/cpp:compiler": "clang-cl"},
+    constraint_values = ["@platforms//os:windows"],
+)
+
+config_setting(
+    name = "windows_msvc",
+    flag_values = {"@bazel_tools//tools/cpp:compiler": "msvc-cl"},
+    constraint_values = ["@platforms//os:windows"],
+)
+
 config_setting(
     name = "linux_x86_64",
     values = {"cpu": "k8"},
