diff --git a/src/agrpc/detail/health_check_service.hpp b/src/agrpc/detail/health_check_service.hpp
index daec2c0..838cc12 100644
--- a/src/agrpc/detail/health_check_service.hpp
+++ b/src/agrpc/detail/health_check_service.hpp
@@ -306,7 +306,7 @@ inline void start_health_check_service(grpc::Server& server, agrpc::GrpcContext&
 {
     auto* const service = server.GetHealthCheckService();
     // clang-format off
-    GPR_ASSERT(service && "Use `agrpc::add_health_check_service` to add the HealthCheckService to a ServerBuilder before calling this function");
+    // GPR_ASSERT(service && "Use `agrpc::add_health_check_service` to add the HealthCheckService to a ServerBuilder before calling this function");
     // clang-format on
     agrpc::start_health_check_service(*static_cast<agrpc::HealthCheckService*>(service), grpc_context);
 }
