diff --git a/src/agrpc/detail/asio_forward.hpp b/src/agrpc/detail/asio_forward.hpp
index efd77ff..4b064c8 100644
--- a/src/agrpc/detail/asio_forward.hpp
+++ b/src/agrpc/detail/asio_forward.hpp
@@ -26,6 +26,7 @@
 #include <asio/associated_executor.hpp>
 #include <asio/async_result.hpp>
 #include <asio/bind_executor.hpp>
+#include <asio/error.hpp>
 #include <asio/execution/allocator.hpp>
 #include <asio/execution/blocking.hpp>
 #include <asio/execution/context.hpp>
@@ -64,6 +65,7 @@
 #include <boost/asio/associated_executor.hpp>
 #include <boost/asio/async_result.hpp>
 #include <boost/asio/bind_executor.hpp>
+#include <boost/asio/error.hpp>
 #include <boost/asio/execution/allocator.hpp>
 #include <boost/asio/execution/blocking.hpp>
 #include <boost/asio/execution/context.hpp>
