diff --git a/frontends/ast/dpicall.cc b/frontends/ast/dpicall.cc
index d76318739..16f79aa03 100644
--- a/frontends/ast/dpicall.cc
+++ b/frontends/ast/dpicall.cc
@@ -24,6 +24,12 @@
 #include <dlfcn.h>
 #include <ffi.h>
 
+#if defined(__APPLE__) || defined(__MACH__)
+#ifndef RTLD_DEFAULT
+#define RTLD_DEFAULT ((void*) -2)
+#endif
+#endif
+
 YOSYS_NAMESPACE_BEGIN
 
 typedef void (*ffi_fptr) ();
@@ -161,4 +167,3 @@ std::unique_ptr<AST::AstNode> AST::dpi_call(AstSrcLocType, const std::string&, c
 YOSYS_NAMESPACE_END
 
 #endif /* YOSYS_ENABLE_PLUGINS */
-
diff --git a/libs/ezsat/ezminisat.cc b/libs/ezsat/ezminisat.cc
index 30df625cb..374e17fd4 100644
--- a/libs/ezsat/ezminisat.cc
+++ b/libs/ezsat/ezminisat.cc
@@ -29,7 +29,10 @@
 
 #include <limits.h>
 #include <stdint.h>
+#if !defined(__APPLE__) && !defined(__MACH__)
 #include <cinttypes>
+#endif
+
 
 #if !defined(_WIN32) && !defined(__wasm)
 #  include <csignal>
@@ -245,4 +248,3 @@ contradiction:
 #endif
 	return true;
 }
-
diff --git a/libs/minisat/IntTypes.h b/libs/minisat/IntTypes.h
index c48816284..4fdbc10f2 100644
--- a/libs/minisat/IntTypes.h
+++ b/libs/minisat/IntTypes.h
@@ -33,6 +33,12 @@ OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWA
 #   include <stdint.h>
 #   include <inttypes.h>
 
+#if defined(__APPLE__) || defined(__MACH__)
+#ifndef PRIi64
+# include <_inttypes.h>
+#endif
+#endif
+
 #endif
 
 #include <limits.h>
diff --git a/passes/cmds/plugin.cc b/passes/cmds/plugin.cc
index a653844b7..1537c4b61 100644
--- a/passes/cmds/plugin.cc
+++ b/passes/cmds/plugin.cc
@@ -22,6 +22,11 @@
 
 #ifdef YOSYS_ENABLE_PLUGINS
 #  include <dlfcn.h>
+#  ifdef __APPLE__
+#    ifndef RTLD_LOCAL
+#      define RTLD_LOCAL 0
+#    endif
+#  endif
 #endif
 
 #ifdef WITH_PYTHON
