--- a/OgreMain/src/OgreRenderQueue.cpp
+++ b/OgreMain/src/OgreRenderQueue.cpp
@@ -312,6 +312,7 @@ namespace Ogre
             Hlms *hlms = mHlmsManager->getHlms( static_cast<HlmsTypes>( i ) );
             if( hlms )
             {
+                if (!mSceneManager->getCurrentShadowNode()) continue;
                 mPassCache[i] = hlms->preparePassHash( mSceneManager->getCurrentShadowNode(), casterPass,
                                                        dualParaboloid, mSceneManager );
             }
