--- a/OgreMain/include/OgreGpuProgramParams.h
+++ b/OgreMain/include/OgreGpuProgramParams.h
@@ -1524,7 +1524,7 @@ namespace Ogre {

     public:
         GpuProgramParameters();
-        ~GpuProgramParameters() {}
+        ~GpuProgramParameters();

         /// Copy constructor
         GpuProgramParameters(const GpuProgramParameters& oth);
@@ -1922,7 +1922,7 @@ namespace Ogre {
         // const bool* getBoolPointer(size_t pos) const { return &mBoolConstants[pos]; }

         /// Get a reference to the list of auto constant bindings
-        const AutoConstantList& getAutoConstantList() const { return mAutoConstants; }
+        const AutoConstantList& getAutoConstantList() const;

         /** Sets up a constant which will automatically be updated by the system.
             @remarks
@@ -1983,7 +1983,7 @@ namespace Ogre {
         /** Gets an iterator over the automatic constant bindings currently in place. */
         AutoConstantIterator getAutoConstantIterator(void) const;
         /// Gets the number of int constants that have been set
-        size_t getAutoConstantCount(void) const { return mAutoConstants.size(); }
+        size_t getAutoConstantCount(void) const;
         /** Gets a specific Auto Constant entry if index is in valid range
             otherwise returns a NULL
             @param index which entry is to be retrieved
@@ -1991,7 +1991,7 @@ namespace Ogre {
         GpuProgramParameters_AutoConstantEntry *getAutoConstantEntry(
             const size_t index );
         /** Returns true if this instance has any automatic constants. */
-        bool hasAutoConstants( void ) const { return !( mAutoConstants.empty() ); }
+        bool hasAutoConstants( void ) const;
         /** Finds an auto constant that's affecting a given logical parameter
             index for floating-point values.
             @note Only applicable for low-level programs.

--- a/OgreMain/src/OgreGpuProgramParams.cpp
+++ b/OgreMain/src/OgreGpuProgramParams.cpp
@@ -940,6 +940,8 @@ namespace Ogre
     {
     }
     //-----------------------------------------------------------------------------
+    GpuProgramParameters::~GpuProgramParameters() {}
+    //-----------------------------------------------------------------------------

     GpuProgramParameters::GpuProgramParameters(const GpuProgramParameters& oth)
     {
@@ -2235,6 +2237,12 @@ namespace Ogre
         }
     }
     //-----------------------------------------------------------------------------
+    bool GpuProgramParameters::hasAutoConstants( void ) const { return !( mAutoConstants.empty() ); }
+    //-----------------------------------------------------------------------------
+    size_t GpuProgramParameters::getAutoConstantCount(void) const { return mAutoConstants.size(); }
+    //-----------------------------------------------------------------------------
+    const GpuProgramParameters::AutoConstantList& GpuProgramParameters::getAutoConstantList() const { return mAutoConstants; }
+    //-----------------------------------------------------------------------------
     void GpuProgramParameters::setAutoConstant(size_t index, AutoConstantType acType, size_t extraInfo)
     {
         // Get auto constant definition for sizing
