--- MODULE.bazel
+++ MODULE.bazel
@@ -6,23 +6,13 @@
 
 module(
     name = "kleidiai",
+    version = "0.0.0-20260506-d41219d3db",
+    bazel_compatibility = [">=7.2.1"],
     repo_name = "com_arm_kleidiai",
 )
 
-kai_local_archive = use_repo_rule("//:kai_defs.bzl", "kai_local_archive")
-
 bazel_dep(name = "bazel_skylib", version = "1.4.2")
+bazel_dep(name = "google_benchmark", version = "1.9.4", repo_name = "com_google_benchmark")
+bazel_dep(name = "googletest", version = "1.17.0", repo_name = "com_google_googletest")
 bazel_dep(name = "platforms", version = "1.0.0")
 bazel_dep(name = "rules_cc", version = "0.2.16")
-
-kai_local_archive(
-    name = "com_google_googletest",
-    archive = "//:third_party/googletest-v1.17.0.zip",
-    strip_prefix = "googletest-1.17.0",
-)
-
-kai_local_archive(
-    name = "com_google_benchmark",
-    archive = "//:third_party/benchmark-v1.9.4.zip",
-    strip_prefix = "benchmark-v1.9.4",
-)
