diff --git a/.bazelrc b/.bazelrc
index 1e36b9b..fa84eda 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -79,7 +79,7 @@ common --lockfile_mode=off
 
 # Allow bazel configuration flags to locate the core rules to avoid adding local
 # override paths to the `MODULE.bazel` file and breaking downstream consumers.
-common --override_module=rules_rust=../../
+# common --override_module=rules_rust=../../
 
 ###############################################################################
 ## Custom user flags
