diff --git a/.bazelrc b/.bazelrc
index 1e36b9b..664791c 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -65,7 +65,7 @@ build --incompatible_merge_fixed_and_default_shell_env
 
 # Disable legacy __init__.py behavior which is known to conflict with
 # modern python versions (3.9+)
-build --incompatible_default_to_explicit_init_py
+# build --incompatible_default_to_explicit_init_py
 
 ###############################################################################
 ## Bzlmod
