diff --git a/MODULE.bazel b/MODULE.bazel
index 842544e..5e5ee92 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -32,13 +32,13 @@ register_toolchains(
 bazel_dep(
     name = "rules_rust",
     version = "0.38.0",
-    # In released versions: dev_dependency = True
+    dev_dependency = True
 )
 
 rust = use_extension(
     "@rules_rust//rust:extensions.bzl",
     "rust",
-    # In released versions: dev_dependency = True
+    dev_dependency = True
 )
 
 rust.toolchain(
@@ -49,13 +49,13 @@ use_repo(rust, "rust_toolchains")
 
 register_toolchains(
     "@rust_toolchains//:all",
-    # In released versions: dev_dependency = True
+    dev_dependency = True
 )
 
 crate = use_extension(
     "@rules_rust//crate_universe:extension.bzl",
     "crate",
-    # In released versions: dev_dependency = True
+    dev_dependency = True
 )
 
 crate.from_cargo(
