diff --git a/cachelib/allocator/memory/Slab.h b/cachelib/allocator/memory/Slab.h
index 4f7603c..e8ee5dc 100644
--- a/cachelib/allocator/memory/Slab.h
+++ b/cachelib/allocator/memory/Slab.h
@@ -17,7 +17,7 @@
 #pragma once
 
 #include <folly/logging/xlog.h>
-#include <gtest/gtest.h>
+#include <gtest/gtest_prod.h>
 
 #include <cstddef>
 #include <cstdint>
diff --git a/cachelib/allocator/CacheAllocator.h b/cachelib/allocator/CacheAllocator.h
index 920202d..49778df 100644
--- a/cachelib/allocator/CacheAllocator.h
+++ b/cachelib/allocator/CacheAllocator.h
@@ -24,7 +24,7 @@
 #include <folly/json/DynamicConverter.h>
 #include <folly/logging/xlog.h>
 #include <folly/synchronization/SanitizeThread.h>
-#include <gtest/gtest.h>
+#include <gtest/gtest_prod.h>
 
 #include <chrono>
 #include <functional>
diff --git a/cachelib/allocator/Handle.h b/cachelib/allocator/Handle.h
index b72064d..b9f9271 100644
--- a/cachelib/allocator/Handle.h
+++ b/cachelib/allocator/Handle.h
@@ -20,7 +20,7 @@
 #include <folly/fibers/Baton.h>
 #include <folly/futures/Future.h>
 #include <folly/logging/xlog.h>
-#include <gtest/gtest.h>
+#include <gtest/gtest_prod.h>
 
 #include <iostream>
 #include <mutex>
