diff --git sdk_core/comm/define.h sdk_core/comm/define.h
index 66996bf..2de41ff 100644
--- sdk_core/comm/define.h
+++ sdk_core/comm/define.h
@@ -31,6 +31,7 @@
 #include <functional>
 #include <vector>
 #include <atomic>
+#include <cstdint>
 
 #include "livox_lidar_def.h"
 
diff --git sdk_core/logger_handler/file_manager.h sdk_core/logger_handler/file_manager.h
index 242a431..14cd76a 100644
--- sdk_core/logger_handler/file_manager.h
+++ sdk_core/logger_handler/file_manager.h
@@ -28,6 +28,7 @@
 #include <string>
 #include <vector>
 #include <map>
+#include <cstdint>
 
 namespace livox {
 namespace lidar {
