--- MODULE.bazel
+++ MODULE.bazel
@@ -0,0 +1,11 @@
+module(
+    name = "fastdds",
+    version = "3.1.1",
+    compatibility_level = 2,
+)
+
+bazel_dep(name = "fastcdr", version = "2.2.2")
+bazel_dep(name = "foonathan_memory", version = "0.7.3.bcr.2")
+bazel_dep(name = "tinyxml2", version = "10.0.0")
+bazel_dep(name = "asio", version = "1.28.2")
+bazel_dep(name = "rules_foreign_cc", version = "0.13.0")
