--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,25 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "websocketpp",
+    hdrs = glob(["websocketpp/**/*.hpp"]),
+    defines = [
+        "ASIO_STANDALONE",
+        "_WEBSOCKETPP_CPP11_STL_",
+    ],
+    includes = ["."],
+    linkopts = select({
+        "@platforms//os:osx": [
+            "-lpthread",
+        ],
+        "//conditions:default": [
+            "-lpthread",
+            "-lrt",
+        ],
+    }),
+    visibility = ["//visibility:public"],
+    deps = [
+        "@asio",
+        "@zlib",
+    ],
+)
