===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,8 +1,8 @@
 module(
     name = "grpc_kotlin",
     # Note: the Publish-to-BCR app will patch this line to stamp the version being published.
-    version = "0.0.0",
+    version = "1.5.0-rc.5",
     compatibility_level = 1,
 )
 
 bazel_dep(name = "protobuf", version = "31.1")  # Max version without causing compiler errors.
