===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -5,3 +5,4 @@
 )
 
 bazel_dep(name = "rules_python", version = "2.0.3")
+bazel_dep(name = "rules_shell", version = "0.8.0")
===================================================================
--- a/tools/BUILD.bazel
+++ b/tools/BUILD.bazel
@@ -1,22 +1,18 @@
 package(default_visibility = ["//visibility:public"])
 
-load("@rules_python//python:defs.bzl", "py_binary")
+load("@rules_shell//shell:sh_binary.bzl", "sh_binary")
 
-py_binary(
+sh_binary(
     name = "assert_manifest",
     srcs = ["assert_manifest.py"],
-    main = "assert_manifest.py",
-    python_version = "PY3",
 )
 
-py_binary(
+sh_binary(
     name = "package_omniverse",
     srcs = ["package_omniverse.py"],
-    main = "package_omniverse.py",
-    python_version = "PY3",
 )
 
 exports_files([
     "assert_manifest.py",
     "package_omniverse.py",
 ])
