From 23581947b670d9c93d67f9768c13fdc0f396f777 Mon Sep 17 00:00:00 2001
From: Corentin Le Molgat <corentinl@google.com>
Date: Fri, 10 Jul 2026 14:58:29 +0200
Subject: [PATCH 2/2] fix_build_bazel

---
 BUILD.bazel | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/BUILD.bazel b/BUILD.bazel
index 6a6589bc46..f192e72517 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -118,6 +118,10 @@ cc_library(
     ],
     linkopts = select({
         "@rules_cc//cc/compiler:msvc-cl": ["-DEFAULTLIB:shell32.lib"],
+        "@platforms//os:linux": [
+          "-Wl,--no-as-needed -ldl",
+          "-pthread",
+        ],
         "//conditions:default": ["-lpthread"],
     }),
     visibility = ["//visibility:public"],
