--- a/gutil/status.h
+++ b/gutil/status.h
@@ -328,9 +328,13 @@
 //     RETURN_IF_ERROR(Foo()).LogError() << "Additional Info";
 //     return absl::OkStatus()
 //   }
-#define RETURN_IF_ERROR(expr)                     \
-  for (absl::Status status = expr; !status.ok();) \
-  return gutil::StatusBuilder(std::move(status))
+// Use a unique variable name to avoid shadowing user variables named `status`.
+#define __RETURN_IF_ERROR_VAL_DIRECT(arg) __RETURN_IF_ERROR_STATUS_##arg
+#define __RETURN_IF_ERROR_VAL(arg) __RETURN_IF_ERROR_VAL_DIRECT(arg)
+#define RETURN_IF_ERROR(expr)                                                \
+  for (absl::Status __RETURN_IF_ERROR_VAL(__LINE__) = expr;                  \
+       !__RETURN_IF_ERROR_VAL(__LINE__).ok();)                               \
+  return gutil::StatusBuilder(std::move(__RETURN_IF_ERROR_VAL(__LINE__)))
 
 // These macros help create unique variable names for ASSIGN_OR_RETURN. Not for
 // public use.
