diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..037be46
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,15 @@
+module(
+    name = "triton-inference-server-common",
+    version = "0.0.0-20240821-578491f",
+    compatibility_level = 0,
+)
+
+bazel_dep(name = "rules_proto", version = "6.0.2")
+bazel_dep(name = "rules_proto_grpc_cpp", version = "5.0.0")
+bazel_dep(name = "toolchains_protoc", version = "0.3.1")
+
+protoc = use_extension("@toolchains_protoc//protoc:extensions.bzl", "protoc")
+protoc.toolchain(
+    google_protobuf = "com_google_protobuf",
+    version = "v27.1",
+)
diff --git a/protobuf/BUILD b/protobuf/BUILD
new file mode 100644
index 0000000..c6cf835
--- /dev/null
+++ b/protobuf/BUILD
@@ -0,0 +1,24 @@
+load("@rules_proto//proto:defs.bzl", "proto_library")
+load("@rules_proto_grpc_cpp//:defs.bzl", "cpp_grpc_library", "cpp_proto_library")
+
+package(default_visibility = ["//visibility:public"])
+
+proto_library(
+    name = "common_proto",
+    srcs = [
+        "grpc_service.proto",
+        "health.proto",
+        "model_config.proto",
+    ],
+    import_prefix = "triton/",
+)
+
+cpp_proto_library(
+    name = "common_cc_proto",
+    protos = [":common_proto"],
+)
+
+cpp_grpc_library(
+    name = "common_cc_grpc",
+    protos = [":common_proto"],
+)
diff --git a/protobuf/grpc_service.proto b/protobuf/grpc_service.proto
index 451dd74..d86c024 100644
--- a/protobuf/grpc_service.proto
+++ b/protobuf/grpc_service.proto
@@ -30,7 +30,7 @@ package inference;
 
 //@@.. cpp:namespace:: inference
 
-import "model_config.proto";
+import "triton/protobuf/model_config.proto";
 
 //@@
 //@@.. cpp:var:: service InferenceService
