From f718b2ac70fd5ec79dcd7a29040998625c106b4b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?F=C3=A6don=20J=C3=B3hannes=20Sinis?= <fs@treble.tech>
Date: Thu, 22 May 2025 12:42:58 +0000
Subject: [PATCH] Remove imgui_impl_opengl3 from polyscope.cpp

---
 src/polyscope.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/polyscope.cpp b/src/polyscope.cpp
index 47dc549..b3190df 100644
--- a/src/polyscope.cpp
+++ b/src/polyscope.cpp
@@ -20,8 +20,6 @@
 #include "nlohmann/json.hpp"
 using json = nlohmann::json;

-#include "backends/imgui_impl_opengl3.h"
-
 namespace polyscope {

 // Note: Storage for global members lives in state.cpp and options.cpp
--
2.34.1
