diff --git a/BUILD.bazel b/BUILD.bazel
index ada6873..58fd99a 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -371,16 +371,17 @@ cc_library(
     name = "_InstructionCounter",
     srcs = glob(["Sources/_InstructionCounter/src/*.c"]),
     hdrs = glob(["Sources/_InstructionCounter/include/*.h"]),
+    aspect_hints = ["@build_bazel_rules_swift//swift:auto_module"],
     includes = ["Sources/_InstructionCounter/include"],
-    tags = ["swift_module=_InstructionCounter"],
     visibility = ["//visibility:private"],
 )
 
 cc_library(
     name = "_SwiftLibraryPluginProviderCShims",
+    srcs = glob(["Sources/_SwiftLibraryPluginProviderCShims/*.c"]),
     hdrs = glob(["Sources/_SwiftLibraryPluginProviderCShims/include/*.h"]),
+    aspect_hints = ["@build_bazel_rules_swift//swift:auto_module"],
     includes = ["Sources/_SwiftLibraryPluginProviderCShims/include"],
-    tags = ["swift_module=_SwiftLibraryPluginProviderCShims"],
     visibility = ["//visibility:private"],
 )
 
@@ -388,8 +389,8 @@ cc_library(
     name = "_SwiftSyntaxCShims",
     srcs = glob(["Sources/_SwiftSyntaxCShims/*.c"]),
     hdrs = glob(["Sources/_SwiftSyntaxCShims/include/*.h"]),
+    aspect_hints = ["@build_bazel_rules_swift//swift:auto_module"],
     includes = ["Sources/_SwiftSyntaxCShims/include"],
-    tags = ["swift_module=_SwiftSyntaxCShims"],
     visibility = ["//visibility:private"],
 )
 
