diff --git a/BUILD.bazel b/BUILD.bazel
index 5a710dd..ada6873 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -11,6 +11,7 @@
 ##===----------------------------------------------------------------------===##
 
 load("@build_bazel_rules_apple//apple/testing/default_runner:ios_xctestrun_runner.bzl", "ios_xctestrun_runner")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//utils/bazel:swift_syntax_library.bzl", "swift_syntax_library", "swift_syntax_test")
 
 package(default_visibility = ["//visibility:public"])
diff --git a/utils/bazel/opt_wrapper.bzl b/utils/bazel/opt_wrapper.bzl
index 63d48df..e684de8 100644
--- a/utils/bazel/opt_wrapper.bzl
+++ b/utils/bazel/opt_wrapper.bzl
@@ -18,6 +18,7 @@ benefits of compiler optimizations.
 """
 
 load("@build_bazel_rules_swift//swift:swift.bzl", "SwiftInfo")
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 def _force_opt_impl(settings, _attr):
     return {
