--- a/python/python3/cxxtest/cxxtestgen.py	2014-06-04 05:57:07.000000000 +0200
+++ b/python/python3/cxxtest/cxxtestgen.py	2026-05-06 09:56:42.219021381 +0200
@@ -298,6 +298,8 @@
     output.write( "\n" )
     if options.xunit_printer:
         output.write( "#include <fstream>\n" )
+        if options.xunit_file.startswith("env::"):
+            output.write( "#include <cstdlib>\n" )
     if options.haveStandardLibrary:
         output.write( "#define _CXXTEST_HAVE_STD\n" )
     if options.haveExceptionHandling:
@@ -335,8 +337,13 @@
     else:
         tester_t = "CxxTest::%s" % (options.runner)
     if options.xunit_printer:
-       output.write( '    std::ofstream ofstr("%s");\n' % options.xunit_file )
-       output.write( '    %s tmp(ofstr);\n' % tester_t )
+        if options.xunit_file.startswith("env::"):
+            env_var = options.xunit_file[5:]
+            output.write( '    const char* _xunit_out_ = std::getenv("%s");\n' % env_var )
+            output.write( '    std::ofstream ofstr(_xunit_out_ ? _xunit_out_ : "TEST-%s.xml");\n' % options.world )
+        else:
+            output.write( '    std::ofstream ofstr("%s");\n' % options.xunit_file )
+        output.write( '    %s tmp(ofstr);\n' % tester_t )
     else:
        output.write( '    %s tmp;\n' % tester_t )
     output.write( '    CxxTest::RealWorldDescription::_worldName = "%s";\n' % options.world )
