diff --git buildrs/opaque_types_generator.rs buildrs/opaque_types_generator.rs
index f0cc1271..59247658 100644
--- buildrs/opaque_types_generator.rs
+++ buildrs/opaque_types_generator.rs
@@ -126,7 +126,8 @@ fn produce_opaque_types_data() -> (String, PathBuf) {
         feature_args.push(feature);
     }
 
-    let mut command = std::process::Command::new("cargo");
+    let cargo = std::env::var_os("CARGO").unwrap_or_else(|| "cargo".into());
+    let mut command = std::process::Command::new(cargo);
     command
         .arg("build")
         .args(feature_args)
