--- /dev/null
+++ a/wsg_32_driver/BUILD.bazel
@@ -0,0 +1,38 @@
+package(
+    default_visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "libwsg32",
+    srcs = [
+        "src/checksum.cpp",
+        "src/cmd.c",
+        "src/common.cpp",
+        "src/functions.cpp",
+        "src/interface.cpp",
+        "src/msg.c",
+        "src/serial.c",
+        "src/tcp.c",
+        "src/udp.c",
+    ],
+    hdrs = [
+        "include/wsg_32/char.h",
+        "include/wsg_32/checksum.h",
+        "include/wsg_32/cmd.h",
+        "include/wsg_32/common.h",
+        "include/wsg_32/functions.h",
+        "include/wsg_32/interface.h",
+        "include/wsg_32/msg.h",
+        "include/wsg_32/serial.h",
+        "include/wsg_32/tcp.h",
+        "include/wsg_32/udp.h",
+    ],
+    copts = [
+        "-fexceptions",
+        "-Wno-c++98-c++11-compat",
+        "-Wno-sizeof-pointer-memaccess",
+    ],
+    includes = [
+        "include",
+    ],
+)
