--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -59,7 +59,7 @@
 bazel_dep(name = "rules_android", version = "0.7.3")
 bazel_dep(name = "rules_cc", version = "0.0.9")
 bazel_dep(name = "rules_java", version = "9.1.0")
-bazel_dep(name = "rules_jvm_external", version = "6.0")
+bazel_dep(name = "rules_jvm_external", version = "6.5")

 android_sdk_repository_extension = use_extension(
     "@rules_android//rules/android_sdk_repository:rule.bzl",
