diff --git a/BUILD.bazel b/BUILD.bazel
index 6d7ac3d..b043ace 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
 licenses(["unencumbered"])  # Public Domain or MIT
 
 exports_files(["LICENSE"])
@@ -14,9 +16,9 @@ cc_library(
         "include/json/allocator.h",
         "include/json/assertions.h",
         "include/json/config.h",
-        "include/json/json_features.h",
         "include/json/forwards.h",
         "include/json/json.h",
+        "include/json/json_features.h",
         "include/json/reader.h",
         "include/json/value.h",
         "include/json/version.h",
diff --git a/include/json/config.h b/include/json/config.h
index a7d9bdf..a85a94e 100644
--- a/include/json/config.h
+++ b/include/json/config.h
@@ -124,3 +124,5 @@
 using LargestUInt = UInt64;
-#define JSON_HAS_INT64
+#ifndef JSON_HAS_INT64
+#define JSON_HAS_INT64
+#endif
 #endif // if defined(JSON_NO_INT64)

