--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,31 @@
+cc_library(
+    name = "hdrhistogram_c",
+    srcs = [
+        "src/hdr_encoding.c",
+        "src/hdr_histogram.c",
+        "src/hdr_histogram_log.c",
+        "src/hdr_interval_recorder.c",
+        "src/hdr_thread.c",
+        "src/hdr_time.c",
+        "src/hdr_writer_reader_phaser.c",
+    ],
+    hdrs = [
+        "src/hdr_atomic.h",
+        "src/hdr_encoding.h",
+        "src/hdr_endian.h",
+        "src/hdr_histogram.h",
+        "src/hdr_histogram_log.h",
+        "src/hdr_interval_recorder.h",
+        "src/hdr_tests.h",
+        "src/hdr_thread.h",
+        "src/hdr_time.h",
+        "src/hdr_writer_reader_phaser.h",
+    ],
+    copts = [
+        "-std=gnu99",
+        "-Wno-implicit-function-declaration",
+        "-Wno-error",
+    ],
+    visibility = ["//visibility:public"],
+    deps = ["@zlib"],
+)
