===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,39 +1,14 @@
-"bazelndf dependencies"
-
 module(
     name = "bazeldnf",
-    version = "v0.5.9",
+    version = "v0.7.0-alpha1",
     compatibility_level = 0,
 )

 bazeldnf = use_extension("//bazeldnf:extensions.bzl", "bazeldnf")
 bazeldnf.toolchain()
-bazeldnf.rpm(
-    name = "bazeldnf_internal_abseil-cpp-devel",
-    sha256 = "1393c28f5a3f3029769bbe3436b3eec58c7b11761c4ad6695c309b21474d9804",
-    urls = [
-        "https://kojipkgs.fedoraproject.org//packages/abseil-cpp/20240722.1/1.fc42/x86_64/abseil-cpp-devel-20240722.1-1.fc42.x86_64.rpm",
-    ],
-)
-bazeldnf.rpm(
-    name = "bazeldnf_internal_libvirt-devel",
-    sha256 = "dba37bbe57903afe49b5666f1781eb50001baa81af4584b355db0b6a2afad9fa",
-    urls = [
-        "https://kojipkgs.fedoraproject.org//packages/libvirt/11.0.0/1.fc42/x86_64/libvirt-devel-11.0.0-1.fc42.x86_64.rpm",
-    ],
-)
-bazeldnf.rpm(
-    name = "bazeldnf_internal_libvirt-libs",
-    sha256 = "aac272a2ace134b5ef60a41e6624deb24331e79c76699ef6cef0dca22d94ac7e",
-    urls = [
-        "https://kojipkgs.fedoraproject.org//packages/libvirt/11.0.0/1.fc42/x86_64/libvirt-libs-11.0.0-1.fc42.x86_64.rpm",
-    ],
-)
+
 use_repo(
     bazeldnf,
-    "bazeldnf_internal_abseil-cpp-devel",
-    "bazeldnf_internal_libvirt-devel",
-    "bazeldnf_internal_libvirt-libs",
     "bazeldnf_toolchains",
 )

@@ -48,9 +23,6 @@
 bazel_dep(name = "gazelle", version = "0.42.0")
 bazel_dep(name = "rules_go", version = "0.52.0")

-bazel_dep(name = "protobuf", version = "27.3", dev_dependency = True)
-bazel_dep(name = "toolchains_protoc", version = "0.3.2", dev_dependency = True)
-
 go_deps = use_extension("@gazelle//:extensions.bzl", "go_deps")
 go_deps.from_file(go_mod = "//:go.mod")
 use_repo(
@@ -66,25 +38,3 @@
     "org_golang_x_crypto",
     "org_golang_x_exp",
 )
-
-# deps only needed for the repo internals
-bazel_dep(name = "bazel_skylib_gazelle_plugin", version = "1.7.1", dev_dependency = True)
-bazel_dep(name = "aspect_bazel_lib", version = "2.8.1", dev_dependency = True)
-bazel_dep(name = "buildifier_prebuilt", version = "6.4.0", dev_dependency = True)
-bazel_dep(name = "rules_pkg", version = "1.0.1", dev_dependency = True)
-bazel_dep(name = "stardoc", version = "0.8.0", dev_dependency = True)
-bazel_dep(name = "rules_proto", version = "6.0.0", dev_dependency = True)
-bazel_dep(name = "rules_java", version = "7.9.0", dev_dependency = True)
-
-http_jar = use_repo_rule("@bazel_tools//tools/build_defs/repo:http.bzl", "http_jar")
-
-http_jar(
-    name = "protobuf-java",
-    integrity = "sha256-0C+GOpCj/8d9Xu7AMcGOV58wx8uY8/OoFP6LiMQ9O8g=",
-    urls = ["https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java/4.27.3/protobuf-java-4.27.3.jar"],
-)
-
-register_toolchains(
-    "//tools/toolchains:all",
-    dev_dependency = True,
-)
