diff --git a/aom_dsp/butteraugli.c b/aom_dsp/butteraugli.c
index b373e12..c704c66 100644
--- a/aom_dsp/butteraugli.c
+++ b/aom_dsp/butteraugli.c
@@ -14,7 +14,7 @@
 
 #include "aom_dsp/butteraugli.h"
 #include "aom_mem/aom_mem.h"
-#include "third_party/libyuv/include/libyuv/convert_argb.h"
+#include "libyuv/convert_argb.h"
 
 int aom_calc_butteraugli(const YV12_BUFFER_CONFIG *source,
                          const YV12_BUFFER_CONFIG *distorted, int bit_depth,
diff --git a/apps/aomdec.c b/apps/aomdec.c
index 9933f6a..f8e9a9a 100644
--- a/apps/aomdec.c
+++ b/apps/aomdec.c
@@ -43,7 +43,7 @@
 #include "common/y4menc.h"
 
 #if CONFIG_LIBYUV
-#include "third_party/libyuv/include/libyuv/scale.h"
+#include "libyuv/scale.h"
 #endif
 
 static const char *exec_name;
diff --git a/apps/aomenc.c b/apps/aomenc.c
index 3caae90..b398f9a 100644
--- a/apps/aomenc.c
+++ b/apps/aomenc.c
@@ -47,7 +47,7 @@
 #include "stats/rate_hist.h"
 
 #if CONFIG_LIBYUV
-#include "third_party/libyuv/include/libyuv/scale.h"
+#include "libyuv/scale.h"
 #endif
 
 /* Swallow warnings about unused results of fread/fwrite */
diff --git a/test/datarate_test.cc b/test/datarate_test.cc
index 3d86128..7a7cf44 100644
--- a/test/datarate_test.cc
+++ b/test/datarate_test.cc
@@ -22,7 +22,7 @@
 #include "aom/aom_codec.h"
 
 #if CONFIG_LIBYUV
-#include "third_party/libyuv/include/libyuv/scale.h"
+#include "libyuv/scale.h"
 #endif
 
 namespace datarate_test {
