diff --git a/BUILD.bazel b/BUILD.bazel
index 84a3990..4f443d5 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -2,13 +2,23 @@
 # Loaders
 ###############################################################################
 
-load("@rules_cc//cc:defs.bzl", "cc_library")
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library")
 load("//:bazel/variables.bzl", "COPTS")
 
 ###############################################################################
 # Libraries
 ###############################################################################
 
+# Aggregate library for maliput_geopackage.
+cc_library(
+    name = "maliput_geopackage",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":builder",
+        ":geopackage",
+    ],
+)
+
 # Private geopackage parser headers (internal implementation)
 cc_library(
     name = "private_geopackage_headers",
