diff --git a/include/boost/histogram/detail/static_vector.hpp b/include/boost/histogram/detail/static_vector.hpp
--- a/include/boost/histogram/detail/static_vector.hpp
+++ b/include/boost/histogram/detail/static_vector.hpp
@@ -9,6 +9,7 @@

 #include <algorithm>
 #include <boost/throw_exception.hpp>
+#include <cassert>
 #include <stdexcept>

 namespace boost {
diff --git a/include/boost/histogram/detail/relaxed_tuple_size.hpp b/include/boost/histogram/detail/relaxed_tuple_size.hpp
--- a/include/boost/histogram/detail/relaxed_tuple_size.hpp
+++ b/include/boost/histogram/detail/relaxed_tuple_size.hpp
@@ -8,6 +8,7 @@
 #define BOOST_HISTOGRAM_DETAIL_RELAXED_TUPLE_SIZE_HPP

 #include <type_traits>
+#include <tuple>

 namespace boost {
 namespace histogram {
diff --git a/include/boost/histogram/detail/optional_index.hpp b/include/boost/histogram/detail/optional_index.hpp
--- a/include/boost/histogram/detail/optional_index.hpp
+++ b/include/boost/histogram/detail/optional_index.hpp
@@ -8,6 +8,7 @@
 #define BOOST_HISTOGRAM_DETAIL_OPTIONAL_INDEX_HPP

 #include <cassert>
+#include <cstddef>
 #include <cstdint>

 namespace boost {
diff --git a/include/boost/histogram/detail/priority.hpp b/include/boost/histogram/detail/priority.hpp
--- a/include/boost/histogram/detail/priority.hpp
+++ b/include/boost/histogram/detail/priority.hpp
@@ -8,6 +8,7 @@
 #define BOOST_HISTOGRAM_DETAIL_PRIORITY_HPP

 #include <cstdint>
+#include <cstddef>

 namespace boost {
 namespace histogram {
