--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,49 @@
+cc_library(
+  name = "cpuid",
+  srcs = [
+    "libcpuid/amd_code_t.h",
+    "libcpuid/asm-bits.c",
+    "libcpuid/asm-bits.h",
+    "libcpuid/centaur_code_t.h",
+    "libcpuid/cpuid_main.c",
+    "libcpuid/intel_code_t.h",
+    "libcpuid/libcpuid_internal.h",
+    "libcpuid/libcpuid_util.c",
+    "libcpuid/libcpuid_util.h",
+    "libcpuid/rdmsr.c",
+    "libcpuid/rdtsc.c",
+    "libcpuid/rdtsc.h",
+    "libcpuid/recog_amd.c",
+    "libcpuid/recog_amd.h",
+    "libcpuid/recog_centaur.c",
+    "libcpuid/recog_centaur.h",
+    "libcpuid/recog_intel.c",
+    "libcpuid/recog_intel.h",
+  ] + select({
+    "@platforms//os:windows": [
+      "libcpuid/msrdriver.c",
+    ],
+    "//conditions:default": [],
+  }),
+  hdrs = [
+    "libcpuid/libcpuid.h",
+    "libcpuid/libcpuid_constants.h",
+    "libcpuid/libcpuid_types.h",
+  ],
+  local_defines = [
+    "VERSION=\\\"0.6.5\\\"",
+  ] + select({
+        "@platforms//os:linux": ["_GNU_SOURCE"],
+        "//conditions:default": [],
+  }),
+  linkopts = select({
+      "@platforms//os:windows": [],
+      "//conditions:default": ["-lpthread"],
+  }),
+  visibility = ["//visibility:public"],
+)
+alias(
+    name = "libcpuid",
+    actual = ":cpuid",
+    visibility = ["//visibility:public"],
+)
