diff --git a/examples/bzlmod/.bazelrc b/examples/bzlmod/.bazelrc
index 45d7065..59786a9 100644
--- a/examples/bzlmod/.bazelrc
+++ b/examples/bzlmod/.bazelrc
@@ -3,10 +3,5 @@ import ../../.bazelrc
 common --enable_bzlmod
 common --incompatible_use_plus_in_repo_names
 
-common --check_direct_dependencies=error
-
 build --java_language_version=11
 build --tool_java_language_version=11
-
-# Verify that we can share a lock file
-common:bazel5 --enable_bzlmod=false
diff --git a/examples/bzlmod/MODULE.bazel b/examples/bzlmod/MODULE.bazel
index ffb675e..367fabb 100644
--- a/examples/bzlmod/MODULE.bazel
+++ b/examples/bzlmod/MODULE.bazel
@@ -5,7 +5,7 @@ module(
 
 # Required because `protobuf` contributes to the default namespace
 bazel_dep(name = "protobuf", version = "29.0", repo_name = None)
-bazel_dep(name = "rules_java", version = "8.13.0")
+bazel_dep(name = "rules_java", version = "8.14.0")
 bazel_dep(name = "rules_jvm_external", version = "0.0")
 local_path_override(
     module_name = "rules_jvm_external",  # matches the name of the `bazel_dep`
