From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matthew Clarkson <matthew.clarkson@arm.com>
Date: Fri, 16 Aug 2024 14:16:40 +0100
Subject: [PATCH] chore: ignore Bazel build symlinks

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 5af55a1..c32d031 100644
--- a/.gitignore
+++ b/.gitignore
@@ -8,3 +8,4 @@
 /build*
 /log*
 /scratch*
+/bazel-*
