From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matthew Clarkson <matthew.clarkson@arm.com>
Date: Mon, 19 Aug 2024 09:56:36 +0100
Subject: [PATCH] build: update module version

---
 MODULE.bazel | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MODULE.bazel b/MODULE.bazel
index a9eb7be..ac9b223 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,5 +1,6 @@
 module(
     name = "libarmgpuinfo",
+    version = "1.1.0+11140bec96",
 )
 
 bazel_dep(name = "rules_cc", version = "0.0.9")
