diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..08961b2
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,30 @@
+load("@bazel_skylib//lib:selects.bzl", "selects")
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "coroutine",
+    srcs = selects.with_or({
+        ("@platforms//os:linux", "@platforms//os:android", "@platforms//os:osx", "@platforms//os:ios", "@platforms//os:watchos", "@platforms//os:tvos"): [
+            "src/posix/stack_traits.cpp",
+        ],
+        "@boost.rules.tools//:windows_x86_64": [
+            "src/windows/stack_traits.cpp",
+        ],
+    }) + [
+        "src/detail/coroutine_context.cpp",
+    ],
+    deps = [
+        "@boost//:assert",
+        "@boost//:config",
+        "@boost//:context",
+        "@boost//:core",
+        "@boost//:exception",
+        "@boost//:move",
+        "@boost//:system",
+        "@boost//:throw_exception",
+        "@boost//:type_traits",
+        "@boost//:utility",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..0a43371
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,25 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.coroutine",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "bazel_skylib", version = "1.5.0")
+
+
+
+
+
+
+
+
+
+
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..e840db0
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:coroutine",
+    "@boost//:bind",
+    "@boost//:foreach",
+    "@boost//:lexical_cast",
+    "@boost//:test",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        deps = DEPS,
+    ),
+)
