--- MODULE.bazel
+++ MODULE.bazel
@@ -1,6 +1,6 @@
 module(
     name = "s2geometry",
-    version = "head",
+    version = "0.13.1",
     bazel_compatibility = [
         ">=7.0.0",
     ],
@@ -13,6 +13,8 @@ bazel_dep(name = "boringssl", version = "0.20250514.0")
 bazel_dep(name = "abseil-cpp", version = "20250814.1")
 bazel_dep(name = "googletest", version = "1.17.0")
 bazel_dep(name = "rules_cc", version = "0.1.1")
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "openssl", version = "3.3.1.bcr.1")

 # Forces usage of the patched re2 version that marks cc_configure_extension as a
 # dev dep as a workaround for https://github.com/bazelbuild/bazel/issues/24426.
