--- s2/util/math/exactfloat/BUILD
+++ s2/util/math/exactfloat/BUILD
@@ -8,8 +8,10 @@ cc_library(
         "//s2/base:logging",
         "@abseil-cpp//absl/log:log",
         "@abseil-cpp//absl/log:absl_check",
-        "@boringssl//:crypto",
-    ],
+    ] + select({
+        "//:use_openssl_setting": ["@openssl//:crypto"],
+        "//conditions:default": ["@boringssl//:crypto"],
+    }),
 )

 cc_test(
