diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..98229ee
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,20 @@
+module(name = "jsonnet", version = "0.20.0.bcr.1")
+
+bazel_dep(name = "googletest", version = "1.11.0", repo_name = "com_google_googletest")
+bazel_dep(name = "rules_jsonnet", version = "0.6.0", repo_name = "io_bazel_rules_jsonnet")
+bazel_dep(name = "rules_python", version = "1.0.0")
+
+PYTHON_VERSIONS = [
+    "3.10",
+    "3.11",
+    "3.12",
+    "3.13",
+]
+python = use_extension("@rules_python//python/extensions:python.bzl", "python")
+[
+    python.toolchain(
+        is_default = python_version == PYTHON_VERSIONS[-1],  # Choose last version as default.
+        python_version = python_version,
+    )
+    for python_version in PYTHON_VERSIONS
+]
diff --git a/python/BUILD b/python/BUILD
index 12936ad..0afe833 100644
--- a/python/BUILD
+++ b/python/BUILD
@@ -4,7 +4,7 @@ cc_binary(
     linkshared = 1,
     deps = [
         "//core:libjsonnet",
-        "@default_python3_headers//:headers",
+        "@rules_python//python/cc:current_py_cc_headers",
     ],
 )
 
