diff --git a/MODULE.bazel b/MODULE.bazel
index 046e33e..fc0dde5 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -31,9 +31,10 @@ python.toolchain(
 )
 use_repo(python, "python3_11_toolchains")
 
-register_toolchains(
-    "@python3_11_toolchains//:all",
-)
+# Dev only. Commented out by BCR publishing patch
+# register_toolchains(
+#     "@python3_11_toolchains//:all",
+# )
 
 pip = use_extension(
     "@rules_python//python:extensions.bzl",
