diff --git a/CMakeLists.txt b/CMakeLists.txt
index c911dd3..1be48f1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,6 +42,9 @@ if (MSVC)
   # MSVC_VERSION = 1920 corresponds to Visual Studio 2019
   if (NOT CMAKE_GENERATOR_PLATFORM AND MSVC_VERSION LESS 1920)
     set (_ARCH "x86")
+  elseif ("${_ARCH}" STREQUAL "")
+    # Updated to set the correct architecture in Bazel Central Registry CI.
+    set (_ARCH "x64")
   endif ()
 
   enable_language (ASM_MASM)
@@ -170,7 +173,7 @@ list(REMOVE_ITEM CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
 include_directories(include src "${PROJECT_BINARY_DIR}/src")
 add_subdirectory (src)
 
-add_library (mimick STATIC ${SOURCE_FILES})
+add_library (mimick ${SOURCE_FILES})
 target_include_directories(mimick INTERFACE $<INSTALL_INTERFACE:include>)
 
 foreach (F ${INTERFACE_FILES})
@@ -209,9 +212,9 @@ function (add_mimick_test _NAME)
   add_mimick_sample(${_NAME})
 endfunction ()
 
-enable_testing ()
-add_subdirectory (test)
-add_subdirectory (sample)
+# enable_testing ()
+# add_subdirectory (test)
+# add_subdirectory (sample)
 
 install(EXPORT mimick-targets DESTINATION lib/cmake/mimick)
 include(CMakePackageConfigHelpers)
