diff --git a/CMakeLists.txt b/CMakeLists.txt
index c911dd3..6c412f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,7 +170,7 @@ list(REMOVE_ITEM CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
 include_directories(include src "${PROJECT_BINARY_DIR}/src")
 add_subdirectory (src)
 
-add_library (mimick STATIC ${SOURCE_FILES})
+add_library (mimick ${SOURCE_FILES})
 target_include_directories(mimick INTERFACE $<INSTALL_INTERFACE:include>)
 
 foreach (F ${INTERFACE_FILES})
@@ -209,9 +209,9 @@ function (add_mimick_test _NAME)
   add_mimick_sample(${_NAME})
 endfunction ()
 
-enable_testing ()
-add_subdirectory (test)
-add_subdirectory (sample)
+# enable_testing ()
+# add_subdirectory (test)
+# add_subdirectory (sample)
 
 install(EXPORT mimick-targets DESTINATION lib/cmake/mimick)
 include(CMakePackageConfigHelpers)
