diff --git a/bazel/private/upb_proto_library_internal/aspect.bzl b/bazel/private/upb_proto_library_internal/aspect.bzl
index cd666cc..5d634c1 100644
--- a/bazel/private/upb_proto_library_internal/aspect.bzl
+++ b/bazel/private/upb_proto_library_internal/aspect.bzl
@@ -1,5 +1,7 @@
 """Implementation of the aspect that powers the upb_*_proto_library() rules."""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
+load("@rules_cc//cc/common:cc_common.bzl", "cc_common")
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
 load("//bazel/common:proto_common.bzl", "proto_common")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
diff --git a/bazel/private/upb_proto_library_internal/cc_library_func.bzl b/bazel/private/upb_proto_library_internal/cc_library_func.bzl
index 4753d18..17386cf 100644
--- a/bazel/private/upb_proto_library_internal/cc_library_func.bzl
+++ b/bazel/private/upb_proto_library_internal/cc_library_func.bzl
@@ -1,5 +1,7 @@
 """A function to compile C/C++ code, like cc_library() but from Starlark."""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
+load("@rules_cc//cc/common:cc_common.bzl", "cc_common")
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain", "use_cpp_toolchain")
 
 def upb_use_cpp_toolchain():
diff --git a/bazel/private/upb_proto_library_internal/rule.bzl b/bazel/private/upb_proto_library_internal/rule.bzl
index feb2929..2a9e7eb 100644
--- a/bazel/private/upb_proto_library_internal/rule.bzl
+++ b/bazel/private/upb_proto_library_internal/rule.bzl
@@ -1,3 +1,4 @@
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 """Internal rule implementation for upb_*_proto_library() rules."""
 
 def _filter_none(elems):
diff --git a/bazel/upb_c_proto_library.bzl b/bazel/upb_c_proto_library.bzl
index ba33dc6..8094485 100644
--- a/bazel/upb_c_proto_library.bzl
+++ b/bazel/upb_c_proto_library.bzl
@@ -1,5 +1,6 @@
 """upb_c_proto_library() exposes upb's generated C API for protobuf (foo.upb.h)"""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("//bazel:upb_minitable_proto_library.bzl", "UpbMinitableCcInfo", "upb_minitable_proto_library_aspect")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
 load("//bazel/private:upb_proto_library_internal/aspect.bzl", "upb_proto_aspect_impl")
diff --git a/bazel/upb_minitable_proto_library.bzl b/bazel/upb_minitable_proto_library.bzl
index ee84cee..1f9172f 100644
--- a/bazel/upb_minitable_proto_library.bzl
+++ b/bazel/upb_minitable_proto_library.bzl
@@ -1,5 +1,7 @@
 """upb_minitable_proto_library() exposes upb's generated minitables (foo.upb_minitable.h)"""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
+load("@rules_cc//cc/common:cc_common.bzl", "cc_common")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
 load("//bazel/private:upb_proto_library_internal/aspect.bzl", "upb_proto_aspect_impl")
 load("//bazel/private:upb_proto_library_internal/cc_library_func.bzl", "upb_use_cpp_toolchain")
diff --git a/bazel/upb_proto_reflection_library.bzl b/bazel/upb_proto_reflection_library.bzl
index 5fa7ccf..8a552d0 100644
--- a/bazel/upb_proto_reflection_library.bzl
+++ b/bazel/upb_proto_reflection_library.bzl
@@ -1,5 +1,6 @@
 """upb_c_proto_reflection_library() exposes upb reflection for protobuf (foo.upbdefs.h)"""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("//bazel:upb_minitable_proto_library.bzl", "UpbMinitableCcInfo", "upb_minitable_proto_library_aspect")
 load("//bazel/common:proto_common.bzl", "proto_common")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
diff --git a/build_defs/cc_proto_blacklist_test.bzl b/build_defs/cc_proto_blacklist_test.bzl
index 260abde..b6c8388 100644
--- a/build_defs/cc_proto_blacklist_test.bzl
+++ b/build_defs/cc_proto_blacklist_test.bzl
@@ -1,5 +1,6 @@
 """Contains a unittest to verify that `cc_proto_library` does not generate code for blacklisted `.proto` sources (i.e. WKPs)."""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("@bazel_skylib//lib:unittest.bzl", "asserts", "unittest")
 
 def _cc_proto_blacklist_test_impl(ctx):
diff --git a/build_defs/compiler_config_setting.bzl b/build_defs/compiler_config_setting.bzl
index f4d1f7b..67ba449 100644
--- a/build_defs/compiler_config_setting.bzl
+++ b/build_defs/compiler_config_setting.bzl
@@ -1,3 +1,4 @@
+load("@rules_cc//cc/common:cc_common.bzl", "cc_common")
 """Creates config_setting that allows selecting based on 'compiler' value."""
 
 def create_compiler_config_setting(name, value, visibility = None):
diff --git a/hpb/bazel/hpb_proto_library.bzl b/hpb/bazel/hpb_proto_library.bzl
index 797c6fc..1ba0a33 100644
--- a/hpb/bazel/hpb_proto_library.bzl
+++ b/hpb/bazel/hpb_proto_library.bzl
@@ -9,6 +9,8 @@
   - hpb_proto_library()
 """
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
+load("@rules_cc//cc/common:cc_common.bzl", "cc_common")
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "use_cpp_toolchain")
 load("//bazel:upb_proto_library.bzl", "GeneratedSrcsInfo", "UpbWrappedCcInfo", "upb_proto_library_aspect")
 load("//bazel/common:proto_common.bzl", "proto_common")
diff --git a/pkg/build_systems.bzl b/pkg/build_systems.bzl
index d53fbd7..0856bd0 100644
--- a/pkg/build_systems.bzl
+++ b/pkg/build_systems.bzl
@@ -1,5 +1,6 @@
 """Starlark utilities for working with other build systems."""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("@bazel_skylib//lib:paths.bzl", "paths")
 load("@rules_pkg//pkg:providers.bzl", "PackageFilegroupInfo", "PackageFilesInfo")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
diff --git a/pkg/cc_dist_library.bzl b/pkg/cc_dist_library.bzl
index c12b37e..60d0050 100644
--- a/pkg/cc_dist_library.bzl
+++ b/pkg/cc_dist_library.bzl
@@ -1,5 +1,7 @@
 # Rules for distributable C++ libraries
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
+load("@rules_cc//cc/common:cc_common.bzl", "cc_common")
 load("@rules_cc//cc:action_names.bzl", cc_action_names = "ACTION_NAMES")
 load("@rules_cc//cc:find_cc_toolchain.bzl", "find_cc_toolchain")
 
diff --git a/rust/bazel/aspects.bzl b/rust/bazel/aspects.bzl
index ff23da4..1bd39b5 100644
--- a/rust/bazel/aspects.bzl
+++ b/rust/bazel/aspects.bzl
@@ -1,5 +1,7 @@
 """This file implements rust_proto_library aspect."""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
+load("@rules_cc//cc/common:cc_common.bzl", "cc_common")
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
 
 # buildifier: disable=bzl-visibility
diff --git a/rust/test/rust_proto_library_unit_test/rust_proto_library_unit_test.bzl b/rust/test/rust_proto_library_unit_test/rust_proto_library_unit_test.bzl
index 1638388..f2fa109 100644
--- a/rust/test/rust_proto_library_unit_test/rust_proto_library_unit_test.bzl
+++ b/rust/test/rust_proto_library_unit_test/rust_proto_library_unit_test.bzl
@@ -1,5 +1,6 @@
 """This module contains unit tests for rust_proto_library and its aspect."""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("@bazel_skylib//lib:unittest.bzl", "analysistest", "asserts")
 load("//bazel:proto_library.bzl", "proto_library")
 load("//rust:defs.bzl", "rust_cc_proto_library", "rust_upb_proto_library")
diff --git a/toolchain/cc_toolchain_config.bzl b/toolchain/cc_toolchain_config.bzl
index 3bfe1bf..958037c 100644
--- a/toolchain/cc_toolchain_config.bzl
+++ b/toolchain/cc_toolchain_config.bzl
@@ -1,3 +1,4 @@
+load("@rules_cc//cc/common:cc_common.bzl", "cc_common")
 load("@bazel_tools//tools/build_defs/cc:action_names.bzl", "ACTION_NAMES")
 load(
     "@bazel_tools//tools/cpp:cc_toolchain_config_lib.bzl",
