diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..d1fa508
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,41 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+
+cc_library(
+    name = "ujson4c",
+    srcs = ["src/ujdecode.c"],
+    hdrs = ["src/ujdecode.h"],
+    includes = ["src"],
+    visibility = ["//visibility:public"],
+    deps = [":ultrajson"],
+)
+
+cc_library(
+    name = "ultrajson",
+    srcs = ["3rdparty/ultrajsondec.c"],
+    hdrs = ["3rdparty/ultrajson.h"],
+    includes = ["3rdparty"],
+    visibility = ["//visibility:private"],
+)
+
+cc_test(
+    name = "tests",
+    size = "small",
+    srcs = ["tests/tests.c"],
+    visibility = ["//visibility:public"],
+    deps = [":ujson4c"],
+)
+
+# Use `bazel test benchmark --test_output=all`
+cc_test(
+    name = "benchmark",
+    size = "small",
+    srcs = ["tests/benchmark.c"],
+    copts = ["-D__BENCHMARK__"],
+    data = ["tests/sample.json"],
+    defines = [
+        "MAX_ITERATIONS=10",
+        'SAMPLE_JSON_PATH=\\"tests/sample.json\\"',
+    ],
+    visibility = ["//visibility:public"],
+    deps = [":ujson4c"],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..0319fe8
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,13 @@
+"""
+Module: ujson4c
+Purpose: Provides the ujson4c library compileable as a Bazel target. Includes tests through Bazel
+"""
+
+module(
+    name = "ujson4c",
+    version = "0.0.0-20140408072550-e14f3fd5207f",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "rules_cc", version = "0.1.1")
+bazel_dep(name = "platforms", version = "0.0.11")
diff --git a/tests/benchmark.c b/tests/benchmark.c
index d3423ae..c57ebd4 100644
--- a/tests/benchmark.c
+++ b/tests/benchmark.c
@@ -193 +193,5 @@ int main ()
-	file = fopen("./sample.json", "rb");
+#ifndef SAMPLE_JSON_PATH
+#define SAMPLE_JSON_PATH "./sample.json"
+#endif
+
+	file = fopen(SAMPLE_JSON_PATH, "rb");
@@ -221,0 +226,6 @@ int main ()
+
+#ifdef MAX_ITERATIONS
+			static int i = 1;
+			if (i++ >= MAX_ITERATIONS)
+				break;
+#endif
diff --git a/tests/tests.c b/tests/tests.c
index 5643da3..2981b6a 100644
--- a/tests/tests.c
+++ b/tests/tests.c
@@ -36 +36 @@ www.github.com/esnme/ultrajson
-#include <malloc.h>
+#include <stdlib.h>
