commit a2a01082f45cbb83deaf04f71680aeb6893b3529 (HEAD -> master)
Author: hmelder <service@hugomelder.com>
Date:   Tue Jul 1 16:07:55 2025 +0200

    feat: Add BUILD.bazel

diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 00000000..0ee4b99f
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,44 @@
+""" Builds libxslt.
+"""
+
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]),
+)
+
+cache_entries = {
+    "CMAKE_INSTALL_LIBDIR": "lib",
+    "CMAKE_POSITION_INDEPENDENT_CODE": "ON",
+    "BUILD_SHARED_LIBS": "OFF",
+    #libxslt specific options.
+    "LIBXSLT_WITH_DEBUGGER": "OFF",
+    "LIBXSLT_WITH_CRYPTO": "OFF",
+    "LIBXSLT_WITH_MODULES": "OFF",
+    "LIBXSLT_WITH_PROFILER": "OFF",
+    "LIBXSLT_WITH_PROGRAMS": "OFF",
+    "LIBXSLT_WITH_PYTHON": "OFF",
+    "LIBXSLT_WITH_TESTS": "OFF",
+}
+
+cmake(
+    name = "libxslt",
+    env = {"CMAKE_BUILD_TYPE": "Release"},
+    lib_source = ":srcs",
+    cache_entries = cache_entries,
+    out_static_libs = select({
+        "@platforms//os:windows": [
+            "libxslts.lib",
+            "libexslts.lib",
+        ],
+        "//conditions:default": [
+            "libxslt.a",
+            "libexslt.a",
+        ],
+    }),
+    visibility = ["//visibility:public"],
+    deps = [
+      "@libxml2//:libxml2",
+    ],
+)
