diff --git a/c/enc/hash_longest_match_simd_inc.h b/c/enc/hash_longest_match_simd_inc.h
--- a/c/enc/hash_longest_match_simd_inc.h
+++ b/c/enc/hash_longest_match_simd_inc.h
@@ -220,11 +220,11 @@
   }
   {
     const uint8_t tag = hash & TAG_HASH_MASK;
-    const size_t head = (num[key] + 1) & self->block_mask_;
+    const size_t head = (num[key] + 1U) & self->block_mask_;
     uint64_t matches =
         GetMatchingTagMask(self->block_size_ / 16, tag, tag_bucket, head);
     /* Mask off any matches from uninitialized tags. */
-    uint16_t n = 65535 - num[key];
+    uint16_t n = (uint16_t)(65535 - num[key]);
     uint64_t block_has_unused_slots = self->block_size_ > n;
     uint64_t mask = (block_has_unused_slots << (n & (64 - 1))) - 1;
     matches &= mask;
diff --git a/c/enc/hash_longest_match64_simd_inc.h b/c/enc/hash_longest_match64_simd_inc.h
--- a/c/enc/hash_longest_match64_simd_inc.h
+++ b/c/enc/hash_longest_match64_simd_inc.h
@@ -247,11 +247,11 @@
     const uint8_t tag = hash & TAG_HASH_MASK;
     const uint32_t first4 = BrotliUnalignedRead32(data + cur_ix_masked);
     const size_t max_length_m4 = max_length - 4;
-    const size_t head = (num[key] + 1) & self->block_mask_;
+    const size_t head = (num[key] + 1U) & self->block_mask_;
     uint64_t matches =
         GetMatchingTagMask(self->block_size_ / 16, tag, tag_bucket, head);
     /* Mask off any matches from uninitialized tags. */
-    uint16_t n = 65535 - num[key];
+    uint16_t n = (uint16_t)(65535 - num[key]);
     uint64_t block_has_unused_slots = self->block_size_ > n;
     uint64_t mask = (block_has_unused_slots << (n & (64 - 1))) - 1;
     matches &= mask;
