diff --git a/test/src/test_grpc_context_20.cpp b/test/src/test_grpc_context_20.cpp
index fa603fdc..e3bc5ec0 100644
--- a/test/src/test_grpc_context_20.cpp
+++ b/test/src/test_grpc_context_20.cpp
@@ -30,7 +30,6 @@
 TEST_CASE("GrpcExecutor fulfills Executor TS concepts")
 {
     CHECK(asio::execution::executor<agrpc::GrpcExecutor>);
-    CHECK(asio::execution::executor_of<agrpc::GrpcExecutor, test::InvocableArchetype>);
 }

 #ifdef AGRPC_ASIO_HAS_SENDER_RECEIVER