diff --git a/proxygen/lib/services/WorkerThread.cpp b/proxygen/lib/services/WorkerThread.cpp
index ba0e182..48f0992 100644
--- a/proxygen/lib/services/WorkerThread.cpp
+++ b/proxygen/lib/services/WorkerThread.cpp
@@ -13,6 +13,7 @@
 #include <folly/String.h>
 #include <folly/io/async/EventBaseManager.h>
 #include <folly/io/async/IoUringBackend.h>
+#include <folly/portability/GFlags.h>
 #include <glog/logging.h>
 #include <signal.h>
 
