diff --git a/MODULE.bazel b/MODULE.bazel
index 5bbfafe7d5..8b24d7d8e2 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -7,12 +7,6 @@ bazel_dep(name = "bazel_skylib", version = "1.7.1")
 bazel_dep(name = "cpptoml", version = "0.1.1")
 bazel_dep(name = "platforms", version = "0.0.10")
 bazel_dep(name = "rules_cc", version = "0.0.9")
-
-bazel_dep(
-    name = "buildifier_prebuilt",
-    version = "7.1.2",
-    dev_dependency = True,
-)
-bazel_dep(name = "googletest", version = "1.15.2", dev_dependency = True)
-
+bazel_dep(name = "buildifier_prebuilt", version = "7.1.2")
+bazel_dep(name = "googletest", version = "1.15.2")
 bazel_dep(name = "ncurses", version = "6.4.20221231")
