diff --git a/common/BUILD b/common/BUILD
index 55d0e92..061e22a 100644
--- a/common/BUILD
+++ b/common/BUILD
@@ -22,6 +22,9 @@ package(default_visibility = ["//visibility:public"])
 licenses(["notice"])  # Apache 2
 
 proto_library(
-  name = "common_protocol_proto_lib",
-  srcs = ["Common.proto"],
+    name = "common_protocol_proto_lib",
+    srcs = [
+        "Command.proto",
+        "Common.proto",
+    ],
 )
