--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,40 @@
+""" Builds libmaxminddb.
+"""
+
+load(
+    "@cmake_configure_file//:cmake_configure_file.bzl",
+    "cmake_configure_file",
+)
+
+cmake_configure_file(
+    name = "generate_config_h",
+    src = "include/maxminddb_config.h.cmake.in",
+    out = "include/maxminddb_config.h",
+    cmakelists = ["CMakeLists.txt"],
+    defines = [
+        "MMDB_UINT128_USING_MODE=0",
+        "MMDB_UINT128_IS_BYTE_ARRAY=1",
+    ],
+    visibility = ["//visibility:private"],
+)
+
+cc_library(
+    name = "libmaxminddb",
+    srcs = [
+        "src/maxminddb.c",
+        "src/data-pool.c",
+        "src/data-pool.h",
+        "src/maxminddb-compat-util.h",
+    ],
+    hdrs = [
+        "include/maxminddb.h",
+        "include/maxminddb_config.h",
+    ],
+    copts = ["-w"],
+    includes = ["include"],
+    local_defines = [
+        "MMDB_LITTLE_ENDIAN=1",
+        'PACKAGE_VERSION=\\"1.10.0\\"',
+    ],
+    visibility = ["//visibility:public"],
+)
