diff --git a/src/user/user_mesh.cc b/src/user/user_mesh.cc
--- a/src/user/user_mesh.cc
+++ b/src/user/user_mesh.cc
@@ -71 +71 @@ extern "C" {
-#include "qhull_ra.h"
+#include "libqhull_r/qhull_ra.h"
diff --git a/src/xml/mjz/mjz_decoder.cc b/src/xml/mjz/mjz_decoder.cc
--- a/src/xml/mjz/mjz_decoder.cc
+++ b/src/xml/mjz/mjz_decoder.cc
@@ -34 +34 @@
-#include <miniz.h>
+#include "miniz.h"
diff --git a/src/xml/mjz/mjz_encoder.cc b/src/xml/mjz/mjz_encoder.cc
--- a/src/xml/mjz/mjz_encoder.cc
+++ b/src/xml/mjz/mjz_encoder.cc
@@ -33 +33 @@
-#include <miniz.h>
+#include "miniz.h"
