--- BUILD.bazel
+++ BUILD.bazel
@@ -92,7 +92,6 @@ cc_library(
         "@gz-math",
         "@gz-msgs",
         "@gz-msgs//:gzmsgs_cc_proto",
-        "@gz-rendering",
         "@gz-transport",
         "@gz-utils//:ImplPtr",
         "@gz-utils//:SuppressWarning",
@@ -243,8 +242,8 @@ gz_sensor_library(
     name = "depth_camera",
     srcs = ["src/DepthCameraSensor.cc"],
     hdrs = ["include/gz/sensors/DepthCameraSensor.hh"],
-    copts = [
-        "-Wno-private-header",  # Needs private headers from :gz-sensors and :camera
+    features = [
+        "-layering_check",  # Needs private headers from :gz-sensors and :camera
     ],
     includes = ["include"],
     visibility = ["//visibility:public"],
@@ -299,7 +298,6 @@ gz_sensor_library(
         "@gz-math",
         "@gz-msgs",
         "@gz-msgs//:gzmsgs_cc_proto",
-        "@gz-rendering",
         "@gz-transport",
         "@gz-utils//:SuppressWarning",
         "@sdformat",
@@ -484,8 +482,8 @@ gz_sensor_library(
         "src/RgbdCameraSensor.cc",
     ],
     hdrs = ["include/gz/sensors/RgbdCameraSensor.hh"],
-    copts = [
-        "-Wno-private-header",  # Needs private headers from :gz-sensors and :camera
+    features = [
+        "-layering_check",  # Needs private headers from :gz-sensors and :camera
     ],
     includes = ["include"],
     visibility = ["//visibility:public"],
