diff --git a/test/src/test_asio_grpc_20.cpp b/test/src/test_asio_grpc_20.cpp
index e6689008..47be847b 100644
--- a/test/src/test_asio_grpc_20.cpp
+++ b/test/src/test_asio_grpc_20.cpp
@@ -227,7 +227,7 @@ TEST_CASE_FIXTURE(test::GrpcClientServerTest, "awaitable run_with_deadline no ca
             const auto not_to_exceed = test::one_second_from_now();
             using namespace asio::experimental::awaitable_operators;
             auto result = co_await (RPC::request(grpc_context, *stub, client_context, request, response) ||
-                                    agrpc::Alarm(grpc_context).wait(not_to_exceed));
+                                    agrpc::Alarm(grpc_context).wait(not_to_exceed, asio::use_awaitable));
             status = std::move(std::get<0>(result));
             CHECK_LT(test::now(), not_to_exceed);
         });
@@ -247,7 +247,7 @@ TEST_CASE_FIXTURE(test::GrpcClientServerTest, "awaitable run_with_deadline and c
             grpc::ServerAsyncResponseWriter<test::msg::Response> writer{&server_context};
             CHECK(co_await agrpc::request(&test::v1::Test::AsyncService::RequestUnary, service, server_context, request,
                                           writer));
-            co_await agrpc::Alarm(grpc_context).wait(test::one_second_from_now());
+            co_await agrpc::Alarm(grpc_context).wait(test::one_second_from_now(), asio::use_awaitable);
             test::msg::Response response;
             server_finish_ok = co_await agrpc::finish(writer, response, grpc::Status::OK);
         },
@@ -258,7 +258,7 @@ TEST_CASE_FIXTURE(test::GrpcClientServerTest, "awaitable run_with_deadline and c
             const auto not_to_exceed = test::one_second_from_now();
             using namespace asio::experimental::awaitable_operators;
             auto result = co_await (RPC::request(grpc_context, *stub, client_context, request, response) ||
-                                    agrpc::Alarm(grpc_context).wait(test::hundred_milliseconds_from_now()));
+                                    agrpc::Alarm(grpc_context).wait(test::hundred_milliseconds_from_now(), asio::use_awaitable));
             is_cancelled = 1 == result.index();
             CHECK_LT(test::now(), not_to_exceed);
         });
