diff --git a/src/loggers/groot2_publisher.cpp b/src/loggers/groot2_publisher.cpp
index 6146507..83bbe74 100644
--- a/src/loggers/groot2_publisher.cpp
+++ b/src/loggers/groot2_publisher.cpp
@@ -1,8 +1,8 @@
 #include "behaviortree_cpp/loggers/groot2_publisher.h"
 #include "behaviortree_cpp/loggers/groot2_protocol.h"
 #include "behaviortree_cpp/xml_parsing.h"
-#include "cppzmq/zmq.hpp"
-#include "cppzmq/zmq_addon.hpp"
+#include <zmq.hpp>
+#include <zmq_addon.hpp>
 
 namespace BT
 {
diff --git a/tools/bt_recorder.cpp b/tools/bt_recorder.cpp
index c652f9a..a1266de 100644
--- a/tools/bt_recorder.cpp
+++ b/tools/bt_recorder.cpp
@@ -3,7 +3,7 @@
 #include <fstream>
 #include <signal.h>
 #include <fstream>
-#include "cppzmq/zmq.hpp"
+#include <zmq.hpp>
 #include "behaviortree_cpp/flatbuffers/BT_logger_generated.h"
 
 // http://zguide.zeromq.org/cpp:interrupt
