diff --git a/extensions.bzl b/extensions.bzl
new file mode 100644
index 0000000..aaa13de
--- /dev/null
+++ b/extensions.bzl
@@ -0,0 +1,13 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def _non_module_dependencies_impl(_ctx):
+    http_archive(
+        name = "com_google_googleapis",
+        sha256 = "979859a238e6626850fee33d30f4240e90e71009786a55a15134df582dbc2dbe",
+        strip_prefix = "googleapis-8d245ac97e058b541f1477b1a85d676b18e80849",
+        urls = ["https://github.com/googleapis/googleapis/archive/8d245ac97e058b541f1477b1a85d676b18e80849.tar.gz"],
+    )
+
+non_module_dependencies = module_extension(
+    implementation = _non_module_dependencies_impl,
+)
diff --git a/googleapis_ext.bzl b/googleapis_ext.bzl
new file mode 100644
index 0000000..9a3ce9d
--- /dev/null
+++ b/googleapis_ext.bzl
@@ -0,0 +1,7 @@
+load("@com_google_googleapis//:repository_rules.bzl", "switched_rules_by_language")
+
+googleapis_ext = module_extension(implementation = lambda x: switched_rules_by_language(
+    name = "com_google_googleapis_imports",
+    cc = True,
+    grpc = True,
+))
