diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..c9114ea
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,22 @@
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "lockfree",
+    deps = [
+        "@boost.align//:align",
+        "@boost.array//:array",
+        "@boost.assert//:assert",
+        "@boost.atomic//:atomic",
+        "@boost.config//:config",
+        "@boost.core//:core",
+        "@boost.integer//:integer",
+        "@boost.iterator//:iterator",
+        "@boost.mpl//:mpl",
+        "@boost.parameter//:parameter",
+        "@boost.predef//:predef",
+        "@boost.static_assert//:static_assert",
+        "@boost.tuple//:tuple",
+        "@boost.type_traits//:type_traits",
+        "@boost.utility//:utility",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..910da0c
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,23 @@
+module(
+    name = "boost.lockfree",
+    version = "1.83.0.bzl.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "boost.align", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.array", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.atomic", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.config", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.core", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.integer", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.iterator", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.mpl", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.parameter", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.predef", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.static_assert", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.tuple", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.type_traits", version = "1.83.0.bzl.1")
+bazel_dep(name = "boost.utility", version = "1.83.0.bzl.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..650cdc5
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost.lockfree//:lockfree",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "lockfree_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/lockfree_test.cc b/bazel_test/lockfree_test.cc
new file mode 100644
index 0000000..1ec09ee
--- /dev/null
+++ b/bazel_test/lockfree_test.cc
@@ -0,0 +1,7 @@
+#include <boost/lockfree/queue.hpp>
+
+int main()
+{
+  boost::lockfree::queue<int> queue(128);
+  return 0;
+}
