--- a/fuzztest/internal/domains/protobuf_domain_impl.h
+++ b/fuzztest/internal/domains/protobuf_domain_impl.h
@@ -1204,7 +1204,7 @@ class ProtobufDomainImpl {
         auto v = self.GetValue(corpus_copy);
         // We need to roundtrip through serialization to really dedup. The
         // reflection API alone doesn't cut it.
-        v->ParsePartialFromString(v->SerializePartialAsString());
+        static_cast<void>(v->ParsePartialFromString(v->SerializePartialAsString()));
         if (v->GetReflection()->FieldSize(*v, field) ==
             domain.GetValue(copy).size()) {
           // The number of entries is the same, so accept the change.
