diff --git a/MODULE.bazel b/MODULE.bazel
index ff711b1..7c17693 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -2,7 +2,7 @@
 
 module(
     name = "rules_helm",
-    version = "0.6.0",
+    version = "0.6.0.bcr.1",
     compatibility_level = 1,
 )
 
@@ -13,7 +13,7 @@ bazel_dep(name = "rules_go", version = "0.45.1", repo_name = "io_bazel_rules_go"
 # This is unfortunately required by `rules_oci`.
 # https://github.com/bazel-contrib/rules_oci/issues/575
 bazel_dep(name = "aspect_bazel_lib", version = "2.7.2", dev_dependency = True)
-bazel_dep(name = "rules_oci", version = "2.0.0", dev_dependency = True)
+bazel_dep(name = "rules_oci", version = "2.0.0")
 bazel_dep(name = "stardoc", version = "0.6.2", dev_dependency = True, repo_name = "io_bazel_stardoc")
 
 go_sdk = use_extension("@io_bazel_rules_go//go:extensions.bzl", "go_sdk")
