diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 000000000..83ea047de
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,41 @@
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "BlocksRuntime",
+    srcs = [
+        "BlocksRuntime/Block_private.h",
+        "BlocksRuntime/data.c",
+        "BlocksRuntime/runtime.c",
+        "config.h",
+    ],
+    hdrs = [
+        "BlocksRuntime/Block.h",
+    ],
+    copts = select({
+        "@rules_cc//cc/compiler:msvc-cl": [],
+        "//conditions:default": [
+            # BlocksRuntime/runtime.c:72:22: warning: unused function 'OSAtomicCompareAndSwapLong' [-Wunused-function]
+            "-Wunused-function",
+        ],
+    }),
+    includes = ["BlocksRuntime"],
+)
+
+# Allows the library to be referenced as @blocksruntime.
+alias(
+    name = "blocksruntime",
+    actual = "BlocksRuntime",
+)
+
+cc_test(
+    name = "sample",
+    srcs = ["sample.c"],
+    copts = ["-fblocks"],
+    # -fblocks requires clang.
+    target_compatible_with = select({
+        "@rules_cc//cc/compiler:clang": [],
+        "//conditions:default": ["@platforms//:incompatible"],
+    }),
+    visibility = ["//visibility:private"],
+    deps = [":BlocksRuntime"],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 000000000..ede89d43b
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,8 @@
+module(
+    name = "blocksruntime",
+    version = "1.0.0",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "platforms", version = "0.0.6")
+bazel_dep(name = "rules_cc", version = "0.0.8")
