--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,26 @@
+# Source: https://github.com/Juniper/ebplane/blob/master/third_party/BUILD.elf
+load("@rules_foreign_cc//tools/build_defs:configure.bzl", "configure_make")
+
++licenses(["notice"])  # The libraries and backends are dual GPLv2+/LGPLv3+. The utilities are GPLv3+.
+
+filegroup(
+    name = "elfutils_all",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
+
+configure_make(
+    name = "libelf",
+    # elfutils can only be built with gcc. What you see below is a hack:
+    # it overrides the environment variables set by bazel to use clang with
+    # values that force detection and use of the host installed gcc.
+    # TODO: fix the build to be more hermetic.
+    configure_options = [
+        "CC=gcc",
+        "AR=''",
+        "CFLAGS=''",
+        "LDFLAGS=''",
+    ],
+    lib_source = ":elfutils_all",
+    out_lib_dir = "lib",
+    shared_libraries = ["libelf.so"],
+    visibility = ["//visibility:public"],
+)
