--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -1,3 +1,7 @@
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+
+package(default_visibility = ["//:__subpackages__"])
+
 exports_files([
     "gtest.BUILD",
     "zlib.BUILD",
@@ -14,17 +18,32 @@ exports_files([
     "protoc-gen-validate.patch",
 ])

-package(default_visibility = ["//:__subpackages__"])
+bool_flag(
+    name = "use_openssl",
+    build_setting_default = False,
+    visibility = ["//visibility:public"],
+)
+
+config_setting(
+    name = "use_openssl_setting",
+    flag_values = {":use_openssl": "true"},
+)

 alias(
     name = "libssl",
-    actual = "@boringssl//:ssl",
+    actual = select({
+        ":use_openssl_setting": "@openssl//:ssl",
+        "//conditions:default": "@boringssl//:ssl",
+    }),
     tags = ["manual"],
 )

 alias(
     name = "libcrypto",
-    actual = "@boringssl//:crypto",
+    actual = select({
+        ":use_openssl_setting": "@openssl//:crypto",
+        "//conditions:default": "@boringssl//:crypto",
+    }),
     tags = ["manual"],
 )

