--- BUILD.bazel
+++ BUILD.bazel
@@ -1,5 +1,18 @@
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+
 package(default_visibility = ["//visibility:public"])

+bool_flag(
+    name = "use_openssl",
+    build_setting_default = False,
+    visibility = ["//visibility:public"],
+)
+
+config_setting(
+    name = "use_openssl_setting",
+    flag_values = {":use_openssl": "true"},
+)
+
 cc_library(
     name = "s2",
     srcs = [
@@ -261,6 +274,7 @@ cc_library(
         "@abseil-cpp//absl/random:bit_gen_ref",
         "@abseil-cpp//absl/status",
     ],
+    includes = [""],
 )

 cc_binary(
