diff --git a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,13 +61,7 @@ channel-api = ["crossbeam"]
 check_asm = []
 decode_test = ["aom-sys"]
 decode_test_dav1d = ["dav1d-sys"]
-default = [
-    "binaries",
-    "asm",
-    "threading",
-    "signal_support",
-    "git_version",
-]
+default = ["capi", "threading"]
 desync_finder = ["backtrace"]
 dump_ivf = ["ivf"]
 dump_lookahead_data = [
