--- MODULE.bazel
+++ MODULE.bazel
@@ -1,18 +1,11 @@
 ## MODULE.bazel
 module(
     name = "gz-plugin",
-    repo_name = "org_gazebosim_gz-plugin",
+    version = "4.0.0-pre1",
+    compatibility_level = 3,
 )
 
 bazel_dep(name = "googletest", version = "1.14.0")
 bazel_dep(name = "rules_license", version = "1.0.0")
-
-# Gazebo Dependencies
 bazel_dep(name = "rules_gazebo", version = "0.0.2")
-bazel_dep(name = "gz-utils")
-
-archive_override(
-    module_name = "gz-utils",
-    strip_prefix = "gz-utils-main",
-    urls = ["https://github.com/gazebosim/gz-utils/archive/refs/heads/main.tar.gz"],
-)
+bazel_dep(name = "gz-utils", version = "4.0.0-pre1")
