diff --git a/MODULE.bazel b/MODULE.bazel
index 652c085..088a157 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,6 @@
 module(
     name = "bzlparty_rules_quickjs",
-    version = "0.0.0",
+    version = "0.1.0",
     compatibility_level = 1,
 )
 
@@ -10,5 +10,3 @@ bazel_dep(name = "aspect_rules_js", version = "1.34.1")
 
 quickjs_extension = use_extension("//quickjs:extensions.bzl", "quickjs_extension")
 use_repo(quickjs_extension, "bzlparty_quickjs")
-
-bazel_dep(name = "buildifier_prebuilt", version = "6.4.0", dev_dependency = True)
