diff --git a/include/boost/geometry/views/enumerate_view.hpp b/include/boost/geometry/views/enumerate_view.hpp
index 65f3beed9..1ded0f65d 100644
--- a/include/boost/geometry/views/enumerate_view.hpp
+++ b/include/boost/geometry/views/enumerate_view.hpp
@@ -9,6 +9,8 @@
 #ifndef BOOST_GEOMETRY_VIEWS_ENUMERATE_VIEW_HPP
 #define BOOST_GEOMETRY_VIEWS_ENUMERATE_VIEW_HPP
 
+#include <algorithm>
+
 #include <boost/iterator/iterator_facade.hpp>
 #include <boost/iterator/iterator_categories.hpp>
 #include <boost/range/begin.hpp>
