diff --git a/BUILD b/BUILD
new file mode 100644
index 0000000..bfaba5a
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,12 @@
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "c4core",
+    srcs = glob(["src/c4/*.cpp"]),
+    hdrs = glob(["src/c4/**/*.hpp"]) + glob(["src/c4/**/*.h"]),
+    strip_include_prefix = "src",
+    defines = ["C4_NO_DEBUG_BREAK"],
+    deps = [
+      "@fast_float//:fast_float"
+    ],
+)
