diff --git a/lib/stdint.in.h b/lib/stdint.in.h
--- a/lib/stdint.in.h
+++ b/lib/stdint.in.h
@@ -99,7 +99,7 @@
    AIX 5.2 <sys/types.h> isn't needed and causes troubles.
    Mac OS X 10.4.6 <sys/types.h> includes <stdint.h> (which is us), but
    relies on the system <stdint.h> definitions, so include
-   <sys/types.h> after @NEXT_STDINT_H@.  */
+   <sys/types.h> after @ NEXT_STDINT_H @.  */
 # if @HAVE_SYS_TYPES_H@ && ! defined _AIX
 #  include <sys/types.h>
 # endif
