diff --git a/include/vsr/detail/vsr_algebra.h b/include/vsr/detail/vsr_algebra.h
index 324a32c..c4c7047 100644
--- a/include/vsr/detail/vsr_algebra.h
+++ b/include/vsr/detail/vsr_algebra.h
@@ -195,7 +195,7 @@ namespace vsr {
            typedef gp_basis_t<typename B::basis, typename A::basis > tmp_basis;      
            //............................................lh...........rh.................return type 
            using x = typename impl::template rot_arrow_t<tmp_basis, typename B::basis, typename A::basis>;                 
-           return x::Arrow::template Make<A>( gp(b, a), Reverse<typename B::basis>::Type::template Make(b) );
+           return x::Arrow::template Make<A>( gp(b, a), Reverse<typename B::basis>::Type::Make(b) );
          }
     
          /// Reflect a by b, return type a
@@ -203,7 +203,7 @@ namespace vsr {
          static constexpr A reflect(const A& a, const B& b) {
           typedef gp_basis_t<typename B::basis, typename A::basis > tmp_basis;
           using x = typename impl::template rot_arrow_t<tmp_basis, typename B::basis, typename A::basis>;
-          return x::Arrow::template Make<A>( gp(b, a.involution() ), Reverse<typename B::basis>::Type::template Make(b) );
+          return x::Arrow::template Make<A>( gp(b, a.involution() ), Reverse<typename B::basis>::Type::Make(b) );
          } 
  
           /// make a type from sum of basis B1 and B2
diff --git a/include/vsr/detail/vsr_multivector.h b/include/vsr/detail/vsr_multivector.h
index 33355d8..d629743 100644
--- a/include/vsr/detail/vsr_multivector.h
+++ b/include/vsr/detail/vsr_multivector.h
@@ -185,7 +185,7 @@ namespace vsr{
 
       /// Reversion \\(\\tilde{A}\\) 
       Multivector operator ~() const {
-        return Reverse< basis >::Type::template Make(*this) ;
+        return Reverse< basis >::Type::Make(*this) ;
       }
       
       /// Inversion \\(\\tilde{A}/A\\tilde{A}\\) 
@@ -430,11 +430,11 @@ Multivector<Algebra,B> Multivector<Algebra,B>::yz = Multivector<Algebra,B>().tem
  *-----------------------------------------------------------------------------*/
 template<typename Algebra, typename B> 
 Multivector<Algebra,B> Multivector<Algebra,B>::conjugation() const{
-	return Conjugate<B>::Type::template Make(*this);
+	return Conjugate<B>::Type::Make(*this);
 }
 template<typename Algebra, typename B> 
 Multivector<Algebra,B> Multivector<Algebra,B>::involution() const{
-	return Involute<B>::Type::template Make(*this);
+	return Involute<B>::Type::Make(*this);
 } 
 
 
