===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -3,8 +3,9 @@
 module(
     name = "aspect_rules_lint",
     bazel_compatibility = [">=7.6.0"],
     compatibility_level = 1,
+    version = "2.0.0-beta.0",
 )
 
 # We don't actually depend directly on aspect_bazel_lib, but we do require this
 # version be used transitively so that it doesn't produce conflicting actions
