--- a/Components/Hlms/Pbs/src/InstantRadiosity/OgreInstantRadiosity.cpp
+++ b/Components/Hlms/Pbs/src/InstantRadiosity/OgreInstantRadiosity.cpp
@@ -58,7 +58,7 @@ THE SOFTWARE.
     OGRE_PLATFORM == OGRE_PLATFORM_FREEBSD)
     #include <random>
 #else
-    #include <tr1/random>
+    #include <random>
 #endif

 namespace Ogre
@@ -72,7 +72,7 @@ namespace Ogre
     OGRE_PLATFORM == OGRE_PLATFORM_FREEBSD
         std::mt19937        mRng;
 #else
-        std::tr1::mt19937   mRng;
+        std::mt19937   mRng;
 #endif

     public:

--- a/OgreMain/include/OgreString.h
+++ b/OgreMain/include/OgreString.h
@@ -212,16 +212,16 @@ namespace Ogre {
 #   elif OGRE_COMP_VER < 430
     typedef ::__gnu_cxx::hash< _StringBase > _StringHash;
 #   else
-    typedef ::std::tr1::hash< _StringBase > _StringHash;
+    typedef ::std::hash< _StringBase > _StringHash;
 #   endif
 #elif OGRE_COMPILER == OGRE_COMPILER_CLANG
 #   if defined(_LIBCPP_VERSION) || __cplusplus >= 201103L
     typedef ::std::hash< _StringBase > _StringHash;
 #   else
-    typedef ::std::tr1::hash< _StringBase > _StringHash;
+    typedef ::std::hash< _StringBase > _StringHash;
 #   endif
 #elif OGRE_COMPILER == OGRE_COMPILER_MSVC && OGRE_COMP_VER >= 1600 && OGRE_COMP_VER < 1910 && !defined(STLPORT) // VC++ 10.0
-    typedef ::std::tr1::hash< _StringBase > _StringHash;
+    typedef ::std::hash< _StringBase > _StringHash;
 #elif !defined( _STLP_HASH_FUN_H )
     typedef stdext::hash_compare< _StringBase, std::less< _StringBase > > _StringHash;
 #else
