diff --git a/MODULE.bazel b/MODULE.bazel
index dd035477..420f0c65 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,7 @@
 module(
     name = "envoy_toolshed",
     version = "0.3.24",
+    bazel_compatibility = [">=7.2.1"],
 )
 
 bazel_dep(name = "aspect_bazel_lib", version = "2.22.0")
diff --git a/website/theme/images/BUILD b/website/theme/images/BUILD
index 725c5477..6c8c3e42 100644
--- a/website/theme/images/BUILD
+++ b/website/theme/images/BUILD
@@ -3,11 +3,14 @@ load("@rules_pkg//pkg:pkg.bzl", "pkg_tar")
 
 pkg_files(
     name = "image_files",
-    srcs = glob([
-        "**/*.ico",
-        "**/*.svg",
-        "**/*.png",
-    ]),
+    srcs = glob(
+        [
+            "**/*.ico",
+            "**/*.svg",
+            "**/*.png",
+        ],
+        allow_empty = True,
+    ),
     prefix = "theme/static/images",
     strip_prefix = "",
     visibility = ["//visibility:public"],
diff --git a/sysroot/BUILD b/sysroot/BUILD
index 7a6dd836..bede90a9 100644
--- a/sysroot/BUILD
+++ b/sysroot/BUILD
@@ -19,6 +19,7 @@ load(":fs.bzl", "sysroots")
 exports_files([
     "build_sysroot.sh",
     "config.yaml",
+    "extensions.bzl",
     "fs.bzl",
     "sysroot.bzl",
 ])
diff --git a/.bazelignore b/. bazelignore
new file mode 100644
index 0000000..c6e5d31
--- /dev/null
+++ b/.bazelignore
@@ -0,0 +1,1 @@
+bcr
diff --git a/website/macros.bzl b/website/macros.bzl
index e07cd362..ff93e2d7 100644
--- a/website/macros.bzl
+++ b/website/macros.bzl
@@ -96,7 +96,7 @@ def static_website(
             TAR_COMMAND=$$(which tar)
         fi
 
-        $$TAR_COMMAND -xf $$SOURCE "$${DECOMPRESS_ARGS}"
+        $$TAR_COMMAND -xf $$SOURCE $${DECOMPRESS_ARGS:+DECOMPRESS_ARGS}
 
         while IFS= read -r CMD; do
             $$CMD
@@ -104,7 +104,7 @@ def static_website(
 
         $$GENERATOR "$$CONTENT"
 
-        $$TAR_COMMAND cfh $@ $$EXCLUDES -C "$$OUTPUT" .
+        $$TAR_COMMAND cfh $@ $${EXCLUDES:+$$EXCLUDES} -C "$$OUTPUT" .
         """ % (name_sources, decompressor_args, generator, content_path, output_path, mapping_commands, exclude_args, url),
         outs = [name_website_tarball],
         srcs = extra_srcs,
