diff --git a/BUILD.bazel b/BUILD.bazel
index c8b6070d0..f028ad5f3 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -18,7 +18,6 @@ cc_library(
     name = "highs",
     srcs = glob([
         "extern/filereaderlp/*.cpp",
-        "extern/zlib/*.cpp",
         "src/interfaces/highs_c_api.cpp",
         "src/io/*.cpp",
         "src/ipm/*.cpp",
@@ -37,7 +36,6 @@ cc_library(
         "src/util/*.cpp",
     ]),
     hdrs = glob([
-        "HConfig.h",
         "**/*.h",
         "src/qpsolver/*.hpp",
         "src/Highs.h",
