diff --git a/.bazelrc b/.bazelrc
index 710c4d81a..4980d66e3 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -1 +1 @@
-common --noenable_bzlmod
+# common --noenable_bzlmod
diff --git a/BUILD.bazel b/BUILD.bazel
index c8b6070d0..f028ad5f3 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -18,7 +18,6 @@ cc_library(
     name = "highs",
     srcs = glob([
         "extern/filereaderlp/*.cpp",
-        "extern/zlib/*.cpp",
         "src/interfaces/highs_c_api.cpp",
         "src/io/*.cpp",
         "src/ipm/*.cpp",
@@ -37,7 +36,7 @@ cc_library(
         "src/util/*.cpp",
     ]),
     hdrs = glob([
-        "HConfig.h",
+#        "HConfig.h",
         "**/*.h",
         "src/qpsolver/*.hpp",
         "src/Highs.h",
diff --git a/MODULE.bazel b/MODULE.bazel
index adb22ca97..624029bd6 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,17 +1,17 @@
 module(
     name = "highs",
-    version = "1.8.1",
+    version = "1.8.1.bcr.1",
 )
 
 bazel_dep(
     name = "bazel_skylib",
-    version = "1.6.1",
+    version = "1.7.1",
 )
 bazel_dep(
     name = "rules_cc",
-    version = "0.0.9",
+    version = "0.0.16",
 )
 bazel_dep(
     name = "zlib",
-    version = "1.3.1.bcr.1",
+    version = "1.3.1.bcr.3",
 )
