diff --git a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -13,6 +13,7 @@
 ##===----------------------------------------------------------------------===##

 load("@build_bazel_rules_apple//apple/testing/default_runner:ios_xctestrun_runner.bzl", "ios_xctestrun_runner")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("//utils/bazel:swift_syntax_library.bzl", "swift_syntax_library", "swift_syntax_test")

 package(default_visibility = ["//visibility:public"])
@@ -208,6 +209,7 @@ swift_syntax_library(
 swift_syntax_test(
     name = "SwiftRefactorTest",
     deps = [
+        ":SwiftIDEUtils",
         ":SwiftRefactor",
         ":_SwiftSyntaxTestSupport",
     ],
@@ -247,6 +249,7 @@ swift_syntax_library(
     name = "SwiftSyntaxMacros",
     deps = [
         ":SwiftDiagnostics",
+        ":SwiftIfConfig",
         ":SwiftParser",
         ":SwiftSyntax",
         ":SwiftSyntaxBuilder",
