diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 000000000..b2eb7c263
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,42 @@
+cc_library(
+    name = "mbedtls",
+    srcs = glob(["library/*.c"]),
+    hdrs = glob(
+        [
+            "include/**/*.h",
+            "library/*.h",
+        ],
+        exclude = ["include/mbedtls/mbedtls_config.h"],
+    ),
+    includes = [
+        "include",
+        "library",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [":mbedtls_config"],
+)
+
+cc_library(
+    name = "mbedtls_default_config",
+    hdrs = ["include/mbedtls/mbedtls_config.h"],
+    includes = [
+        "include",
+    ],
+)
+
+label_flag(
+    name = "mbedtls_config",
+    build_setting_default = ":mbedtls_default_config",
+    visibility = ["//visibility:public"],
+)
+
+cc_binary(
+    name = "linktest",
+    linkopts = select({
+        "@platforms//os:macos": ["-all_load"],
+        "//conditions:default": [],
+    }),
+    linkshared = True,
+    visibility = ["//visibility:private"],
+    deps = ["mbedtls"],
+)
