diff --git a/swc/private/swc.bzl b/swc/private/swc.bzl
index 67f8845..05c8584 100644
--- a/swc/private/swc.bzl
+++ b/swc/private/swc.bzl
@@ -40,7 +40,7 @@ def _impl(ctx):
             fail("Under output_dir, there must be a single entry in srcs")
         if not ctx.files.srcs[0].is_directory:
             fail("Under output_dir, the srcs must be directories, not files")
-        out = ctx.actions.declare_file(ctx.label.name)
+        out = ctx.actions.declare_directory(ctx.label.name)
         outputs.append(out)
         args.add_all([
             ctx.files.srcs[0].path,
