--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,19 @@
+load("@rules_python//python:defs.bzl", "py_binary")
+
+py_binary(
+    name = "meson",
+    srcs = ["meson.py"],
+    python_version = "PY3",
+    visibility = ["//visibility:public"],
+    deps = [":meson_lib"],
+)
+
+py_library(
+    name = "meson_lib",
+    srcs = glob(["mesonbuild/**/*.py"]),
+    data = glob(
+        ["mesonbuild/**/*"],
+        exclude = ["mesonbuild/**/*.py"],
+    ),
+    srcs_version = "PY3",
+)
