diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..b4e9320
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,14 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "container_hash",
+    # defines = ["BOOST_NO_CXX98_FUNCTION_BASE"], # TODO this shouldn"t be needed anymore - https://github.com/boostorg/container_hash/issues/22
+    deps = [
+        "@boost//:config",
+        "@boost//:describe",
+        "@boost//:mp11",
+        "@boost//:type_traits",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..a4b0fd3
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.container_hash",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..ce8c949
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,19 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:container_hash",
+    "@boost//:core",
+    "@boost//:filesystem",
+    "@boost//:unordered",
+    "@boost//:utility",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        negative_test_names = [
+            "namespace_fail_test",
+        ],
+        deps = DEPS,
+    ),
+)
