diff --git modules/federated_language/0.1.0/MODULE.bazel modules/federated_language/0.1.0/MODULE.bazel
index 060de133..56612611 100644
--- modules/federated_language/0.1.0/MODULE.bazel
+++ modules/federated_language/0.1.0/MODULE.bazel
@@ -12,7 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-module(name = "federated_language")
+module(
+    name = "federated_language",
+    version = "0.1.0",
+    compatibility_level = 0,
+)
 
 bazel_dep(name = "protobuf", version = "28.3")
 bazel_dep(name = "rules_license", version = "1.0.0")

