diff --git MODULE.bazel MODULE.bazel
index 060de13..5661261 100644
--- MODULE.bazel
+++ MODULE.bazel
@@ -12,7 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-module(name = "federated_language")
+module(
+    name = "federated_language",
+    version = "0.1.0",
+    compatibility_level = 0,
+)
 
 bazel_dep(name = "protobuf", version = "28.3")
 bazel_dep(name = "rules_license", version = "1.0.0")

