===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,9 +1,9 @@
 "Bazel rules for ShellSpec - BDD testing framework for shell scripts"
 
 module(
     name = "rules_shellspec",
-    version = "",
+    version = "0.1.5",
     # NOTE:
     #
     # Always leave version unset or set to "" (the default). The default value
     # can prevent issues when the module is used via non-registry overrides
