diff --git a/bazel/generate_cc.bzl b/bazel/generate_cc.bzl
index bd1d545..10d7750 100644
--- a/bazel/generate_cc.bzl
+++ b/bazel/generate_cc.bzl
@@ -187,7 +187,7 @@ _generate_cc = rule(
             mandatory = False,
         ),
         "_protoc": attr.label(
-            default = Label("//external:protocol_compiler"),
+            default = Label("//third_party:protocol_compiler"),
             executable = True,
             cfg = "host",
         ),
diff --git a/bazel/generate_objc.bzl b/bazel/generate_objc.bzl
index b0e9211..22aa0fa 100644
--- a/bazel/generate_objc.bzl
+++ b/bazel/generate_objc.bzl
@@ -180,7 +180,7 @@ generate_objc = rule(
             default = "@com_google_protobuf//:well_known_protos",
         ),
         "_protoc": attr.label(
-            default = Label("//external:protocol_compiler"),
+            default = Label("//third_party:protocol_compiler"),
             executable = True,
             cfg = "host",
         ),
diff --git a/bazel/python_rules.bzl b/bazel/python_rules.bzl
index a39159a..967a26c 100644
--- a/bazel/python_rules.bzl
+++ b/bazel/python_rules.bzl
@@ -104,7 +104,7 @@ _gen_py_aspect = aspect(
     fragments = ["py"],
     attrs = {
         "_protoc": attr.label(
-            default = Label("//external:protocol_compiler"),
+            default = Label("//third_party:protocol_compiler"),
             providers = ["files_to_run"],
             executable = True,
             cfg = "host",
@@ -160,7 +160,7 @@ py_proto_library = rule(
             aspects = [_gen_py_aspect],
         ),
         "_protoc": attr.label(
-            default = Label("//external:protocol_compiler"),
+            default = Label("//third_party:protocol_compiler"),
             providers = ["files_to_run"],
             executable = True,
             cfg = "host",
@@ -248,7 +248,7 @@ _generate_pb2_grpc_src = rule(
             executable = True,
             providers = ["files_to_run"],
             cfg = "host",
-            default = Label("//external:protocol_compiler"),
+            default = Label("//third_party:protocol_compiler"),
         ),
         "_grpc_library": attr.label(
             default = Label("//src/python/grpcio/grpc:grpcio"),
