diff --git a/gdk-pixbuf/gdk-pixbuf-io.c b/gdk-pixbuf/gdk-pixbuf-io.c
--- a/gdk-pixbuf/gdk-pixbuf-io.c
+++ b/gdk-pixbuf/gdk-pixbuf-io.c
@@ -609,6 +609,9 @@
 #ifdef INCLUDE_ani
         load_one_builtin_module (ani);
 #endif
+#ifdef INCLUDE_svg
+        load_one_builtin_module (svg);
+#endif
 #ifdef INCLUDE_png
         load_one_builtin_module (png);
 #endif
@@ -714,6 +717,7 @@
   extern void _gdk_pixbuf__##type##_fill_info   (GdkPixbufFormat *info);   \
   extern void _gdk_pixbuf__##type##_fill_vtable (GdkPixbufModule *module)
 
+module (svg);
 module (png);
 module (jpeg);
 module (gif);
@@ -824,6 +828,9 @@
 	try_module (wbmp,android_wbmp);
 	try_module (heif,android_heif);
 #endif
+#ifdef INCLUDE_svg
+        try_module (svg,svg);
+#endif
 #ifdef INCLUDE_png
         try_module (png,png);
 #endif
