diff --git a/.bazelignore b/.bazelignore
index 4ac6f4f..a94eadf 100644
--- a/.bazelignore
+++ b/.bazelignore
@@ -1,3 +1,5 @@
-# ignore the examples which are used to document suggested usage, validate this project, and -- in future -- repro issues
-examples/**
+# ignore the examples which are used to document suggested usage, validate this project,
+# and -- in future -- repro issues that may arise
+docs
+examples
 
