--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -5,12 +5,10 @@
 # For more details, please check https://github.com/bazelbuild/bazel/issues/18958
 ###############################################################################
 
-CPU_FEATURES_VERSION = "0.11.0"
-
 module(
     name = "cpu_features",
     repo_name = "com_google_cpufeatures",
-    version = CPU_FEATURES_VERSION,
+    version = "0.11.0",
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.9.0")
