--- a/MODULE.bazel
+++ a/MODULE.bazel
@@ -0,0 +1,11 @@
+module(
+    name = "protobuf",
+    version = "3.19.2",
+    compatibility_level = 1,
+)
+bazel_dep(name = "bazel_skylib", version = "1.0.3")
+bazel_dep(name = "zlib", version = "1.2.12")
+bazel_dep(name = "rules_python", version = "0.4.0")
+bazel_dep(name = "rules_cc", version = "0.0.1")
+bazel_dep(name = "rules_proto", version = "4.0.0")
+bazel_dep(name = "rules_java", version = "4.0.0")
