diff --git BUILD.bazel BUILD.bazel
--- BUILD.bazel
+++ BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 cc_library(
     name = "modern-cpp-kafka-api",
 
diff --git examples/BUILD.bazel examples/BUILD.bazel
--- examples/BUILD.bazel
+++ examples/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "kafka_sync_producer",
 
diff --git tools/console_clients/BUILD.bazel tools/console_clients/BUILD.bazel
--- tools/console_clients/BUILD.bazel
+++ tools/console_clients/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+
 cc_binary(
     name = "kafka-topics",
 
diff --git tests/BUILD.bazel tests/BUILD.bazel
--- tests/BUILD.bazel
+++ tests/BUILD.bazel
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 cc_test(
     name = "kafka-unit-test",
 
