--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,24 @@
+load("@protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
+load("@protobuf//bazel:proto_library.bzl", "proto_library")
+
+proto_library(
+    name = "zipkin_proto",
+    srcs = [
+        "zipkin.proto",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+proto_library(
+    name = "zipkin_jsonv2_proto",
+    srcs = [
+        "zipkin-jsonv2.proto",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_proto_library(
+    name = "zipkin-api",
+    visibility = ["//visibility:public"],
+    deps = [":zipkin_proto"],
+)
