--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,18 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+licenses(["notice"])  # Apache 2
+
+package(
+    default_visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "spdlog",
+    hdrs = glob([
+        "include/**/*.cc",
+        "include/**/*.h",
+    ]),
+    defines = ["SPDLOG_FMT_EXTERNAL"],
+    includes = ["include"],
+    deps = ["@fmt"],
+)
