--- libprjoxide/prjoxide/Cargo.toml
+++ libprjoxide/prjoxide/Cargo.toml
@@ -7,7 +7,7 @@
 
 [features]
 default = []
-interchange = ["capnp", "flate2", "capnpc"]
+interchange = []
 
 [dependencies]
 regex = "1"
@@ -22,11 +22,11 @@
 log = "0.4.11"
 clap = { version = "3.1", features = ["derive"] }
 include_dir = "0.6.0"
-capnp = {version = "0.14", optional = true }
-flate2 = {version = "1.0", optional = true }
+capnp = "0.14"
+flate2 = "1.0"
 
 [build-dependencies]
-capnpc = {version = "0.14", optional = true }
+capnpc = "0.14"
 
 [lib]
 name = "prjoxide"
