diff --git a/BUILD.bazel b/BUILD.bazel
index 1d9685d..6abc1ad 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -300,7 +300,7 @@ cc_library(
     }),
     includes = [prefix_dir + "include"],
     deps = select({
-        ":has_absl": ["@com_google_absl//absl/synchronization"],
+        ":has_absl": ["@abseil-cpp//absl/synchronization"],
         "//conditions:default": [],
     }),
 )
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..3e1a3dc
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,8 @@
+module(
+    name = "openfst",
+    version = "1.8.4",
+    compatibility_level = 0,
+)
+
+bazel_dep(name = "abseil-cpp", version = "20250512.1")
+bazel_dep(name = "rules_cc", version = "0.1.2")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
deleted file mode 100644
index 9b16a28..0000000
--- a/WORKSPACE.bazel
+++ /dev/null
@@ -1,33 +0,0 @@
-# Copyright 2015-2024 Google LLC. All Rights Reserved.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-# Bazel (https://bazel.build/) workspace.
-
-workspace(name = "openfst")
-
-load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
-
-http_archive(
-    # Name has to match workspace name in:
-    #   https://github.com/abseil/abseil-cpp/blob/master/WORKSPACE
-    name = "com_google_absl",
-    strip_prefix = "abseil-cpp-master",
-    urls = ["https://github.com/abseil/abseil-cpp/archive/master.zip"],
-)
-
-http_archive(
-    name = "com_google_googletest",
-    strip_prefix = "googletest-main",
-    urls = ["https://github.com/google/googletest/archive/main.zip"],
-)
