--- a/MODULE.bazel
+++ a/MODULE.bazel
@@ -2,9 +2,6 @@
     name = "rules_go",
     version = "0.33.0",
     compatibility_level = 0,
-    toolchains_to_register = [
-        "@go_default_sdk//:all",
-    ],
 )
 
 print("WARNING: The rules_go Bazel module is still highly experimental and subject to change at any time. Only use it to try out bzlmod for now.")
@@ -21,3 +18,4 @@
 go_sdk = use_extension("@rules_go//go:extensions.bzl", "go_sdk")
 go_sdk.download(name = "go_default_sdk", version = "1.18.3")
 use_repo(go_sdk, "go_default_sdk")
+register_toolchains("@go_default_sdk//:all")
