--- MODULE.bazel
+++ MODULE.bazel
@@ -1,39 +1,14 @@
-## MODULE.bazel
 module(
     name = "gz-math",
-    repo_name = "org_gazebosim_gz-math",
+    version = "9.0.0-pre1",
+    compatibility_level = 0,
 )
-
-bazel_dep(name = "buildifier_prebuilt", version = "7.3.1")
+bazel_dep(name = "buildifier_prebuilt", version = "8.2.0.2")
 bazel_dep(name = "googletest", version = "1.14.0")
 bazel_dep(name = "rules_license", version = "1.0.0")
 bazel_dep(name = "eigen", version = "3.4.0.bcr.3")  # workaround for https://github.com/bazelbuild/bazel-central-registry/issues/4355
 bazel_dep(name = "pybind11_bazel", version = "2.13.6")
 bazel_dep(name = "rules_python", version = "0.36.0")
 bazel_dep(name = "platforms", version = "0.0.11")
-
-# Gazebo Dependencies
 bazel_dep(name = "rules_gazebo", version = "0.0.6")
-bazel_dep(name = "gz-utils")
-archive_override(
-    module_name = "gz-utils",
-    strip_prefix = "gz-utils-main",
-    urls = ["https://github.com/gazebosim/gz-utils/archive/refs/heads/main.tar.gz"],
-)
-
-PYTHON_VERSIONS = [
-    "3.9",
-    "3.10",
-    "3.11",
-    "3.12",
-]
-
-python = use_extension("@rules_python//python/extensions:python.bzl", "python")
-
-[
-    python.toolchain(
-        is_default = python_version == PYTHON_VERSIONS[-1],
-        python_version = python_version,
-    )
-    for python_version in PYTHON_VERSIONS
-]
+bazel_dep(name = "gz-utils", version = "4.0.0-pre1")
