--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,18 @@
+""" Builds units.
+"""
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+
+licenses(["notice"])  # MIT License
+
+cc_library(
+    name = "units",
+    hdrs = ["include/units.h"],
+    visibility = ["//visibility:public"],
+    includes = ["include"],
+)
+
+cc_test(
+    name = "units_test",
+    srcs = ["unitTests/main.cpp"],
+    deps = ["@googletest//:gtest"]
+)
