--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,16 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+
+cc_library(
+    name = "darts-clone",
+    hdrs = glob(["include/*.h"]),
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+)
+
+cc_test(
+    name = "darts_clone_compat_test",
+    srcs = ["darts_clone_compat_test.cc"],
+    deps = [
+        ":darts-clone",
+    ],
+)
