--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,7 +1,7 @@
 module(
     name = "lobster",
     compatibility_level = 1,
-    version = "0.13.2",
+    version = "1.0.3",
 )
 
 bazel_dep(
@@ -29,7 +29,7 @@
 uv.configure(version = "0.8.24")
 
 bazel_dep(name = "googletest", version = "1.13.0")
-bazel_dep(name = "trlc", version = "0.0.0")
+bazel_dep(name = "trlc", version = "2.0.5")
 
 pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
 
@@ -51,12 +51,6 @@
 
 bazel_dep(name = "gazelle", version = "0.44.0", repo_name = "bazel_gazelle")
 
-git_override(
-    commit = "650b51a47264a4f232b3341f473527710fc32669",
-    module_name = "trlc",
-    remote = "https://github.com/bmw-software-engineering/trlc.git",
-)
-
 pip.parse(
     experimental_requirement_cycles = {
         "sphinx": [
