--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,40 @@
+""" Builds Sophus.
+"""
+
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "sophus",
+    hdrs = [
+        "sophus/average.hpp",
+        "sophus/cartesian.hpp",
+        "sophus/ceres_local_parameterization.hpp",
+        "sophus/ceres_manifold.hpp",
+        "sophus/ceres_typetraits.hpp",
+        "sophus/common.hpp",
+        "sophus/geometry.hpp",
+        "sophus/interpolate.hpp",
+        "sophus/interpolate_details.hpp",
+        "sophus/num_diff.hpp",
+        "sophus/rotation_matrix.hpp",
+        "sophus/rxso2.hpp",
+        "sophus/rxso3.hpp",
+        "sophus/se2.hpp",
+        "sophus/se3.hpp",
+        "sophus/sim2.hpp",
+        "sophus/sim3.hpp",
+        "sophus/sim_details.hpp",
+        "sophus/so2.hpp",
+        "sophus/so3.hpp",
+        "sophus/spline.hpp",
+        "sophus/types.hpp",
+        "sophus/velocities.hpp"
+    ],
+    deps = [
+        "@eigen",
+        "@fmt",
+    ],
+    includes = ["."],
+    visibility = ["//visibility:public"],
+)
+
