diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..bbc54ee
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,8 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "saru",
+    hdrs = glob(["include/**/*.h"]), # Currently only providing the saru .h headers but more of the tools can be added if needed
+    includes = ["include"],  # Expose the include directory files without prefix
+    visibility = ["//visibility:public"],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..fa808b8
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,13 @@
+"""
+Module: saru
+Purpose: Provides the Saru testing framework for use with other bazel modules.
+Notes: The Saru testing framework isn't currently actively maintained, but is used in the tests of mikeando/fastjson
+"""
+
+module(
+    name = "saru",
+    version = "0.0.0-20130617092049-c11c375fefd7",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "rules_cc", version = "0.1.1")
