--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,43 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "make")
+
+config_setting(
+    name = "msvc_compiler",
+    flag_values = {"@bazel_tools//tools/cpp:compiler": "msvc-cl"},
+)
+
+config_setting(
+    name = "macos",
+    constraint_values = ["@platforms//os:macos"],
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "pfm_srcs",
+    srcs = glob(["**"]),
+)
+
+make(
+    name = "libpfm",
+    lib_source = ":pfm_srcs",
+    lib_name = "libpfm",
+    copts = select({
+        "macos": [
+            "-fPIC"
+        ],
+        "//conditions:default": [
+            "-Wno-format-truncation",
+            "-Wno-use-after-free",
+            "-fPIC"]
+    }),
+    visibility = [
+        "//visibility:public",
+    ],
+    # libevent script uses it's own libtool for linking;
+    # so do not specify linker tool for it
+    # (otherwise, if the libtool from bazel's toolchain is supplied,
+    # the build script has problems with passing output file to libtool)
+    # see #315
+    env = {
+        "AR": "",
+    }
+)
