--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,20 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "make")
+
+filegroup(
+    name = "pfm_srcs",
+    srcs = glob(["**"]),
+)
+
+make(
+    name = "libpfm",
+    lib_source = ":pfm_srcs",
+    lib_name = "libpfm",
+    copts = [
+        "-Wno-format-truncation",
+        "-Wno-use-after-free",
+        "-fPIC",
+    ],
+    visibility = [
+        "//visibility:public",
+    ],
+)
