--- lib/Makefile
+++ lib/Makefile
@@ -19,10 +19,11 @@
 # CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
 # OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #
-TOPDIR  := $(shell if [ "$$PWD" != "" ]; then echo $$PWD; else pwd; fi)/..
+SELF_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
+TOP_DIR := $(SELF_DIR)/..
 
-include $(TOPDIR)/config.mk
-include $(TOPDIR)/rules.mk
+include $(TOP_DIR)/config.mk
+include $(TOP_DIR)/rules.mk
 
 #
 # Common files
@@ -379,9 +380,14 @@
 
 $(OBJS) $(SOBJS): $(TOPDIR)/config.mk $(TOPDIR)/rules.mk Makefile $(INCDEP)
 
+AR_CMDLINE=cq 
+ifeq ($(SYS),Darwin) 
+AR_CMDLINE=-static
+endif
+
 libpfm.a:  $(OBJS)
 	$(RM) $@
-	$(AR) cq $@ $(OBJS)
+	$(AR) $(AR_CMDLINE) $@ $(OBJS)
 
 $(SLIBPFM):  $(SOBJS)
 	$(CC) $(CFLAGS) $(SLDFLAGS)  -o $@ $(SOBJS)
