--- MODULE.bazel
+++ MODULE.bazel
@@ -1,19 +1,11 @@
-## MODULE.bazel
 module(
     name = "gz-math",
-    repo_name = "org_gazebosim_gz-math",
+    version = "8.1.0",
+    compatibility_level = 0,
 )
-
 bazel_dep(name = "buildifier_prebuilt", version = "7.3.1")
 bazel_dep(name = "googletest", version = "1.14.0")
 bazel_dep(name = "rules_license", version = "1.0.0")
 bazel_dep(name = "eigen", version = "3.4.0")
-
-# Gazebo Dependencies
 bazel_dep(name = "rules_gazebo", version = "0.0.2")
-bazel_dep(name = "gz-utils")
-archive_override(
-    module_name = "gz-utils",
-    strip_prefix = "gz-utils-gz-utils3",
-    urls = ["https://github.com/gazebosim/gz-utils/archive/refs/heads/gz-utils3.tar.gz"],
-)
+bazel_dep(name = "gz-utils", version = "3.1.0")
