diff --git tests/0113-cooperative_rebalance.cpp tests/0113-cooperative_rebalance.cpp
index c54619d7..e91c4129 100644
--- tests/0113-cooperative_rebalance.cpp
+++ tests/0113-cooperative_rebalance.cpp
@@ -26,10 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-extern "C" {
-#include "../src/rdkafka_protocol.h"
-#include "test.h"
-}
 #include <iostream>
 #include <map>
 #include <set>
@@ -39,6 +35,10 @@ extern "C" {
 #include <assert.h>
 #include "testcpp.h"
 #include <fstream>
+extern "C" {
+  #include "../src/rdkafka_protocol.h"
+  #include "test.h"
+}
 
 using namespace std;
 
