diff --git a/MODULE.bazel b/MODULE.bazel
index c298a32..658794e 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -18,8 +18,8 @@ module(name = "helly25_mbo", version = "0.4.0", repo_name = "com_helly25_mbo")
 
 # For local development we include LLVM and Hedron-Compile-Commands.
 # For bazelmod usage these have to be provided by the main module - if necessary.
-include("//bazelmod:hedron.MODULE.bazel")
-include("//bazelmod:llvm.MODULE.bazel")
+# include("//bazelmod:hedron.MODULE.bazel")
+# include("//bazelmod:llvm.MODULE.bazel")
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
 bazel_dep(name = "rules_cc", version = "0.1.1")
diff --git a/mbo/mope/mope.bzl b/mbo/mope/mope.bzl
index d80d788..7aeac41 100644
--- a/mbo/mope/mope.bzl
+++ b/mbo/mope/mope.bzl
@@ -35,7 +35,7 @@ load("//mbo/diff:diff.bzl", "diff_test")
 #    c) `$(which "clang_format")`
 #    d) `clang-format-23` ... `clang-format-15`, `clang-format`
 #    e) If even (d) fails, then we will just copy the file.
-_CLANG_FORMAT_BINARY = ""  # Ignore clang-format from repo with: "clang-format-auto"
+_CLANG_FORMAT_BINARY = "clang-format-auto"  # Ignore clang-format from repo with: "clang-format-auto"
 
 def _get_clang_format(ctx):
     """Get the selected clang-format from `--//mbo/mope:clang_format` bazel flag."""
