diff --git a/MODULE.bazel b/MODULE.bazel
index bb91a62..1d31850 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -14,7 +14,7 @@
 # limitations under the License.
 
 # Due to flag and test references to the repo name we use the old name for now.
-module(name = "helly25_mbo", version = "0.4.0", repo_name = "com_helly25_mbo")
+module(name = "helly25_mbo", version = "0.4.0-rc.1", repo_name = "com_helly25_mbo")
 
 # For local development we include LLVM and Hedron-Compile-Commands.
 # For bazelmod usage these have to be provided by the main module - if necessary.
