diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..62b5c04
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,50 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "graph",
+    deps = [
+        "@boost//:algorithm",
+        "@boost//:any",
+        "@boost//:array",
+        "@boost//:assert",
+        "@boost//:bimap",
+        "@boost//:bind",
+        "@boost//:concept_check",
+        "@boost//:config",
+        "@boost//:container_hash",
+        "@boost//:conversion",
+        "@boost//:core",
+        "@boost//:detail",
+        "@boost//:foreach",
+        "@boost//:function",
+        "@boost//:integer",
+        "@boost//:iterator",
+        "@boost//:lexical_cast",
+        "@boost//:math",
+        "@boost//:move",
+        "@boost//:mpl",
+        "@boost//:multi_index",
+        "@boost//:optional",
+        "@boost//:parameter",
+        "@boost//:preprocessor",
+        "@boost//:property_map",
+        "@boost//:property_tree",
+        "@boost//:random",
+        "@boost//:range",
+        "@boost//:regex",
+        "@boost//:serialization",
+        "@boost//:smart_ptr",
+        "@boost//:spirit",
+        "@boost//:static_assert",
+        "@boost//:throw_exception",
+        "@boost//:tti",
+        "@boost//:tuple",
+        "@boost//:type_traits",
+        "@boost//:typeof",
+        "@boost//:unordered",
+        "@boost//:utility",
+        "@boost//:xpressive",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..3342d5f
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.graph",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..a54fcdd
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,19 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:graph",
+    "@boost//:assign",
+    "@boost//:filesystem",
+    "@boost//:timer",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        data = glob([
+            "planar_input_graphs/**",
+            "prgen_input_graphs/**",
+        ]),
+        deps = DEPS,
+    ),
+)
