diff --git build.rs build.rs
index 526deb6..91dda59 100644
--- build.rs
+++ build.rs
@@ -265,17 +265,7 @@ fn get_opaque_type_docs() -> HashMap<String, Vec<String>> {
 // See: https://github.com/rust-lang/cargo/issues/545
 fn cargo_target_dir() -> PathBuf {
     let out_dir = PathBuf::from(env::var("OUT_DIR").expect("OUT_DIR should be set"));
-    let profile = env::var("PROFILE").expect("PROFILE should be set");
-
-    let mut target_dir = None;
-    let mut out_dir_path = out_dir.as_path();
-    while let Some(parent) = out_dir_path.parent() {
-        if parent.ends_with(&profile) {
-            target_dir = Some(parent);
-            break;
-        }
-        out_dir_path = parent;
-    }
+    let target_dir = out_dir.as_path().parent();
 
     target_dir
         .expect("OUT_DIR should be a child of a PROFILE directory")
