--- /dev/null
+++ tests/libutil/BUILD.bazel
@@ -0,0 +1,48 @@
+load("@rules_cc//cc:defs.bzl", "cc_test")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS", "LINKOPTS")
+
+# TODO: Fix me! TESTPATH needs to be runfiles directory.
+# cc_test(
+#     name = "str_table",
+#     srcs = ["str_table.c"],
+#     copts = COPTS,
+#     data = ["words.txt"],
+#     linkopts = LINKOPTS,
+#     linkstatic = True,
+#     local_defines = DEFINES + [
+#         "TESTPATH=tests/libutil",
+#     ],
+#     deps = [
+#         "//lib/compat",
+#         "//lib/io",
+#         "//lib/util",
+#     ],
+# )
+
+[
+    cc_test(
+        name = "{}".format(file),
+        srcs = ["{}.c".format(file)],
+        copts = COPTS,
+        linkopts = LINKOPTS,
+        linkstatic = True,
+        local_defines = DEFINES,
+        deps = [
+            "//lib/compat",
+            "//lib/util",
+        ],
+    )
+    for file in (
+        "rbtree",
+        "xxhash",
+        "threadpool",
+        "is_memory_zero",
+        "canonicalize_name",
+        "filename_sane",
+        "epoch",
+        "hex_decode",
+        "base64_decode",
+        "fix_win32_filename",
+    )
+]
