--- /dev/null
+++ bin/rdsquashfs/BUILD.bazel
@@ -0,0 +1,24 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS", "LINKOPTS")
+
+cc_binary(
+    name = "rdsquashfs",
+    srcs = glob([
+        "**/*.c",
+        "**/*.h",
+    ]),
+    copts = COPTS,
+    linkopts = LINKOPTS,
+    linkstatic = True,
+    local_defines = DEFINES,
+    visibility = ["//visibility:public"],
+    deps = [
+        "//lib/common",
+        "//lib/compat",
+        "//lib/fstree",
+        "//lib/io",
+        "//lib/sqfs:squashfs",
+        "//lib/util",
+    ],
+)
