--- /dev/null
+++ tests/libio/BUILD.bazel
@@ -0,0 +1,43 @@
+load("@rules_cc//cc:defs.bzl", "cc_test")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS", "LINKOPTS")
+
+cc_test(
+    name = "get_line",
+    srcs = ["get_line.c"],
+    copts = COPTS,
+    data = ["get_line.txt"],
+    linkopts = LINKOPTS,
+    linkstatic = True,
+    local_defines = DEFINES + [
+        "TESTFILE=$(location get_line.txt)",
+    ],
+    deps = [
+        "//lib/compat",
+        "//lib/io",
+    ],
+)
+
+[
+    cc_test(
+        name = "{}".format(file),
+        srcs = ["uncompress.c"],
+        copts = COPTS,
+        linkopts = LINKOPTS,
+        linkstatic = True,
+        local_defines = DEFINES + defines,
+        deps = [
+            "//lib/compat",
+            "//lib/io",
+        ],
+    )
+    for file, defines in {
+        "xfrm_bzip2": ["TEST_BZIP2=1"],
+        "xfrm_bzip22": ["TEST_BZIP22=1"],
+        "xfrm_xz": ["TEST_XZ=1"],
+        "xfrm_xz2": ["TEST_XZ2=1"],
+        "xfrm_gzip": ["TEST_GZIP=1"],
+        "xfrm_zstd": ["TEST_ZSTD=1"],
+        "xfrm_zstd2": ["TEST_ZSTD2=1"],
+    }.items()
+]
