--- /dev/null
+++ extras/BUILD.bazel
@@ -0,0 +1,23 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS", "LINKOPTS")
+
+[
+    cc_binary(
+        name = "{}".format(file),
+        srcs = ["{}.c".format(file)],
+        copts = COPTS,
+        linkopts = LINKOPTS,
+        linkstatic = True,
+        local_defines = DEFINES,
+        deps = [
+            "//lib/sqfs:squashfs",
+        ],
+    )
+    for file in (
+        "mknastyfs",
+        "mk42sqfs",
+        "list_files",
+        "extract_one",
+    )
+]
