--- /dev/null
+++ tests/gensquashfs/BUILD.bazel
@@ -0,0 +1,129 @@
+load("@rules_cc//cc:defs.bzl", "cc_test")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS", "LINKOPTS")
+
+cc_test(
+    name = "filemap_xattr",
+    srcs = [
+        "filemap_xattr.c",
+    ],
+    copts = COPTS,
+    data = ["xattr1.txt"],
+    linkopts = LINKOPTS,
+    linkstatic = True,
+    local_defines = DEFINES + [
+        "TESTPATH=$(location xattr1.txt)",
+    ],
+    deps = [
+        "//bin/gensquashfs:filemap_xattr",
+        "//lib/compat",
+        "//lib/fstree",
+        "//lib/io",
+        "//lib/sqfs:squashfs",
+        "//lib/util",
+    ],
+)
+
+cc_test(
+    name = "fstree_from_file",
+    srcs = [
+        "fstree_from_file.c",
+    ],
+    copts = COPTS,
+    data = ["fstree1.txt"],
+    linkopts = LINKOPTS,
+    linkstatic = True,
+    local_defines = DEFINES + [
+        "TESTPATH=$(location fstree1.txt)",
+    ],
+    deps = [
+        "//bin/gensquashfs:fstree",
+        "//lib/compat",
+        "//lib/fstree",
+        "//lib/io",
+        "//lib/util",
+    ],
+)
+
+cc_test(
+    name = "fstree_glob1",
+    srcs = [
+        "fstree_glob1.c",
+    ],
+    copts = COPTS,
+    data = [
+        "testdir",
+        "//tests/libtar:data",
+    ] + glob(["**/*"]),
+    linkopts = LINKOPTS,
+    linkstatic = True,
+    local_defines = DEFINES + [
+        "TESTPATH=$(location testdir)/../",
+    ],
+    deps = [
+        "//bin/gensquashfs:fstree",
+        "//lib/compat",
+        "//lib/fstree",
+        "//lib/io",
+        "//lib/util",
+    ],
+)
+
+cc_test(
+    name = "fstree_from_dir",
+    srcs = [
+        "fstree_from_dir.c",
+    ],
+    copts = COPTS,
+    data = ["testdir"],
+    linkopts = LINKOPTS,
+    linkstatic = True,
+    local_defines = DEFINES + [
+        "TESTPATH='$(location testdir)'",
+    ],
+    deps = [
+        "//bin/gensquashfs:fstree_from_dir",
+        "//lib/compat",
+        "//lib/fstree",
+        "//lib/util",
+    ],
+)
+
+cc_test(
+    name = "sort_file",
+    srcs = [
+        "sort_file.c",
+    ],
+    copts = COPTS,
+    linkopts = LINKOPTS,
+    linkstatic = True,
+    local_defines = DEFINES,
+    deps = [
+        "//bin/gensquashfs:fstree",
+        "//bin/gensquashfs:sort_by_file",
+        "//lib/compat",
+        "//lib/fstree",
+        "//lib/io",
+        "//lib/util",
+    ],
+)
+
+cc_test(
+    name = "fstree_fuzz",
+    srcs = [
+        "fstree_fuzz.c",
+    ],
+    args = ["$(location fstree1.txt)"],
+    copts = COPTS,
+    data = ["fstree1.txt"],
+    linkopts = LINKOPTS,
+    linkstatic = True,
+    local_defines = DEFINES,
+    deps = [
+        "//bin/gensquashfs:fstree",
+        "//lib/compat",
+        "//lib/fstree",
+        "//lib/io",
+        "//lib/util",
+    ],
+)
