--- /dev/null
+++ bin/gensquashfs/BUILD.bazel
@@ -0,0 +1,71 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS", "LINKOPTS")
+
+[
+    cc_library(
+        name = "{}".format(name),
+        srcs = [
+            "{}.c".format(name),
+            "//:config",
+        ],
+        hdrs = [
+            "mkfs.h",
+        ],
+        copts = COPTS,
+        includes = ["."],
+        linkopts = LINKOPTS,
+        linkstatic = True,
+        local_defines = DEFINES,
+        visibility = ["//tests/gensquashfs:__pkg__"],
+        deps = [
+            "//:hdrs",
+        ],
+    )
+    for name in (
+        "filemap_xattr",
+        "fstree_from_dir",
+        "sort_by_file",
+    )
+]
+
+cc_library(
+    name = "fstree",
+    srcs = [
+        "fstree_from_dir.c",
+        "fstree_from_file.c",
+        "//:config",
+    ],
+    hdrs = [
+        "mkfs.h",
+    ],
+    copts = COPTS,
+    includes = ["."],
+    linkopts = LINKOPTS,
+    linkstatic = True,
+    local_defines = DEFINES,
+    visibility = ["//tests/gensquashfs:__pkg__"],
+    deps = [
+        "//:hdrs",
+    ],
+)
+
+cc_binary(
+    name = "gensquashfs",
+    srcs = glob([
+        "**/*.c",
+        "**/*.h",
+    ]),
+    copts = COPTS,
+    linkopts = LINKOPTS,
+    local_defines = DEFINES,
+    visibility = ["//visibility:public"],
+    deps = [
+        "//lib/common",
+        "//lib/compat",
+        "//lib/fstree",
+        "//lib/io",
+        "//lib/sqfs:squashfs",
+        "//lib/util",
+    ],
+)
