--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,24 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+genrule(
+    name = "config",
+    srcs = ["//config"],
+    outs = ["config.h"],
+    cmd = "$(CP) $< $@",
+    toolchains = ["@rules_coreutils//coreutils/toolchain/cp:resolved"],
+    visibility = ["//:__subpackages__"],
+)
+
+cc_library(
+    name = "hdrs",
+    hdrs = glob(["include/**/*.h"]),
+    strip_include_prefix = "include",
+    visibility = ["//:__subpackages__"],
+)
+
+cc_library(
+    name = "sqfs_hdrs",
+    hdrs = glob(["include/sqfs/**/*.h"]),
+    strip_include_prefix = "include",
+    visibility = ["//:__subpackages__"],
+)
