--- /dev/null
+++ lib/sqfs/BUILD.bazel
@@ -0,0 +1,46 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS", "LINKOPTS")
+
+cc_library(
+    name = "squashfs",
+    srcs = glob(
+        [
+            "**/*.c",
+            "**/*.h",
+        ],
+        exclude = [
+            "unix/**/*",
+            "win32/**/*",
+        ],
+    ) + [
+        "//lib/compat:srcs",
+        "//lib/util:srcs",
+    ] + select({
+        "@platforms//os:windows": glob([
+            "win32/**/*.c",
+            "win32/**/*.h",
+        ]),
+        "//conditions:default": glob([
+            "unix/**/*.c",
+            "unix/**/*.h",
+        ]),
+    }) + ["//:config"],
+    hdrs = [
+        "//:sqfs_hdrs",
+    ],
+    copts = COPTS,
+    linkopts = LINKOPTS,
+    local_defines = DEFINES,
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:hdrs",
+        "//lib/compat",
+        "//lib/util",
+        "@lz4",
+        "@lz4//:lz4_hc",
+        "@xz//:lzma",
+        "@zlib",
+        "@zstd",
+    ],
+)
