--- /dev/null
+++ lib/tar/BUILD.bazel
@@ -0,0 +1,21 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS")
+
+cc_library(
+    name = "tar",
+    srcs = glob(
+        [
+            "**/*.c",
+            "**/*.h",
+        ],
+    ) + ["//:config"],
+    copts = COPTS,
+    linkstatic = True,
+    local_defines = DEFINES,
+    visibility = ["//:__subpackages__"],
+    deps = [
+        "//:hdrs",
+        "//lib/io",
+    ],
+)
