--- /dev/null
+++ tests/libsqfs/BUILD.bazel
@@ -0,0 +1,32 @@
+load("@rules_cc//cc:defs.bzl", "cc_test")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS", "LINKOPTS")
+
+[
+    cc_test(
+        name = "{}".format(file),
+        srcs = ["{}.c".format(file)],
+        args = args,
+        copts = COPTS,
+        linkopts = LINKOPTS,
+        linkstatic = True,
+        local_defines = DEFINES,
+        deps = [
+            "//lib/common",
+            "//lib/compat",
+            "//lib/sqfs:squashfs",
+        ],
+    )
+    for file, args in {
+        "abi": [],
+        "get_node_path": [],
+        "table": [],
+        "xattr_benchmark": [
+            "-b",
+            "10",
+            "-g",
+            "4",
+        ],
+        "xattr_writer": [],
+    }.items()
+]
