--- /dev/null
+++ lib/compat/BUILD.bazel
@@ -0,0 +1,29 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS")
+
+filegroup(
+    name = "srcs",
+    srcs = [
+        "strchrnul.c",
+        "strndup.c",
+    ],
+    visibility = ["//:__subpackages__"],
+)
+
+cc_library(
+    name = "compat",
+    srcs = glob(
+        [
+            "**/*.c",
+            "**/*.h",
+        ],
+    ) + ["//:config"],
+    copts = COPTS,
+    linkstatic = True,
+    local_defines = DEFINES,
+    visibility = ["//:__subpackages__"],
+    deps = [
+        "//:hdrs",
+    ],
+)
