--- /dev/null
+++ lib/io/BUILD.bazel
@@ -0,0 +1,37 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS")
+
+cc_library(
+    name = "io",
+    srcs = glob(
+        [
+            "**/*.c",
+            "**/*.h",
+        ],
+        exclude = [
+            "unix/**/*",
+            "win32/**/*",
+        ],
+    ) + select({
+        "@platforms//os:windows": glob([
+            "win32/**/*.c",
+            "win32/**/*.h",
+        ]),
+        "//conditions:default": glob([
+            "unix/**/*.c",
+            "unix/**/*.h",
+        ]),
+    }) + ["//:config"],
+    copts = COPTS,
+    linkstatic = True,
+    local_defines = DEFINES,
+    visibility = ["//:__subpackages__"],
+    deps = [
+        "//:hdrs",
+        "@bzip2//:bz2",
+        "@xz//:lzma",
+        "@zlib",
+        "@zstd",
+    ],
+)
