--- /dev/null
+++ lib/util/BUILD.bazel
@@ -0,0 +1,37 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS")
+
+filegroup(
+    name = "srcs",
+    srcs = [
+        "alloc.c",
+        "array.c",
+        "file_cmp.c",
+        "hash_table.c",
+        "is_memory_zero.c",
+        "mempool.c",
+        "rbtree.c",
+        "str_table.c",
+        "threadpool.c",
+        "xxhash.c",
+    ],
+    visibility = ["//:__subpackages__"],
+)
+
+cc_library(
+    name = "util",
+    srcs = glob(
+        [
+            "**/*.c",
+            "**/*.h",
+        ],
+    ) + ["//:config"],
+    copts = COPTS,
+    linkstatic = True,
+    local_defines = DEFINES,
+    visibility = ["//:__subpackages__"],
+    deps = [
+        "//:hdrs",
+    ],
+)
