--- /dev/null
+++ tests/libtar/BUILD.bazel
@@ -0,0 +1,312 @@
+load("@rules_cc//cc:defs.bzl", "cc_test")
+load("//:defines.bzl", "DEFINES")
+load("//:pthread.bzl", "COPTS", "LINKOPTS")
+
+exports_files(
+    srcs = ["data"],
+    visibility = ["//tests:__subpackages__"],
+)
+
+[
+    cc_test(
+        name = "{}".format(name),
+        srcs = ["{}".format(file)],
+        args = args,
+        copts = COPTS,
+        data = [
+            ":data",
+        ],
+        linkopts = LINKOPTS,
+        linkstatic = True,
+        local_defines = DEFINES + defines,
+        deps = [
+            "//lib/compat",
+            "//lib/io",
+            "//lib/tar",
+            "//lib/util",
+        ],
+    )
+    for name, file, args, defines in [
+        (
+            "tar_gnu0",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=format-acceptance/gnu.tar",
+            ],
+        ),
+        (
+            "tar_gnu1",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=format-acceptance/gnu-g.tar",
+            ],
+        ),
+        (
+            "tar_gnu2",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=user-group-largenum/gnu.tar",
+                "TESTUID=0x80000000",
+                "TESTGID=0x80000000",
+                "TESTTS=1542995392",
+            ],
+        ),
+        (
+            "tar_gnu3",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=negative-mtime/gnu.tar",
+                "TESTTS=-315622800",
+            ],
+        ),
+        (
+            "tar_gnu4",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=long-paths/gnu.tar",
+                "LONG_NAME_TEST",
+                "TESTTS=1542909670",
+            ],
+        ),
+        (
+            "tar_gnu5",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=large-mtime/gnu.tar",
+                "TESTTS=8589934592L",
+            ],
+        ),
+        (
+            "tar_gnu6",
+            "tar_big_file.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=file-size/gnu.tar",
+            ],
+        ),
+        (
+            "tar_pax0",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=format-acceptance/pax.tar",
+            ],
+        ),
+        (
+            "tar_pax1",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=user-group-largenum/pax.tar",
+                "TESTUID=2147483648UL",
+                "TESTGID=2147483648UL",
+                "TESTTS=1542995392",
+            ],
+        ),
+        (
+            "tar_pax2",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=large-mtime/pax.tar",
+                "TESTTS=8589934592L",
+            ],
+        ),
+        (
+            "tar_pax3",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=negative-mtime/pax.tar",
+                "TESTTS=-315622800",
+            ],
+        ),
+        (
+            "tar_pax4",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=long-paths/pax.tar",
+                "LONG_NAME_TEST",
+                "TESTTS=1542909670",
+            ],
+        ),
+        (
+            "tar_pax5",
+            "tar_big_file.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=file-size/pax.tar",
+            ],
+        ),
+        (
+            "tar_ustar0",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=format-acceptance/ustar.tar",
+            ],
+        ),
+        (
+            "tar_ustar1",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=format-acceptance/ustar-pre-posix.tar",
+            ],
+        ),
+        (
+            "tar_ustar2",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=format-acceptance/v7.tar",
+            ],
+        ),
+        (
+            "tar_ustar3",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=user-group-largenum/8-digit.tar",
+                "TESTUID=8388608",
+                "TESTGID=8388608",
+                "TESTTS=1542995392",
+            ],
+        ),
+        (
+            "tar_ustar4",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=large-mtime/12-digit.tar",
+                "TESTTS=8589934592L",
+            ],
+        ),
+        (
+            "tar_ustar5",
+            "tar_simple.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=long-paths/ustar.tar",
+                "LONG_NAME_TEST",
+                "TESTTS=1542909670",
+            ],
+        ),
+        (
+            "tar_ustar6",
+            "tar_big_file.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=file-size/12-digit.tar",
+            ],
+        ),
+        (
+            "tar_target_filled",
+            "tar_target_filled.c",
+            [],
+            ["TESTPATH=$(location :data)"],
+        ),
+        (
+            "tar_sparse_gnu",
+            "tar_sparse_gnu.c",
+            [],
+            ["TESTPATH=$(location :data)"],
+        ),
+        (
+            "tar_sparse_gnu0",
+            "tar_sparse.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=sparse-files/pax-gnu0-0.tar",
+            ],
+        ),
+        (
+            "tar_sparse_gnu1",
+            "tar_sparse.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=sparse-files/pax-gnu0-1.tar",
+            ],
+        ),
+        (
+            "tar_sparse_gnu2",
+            "tar_sparse.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=sparse-files/pax-gnu1-0.tar",
+            ],
+        ),
+        (
+            "tar_sparse_gnu3",
+            "tar_sparse.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=sparse-files/gnu.tar",
+            ],
+        ),
+        (
+            "tar_xattr_bsd",
+            "tar_xattr.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=xattr/xattr-libarchive.tar",
+            ],
+        ),
+        (
+            "tar_xattr_schily",
+            "tar_xattr.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=xattr/xattr-schily.tar",
+            ],
+        ),
+        (
+            "tar_xattr_schily_bin",
+            "tar_xattr_bin.c",
+            [],
+            [
+                "TESTPATH=$(location :data)",
+                "TESTFILE=xattr/xattr-schily-binary.tar",
+            ],
+        ),
+        (
+            "tar_fuzz",
+            "tar_fuzz.c",
+            ["$(location :data)/sparse-files/gnu-small.tar"],
+            ["TESTPATH=$(location :data)"],
+        ),
+    ]
+]
