===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -5,9 +5,9 @@
     bazel_compatibility = [">=7.2.1"],
 )
 
 # Only needed for well-known protos.
-bazel_dep(name = "protobuf", version = "21.7", repo_name = "com_google_protobuf")
+bazel_dep(name = "protobuf", version = "test-tag-5", repo_name = "com_google_protobuf")
 
 # Only needed for proto_library.
 bazel_dep(name = "rules_proto", version = "5.3.0-21.7")
 
