diff --git a/lcm-bazel/private/lcm_c_library_srcs.bzl b/lcm-bazel/private/lcm_c_library_srcs.bzl
index f0c7967..3151f00 100644
--- a/lcm-bazel/private/lcm_c_library_srcs.bzl
+++ b/lcm-bazel/private/lcm_c_library_srcs.bzl
@@ -45,7 +45,7 @@ lcm_c_library_srcs = rule(
             providers = [LcmInfo],
         ),
         "lcmgen": attr.label(
-            cfg = "host",
+            cfg = "exec",
             executable = True,
             default = Label("//lcmgen:lcm-gen"),
         ),
diff --git a/lcm-bazel/private/lcm_cc_library_srcs.bzl b/lcm-bazel/private/lcm_cc_library_srcs.bzl
index 70a752a..442f61a 100644
--- a/lcm-bazel/private/lcm_cc_library_srcs.bzl
+++ b/lcm-bazel/private/lcm_cc_library_srcs.bzl
@@ -43,7 +43,7 @@ lcm_cc_library_srcs = rule(
             providers = [LcmInfo],
         ),
         "lcmgen": attr.label(
-            cfg = "host",
+            cfg = "exec",
             executable = True,
             default = Label("//lcmgen:lcm-gen"),
         ),
diff --git a/lcm-bazel/private/lcm_java_library_srcs.bzl b/lcm-bazel/private/lcm_java_library_srcs.bzl
index f28bf7b..ea1d05f 100644
--- a/lcm-bazel/private/lcm_java_library_srcs.bzl
+++ b/lcm-bazel/private/lcm_java_library_srcs.bzl
@@ -43,7 +43,7 @@ lcm_java_library_srcs = rule(
             providers = [LcmInfo],
         ),
         "lcmgen": attr.label(
-            cfg = "host",
+            cfg = "exec",
             executable = True,
             default = Label("//lcmgen:lcm-gen"),
         ),
diff --git a/lcm-bazel/private/lcm_py_library_srcs.bzl b/lcm-bazel/private/lcm_py_library_srcs.bzl
index 31f6962..b9736fd 100644
--- a/lcm-bazel/private/lcm_py_library_srcs.bzl
+++ b/lcm-bazel/private/lcm_py_library_srcs.bzl
@@ -45,7 +45,7 @@ lcm_py_library_srcs = rule(
             providers = [LcmInfo],
         ),
         "lcmgen": attr.label(
-            cfg = "host",
+            cfg = "exec",
             executable = True,
             default = Label("//lcmgen:lcm-gen"),
         ),
