From 7de898d7ccbf8cf3b0cabcd5ead1858573480ab3 Mon Sep 17 00:00:00 2001
From: Patrick Balestra <me@patrickbalestra.com>
Date: Fri, 23 Dec 2022 10:28:51 +0100
Subject: [PATCH] Add dependency on platforms

---
 MODULE.bazel | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MODULE.bazel b/MODULE.bazel
index 61b449d..392a75e 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -11,6 +11,7 @@ module(
 bazel_dep(name = "bazel_skylib", version = "1.3.0")
 bazel_dep(name = "apple_support", repo_name = "build_bazel_apple_support", version = "1.3.2")
 bazel_dep(name = "rules_cc", version = "0.0.2")
+bazel_dep(name = "platforms", version = "0.0.5")
 bazel_dep(name = "protobuf", repo_name = "com_google_protobuf", version = "3.19.2")  # To be removed once rules_proto is bzlmod-ready.
 
 non_module_deps = use_extension("//swift:extensions.bzl", "non_module_deps")
-- 
2.37.0 (Apple Git-136)

