diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..0d09f54
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,32 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "unittest-cpp",
+    srcs = glob([
+        "UnitTest++/*.h",
+        "UnitTest++/*.cpp",
+    ]) + select({
+        "@platforms//os:windows": glob([
+            "UnitTest++/Win32/*.h",
+            "UnitTest++/Win32/*.cpp",
+        ]),
+        "//conditions:default": glob([
+            "UnitTest++/Posix/*.h",
+            "UnitTest++/Posix/*.cpp",
+        ]),
+    }),
+    hdrs = [
+        "UnitTest++/UnitTest++.h",
+        "UnitTest++/UnitTestPP.h",
+    ],
+    copts = select({
+        "@platforms//os:windows": [
+            "-D_CRT_SECURE_NO_DEPRECATE",  # Windows-specific define
+        ],
+        "//conditions:default": [
+            "-std=c++11",  # Default to C++11 for POSIX systems
+        ],
+    }),
+    includes = ["UnitTest++"],
+    visibility = ["//visibility:public"],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..244e956
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,13 @@
+"""
+Module: unittest-cpp
+Purpose: Provides the unittest++ library compileable as a Bazel target. Includes tests through Bazel
+"""
+
+module(
+    name = "unittest-cpp",
+    version = "2.0.1-0.20201113052638-10e50ad70c69",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "rules_cc", version = "0.1.1")
+bazel_dep(name = "platforms", version = "0.0.11")
diff --git a/tests/BUILD.bazel b/tests/BUILD.bazel
new file mode 100644
index 0000000..b719333
--- /dev/null
+++ b/tests/BUILD.bazel
@@ -0,0 +1,10 @@
+load("@rules_cc//cc:defs.bzl", "cc_test")
+
+cc_test(
+    name = "all_tests",
+    srcs = glob([
+        "*.h",
+        "*.cpp",
+    ]),
+    deps = ["//:unittest-cpp"],
+)
