diff --git a/tools/cpp/logger/platform.hpp b/tools/cpp/logger/platform.hpp
index 0494d2b..9921c39 100644
--- a/tools/cpp/logger/platform.hpp
+++ b/tools/cpp/logger/platform.hpp
@@ -8,7 +8,11 @@
 #else
 # include <sys/time.h>
 # include <unistd.h>
+#if defined(__linux__)
 # include <linux/limits.h>
+#else
+# include <limits.h>
+#endif
 #endif
 
 struct Platform
@@ -18,7 +22,11 @@ struct Platform
         ::fflush(fp);
 #ifndef WIN32
         // Perform a full fsync operation after flush
+        #if defined(__APPLE__)
+        fsync(fileno(fp));
+#else
         fdatasync(fileno(fp));
+#endif
 #endif
     }
 
diff --git a/tools/cpp/transcoder/main.cpp b/tools/cpp/transcoder/main.cpp
index 7710155..384fa0b 100644
--- a/tools/cpp/transcoder/main.cpp
+++ b/tools/cpp/transcoder/main.cpp
@@ -97,7 +97,7 @@ int main(int argc, char* argv[])
         return 1;
     }
     fseeko(inlog.getFilePtr(), 0, SEEK_END);
-    off64_t logSize = ftello(inlog.getFilePtr());
+    off_t logSize = ftello(inlog.getFilePtr());
     fseeko(inlog.getFilePtr(), 0, SEEK_SET);
 
     zcm::LogFile outlog(args.outlog, "w");
@@ -124,7 +124,7 @@ int main(int argc, char* argv[])
 
     size_t numInEvents = 0, numOutEvents = 0;
     const zcm::LogEvent* evt;
-    off64_t offset;
+    off_t offset;
 
     while (1) {
         offset = ftello(inlog.getFilePtr());
diff --git a/zcm/transport/transport_can.cpp b/zcm/transport/transport_can.cpp
index b2a70ab..15d0be4 100644
--- a/zcm/transport/transport_can.cpp
+++ b/zcm/transport/transport_can.cpp
@@ -19,10 +19,12 @@
 #include <net/if.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#ifdef __linux__
 #include <linux/sockios.h>
 
 #include <linux/can.h>
 #include <linux/can/raw.h>
+#endif
 
 // Define this the class name you want
 #define ZCM_TRANS_CLASSNAME TransportCan
@@ -38,7 +40,9 @@ struct ZCM_TRANS_CLASSNAME : public zcm_trans_t
 
     int soc = -1;
     bool socSettingsGood = false;
+    #ifdef __linux__
     struct sockaddr_can addr;
+#endif
 	struct ifreq ifr;
 
     zcm_trans_t* gst = nullptr;
diff --git a/zcm/transport/transport_serial.cpp b/zcm/transport/transport_serial.cpp
index 5220855..b5ee096 100644
--- a/zcm/transport/transport_serial.cpp
+++ b/zcm/transport/transport_serial.cpp
@@ -13,7 +13,9 @@
 #include <errno.h>
 #include <termios.h>
 #include <sys/ioctl.h>
+#ifdef __linux__
 #include <linux/usbdevice_fs.h>
+#endif
 
 #include <cassert>
 #include <cstring>
@@ -97,7 +99,9 @@ bool Serial::open(const string& port_, int baud, bool hwFlowControl)
     // this call may fail: but we don't care
     // occasionally the USB bus can get in a weird state
     // this call will reset the USB in that situation
+    #ifdef __linux__
     ioctl(fd, USBDEVFS_RESET, 0);
+#endif
 
     struct termios opts;
 
diff --git a/zcm/transport/udpm/udpmsocket.cpp b/zcm/transport/udpm/udpmsocket.cpp
index ae5ca43..f38284c 100644
--- a/zcm/transport/udpm/udpmsocket.cpp
+++ b/zcm/transport/udpm/udpmsocket.cpp
@@ -127,7 +127,7 @@ bool UDPMSocket::bindPort(u16 port)
     addr.sin_addr.s_addr = INADDR_ANY;
     addr.sin_port = htons(port);
 
-    if (bind(fd, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
+    if (::bind(fd, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
         perror("bind");
         return false;
     }
