--- a/libswscale/format.c
+++ b/libswscale/format.c
@@ -1434,6 +1434,8 @@
     return c;
 }
 
+#endif /* CONFIG_UNSTABLE */
+
 void ff_sws_chroma_pos(const SwsFormat *fmt, bool *incomplete,
                        int *out_x_pos, int *out_y_pos)
 {
@@ -1475,6 +1477,8 @@
     *out_y_pos = y_pos;
 }
 
+#if CONFIG_UNSTABLE
+
 int ff_sws_decode_colors(SwsContext *ctx, SwsPixelType type,
                          SwsOpList *ops, const SwsFormat *fmt, bool *incomplete)
 {
