diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..2067eee
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,22 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+boost_library(
+    name = "program_options",
+    deps = [ 
+        "@boost//:any",
+        "@boost//:bind",
+        "@boost//:config",
+        "@boost//:core",
+        "@boost//:detail",
+        "@boost//:function",
+        "@boost//:iterator",
+        "@boost//:lexical_cast",
+        "@boost//:smart_ptr",
+        "@boost//:static_assert",
+        "@boost//:throw_exception",
+        "@boost//:tokenizer",
+        "@boost//:type_traits",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..e03a3c2
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.program_options",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..80b50c9
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:program_options",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "program_options_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/program_options_test.cc b/bazel_test/program_options_test.cc
new file mode 100644
index 0000000..f392cb8
--- /dev/null
+++ b/bazel_test/program_options_test.cc
@@ -0,0 +1,24 @@
+#include <boost/program_options.hpp>
+
+namespace po = boost::program_options;
+
+int main()
+{
+  int ac = 3;
+  const char *const av[] = {"", "--compression", "7"};
+
+  po::options_description desc("Allowed options");
+  desc.add_options()
+      ("compression", po::value<int>(), "set compression level")
+  ;
+
+  po::variables_map vm;
+  po::store(po::parse_command_line(ac, av, desc), vm);
+  po::notify(vm);
+
+  if (!vm.count("compression") || vm["compression"].as<int>() != 7) {
+      return 1;
+  }
+
+  return 0;
+}
