diff --git a/folly/testing/TestUtil.cpp b/folly/testing/TestUtil.cpp
index 42d05bb..6522b9d 100644
--- a/folly/testing/TestUtil.cpp
+++ b/folly/testing/TestUtil.cpp
@@ -40,0 +41,2 @@
+#include "tools/cpp/runfiles/runfiles.h"
+
@@ -240,5 +242,9 @@ fs::path find_resource(std::string_view resource) {
-  auto const ext = folly::ext::test_find_resource;
-  auto fn = ext //
-      ? fs::path(ext(resource)) // hooked, eg via internal extension
-      : fs::executable_path().parent_path() //
-          / std::string(resource); // current cmake build
+  using bazel::tools::cpp::runfiles::Runfiles;
+  std::string error;
+  std::unique_ptr<Runfiles> runfiles(
+      Runfiles::CreateForTest(BAZEL_CURRENT_REPOSITORY, &error));
+  if (runfiles == nullptr) {
+    throw std::runtime_error("error initializing bazel runfiles: " + error);
+  }
+  const fs::path fn =
+      runfiles->Rlocation(std::string("folly/") + std::string(resource));
