diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..cd92032
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,21 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+BOOST_CONTAINER_INCLUDES_WITH_SRC_EXTENSION = [
+    "src/dlmalloc_ext_2_8_6.c",
+    "src/dlmalloc_2_8_6.c",
+]
+
+boost_library(
+    name = "container",
+    hdrs = BOOST_CONTAINER_INCLUDES_WITH_SRC_EXTENSION,
+    exclude_src = BOOST_CONTAINER_INCLUDES_WITH_SRC_EXTENSION,
+    deps = [
+        "@boost//:assert",
+        "@boost//:config",
+        "@boost//:intrusive",
+        "@boost//:move",
+        "@boost//:static_assert",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..084228f
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,14 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.container",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/bazel_test/BUILD.bazel b/bazel_test/BUILD.bazel
new file mode 100644
index 0000000..f4ff8e7
--- /dev/null
+++ b/bazel_test/BUILD.bazel
@@ -0,0 +1,16 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:container",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        file_extensions = ".cc",
+        names = [
+            "container_test",
+        ],
+        deps = DEPS,
+    ),
+)
diff --git a/bazel_test/container_test.cc b/bazel_test/container_test.cc
new file mode 100644
index 0000000..daf5f57
--- /dev/null
+++ b/bazel_test/container_test.cc
@@ -0,0 +1,10 @@
+#include <boost/container/allocator.hpp>
+#include <boost/container/flat_map.hpp>
+
+int main()
+{
+  boost::container::flat_map<int, int, std::less<int>, boost::container::allocator<std::pair<int, int>>> m;
+  m.emplace(1, 42);
+  m.emplace(2, 0);
+  return m.at(2);
+}
