diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..cd92032
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,21 @@
+# Hey there! Before you start modifying this file, go check out https://github.com/dynacondev/boost.rules.tools for the tool you need!
+
+load("@boost.rules.tools//:tools.bzl", "boost_library")
+
+BOOST_CONTAINER_INCLUDES_WITH_SRC_EXTENSION = [
+    "src/dlmalloc_ext_2_8_6.c",
+    "src/dlmalloc_2_8_6.c",
+]
+
+boost_library(
+    name = "container",
+    hdrs = BOOST_CONTAINER_INCLUDES_WITH_SRC_EXTENSION,
+    exclude_src = BOOST_CONTAINER_INCLUDES_WITH_SRC_EXTENSION,
+    deps = [
+        "@boost//:assert",
+        "@boost//:config",
+        "@boost//:intrusive",
+        "@boost//:move",
+        "@boost//:static_assert",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100755
index 0000000..ef8936f
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,17 @@
+"""
+Hey there! Before editing, check out the tool at https://github.com/dynacondev/boost.rules.tools
+Note: These modules follow a circular dependency structure on the boost module https://github.com/dynacondev/boost.rules to keep boost releases from mixing!
+"""
+
+module(
+    name = "boost.container",
+    version = "1.83.0.bcr.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "boost", version = "1.83.0.bcr.1")
+bazel_dep(name = "boost.rules.tools", version = "1.0.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+
+# Required for testing
+bazel_dep(name = "boost.intrusive", version = "1.83.0.bcr.1")
diff --git a/WORKSPACE.bazel b/WORKSPACE.bazel
new file mode 100644
index 0000000..be0754f
--- /dev/null
+++ b/WORKSPACE.bazel
@@ -0,0 +1 @@
+# SEE: MODULE.bazel
diff --git a/test/BUILD.bazel b/test/BUILD.bazel
new file mode 100644
index 0000000..bc3481c
--- /dev/null
+++ b/test/BUILD.bazel
@@ -0,0 +1,19 @@
+load("@boost.rules.tools//:tools.bzl", "boost_test_set")
+
+DEPS = [
+    "@boost//:algorithm",
+    "@boost//:container",
+    "@boost//:core",
+    "@boost//:intrusive",
+    "@boost//:iterator",
+    "@boost//:tuple",
+    "@boost//:utility",
+    "@boost.intrusive//test:iterator_test",
+]
+
+test_suite(
+    name = "tests",
+    tests = boost_test_set(
+        deps = DEPS,
+    ),
+)
diff --git a/test/deque_test.cpp b/test/deque_test.cpp
index 907bee5..21f684b 100644
--- a/test/deque_test.cpp
+++ b/test/deque_test.cpp
@@ -28,7 +28,7 @@
 #include "propagate_allocator_test.hpp"
 #include "vector_test.hpp"
 #include "default_init_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 using namespace boost::container;
 
diff --git a/test/devector_test.cpp b/test/devector_test.cpp
index 97f0bff..353b724 100644
--- a/test/devector_test.cpp
+++ b/test/devector_test.cpp
@@ -37,7 +37,7 @@
 #include "input_iterator.hpp"
 #include "vector_test.hpp"
 #include "default_init_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 using namespace boost::container;
 
diff --git a/test/flat_map_test.cpp b/test/flat_map_test.cpp
index 2dd17a1..fa6154c 100644
--- a/test/flat_map_test.cpp
+++ b/test/flat_map_test.cpp
@@ -21,7 +21,7 @@
 #include "propagate_allocator_test.hpp"
 #include "container_common_tests.hpp"
 #include "emplace_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 #include <map>
 #include <utility>
diff --git a/test/flat_set_test.cpp b/test/flat_set_test.cpp
index 00dfe07..bd61797 100644
--- a/test/flat_set_test.cpp
+++ b/test/flat_set_test.cpp
@@ -23,7 +23,7 @@
 #include "propagate_allocator_test.hpp"
 #include "emplace_test.hpp"
 #include "container_common_tests.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 using namespace boost::container;
 
diff --git a/test/list_test.cpp b/test/list_test.cpp
index c4ec219..c1dfd35 100644
--- a/test/list_test.cpp
+++ b/test/list_test.cpp
@@ -17,7 +17,7 @@
 #include "list_test.hpp"
 #include "propagate_allocator_test.hpp"
 #include "emplace_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 using namespace boost::container;
 
diff --git a/test/map_test.cpp b/test/map_test.cpp
index 67f893c..410b8ad 100644
--- a/test/map_test.cpp
+++ b/test/map_test.cpp
@@ -18,7 +18,7 @@
 #include "map_test.hpp"
 #include "propagate_allocator_test.hpp"
 #include "emplace_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 using namespace boost::container;
 
diff --git a/test/set_test.cpp b/test/set_test.cpp
index 2c7c7ca..95552e8 100644
--- a/test/set_test.cpp
+++ b/test/set_test.cpp
@@ -17,7 +17,7 @@
 #include "set_test.hpp"
 #include "propagate_allocator_test.hpp"
 #include "emplace_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 #include <utility> //for std::pair
 
 using namespace boost::container;
diff --git a/test/slist_test.cpp b/test/slist_test.cpp
index 16fd8a0..c093744 100644
--- a/test/slist_test.cpp
+++ b/test/slist_test.cpp
@@ -16,7 +16,7 @@
 #include "list_test.hpp"
 #include "propagate_allocator_test.hpp"
 #include "emplace_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 using namespace boost::container;
 
diff --git a/test/small_vector_test.cpp b/test/small_vector_test.cpp
index 55d737c..6a7bcf2 100644
--- a/test/small_vector_test.cpp
+++ b/test/small_vector_test.cpp
@@ -12,7 +12,7 @@
 #include "movable_int.hpp"
 #include "propagate_allocator_test.hpp"
 #include "default_init_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 #include <boost/container/allocator.hpp>
 #include <boost/container/allocator_traits.hpp>
diff --git a/test/stable_vector_test.cpp b/test/stable_vector_test.cpp
index 93b6ce7..857ac90 100644
--- a/test/stable_vector_test.cpp
+++ b/test/stable_vector_test.cpp
@@ -21,7 +21,7 @@
 #include "propagate_allocator_test.hpp"
 #include "vector_test.hpp"
 #include "default_init_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 using namespace boost::container;
 
diff --git a/test/static_vector_test.cpp b/test/static_vector_test.cpp
index 3046c69..9c40bfe 100644
--- a/test/static_vector_test.cpp
+++ b/test/static_vector_test.cpp
@@ -11,7 +11,7 @@
 #include <boost/container/vector.hpp>
 #include <boost/container/list.hpp>
 #include <boost/container/detail/iterator.hpp>
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 #include <boost/container/detail/config_begin.hpp>
 
 #include "static_vector_test.hpp"
diff --git a/test/string_test.cpp b/test/string_test.cpp
index 8e63f12..effca47 100644
--- a/test/string_test.cpp
+++ b/test/string_test.cpp
@@ -24,7 +24,7 @@
 #include "propagate_allocator_test.hpp"
 #include "default_init_test.hpp"
 #include "comparison_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 #include <boost/utility/string_view.hpp>
 #include <boost/core/lightweight_test.hpp>
 
diff --git a/test/vector_test.cpp b/test/vector_test.cpp
index 5dc3631..95b180b 100644
--- a/test/vector_test.cpp
+++ b/test/vector_test.cpp
@@ -29,7 +29,7 @@
 #include "propagate_allocator_test.hpp"
 #include "vector_test.hpp"
 #include "default_init_test.hpp"
-#include "../../intrusive/test/iterator_test.hpp"
+#include "intrusive/test/iterator_test.hpp"
 
 using namespace boost::container;
 
