diff --git a/BUILD.bazel b/BUILD.bazel
index 003c1b9..9be32e9 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,6 +1,9 @@
 # Description:
 #   Brotli is a generic-purpose lossless compression algorithm.
 
+load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(
     default_visibility = ["//visibility:public"],
 )
@@ -26,7 +29,6 @@ config_setting(
 )
 
 STRICT_C_OPTIONS = select({
-    ":msvc": [],
     ":clang-cl": [
         "/W4",
         "-Wconversion",
@@ -38,6 +40,7 @@ STRICT_C_OPTIONS = select({
         "-Wsign-compare",
         "-Wno-sign-conversion",
     ],
+    ":msvc": [],
     "//conditions:default": [
         "--pedantic-errors",
         "-Wall",
