--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,48 @@
+cc_library(
+    name = "reflect-cpp",
+    hdrs = glob(
+        [
+            "include/rfl.hpp",
+            "include/rfl/*.hpp",
+            "include/rfl/generic/**",
+            "include/rfl/internal/**",
+            "include/rfl/io/**",
+            "include/rfl/json/**",
+            "include/rfl/capnproto/**",
+            "include/rfl/parsing/**",
+            "include/rfl/thirdparty/**",
+            "include/rfl/yaml/**",
+        ],
+        exclude = [
+            "include/rfl/avro.hpp",
+            "include/rfl/bson.hpp",
+            "include/rfl/cbor.hpp",
+            "include/rfl/flexbuf.hpp",
+            "include/rfl/msgpack.hpp",
+            "include/rfl/toml.hpp",
+            "include/rfl/ubjson.hpp",
+            "include/rfl/xml.hpp",
+        ]
+    ),
+    srcs = glob([
+        "src/rfl/Generic.cpp",
+        "src/rfl/generic/**",
+        "src/rfl/internal/**",
+        "src/rfl/json/**",
+        "src/rfl/capnproto/**",
+        "src/rfl/parsing/**",
+        "src/rfl/yaml/**",
+    ]),
+    includes = ["include"],
+    deps = [
+        "@yaml-cpp",
+        "@yyjson",
+        "@capnp-cpp//src/capnp",
+        "@capnp-cpp//src/capnp:capnpc",
+    ],
+    copts = ["--std=c++20"],
+    visibility = ["//visibility:public"],
+    defines = [
+      "REFLECT_CPP_C_ARRAYS_OR_INHERITANCE",
+    ]
+)
