--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,15 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "plthook",
+    hdrs = ["plthook.h"],
+    srcs = select({
+        "@platforms//os:linux": ["plthook_elf.c"],
+        "@platforms//os:android": ["plthook_elf.c"],
+        "@platforms//os:macos": ["plthook_osx.c"],
+        "@platforms//os:windows": ["plthook_win32.c"],
+        "//conditions:default": ["plthook_elf.c"],
+    }),
+    includes = ["."],
+    visibility = ["//visibility:public"],
+)
