
diff --git s2/base/BUILD s2/base/BUILD
index a6770f1..3574511 100644
--- s2/base/BUILD
+++ s2/base/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/testing/BUILD s2/testing/BUILD
index cbd18f3..82c9f06 100644
--- s2/testing/BUILD
+++ s2/testing/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/bitmap/BUILD s2/util/bitmap/BUILD
index 93ccb49..61c3e38 100644
--- s2/util/bitmap/BUILD
+++ s2/util/bitmap/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/bits/BUILD s2/util/bits/BUILD
index 5d89528..3cf75ef 100644
--- s2/util/bits/BUILD
+++ s2/util/bits/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/coding/BUILD s2/util/coding/BUILD
index 5f84624..9bba12c 100644
--- s2/util/coding/BUILD
+++ s2/util/coding/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/endian/BUILD s2/util/endian/BUILD
index 7bc114c..31378bd 100644
--- s2/util/endian/BUILD
+++ s2/util/endian/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/gtl/BUILD s2/util/gtl/BUILD
index b6c0dca..28c2c7c 100644
--- s2/util/gtl/BUILD
+++ s2/util/gtl/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/hash/BUILD s2/util/hash/BUILD
index 363f501..7526769 100644
--- s2/util/hash/BUILD
+++ s2/util/hash/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/math/BUILD s2/util/math/BUILD
index ef27577..8911295 100644
--- s2/util/math/BUILD
+++ s2/util/math/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/math/exactfloat/BUILD s2/util/math/exactfloat/BUILD
index 770c246..fa347ab 100644
--- s2/util/math/exactfloat/BUILD
+++ s2/util/math/exactfloat/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/random/BUILD s2/util/random/BUILD
index c0203c5..5d28a90 100644
--- s2/util/random/BUILD
+++ s2/util/random/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
diff --git s2/util/units/BUILD s2/util/units/BUILD
index 6d4bf7d..3735773 100644
--- s2/util/units/BUILD
+++ s2/util/units/BUILD
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
