diff --git BUILD.bazel BUILD.bazel
new file mode 100644
index 0000000..b34de7e
--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,16 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "boost.type_traits",
+    hdrs = glob([
+        "include/**/*.hpp",
+        "include/**/*.h",
+    ]),
+    includes = ["include"],
+    deps = [
+        "@boost.config",
+        "@boost.static_assert",
+    ],
+)
