diff --git a/src/sat/cadical/cadical_kitten.c b/src/sat/cadical/cadical_kitten.c
index da37389..8b0185b 100644
--- a/src/sat/cadical/cadical_kitten.c
+++ b/src/sat/cadical/cadical_kitten.c
@@ -109,8 +109,6 @@ struct klause {
   unsigned lits[1];
 };
 
-typedef struct statistics statistics;
-
 struct statistics {
   uint64_t learned;
   uint64_t original;
@@ -1929,13 +1927,13 @@ unsigned cadical_kitten_compute_clausal_core (cadical_kitten *cadical_kitten,
       //goto DONE;
         if (learned_ptr)
           *learned_ptr = learned;
-        
+
         LOG ("clausal core of %u original clauses", original);
         LOG ("clausal core of %" PRIu64 " learned clauses", learned);
         cadical_kitten->statistics.original = original;
         cadical_kitten->statistics.learned = 0;
         UPDATE_STATUS (21);
-        
+
         return original;
     }
   }
