diff --git a/protobuf.bzl b/protobuf.bzl
index 3c8afcc17..a5a125944 100644
--- a/protobuf.bzl
+++ b/protobuf.bzl
@@ -501,8 +501,8 @@ def internal_ruby_proto_library(
         srcs = [],
         deps = [],
         includes = ["."],
-        default_runtime = "@com_google_protobuf//ruby:protobuf",
-        protoc = "@com_google_protobuf//:protoc",
+        default_runtime = Label("//ruby:protobuf"),
+        protoc = Label("//:protoc"),
         testonly = None,
         visibility = ["//visibility:public"],
         **kwargs):
