===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -3,8 +3,9 @@
 module(
     name = "aspect_rules_js",
     bazel_compatibility = [">=6.0.0"],
     compatibility_level = 1,
+    version = "2.8.2",
 )
 
 # Lower-bounds (minimum) versions for direct runtime dependencies.
 # Do not bump these unless rules_js requires a newer version to function.
