diff --git a/perl/binary_wrapper.tpl b/perl/binary_wrapper.tpl
index f1cb98d..16e60b5 100644
--- a/perl/binary_wrapper.tpl
+++ b/perl/binary_wrapper.tpl
@@ -7,8 +7,7 @@ elif [ -s `dirname $0`/../../MANIFEST ]; then
 elif [ -d $0.runfiles ]; then
   PATH_PREFIX=`cd $0.runfiles; pwd`/{workspace_name}/
 else
-  echo "WARNING: it does not look to be at the .runfiles directory" >&2
-  exit 1
+  PATH_PREFIX=./
 fi
 
 export PERL5LIB="$PERL5LIB{PERL5LIB}"
diff --git a/test/data_dep/BUILD b/test/data_dep/BUILD
new file mode 100644
index 0000000..00fc8a3
--- /dev/null
+++ b/test/data_dep/BUILD
@@ -0,0 +1,35 @@
+genrule(
+    name = "gen_program_sh",
+    srcs = ["@genhtml//:genhtml_bin"],
+    outs = ["program.sh"],
+    cmd = """\
+cat <<"EOF" >$@
+#!/bin/bash
+set -euxo pipefail
+genhtml='$(rootpath @genhtml//:genhtml_bin)'
+test "$$("$$genhtml" --version)" == "genhtml: LCOV version 1.0"
+EOF
+""",
+)
+
+sh_binary(
+    name = "program_bin",
+    srcs = ["program.sh"],
+    data = ["@genhtml//:genhtml_bin"],
+)
+
+genrule(
+    name = "gen_program_test_sh",
+    srcs = [":program_bin"],
+    outs = ["program_test.sh"],
+    cmd = """\
+echo '#!/bin/sh' >$@
+echo 'exec env --ignore-environment test/data_dep/program_bin' >>$@
+""",
+)
+
+sh_test(
+    name = "program_test",
+    srcs = ["program_test.sh"],
+    data = [":program_bin"],
+)
