--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,17 @@
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "fmi4cpp",
+    srcs = glob([
+        "src/**/*.hpp",
+        "src/**/*.cpp",
+        "src/**/*.h",
+    ]),
+    hdrs = glob([
+        "include/**/*.hpp",
+        "include/**/*.h"
+    ]),
+    includes = ["include", "src"],
+    deps = ["@pugixml//:pugixml"],
+    visibility = ["//visibility:public"]
+)
