diff --git a/folly/io/async/fdsock/test/AsyncFdSocketTest.cpp b/folly/io/async/fdsock/test/AsyncFdSocketTest.cpp
index 197e012e8..c03994f2b 100644
--- a/folly/io/async/fdsock/test/AsyncFdSocketTest.cpp
+++ b/folly/io/async/fdsock/test/AsyncFdSocketTest.cpp
@@ -245,7 +245,7 @@ TEST_F(AsyncFdSocketTest, MultiPartSend) {
   }
   rcb_.verifyData(data.data(), data.size());
   rcb_.clearData();
-  EXPECT_EQ(numSendParts, sendSock.numWrites_);
+  EXPECT_GE(sendSock.numWrites_, numSendParts);
 
   // There are no more data or FDs
   evb_.loopOnce(EVLOOP_NONBLOCK);
