diff --git a/MODULE.bazel b/MODULE.bazel
index 80c40ad..d13eda5 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -2,7 +2,7 @@
 
 module(
     name = "rules_swift",
-    version = "0",
+    version = "3.5.0",
     bazel_compatibility = [">=7.0.0"],
     compatibility_level = 3,
     repo_name = "build_bazel_rules_swift",
@@ -18,7 +18,7 @@ bazel_dep(name = "protobuf", version = "33.4", repo_name = "com_google_protobuf"
 bazel_dep(name = "nlohmann_json", version = "3.12.0.bcr.1", repo_name = "com_github_nlohmann_json")
 bazel_dep(
     name = "swift_argument_parser",
-    version = "1.3.1.2",
+    version = "1.7.0",
     repo_name = "com_github_apple_swift_argument_parser",
 )
 
