diff --git a/plugin/obj_decoder/obj_decoder.cc b/plugin/obj_decoder/obj_decoder.cc
--- a/plugin/obj_decoder/obj_decoder.cc
+++ b/plugin/obj_decoder/obj_decoder.cc
@@ -20,7 +20,7 @@
 #include <mujoco/mjplugin.h>
 #include <mujoco/mjspec.h>
 #include <mujoco/mujoco.h>
-#include <tiny_obj_loader.h>
+#include "tiny_obj_loader.h"
 
 namespace {
 
diff --git a/src/user/user_mesh.cc b/src/user/user_mesh.cc
--- a/src/user/user_mesh.cc
+++ b/src/user/user_mesh.cc
@@ -69,7 +69,7 @@
 #include "user/user_util.h"
 
 extern "C" {
-#include "qhull_ra.h"
+#include "libqhull_r/qhull_ra.h"
 }
 
 namespace {
diff --git a/src/xml/mjz/mjz_encoder.cc b/src/xml/mjz/mjz_encoder.cc
--- a/src/xml/mjz/mjz_encoder.cc
+++ b/src/xml/mjz/mjz_encoder.cc
@@ -31,7 +31,7 @@
   #pragma GCC diagnostic push
   #pragma GCC diagnostic ignored "-Wunused-function"
 #endif
-#include <miniz.h>
+#include "miniz.h"
 #if defined(__GNUC__) || defined(__clang__)
   #pragma GCC diagnostic pop
 #endif
diff --git a/src/xml/mjz/mjz_archive_provider.cc b/src/xml/mjz/mjz_archive_provider.cc
--- a/src/xml/mjz/mjz_archive_provider.cc
+++ b/src/xml/mjz/mjz_archive_provider.cc
@@ -31,7 +31,7 @@
   #pragma GCC diagnostic push
   #pragma GCC diagnostic ignored "-Wunused-function"
 #endif
-#include <miniz.h>
+#include "miniz.h"
 #if defined(__clang__)
   #pragma clang diagnostic pop
 #elif defined(__GNUC__)
diff --git a/test/xml/mjz/mjz_decoder_test.cc b/test/xml/mjz/mjz_decoder_test.cc
--- a/test/xml/mjz/mjz_decoder_test.cc
+++ b/test/xml/mjz/mjz_decoder_test.cc
@@ -29,7 +29,7 @@
   #pragma GCC diagnostic push
   #pragma GCC diagnostic ignored "-Wunused-function"
 #endif
-#include <miniz.h>
+#include "miniz.h"
 #if defined(__GNUC__) || defined(__clang__)
   #pragma GCC diagnostic pop
 #endif
diff --git a/test/xml/mjz/mjz_encoder_test.cc b/test/xml/mjz/mjz_encoder_test.cc
--- a/test/xml/mjz/mjz_encoder_test.cc
+++ b/test/xml/mjz/mjz_encoder_test.cc
@@ -34,7 +34,7 @@
   #pragma GCC diagnostic push
   #pragma GCC diagnostic ignored "-Wunused-function"
 #endif
-#include <miniz.h>
+#include "miniz.h"
 #if defined(__GNUC__) || defined(__clang__)
   #pragma GCC diagnostic pop
 #endif
diff --git a/test/xml/mjz/mjz_archive_provider_test.cc b/test/xml/mjz/mjz_archive_provider_test.cc
--- a/test/xml/mjz/mjz_archive_provider_test.cc
+++ b/test/xml/mjz/mjz_archive_provider_test.cc
@@ -26,7 +26,7 @@
 
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
-#include <miniz.h>
+#include "miniz.h"
 #include <mujoco/mujoco.h>
 #include "test/fixture.h"
 
