diff --git a/MODULE.bazel b/MODULE.bazel
index 1610bed..584a3ce 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,11 +1,14 @@
 module(
     name = "rules_sh",
-    version = "0.2.0",
+    version = "0.3.0",
     compatibility_level = 0,
-    toolchains_to_register = ["@local_posix_config//:local_posix_toolchain"],
 )
+
 bazel_dep(name = "bazel_skylib", version = "1.0.3")
 bazel_dep(name = "platforms", version = "0.0.4")
 
-sh_configure = use_extension("@rules_sh//bzlmod:extensions.bzl", "sh_configure")
-use_repo(sh_configure, "local_posix_config")
+sh_configure = use_extension("//bzlmod:extensions.bzl", "sh_configure")
+
+use_repo(sh_configure, "local_posix_config", "rules_sh_shim_exe")
+
+register_toolchains("@local_posix_config//:local_posix_toolchain")
diff --git a/bzlmod/extensions.bzl b/bzlmod/extensions.bzl
index d54f5c2..90a7f42 100644
--- a/bzlmod/extensions.bzl
+++ b/bzlmod/extensions.bzl
@@ -1,6 +1,14 @@
 load("//sh:posix.bzl", "sh_posix_configure")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_file")
 
 def _sh_configure_impl(ctx):
     sh_posix_configure(register = False)
+    http_file(
+        name = "rules_sh_shim_exe",
+        sha256 = "cb440b8a08a2095a59666a859b35aa5a1524b140b909ecc760f38f3baccf80e6",
+        urls = ["https://github.com/ScoopInstaller/Shim/releases/download/v1.0.1/shim.exe"],
+        downloaded_file_path = "shim.exe",
+        executable = True,
+    )
 
 sh_configure = module_extension(implementation = _sh_configure_impl)
