--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -2,7 +2,7 @@
     name = "grpc-java",
     compatibility_level = 0,
     repo_name = "io_grpc_grpc_java",
-    version = "1.78.0",  # CURRENT_GRPC_VERSION
+    version = "1.78.0.bcr.1",  # CURRENT_GRPC_VERSION
 )

 # GRPC_DEPS_START
@@ -48,14 +48,13 @@

 bazel_dep(name = "bazel_jar_jar", version = "0.1.7")
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
-bazel_dep(name = "googleapis", repo_name = "com_google_googleapis", version = "0.0.0-20240326-1c8d509c5")
+bazel_dep(name = "googleapis", repo_name = "com_google_googleapis", version = "0.0.0-20240819-fe8ba054a")
 bazel_dep(name = "grpc-proto", repo_name = "io_grpc_grpc_proto", version = "0.0.0-20240627-ec30f58")
-# Protobuf 25.5+ is incompatible with Bazel 7 with bzlmod
-bazel_dep(name = "protobuf", repo_name = "com_google_protobuf", version = "24.4")
-bazel_dep(name = "rules_cc", version = "0.0.9")
-bazel_dep(name = "rules_java", version = "5.3.5")
+bazel_dep(name = "protobuf", repo_name = "com_google_protobuf", version = "29.0")
+bazel_dep(name = "rules_cc", version = "0.0.17")
+bazel_dep(name = "rules_java", version = "8.5.1")
 bazel_dep(name = "rules_jvm_external", version = "6.0")
-bazel_dep(name = "rules_proto", version = "5.3.0-21.7")
+bazel_dep(name = "rules_proto", version = "7.0.2")

 maven = use_extension("@rules_jvm_external//:extensions.bzl", "maven")

@@ -183,3 +182,7 @@
     coordinates = "io.grpc:grpc-util",
     target = "@io_grpc_grpc_java//util",
 )
+
+switched_rules = use_extension("@com_google_googleapis//:extensions.bzl", "switched_rules")
+
+switched_rules.use_languages(java = True)

--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -15,6 +15,7 @@
 load("@rules_java//java:defs.bzl", "java_library", "java_plugin", "java_proto_library")
 load("@rules_jvm_external//:defs.bzl", "artifact")
 load(":java_grpc_library.bzl", "java_grpc_library")
+load("@com_google_protobuf//bazel:java_proto_library.bzl", "java_proto_library")

 java_proto_library(
     name = "api_proto_java",
