diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..e08abe7
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,33 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
+
+cc_library(
+    name = "jsmn",
+    hdrs = ["jsmn.h"],
+    visibility = ["//visibility:public"],
+)
+
+cc_test(
+    name = "test",
+    srcs = [
+        "test/test.h",
+        "test/tests.c",
+        "test/testutil.h",
+    ],
+    copts = [
+        "-DJSMN_STRICT=1",
+        "-DJSMN_PARENT_LINKS=1",
+    ],
+    deps = [":jsmn"],
+)
+
+cc_binary(
+    name = "example-simple",
+    srcs = ["example/simple.c"],
+    deps = [":jsmn"],
+)
+
+cc_binary(
+    name = "example-jsondump",
+    srcs = ["example/jsondump.c"],
+    deps = [":jsmn"],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..3447de6
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,13 @@
+"""
+Module: jsmn
+Purpose: Provides the jsmn library compileable as a Bazel target. Includes tests through Bazel and the examples as binary targets
+"""
+
+module(
+    name = "jsmn",
+    version = "1.1.1-0.20211014115138-25647e692c79",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "rules_cc", version = "0.1.1")
+bazel_dep(name = "platforms", version = "0.0.11")
