--- MODULE.bazel
+++ MODULE.bazel
@@ -1,8 +1,6 @@
 module(
     name = "grpc_ecosystem_grpc_gateway",
-    # TODO: Change this to the actual version on each release.
-    #       This can wait until we publish this project on the Bazel registry.
-    version = "0.0.0",
+    version = "2.29.0",
 )
 
 # Bazel Central Registry modules.
@@ -17,6 +15,9 @@ bazel_dep(name = "bazel_skylib", version = "1.9.0")
 bazel_dep(name = "protobuf", version = "34.1", repo_name = "com_google_protobuf")
 bazel_dep(name = "googleapis", version = "0.0.0-20260402-c8ca5bce")
 
+# This is required as an indirect dependency to fix incompatibility issues.
+bazel_dep(name = "rules_android", version = "0.7.1", repo_name = None)
+
 # This is required as a transitive dependency and not directly needed by this module.
 # We have this version pinned to solve for differences in the MODULE.bazel.lock file
 # when running CI.
